<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />

<meta name="date" content="2022-10-05" />

<title>ADVANZ-4 status report</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ADVANZ-4 status report</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">2022-10-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
Get_DistMats
</li>
<li>
wilcox_across_group
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>010_Basic_Report_cache</code> and re-run the analysis. To have
workflowr automatically delete the cache directory prior to building the
file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree679dfd549a7abbc250f490fac82dc764a6fc7c65targetblank679dfd5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MicrobialGenomics/advanz4/tree/679dfd549a7abbc250f490fac82dc764a6fc7c65" target="_blank">679dfd5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree679dfd549a7abbc250f490fac82dc764a6fc7c65targetblank679dfd5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MicrobialGenomics/advanz4/tree/679dfd549a7abbc250f490fac82dc764a6fc7c65" target="_blank">679dfd5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Metadata/MetadataManagement/Old_Metadata/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/010_Basic_Report_cache/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  analysis/605_Adenovirus_Analysis.Rmd

Unstaged changes:
    Modified:   advanz4.Rproj
    Modified:   analysis/010_Basic_Report.Rmd
    Modified:   analysis/320_Beta_Diversity.Rmd
    Modified:   analysis/500_Gene_function_exploratory.Rmd
    Deleted:    analysis/505_Adenovirus_Analysis.Rmd
    Modified:   analysis/_site.yml
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/010_Basic_Report.Rmd</code>) and
HTML (<code>docs/010_Basic_Report.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c208fe48af079c31bc6b5026a111abfefbaf9434/analysis/010_Basic_Report.Rmd" target="_blank">c208fe4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-16
</td>
<td>
created basic report Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="sequencing-throupghput-analysis-and-gene-richness"
class="section level1">
<h1>Sequencing throupghput analysis and Gene Richness</h1>
<p>The median sequencing sample throughput was 20789049 reads, with a
quasi-normal distribution. Samples were rarefied to the 98th percentile,
which accounted to a threshold of 11888709 (Figure1). 6 samples were
below this threshold and hence were removed.</p>
<pre><code>suggested package selbal vsuggested package DataExplorer vsuggested package lme4 vsuggested package merTools v</code></pre>
<pre class="r"><code>phy &lt;- get_phyloseq(mymre, type=&quot;metaphlan&quot;)</code></pre>
<pre class="r"><code>cat_df &lt;- cat_df %&gt;%
  filter(!CategoricalVariable == &quot;CD4diff_96&quot;)
cat_var &lt;- &quot;group&quot;



link_var &lt;- &quot;record_id&quot;
id_var &lt;- &quot;SampleID&quot;

myPal&lt;-RColorBrewer::brewer.pal(name =  cat_df$PaletteName[1],
                                n=length(unique(metadata[,cat_var])))
BiPlotCoordsDF&lt;-BiPlotCoordsDF %&gt;%
  mutate(Dim1s = Dim1*3,
         Dim2s = Dim2*3)


myNMDS_DF &lt;- myNMDS %&gt;%
  rownames_to_column(var = id_var) %&gt;%
  dplyr::full_join(., metadata, by =id_var) %&gt;%
  dplyr::select(SampleID = id_var,
                NMDS1 = MDS1,
                NMDS2 = MDS2,
                linkVar = !!sym(link_var),
                catVar = !!sym(cat_var),
                LongVar = long_var)


  myXmin&lt;-min(myNMDS$MDS1)
  # +(max(dim_NMDS$points[,1])-min(dim_NMDS$points[,1]))/20
  myYmin&lt;-max(myNMDS$MDS2)-(max(myNMDS$MDS1)-min(myNMDS$MDS1))




myAdonis&lt;- adonis(myBrayDistMat~ catVar, data = myNMDS_DF) %&gt;%
  pluck(&quot;aov.tab&quot;) %&gt;%
  as.data.frame() %&gt;%
  slice(1L) %&gt;%
  magrittr::set_rownames(cat_var)


myNMDSplot&lt;-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point() +
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour=&quot;black&quot;),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face=&quot;italic&quot;),
          legend.title=element_text(size=11))+

    ggtitle(expression(atop(&quot;Species composition&quot;,atop(italic(&quot;all samples&quot;))))) +
    theme(plot.title=element_text(lineheight=1,face=&quot;bold&quot;,size=13))+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var)




myNMDSBiplot&lt;-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point(aes(colour=catVar, shape = as.character(LongVar))) +
    scale_colour_manual(values = myPal, labels = c(&quot;DGT&quot;, &quot;DRV/r&quot;)) +
    scale_fill_manual(values=myPal, labels = c(&quot;DGT&quot;, &quot;DRV/r&quot;))+
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour=&quot;black&quot;),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face=&quot;italic&quot;),
          legend.title=element_text(size=11))+

    ggtitle(expression(atop(&quot;Species composition&quot;,atop(italic(&quot;all samples&quot;))))) +
    theme(plot.title=element_text(lineheight=1,face=&quot;bold&quot;,size=13))+
    stat_ellipse(geom=&quot;polygon&quot;,
                 alpha=0.15,color=&quot;black&quot;,
                 aes(fill=catVar),level=0.95)+
    # scale_size_manual(values = c(1:length(unique(myNMDS_DF[,&quot;LongVar&quot;])))/1.5) +
    geom_segment(data=BiPlotCoordsDF,aes(x=0,xend=Dim1s,y=0,yend=Dim2s),
                 arrow = arrow(length = unit(0.2,&quot;cm&quot;)),colour=&quot;darkgrey&quot;)+
    geom_text_repel(data=BiPlotCoordsDF,aes(Dim1s,Dim2s,label=species),
                    size=2.5, box.padding = unit(0.25, &quot;lines&quot;),
                    segment.colour=&quot;black&quot;,
                    segment.alpha=0.5,
                    segment.size=0.10)+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var) +
  annotation_custom(gridExtra::tableGrob(round(myAdonis,3),
                                         theme = ttheme_default(base_size = 8)),ymin = 3.5, xmin=2.4)</code></pre>
<pre class="r"><code>phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;) %&gt;% 
      phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


otu_df &lt;- 
  phyloseq::otu_table(phy) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy)[,&quot;Species&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;))


stat_list &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    
    c(0,24,48,96) %&gt;% 
      set_names() %&gt;% 
      purrr::map_dfr(function(tp){
        otu_df %&gt;%
        dplyr::select(-SampleID) %&gt;%
        colnames() %&gt;%
        purrr::set_names() %&gt;%
        purrr::map_dfr(function(sp) {
          df &lt;-
            otu_df %&gt;%
            dplyr::select(SampleID, !!sym(sp)) %&gt;%
            dplyr::left_join(metadata[, c(&quot;SampleID&quot;, link_var, cv, long_var), drop = F], 
                             by = &quot;SampleID&quot;) %&gt;% 
            dplyr::filter(!!sym(long_var) %in% tp) 

          
          if (length(unique(dplyr::pull(.data = df, !!sym(cv)))) == 2) {
            test &lt;- df %&gt;%
              rstatix::wilcox_test(., formula(paste(sp, &quot;~&quot;, cv)), paired = F) %&gt;%
              dplyr::mutate(method = &quot;Wilcoxon&quot;) 
            
          } else{
            test &lt;- df %&gt;%
              rstatix::kruskal_test(., formula(paste(sp, &quot;~&quot;, cv))) %&gt;%
              dplyr::mutate(method = &quot;Kruskal-Wallis&quot;)
          }
          
        }) %&gt;% 
      rename(species = .y.) %&gt;% 
      dplyr::mutate(long_var = tp) %&gt;% 
      group_by(long_var) %&gt;% 
      mutate(p.adj = p.adjust(p, &quot;BH&quot;)) %&gt;% 
      # filter(p.adj &lt; .05 ) %&gt;% 
      group_by(species) 
      }) 
}) 


DAsp &lt;- stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species)


## --- init MNJ
## It is interesting that Bifidobacterium species (adolescentis, longum) are consistently found as DA in all (most) timepoints
## Produce a table showing 
sp_detected &lt;- 
stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species) %&gt;% 
  tally() %&gt;% 
  filter(n&gt;2) %&gt;%
  arrange(desc(n))## Species being significant (Wilcoxon unadjusted &lt;0.05)

sp_detected</code></pre>
<pre><code># A tibble: 3 x 2
  species                          n
  &lt;chr&gt;                        &lt;int&gt;
1 Bifidobacterium_adolescentis     3
2 Bifidobacterium_longum           3
3 Olsenella_scatoligenes           3</code></pre>
<pre class="r"><code>### Show how many species are differential in at least two of the timepoints in each cat var. Risk_group should result in many species showing that kins of signal. See Venn section below.
### --- end MNJ</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Bifidobacterium adolescentis, Bifidobacterium longum and Olsenella
scatoligenes were significant in all timepoints. Let’s check the actual
abundances per group and timepoint:</p>
<p>Error in unique.default(x, nmax = nmax) : unique() applies only to
vectors</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.4        testthat_3.0.1     ggrepel_0.9.1      gridExtra_2.3     
 [5] vegan_2.5-7        lattice_0.20-41    permute_0.9-7      lmerTest_3.1-3    
 [9] lme4_1.1-27.1      Matrix_1.2-18      rstatix_0.7.0      ggpubr_0.4.0      
[13] RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0      dplyr_1.0.8       
[17] purrr_0.3.4        readr_2.0.1        tidyr_1.2.0        tibble_3.1.6      
[21] ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             tidyselect_1.1.2       htmlwidgets_1.5.4     
  [4] grid_4.0.2             devtools_2.3.2         aws.signature_0.6.0   
  [7] munsell_0.5.0          codetools_0.2-16       withr_2.5.0           
 [10] colorspace_2.0-2       Biobase_2.50.0         phyloseq_1.34.0       
 [13] logger_0.2.2           knitr_1.38             rstudioapi_0.13       
 [16] stats4_4.0.2           ggsignif_0.6.3         labeling_0.4.2        
 [19] emmeans_1.7.1-1        git2r_0.28.0           polyclip_1.10-0       
 [22] bit64_4.0.5            farver_2.1.0           datawizard_0.6.2      
 [25] rhdf5_2.34.0           rprojroot_2.0.3        coda_0.19-4           
 [28] vctrs_0.3.8            generics_0.1.2         xfun_0.30             
 [31] R6_2.5.1               arm_1.12-2             rhdf5filters_1.2.1    
 [34] cachem_1.0.1           assertthat_0.2.1       promises_1.2.0.1      
 [37] networkD3_0.4          scales_1.2.0           vroom_1.5.7           
 [40] merTools_0.5.2         gtable_0.3.0           processx_3.5.2        
 [43] workflowr_1.7.0        rlang_1.0.2            zeallot_0.1.0         
 [46] splines_4.0.2          broom_0.8.0            yaml_2.3.5            
 [49] reshape2_1.4.4         abind_1.4-5            modelr_0.1.8          
 [52] backports_1.4.1        httpuv_1.6.3           tools_4.0.2           
 [55] usethis_2.0.0          logging_0.10-108       ellipsis_0.3.2        
 [58] jquerylib_0.1.4        biomformat_1.18.0      selbal_0.1.0          
 [61] BiocGenerics_0.36.1    sessioninfo_1.1.1      Rcpp_1.0.7            
 [64] plyr_1.8.6             base64enc_0.1-3        zlibbioc_1.36.0       
 [67] ps_1.5.0               prettyunits_1.1.1      correlation_0.8.2     
 [70] S4Vectors_0.28.1       haven_2.3.1            cluster_2.1.0         
 [73] fs_1.5.0               here_1.0.1             DataExplorer_0.8.2    
 [76] magrittr_2.0.2         data.table_1.14.2      reprex_2.0.1          
 [79] mvtnorm_1.1-2          whisker_0.4            pkgload_1.2.4         
 [82] hms_1.1.1              patchwork_1.1.1        mime_0.11             
 [85] evaluate_0.15          xtable_1.8-4           broom.mixed_0.2.7     
 [88] readxl_1.3.1           IRanges_2.24.1         compiler_4.0.2        
 [91] ggstatsplot_0.9.4      crayon_1.5.1           minqa_1.2.4           
 [94] htmltools_0.5.2        mgcv_1.8-31            later_1.3.0           
 [97] tzdb_0.3.0             lubridate_1.8.0        aws.s3_0.3.21         
[100] DBI_1.1.1              tweenr_1.0.2           dbplyr_2.1.1          
[103] MASS_7.3-51.6          boot_1.3-25            ade4_1.7-18           
[106] car_3.0-12             wesanderson_0.3.6      cli_3.0.1             
[109] parallel_4.0.2         insight_0.18.4         igraph_1.2.9          
[112] pkgconfig_2.0.3        statsExpressions_1.3.3 numDeriv_2016.8-1.1   
[115] paletteer_1.4.0        xml2_1.3.2             foreach_1.5.2         
[118] bslib_0.3.1            blme_1.0-5             multtest_2.46.0       
[121] XVector_0.30.0         estimability_1.3       rvest_1.0.1           
[124] callr_3.7.0            digest_0.6.29          parameters_0.18.2     
[127] Biostrings_2.58.0      rmarkdown_2.11         cellranger_1.1.0      
[130] curl_4.3               shiny_1.7.1            nloptr_1.2.2.3        
[133] lifecycle_1.0.1        nlme_3.1-148           jsonlite_1.7.2        
[136] Rhdf5lib_1.12.1        carData_3.0-5          desc_1.4.1            
[139] fansi_0.5.0            pillar_1.7.0           fastmap_1.1.0         
[142] httr_1.4.2             pkgbuild_1.2.0         survival_3.1-12       
[145] glue_1.5.1             remotes_2.4.0          bayestestR_0.12.1     
[148] iterators_1.0.14       bit_4.0.4              performance_0.10.0    
[151] ggforce_0.3.3          stringi_1.7.6          sass_0.4.0            
[154] rematch2_2.1.2         memoise_2.0.0          ape_5.5               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
