<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>MetaExplorer Report Generation</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">MetaExplorer Report Generation</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">3/23/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<<<<<<< Updated upstream
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treec51687a773a5cb2f278af76950c55a802ed234cdtargetblankc51687aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/c51687a773a5cb2f278af76950c55a802ed234cd" target="_blank">c51687a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treec51687a773a5cb2f278af76950c55a802ed234cdtargetblankc51687aa" class="panel-collapse collapse">
=======
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree68ef5b36b734e7ad7444690b6aee98e6029cbfc9targetblank68ef5b3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/68ef5b36b734e7ad7444690b6aee98e6029cbfc9" target="_blank">68ef5b3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree68ef5b36b734e7ad7444690b6aee98e6029cbfc9targetblank68ef5b3a" class="panel-collapse collapse">
>>>>>>> Stashed changes
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
<<<<<<< Updated upstream
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/c51687a773a5cb2f278af76950c55a802ed234cd" target="_blank">c51687a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
=======
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/68ef5b36b734e7ad7444690b6aee98e6029cbfc9" target="_blank">68ef5b3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
>>>>>>> Stashed changes
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  analysis/304_Taxonomy_DA.Rmd

Unstaged changes:
<<<<<<< Updated upstream
    Deleted:    Metadata/.~lock.2022_03_25_clean_metadata_LIMS.csv#
    Deleted:    Metadata/MetadataManagement/.~lock.Informacio_LIMS.xlsx#
    Deleted:    Metadata/MetadataManagement/.~lock.LabData.xls#
    Modified:   analysis/_site.yml
=======
    Modified:   .DS_Store
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/Heatmaps/Species/AbundanceHeatmap_50.pdf
>>>>>>> Stashed changes

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/020_MetaExplorer_Report.Rmd</code>) and HTML (<code>docs/020_MetaExplorer_Report.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<<<<<<< Updated upstream
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/020_MetaExplorer_Report.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/020_MetaExplorer_Report.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
=======
>>>>>>> Stashed changes
<a href="https://github.com/MicrobialGenomics/advanz4/blob/440936112631b3ae0f4f3911ce3a45b9ae62d1bb/analysis/020_MetaExplorer_Report.Rmd" target="_blank">4409361</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-25
</td>
<td>
initi mre
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/440936112631b3ae0f4f3911ce3a45b9ae62d1bb/docs/020_MetaExplorer_Report.html" target="_blank">4409361</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-25
</td>
<td>
initi mre
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3cccd896f26607b09427747a7fc680f507d4f783/analysis/020_MetaExplorer_Report.Rmd" target="_blank">3cccd89</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-23
</td>
<td>
initial mre
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="metar-explorer-report" class="section level2">
<h2>MetaR Explorer Report</h2>
<p>This Rmd is intended to read MRE object (no writing) and produce local report for exploratory and analysis plan meeting</p>
<pre class="r"><code>require(phyloseq)</code></pre>
<pre><code>Loading required package: phyloseq</code></pre>
<pre class="r"><code>require(ggstatsplot)</code></pre>
<pre><code>Loading required package: ggstatsplot</code></pre>
<pre><code>Warning: package &#39;ggstatsplot&#39; was built under R version 4.1.2</code></pre>
<pre><code>You can cite this package as:
     Patil, I. (2021). Visualizations with statistical details: The &#39;ggstatsplot&#39; approach.
     Journal of Open Source Software, 6(61), 3167, doi:10.21105/joss.03167</code></pre>
<pre class="r"><code>require(ggplot2)</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre class="r"><code>require(here)</code></pre>
<pre><code>Loading required package: here</code></pre>
<pre><code>here() starts at /Users/mnoguera/Documents/Work/Development/advanz4</code></pre>
<pre class="r"><code>bucket&lt;-&quot;s3://mistral-wp6-advanz4/&quot;
bucketDir&lt;-&quot;metagenome/WMGS&quot;

# Load package from local copy, needs to handle some of the dependencies manually.
devtools::load_all(&quot;~/Documents/Work/Development/WMGSPipeline/&quot;)### Change to require/library/load local package</code></pre>
<pre><code>ℹ Loading metar</code></pre>
<pre><code>Warning: package &#39;testthat&#39; was built under R version 4.1.2</code></pre>
<pre><code>── Attaching ──────────────────────────────────────────────── metar v-0.0.0.9 ──</code></pre>
<pre><code>suggested package selbal ✓</code></pre>
<pre><code></code></pre>
<pre><code>suggested package DataExplorer ✓</code></pre>
<pre><code></code></pre>
<pre><code>suggested package lme4 ✓</code></pre>
<pre><code></code></pre>
<pre><code>suggested package merTools ✓</code></pre>
<pre><code></code></pre>
<pre class="r"><code>devtools::load_all(&quot;~/Documents/Work/Development/metaExplorer/&quot;)### Change to require/library/load local package</code></pre>
<pre><code>ℹ Loading metaExplorer</code></pre>
<pre><code>Registered S3 method overwritten by &#39;ggside&#39;:
  method from   
  +.gg   ggplot2</code></pre>
<pre class="r"><code># myMREObject&lt;-readRDS(&quot;data/processed/mreObject.rds&quot;)
mre&lt;-aws.s3::s3readRDS(bucket=bucket,object = paste0(bucketDir,&quot;/MREObject.rds&quot;))

### Execute on the fly until memory space for rarefaction issue is solved
mre&lt;-mre %&gt;% metar::igc_lmm() %&gt;% igc_rarefaction()</code></pre>
<pre><code>Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 8 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).
Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning in max(ids, na.rm = TRUE): no non-missing arguments to max; returning
-Inf

Warning in max(ids, na.rm = TRUE): no non-missing arguments to max; returning
-Inf</code></pre>
<pre><code>Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 27 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 27 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 27 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 27 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 27 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning: Removed 4 row(s) containing missing values (geom_path).</code></pre>
<pre><code>Warning in max(ids, na.rm = TRUE): no non-missing arguments to max; returning
-Inf</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre class="r"><code>metaExplorer::export_report(
mre,
diversity_slot = &quot;igc&quot;,
taxa_slot = &quot;metaphlan&quot;,
metadata_vars = get_metadata_vars(mre),
report_title = &quot;ADVANZ4 Exploratory Report&quot;,
output_file = paste0(here::here(),&quot;/docs/mistral_advanz4_report.pdf&quot;)
)</code></pre>
<pre><code>There was an error in &#39;add_p()/add_difference()&#39; for variable &#39;center&#39;, p-value omitted:
Error in stats::fisher.test(structure(c(2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, : FEXACT error 7(location). LDSTP=18600 is too small for this problem,
  (pastp=53.819, ipn_0:=ipoin[itp=60]=419, stp[ipn_0]=56.5565).
Increase workspace or consider using &#39;simulate.p.value=TRUE&#39;
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
<<<<<<< Updated upstream
[1] devtools_2.4.2  usethis_2.1.3   workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] xfun_0.30         bslib_0.3.1       remotes_2.4.1     purrr_0.3.4      
 [5] vctrs_0.3.8       testthat_3.1.2    htmltools_0.5.2   yaml_2.3.5       
 [9] utf8_1.2.2        rlang_1.0.2       pkgbuild_1.2.0    later_1.3.0      
[13] pillar_1.7.0      jquerylib_0.1.4   glue_1.6.2        withr_2.5.0      
[17] sessioninfo_1.2.1 lifecycle_1.0.1   stringr_1.4.0     evaluate_0.15    
[21] memoise_2.0.0     knitr_1.37        callr_3.7.0       fastmap_1.1.0    
[25] httpuv_1.6.3      ps_1.6.0          fansi_1.0.2       Rcpp_1.0.8       
[29] promises_1.2.0.1  cachem_1.0.6      desc_1.4.1        pkgload_1.2.4    
[33] jsonlite_1.8.0    fs_1.5.0          brio_1.1.3        digest_0.6.29    
[37] stringi_1.7.6     processx_3.5.2    getPass_0.2-2     rprojroot_2.0.2  
[41] here_1.0.1        cli_3.2.0         tools_4.0.5       sass_0.4.0       
[45] magrittr_2.0.2    tibble_3.1.6      crayon_1.5.0      whisker_0.4      
[49] pkgconfig_2.0.3   ellipsis_0.3.2    prettyunits_1.1.1 rmarkdown_2.11   
[53] httr_1.4.2        rstudioapi_0.13   R6_2.5.1          git2r_0.29.0     
[57] compiler_4.0.5   </code></pre>
=======
[1] patchwork_1.1.1         metaExplorer_0.0.0.9000 metar_0.1.5            
[4] testthat_3.1.2          here_1.0.1              ggplot2_3.3.5          
[7] ggstatsplot_0.9.1       phyloseq_1.38.0        

loaded via a namespace (and not attached):
  [1] coda_0.19-4                 tidyr_1.2.0                
  [3] gtsummary_1.5.2             bit64_4.0.5                
  [5] knitr_1.37                  multcomp_1.4-18            
  [7] DelayedArray_0.18.0         rpart_4.1.16               
  [9] wesanderson_0.3.6           data.table_1.14.2          
 [11] DataExplorer_0.8.2          RCurl_1.98-1.6             
 [13] doParallel_1.0.17           generics_0.1.2             
 [15] BiocGenerics_0.40.0         TH.data_1.1-0              
 [17] callr_3.7.0                 usethis_2.1.5              
 [19] mixOmics_6.16.3             commonmark_1.8.0           
 [21] correlation_0.8.0           bit_4.0.4                  
 [23] tzdb_0.2.0                  xml2_1.3.3                 
 [25] httpuv_1.6.5                SummarizedExperiment_1.22.0
 [27] assertthat_0.2.1            WRS2_1.1-3                 
 [29] xfun_0.30                   jquerylib_0.1.4            
 [31] evaluate_0.15               promises_1.2.0.1           
 [33] fansi_1.0.2                 igraph_1.2.11              
 [35] DBI_1.1.2                   htmlwidgets_1.5.4          
 [37] reshape_0.8.8               kSamples_1.2-9             
 [39] rARPACK_0.11-0              stats4_4.1.1               
 [41] Rmpfr_0.8-7                 paletteer_1.4.0            
 [43] purrr_0.3.4                 ellipsis_0.3.2             
 [45] RSpectra_0.16-0             dplyr_1.0.8                
 [47] ggpubr_0.4.0                backports_1.4.1            
 [49] insight_0.16.0              permute_0.9-7              
 [51] prismatic_1.1.0             libcoin_1.0-9              
 [53] MatrixGenerics_1.4.3        vctrs_0.3.8                
 [55] Biobase_2.54.0              remotes_2.4.2              
 [57] Cairo_1.5-15                abind_1.4-5                
 [59] cachem_1.0.6                withr_2.5.0                
 [61] aws.signature_0.6.0         ggforce_0.3.3              
 [63] checkmate_2.0.0             vroom_1.5.7                
 [65] vegan_2.5-7                 prettyunits_1.1.1          
 [67] cluster_2.1.2               ape_5.6-2                  
 [69] lazyeval_0.2.2              crayon_1.5.0               
 [71] ellipse_0.4.2               SuppDists_1.1-9.7          
 [73] labeling_0.4.2              pkgconfig_2.0.3            
 [75] tweenr_1.0.2                GenomeInfoDb_1.30.1        
 [77] nlme_3.1-155                statsExpressions_1.3.0     
 [79] pkgload_1.2.4               nnet_7.3-17                
 [81] blme_1.0-5                  devtools_2.4.3             
 [83] rlang_1.0.2                 lifecycle_1.0.1            
 [85] sandwich_3.0-1              selbal_0.1.0               
 [87] rprojroot_2.0.2             polyclip_1.10-0            
 [89] matrixStats_0.61.0          datawizard_0.3.0           
 [91] Matrix_1.4-0                mc2d_0.1-21                
 [93] carData_3.0-5               zoo_1.8-9                  
 [95] Rhdf5lib_1.16.0             boot_1.3-28                
 [97] base64enc_0.1-3             whisker_0.4                
 [99] GlobalOptions_0.1.2         processx_3.5.2             
[101] png_0.1-7                   viridisLite_0.4.0          
[103] rjson_0.2.21                PMCMRplus_1.9.3            
[105] parameters_0.17.0           bitops_1.0-7               
[107] shinydashboard_0.7.2        pROC_1.18.0                
[109] rhdf5filters_1.6.0          Biostrings_2.62.0          
[111] workflowr_1.7.0             shape_1.4.6                
[113] multcompView_0.1-8          stringr_1.4.0              
[115] coin_1.4-2                  arm_1.12-2                 
[117] lefser_1.2.0                jpeg_0.1-9                 
[119] rstatix_0.7.0               S4Vectors_0.32.3           
[121] ggsignif_0.6.3              aws.s3_0.3.21              
[123] scales_1.1.1                memoise_2.0.1              
[125] magrittr_2.0.2              plyr_1.8.6                 
[127] zlibbioc_1.40.0             compiler_4.1.1             
[129] tinytex_0.37                RColorBrewer_1.1-2         
[131] clue_0.3-60                 lme4_1.1-28                
[133] cli_3.2.0                   ade4_1.7-18                
[135] XVector_0.34.0              ps_1.6.0                   
[137] htmlTable_2.4.0             Formula_1.2-4              
[139] ggside_0.2.0                MASS_7.3-55                
[141] mgcv_1.8-39                 tidyselect_1.1.2           
[143] stringi_1.7.6               forcats_0.5.1              
[145] yaml_2.3.5                  latticeExtra_0.6-29        
[147] ggrepel_0.9.1               grid_4.1.1                 
[149] sass_0.4.0                  tools_4.1.1                
[151] parallel_4.1.1              circlize_0.4.14            
[153] rstudioapi_0.13             foreign_0.8-82             
[155] foreach_1.5.2               logging_0.10-108           
[157] git2r_0.30.1                gridExtra_2.3              
[159] farver_2.1.0                digest_0.6.29              
[161] shiny_1.7.1                 BWStest_0.2.2              
[163] networkD3_0.4               Rcpp_1.0.8.3               
[165] GenomicRanges_1.44.0        car_3.0-12                 
[167] broom_0.7.12                performance_0.8.0          
[169] later_1.3.0                 shinyWidgets_0.6.4         
[171] httr_1.4.2                  ComplexHeatmap_2.8.0       
[173] effectsize_0.6.0.1          colorspace_2.0-3           
[175] brio_1.1.3                  fs_1.5.2                   
[177] IRanges_2.28.0              splines_4.1.1              
[179] rematch2_2.1.2              multtest_2.50.0            
[181] shinythemes_1.2.0           plotly_4.10.0              
[183] sessioninfo_1.2.2           gmp_0.6-4                  
[185] xtable_1.8-4                jsonlite_1.8.0             
[187] nloptr_2.0.0                corpcor_1.6.10             
[189] zeallot_0.1.0               modeltools_0.2-23          
[191] gt_0.4.0                    R6_2.5.1                   
[193] Hmisc_4.6-0                 broom.mixed_0.2.7          
[195] pillar_1.7.0                htmltools_0.5.2            
[197] mime_0.12                   BiocParallel_1.28.3        
[199] glue_1.6.2                  fastmap_1.1.0              
[201] minqa_1.2.4                 DT_0.21                    
[203] codetools_0.2-18            pkgbuild_1.3.1             
[205] mvtnorm_1.1-3               utf8_1.2.2                 
[207] lattice_0.20-45             bslib_0.3.1                
[209] tibble_3.1.6                logger_0.2.2               
[211] curl_4.3.2                  attempt_0.3.1              
[213] config_0.3.1                golem_0.3.2                
[215] survival_3.3-1              roxygen2_7.1.2             
[217] rmarkdown_2.13              desc_1.4.1                 
[219] biomformat_1.22.0           merTools_0.5.2             
[221] munsell_0.5.0               GetoptLong_1.0.5           
[223] rhdf5_2.38.0                GenomeInfoDbData_1.2.7     
[225] iterators_1.0.14            broom.helpers_1.6.0        
[227] reshape2_1.4.4              gtable_0.3.0               
[229] bayestestR_0.11.5          </code></pre>
>>>>>>> Stashed changes
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
