<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-03-24" />

<title>Diversity_GeneRichness</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="331_GR_BMI.html">BMI vs diversity</a>
</li>
<li>
  <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
</li>
<li>
  <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
</li>
<li>
  <a href="317_GeneRichness_vs_Inflammation.html">Gene richness vs inflammation</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Diversity_GeneRichness</h1>
<h4 class="date">2023-03-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree50067574928e35bb52808278f7a4544a5a129d9atargetblank5006757a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MicrobialGenomics/advanz4/tree/50067574928e35bb52808278f7a4544a5a129d9a" target="_blank">5006757</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree50067574928e35bb52808278f7a4544a5a129d9atargetblank5006757a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MicrobialGenomics/advanz4/tree/50067574928e35bb52808278f7a4544a5a129d9a" target="_blank">5006757</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/010_Basic_Report_cache/
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    analysis/mre/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  output/figures/

Unstaged changes:
    Modified:   analysis/250_BL_stratification.Rmd
    Modified:   analysis/310_IGC_analysis.Rmd
    Modified:   output/general_boxplot_stats
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf
    Modified:   output/overall_pvals.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/310_IGC_analysis.Rmd</code>) and
HTML (<code>docs/310_IGC_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/d9a89632291ef86685f259291ca1f55b74b07ff0/analysis/310_IGC_analysis.Rmd" target="_blank">d9a8963</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-22
</td>
<td>
Solved local conflicts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/d9a89632291ef86685f259291ca1f55b74b07ff0/docs/310_IGC_analysis.html" target="_blank">d9a8963</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-22
</td>
<td>
Solved local conflicts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9e2fd42198f450579592f1e076394255d450567/analysis/310_IGC_analysis.Rmd" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
<td>
Fixed rendering bugs on 310.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/b9e2fd42198f450579592f1e076394255d450567/docs/310_IGC_analysis.html" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
<td>
Fixed rendering bugs on 310.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/7cbd79ee787541d35b45bc739934a2f1628923c4/analysis/310_IGC_analysis.Rmd" target="_blank">7cbd79e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-21
</td>
<td>
Added stats table to IGC doc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/7cbd79ee787541d35b45bc739934a2f1628923c4/docs/310_IGC_analysis.html" target="_blank">7cbd79e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-21
</td>
<td>
Added stats table to IGC doc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ec0749dc797b4eb1e7cb4cf5e8f4b05bf699c921/analysis/310_IGC_analysis.Rmd" target="_blank">ec0749d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added BMI tests for week 24
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/ec0749dc797b4eb1e7cb4cf5e8f4b05bf699c921/docs/310_IGC_analysis.html" target="_blank">ec0749d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added BMI tests for week 24
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/40fda18774f1450d6a209987f551e8653f7ca40a/docs/310_IGC_analysis.html" target="_blank">40fda18</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
<td>
Readded delta BMI vs GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/160c5b629ee12b8ed055587786942974f97fd8e7/analysis/310_IGC_analysis.Rmd" target="_blank">160c5b6</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-02-19
</td>
<td>
updated data input
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c1a9770c3ca2f9505837e8532b1cbd108d865a0c/docs/310_IGC_analysis.html" target="_blank">c1a9770</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
<td>
updated website and CMV
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/024b1077514245ecd47e073de16b3f022d37ef4a/analysis/310_IGC_analysis.Rmd" target="_blank">024b107</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-12-01
</td>
<td>
Started CMV and updated metadata
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a0f19e26ed38110e625ec87f05921c88274a473c/analysis/310_IGC_analysis.Rmd" target="_blank">a0f19e2</a>
</td>
<td>
cblazquez
</td>
<td>
2022-11-25
</td>
<td>
moved files to code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/a0f19e26ed38110e625ec87f05921c88274a473c/docs/310_IGC_analysis.html" target="_blank">a0f19e2</a>
</td>
<td>
cblazquez
</td>
<td>
2022-11-25
</td>
<td>
moved files to code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/074d657fa84a274ad41976f978b04c48c7415fa2/docs/310_IGC_analysis.html" target="_blank">074d657</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-25
</td>
<td>
Added comparisons in GR and CD4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/24d99edede71a5f7f1a8ddcafdca5370db6681ba/analysis/310_IGC_analysis.Rmd" target="_blank">24d99ed</a>
</td>
<td>
cblazquez
</td>
<td>
2022-11-11
</td>
<td>
Added first draft of metagenome vs clinical parameters
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/analysis/310_IGC_analysis.Rmd" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
<td>
Rendered new workflowr site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/docs/310_IGC_analysis.html" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
<td>
Rendered new workflowr site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e660a2852bdbb9838f9fa3423447ab5659a5aff5/analysis/310_IGC_analysis.Rmd" target="_blank">e660a28</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-10-27
</td>
<td>
updates on various scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/19b15b838b9f3f9a6e8e23c1913d8d8adae38763/analysis/310_IGC_analysis.Rmd" target="_blank">19b15b8</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-10-14
</td>
<td>
Added Functional DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/19b15b838b9f3f9a6e8e23c1913d8d8adae38763/docs/310_IGC_analysis.html" target="_blank">19b15b8</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-10-14
</td>
<td>
Added Functional DA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b5caf01f692d7775acc0d194e6941319bce519cd/analysis/310_IGC_analysis.Rmd" target="_blank">b5caf01</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-28
</td>
<td>
added sCD14 and sCD163 wk 96 to metadata, and older metadata versions to
secondary folder
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/b5caf01f692d7775acc0d194e6941319bce519cd/docs/310_IGC_analysis.html" target="_blank">b5caf01</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-28
</td>
<td>
added sCD14 and sCD163 wk 96 to metadata, and older metadata versions to
secondary folder
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c208fe48af079c31bc6b5026a111abfefbaf9434/analysis/310_IGC_analysis.Rmd" target="_blank">c208fe4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-16
</td>
<td>
created basic report Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c208fe48af079c31bc6b5026a111abfefbaf9434/docs/310_IGC_analysis.html" target="_blank">c208fe4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-16
</td>
<td>
created basic report Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/272d281b6b6d84b83e8c8094b672ad9979725f49/analysis/310_IGC_analysis.Rmd" target="_blank">272d281</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-13
</td>
<td>
added stats to IGC boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/272d281b6b6d84b83e8c8094b672ad9979725f49/docs/310_IGC_analysis.html" target="_blank">272d281</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-13
</td>
<td>
added stats to IGC boxplots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/972f1c0f5d871c6bc6035b744caa04776f18d3e4/analysis/310_IGC_analysis.Rmd" target="_blank">972f1c0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-09-12
</td>
<td>
Added gene richness boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/972f1c0f5d871c6bc6035b744caa04776f18d3e4/docs/310_IGC_analysis.html" target="_blank">972f1c0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-09-12
</td>
<td>
Added gene richness boxplots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0aa7922e8a20e42a68c244562eec53d699c4efdd/analysis/310_IGC_analysis.Rmd" target="_blank">0aa7922</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-07
</td>
<td>
Improved S.parasanguinis interaction plots with inflammation markers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/0aa7922e8a20e42a68c244562eec53d699c4efdd/docs/310_IGC_analysis.html" target="_blank">0aa7922</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-07
</td>
<td>
Improved S.parasanguinis interaction plots with inflammation markers
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/analysis/310_IGC_analysis.Rmd" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
<td>
Added species v inflammation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/310_IGC_analysis.html" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
<td>
Added species v inflammation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/9713823b00b303b0cb85954d9a9ca081ad4a2c8d/docs/310_IGC_analysis.html" target="_blank">9713823</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-27
</td>
<td>
Added longitudinal DA with ANCOM in 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/28f4f36a86ce137e63a5bdce76ff1deb56282797/analysis/310_IGC_analysis.Rmd" target="_blank">28f4f36</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-23
</td>
<td>
started with ADV+Entero analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/2a1fe845b3637c85682f64ab5aea8d2164e1192f/analysis/310_IGC_analysis.Rmd" target="_blank">2a1fe84</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Fixed rmd integration to wflow_build()
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/2a1fe845b3637c85682f64ab5aea8d2164e1192f/docs/310_IGC_analysis.html" target="_blank">2a1fe84</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Fixed rmd integration to wflow_build()
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f213c10f0f5907550448cf2733d862a4929fa1cd/analysis/310_IGC_analysis.Rmd" target="_blank">f213c10</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
reworked basal analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/analysis/310_IGC_analysis.Rmd" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/docs/310_IGC_analysis.html" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/7d67e12d5f11c669abfa93c04ddc09ad23be5713/analysis/310_IGC_analysis.Rmd" target="_blank">7d67e12</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-03-29
</td>
<td>
Fixed commented bugs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/7d67e12d5f11c669abfa93c04ddc09ad23be5713/docs/310_IGC_analysis.html" target="_blank">7d67e12</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-03-29
</td>
<td>
Fixed commented bugs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/eba3a180ec68006819926efda2b5ca478bd6f808/analysis/310_IGC_analysis.Rmd" target="_blank">eba3a18</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-29
</td>
<td>
comments for Carlos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e253850975c479f491140a3a60d9e54a3eca44ff/docs/310_IGC_analysis.html" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/310_IGC_analysis.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/310_IGC_analysis.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/11c1e7d2e4c2b76ccb9e08a96305fa7f34ee48de/analysis/310_IGC_analysis.Rmd" target="_blank">11c1e7d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
Code reorganized towards new structure
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="sequencing-throupghput-analysis" class="section level1">
<h1>Sequencing throupghput analysis</h1>
<p>First we’ll check how well the sequencing went by looking at the read
count distribution among all samples and how many of them will be
dropped once we rarefy the samples</p>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all(&quot;../WMGSPipeline/&quot;)


mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as_tibble(metadata)

mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/get_GeneRichness_Data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>only 6 samples have been discarded by low read count. The threshold
used for filtering was the 2nd percentile, which equals 11888709
reads.</p>
<div id="rarefaction-and-sequencing-depth" class="section level2">
<h2>Rarefaction and sequencing depth</h2>
<p>We’ll control first for the effect of sequencing depth over Gene
richness and try to stablish a proper rarefaction threshold.</p>
<p><img src="figure/310_IGC_analysis.Rmd/Plot_Rarefaction-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="rarefaction-and-sequencing-depth-1" class="section level2">
<h2>Rarefaction and sequencing depth</h2>
<p>We’ll control first for the effect of sequencing depth over Gene
richness and try to stablish a proper rarefaction threshold.</p>
<p>Data shows there is no bias created by sequencing depth between
groups, and a threshold over the 2nd percentile is appropiate.</p>
</div>
<div id="longitudinal-evolution-lmms" class="section level2">
<h2>Longitudinal evolution: LMMs</h2>
<p>We’ll start studying the longitudinal effect of treatment group over
gene richness. For such purpose, we’ll use Linear mixed models using
patients as random effect and an interaction of treatment group and time
as fixed effects with the following lme4 formula:</p>
<p><code>richness ~ group*time_point + (1|ID)</code></p>
<p>We found trouble fiting the LMMs for ethnic groups due to patient
A024 being the only one with the category “unknown”. As the model can’t
estimate variance from only one observation, the code breaks.</p>
<p><img src="figure/310_IGC_analysis.Rmd/plot_LMMs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There is a pretty clear evolution in every one of the treatment
groups. However the evolution rate seems pretty similar between both
treatment groups. The models fail to find a great difference in gene
richness evolution, althougha slight yet significant increasing trend
(slope value = 0.027), can be found in the DGT group (Group 1).</p>
<p>However, creating a single LMM with treatment group as a random
effect we can measure the actual difference in the effect of timepoint
over gene richness between two groups.</p>
<pre><code> [1] &quot;group&quot;        &quot;risk_group&quot;   &quot;center&quot;       &quot;gender&quot;       &quot;ethnic_group&quot;
 [6] &quot;CD4diff_48&quot;   &quot;CD8diff_48&quot;   &quot;CD4after_48&quot;  &quot;CD8after_48&quot;  &quot;CD4diff_96&quot;  
[11] &quot;CD8diff_96&quot;   &quot;CD4after_96&quot;  &quot;CD8after_96&quot; </code></pre>
<pre><code>[1] &quot;group&quot;
[1] &quot;risk_group&quot;
[1] &quot;center&quot;
[1] &quot;gender&quot;
[1] &quot;ethnic_group&quot;
[1] &quot;CD4diff_48&quot;
[1] &quot;CD8diff_48&quot;
[1] &quot;CD4after_48&quot;
[1] &quot;CD8after_48&quot;
[1] &quot;CD4diff_96&quot;
[1] &quot;CD8diff_96&quot;
[1] &quot;CD4after_96&quot;
[1] &quot;CD8after_96&quot;</code></pre>
<p>This shows a slight effect of treatment, where the effect of time
over richness in group group2 is 0.025 (St.err = 0.025) times lower than
the same for group 1 (ANOVA p=0.31)</p>
<pre><code>$group</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$risk_group</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$center</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$gender</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$ethnic_group</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4diff_48</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8diff_48</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4after_48</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8after_48</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4diff_96</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8diff_96</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4after_96</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8after_96</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/Show_plot_list-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dat &lt;- TableList %&gt;% pluck(&quot;group&quot;)
source(&quot;code/Paired_wilcox_NA_Handler.R&quot;)

long_test &lt;- PairedWilcox_withNAs(myDF = dat,Variable = &quot;richness&quot;,Response = &quot;long_var&quot;,IDs = &quot;link_var&quot;) %&gt;% 
  rstatix::adjust_pvalue(p.col = &quot;p&quot;) %&gt;% 
  rstatix::add_significance(p.col = &quot;p.adj&quot;) %&gt;% 
  dplyr::mutate(y.position = y.position + 250000)

all_boxplot &lt;- dat %&gt;% 
  ggplot(., aes(x = as.factor(long_var),y=richness)) + 
  geom_boxplot(fill=&quot;lightgray&quot;) +
  theme_bw() +
  ggpubr::stat_pvalue_manual(data = long_test, label = &quot;p.adj.signif&quot;, hide.ns = T, bracket.size = 1.1, label.size = 12) +
  labs(x = &quot;Week&quot;) +
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 14))

all_boxplot</code></pre>
<pre><code>Warning: Removed 20 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/general_boxplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.csv(long_test, file = &quot;output/general_boxplot_stats&quot;)</code></pre>
</div>
<div id="longitudinal-boxplots" class="section level2">
<h2>Longitudinal Boxplots</h2>
<p>There is a general increase even without stratifying for any
confounding variable.</p>
<pre class="r"><code>source(&quot;code/group_comparisons.R&quot;)
mockdat &lt;- dat %&gt;% 
  mutate(mock = &quot;mock&quot;) 

combn(unique(dat$long_var),2, simplify = F) %&gt;% 
  map_dfr(function(comp){
    comp &lt;- sort(comp)
    mockdat %&gt;% 
    get_group_comparisons(dat = .,
                      link_var = &quot;link_var&quot;,
                      long_var = &quot;long_var&quot;,
                      cat_vector = &quot;mock&quot;,
                      num_vector = &quot;richness&quot;,
                      comps = comp,
                      type = &quot;longitudinal&quot;
                      ) %&gt;% 
  purrr::pluck(&quot;mock&quot;,&quot;richness&quot;,&quot;stats&quot;) %&gt;% 
      dplyr::mutate(group1 = comp[1], 
                    group2 = comp[2], .after=1)

  }) %&gt;% 
      dplyr::arrange(group1,group2) %&gt;% 
  dplyr::mutate(group1 = as.character(group1), 
                group2 = as.character(group2)) %&gt;% 
  dplyr::full_join(long_test, by=c(&quot;group1&quot;,&quot;group2&quot;)) %&gt;% 
  dplyr::select(cat_var, contains(&quot;group&quot;), median1, iqr2, median2, iqr2, medianChange, iqr_change, n1, n2, statistic, p.adj) %&gt;% 


write.csv(.,&quot;output/overall_pvals.csv&quot;,quote = T,row.names = F)  </code></pre>
<pre><code>Warning: Removed 20 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/group_boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 20 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 13 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite values (stat_bracket).</code></pre>
<p><img src="figure/310_IGC_analysis.Rmd/group_boxplots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>This is even more intereseting, as most of the significant tchange
happens in the DTG group, with significant differences in all time
points, and significantly higher richnes in DTG vs DRV/r is found in al
timepoints except 0.</p>
<pre class="r"><code>igc_df &lt;- TableList$group


obj &lt;-
  igc_df %&gt;%
  dplyr::filter(!is.na(richness), !is.na(cat_var)) %&gt;%
  get_group_comparisons(
    .,
    link_var = &quot;link_var&quot;,
    long_var = &quot;long_var&quot;,
    cat_vector = &quot;cat_var&quot;,
    num_vector = &quot;richness&quot;,
    type = &quot;categorical&quot;,
    comps = unique(igc_df$cat_var)
  ) 

objtest &lt;- obj %&gt;% pluck(&quot;cat_var&quot;, &quot;richness&quot;, &quot;test&quot;) %&gt;% 
  rstatix::adjust_pvalue(p.col = &quot;p&quot;, method = &quot;BH&quot;)
objstats &lt;- obj %&gt;% pluck(&quot;cat_var&quot;, &quot;richness&quot;, &quot;stats&quot;) %&gt;%
  mutate(stat = paste(round(median, 2), &quot; [&quot;, round(iqr, 2), &quot;]&quot;, sep = &quot;&quot;))

cat_table &lt;- 
objtest %&gt;%
  pivot_longer(cols = c(&quot;group1&quot;, &quot;group2&quot;)) %&gt;%
  dplyr::rename(cat_var = value) %&gt;%
  dplyr::left_join(objstats[, c(&quot;cat_var&quot;, &quot;long_var&quot;, &quot;stat&quot;)], by = c(&quot;cat_var&quot;, &quot;long_var&quot;)) %&gt;%
  pivot_wider(names_from = &quot;name&quot;,
              values_from = c(&quot;cat_var&quot;, &quot;stat&quot;)) %&gt;%
  dplyr::select(long_var,
                group1 = cat_var_group1,
                group2 = cat_var_group2,
                contains(&quot;stat&quot;),
                p.adj) %&gt;% 
  dplyr::filter(p.adj &lt; 0.05)







obj &lt;-
  igc_df %&gt;%
  dplyr::filter(!is.na(richness), !is.na(cat_var)) %&gt;%
  get_group_comparisons(
    .,
    link_var = &quot;link_var&quot;,
    long_var = &quot;long_var&quot;,
    cat_vector = &quot;cat_var&quot;,
    num_vector = &quot;richness&quot;,
    type = &quot;longitudinal&quot;,
    comps = c(0, 96)
  )
long_table &lt;- 
obj %&gt;%
  pluck(&quot;cat_var&quot;, &quot;richness&quot;, &quot;test&quot;) %&gt;%
  rstatix::adjust_pvalue(method = &quot;BH&quot;, p.col = &quot;p&quot;) %&gt;%
  full_join(obj$cat_var$richness$stats, by = &quot;cat_var&quot;) %&gt;%
  # dplyr::mutate(.y. = nv) %&gt;%
  dplyr::select(cat_var,
                group1,
                group2,
                num_var = .y.,
                contains(&quot;median&quot;),
                contains(&quot;iqr&quot;),
                &quot;p.adj&quot;)

cat_table</code></pre>
<pre><code># A tibble: 0 x 7
# ... with 7 variables: long_var &lt;int&gt;, group1 &lt;chr&gt;, group2 &lt;chr&gt;,
#   statistic &lt;dbl&gt;, stat_group1 &lt;chr&gt;, stat_group2 &lt;chr&gt;, p.adj &lt;dbl&gt;</code></pre>
<pre class="r"><code>long_table</code></pre>
<pre><code># A tibble: 2 x 11
  cat_var group1 group2 num_var medianCh~1 median1 median2   iqr1   iqr2 iqr_c~2
  &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
1 bDRV    0      96     num_var     58874   755107  790849 3.44e5 3.50e5 182579.
2 DTG     0      96     num_var    154320.  786304  965992 2.30e5 2.21e5 204709.
# ... with 1 more variable: p.adj &lt;dbl&gt;, and abbreviated variable names
#   1: medianChange, 2: iqr_change</code></pre>
<pre class="r"><code>source(&quot;code/group_comparisons.R&quot;)


cat_vector %&gt;% 
  set_names() %&gt;% 
  map(function(cv){
    combn(unique(dat$long_var),2, simplify = F) %&gt;% 
  map_dfr(function(comp){
    comp &lt;- sort(comp)
    
    list_res &lt;- boxplots %&gt;%
      pluck(cv, &quot;data&quot;) %&gt;% 
    get_group_comparisons(dat = .,
                      link_var = &quot;link_var&quot;,
                      long_var = &quot;long_var&quot;,
                      cat_vector = &quot;cat_var&quot;,
                      num_vector = &quot;richness&quot;,
                      comps = comp,
                      type = &quot;longitudinal&quot;
                      )  

  stat_df &lt;- 
    list_res %&gt;% 
      pluck(&quot;cat_var&quot;,&quot;richness&quot;,&quot;stats&quot;) %&gt;% 
      dplyr::mutate(group1 = comp[1], 
                    group2 = comp[2], .after=1) %&gt;% 
      dplyr::mutate(group1 = as.character(group1), 
                group2 = as.character(group2))

  test_df &lt;- 
    list_res %&gt;% 
    pluck(&quot;cat_var&quot;,&quot;richness&quot;,&quot;test&quot;) 
   
 
   res_df &lt;- 
     dplyr::full_join(stat_df, test_df, by=c(&quot;cat_var&quot;,&quot;group1&quot;,&quot;group2&quot;)) %&gt;% 
  
  dplyr::arrange(cat_var,group1,group2) %&gt;% 
  dplyr::select(cat_var, contains(&quot;group&quot;), median1, iqr2, median2, iqr2, medianChange, iqr_change, n1, n2, statistic, p) 


  }) 
  })</code></pre>
<pre><code>$group
# A tibble: 12 x 12
   cat_var group1 group2 median1    iqr2 median2 medianCha~1 iqr_c~2    n1    n2
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt; &lt;int&gt; &lt;int&gt;
 1 bDRV    0      24     755107  374478. 654897       25499  222805.    15    15
 2 DTG     0      24     834666. 312567. 973488.      98052  176412.    21    23
 3 bDRV    24     48     751418. 467287. 671618       -6817  190205.    16    14
 4 DTG     24     48     909257  239801. 964298       51746  179725     27    27
 5 bDRV    24     96     755848  375121  815363       33540  161676     15    17
 6 DTG     24     96     886425  247723. 958014.      47761  288152     23    23
 7 bDRV    0      48     751746  354480. 799986       47364. 199075     29    27
 8 DTG     0      48     788174  279461  935456      112426  184921.    26    29
 9 bDRV    0      96     755107  350377. 790849       58874  182579.    26    26
10 DTG     0      96     786304  221308. 965992      154320. 204709.    26    29
11 bDRV    48     96     773533  217306. 800115       27639  176071     24    27
12 DTG     48     96     917476  234882  947144       21814  215577.    30    30
# ... with 2 more variables: statistic &lt;dbl&gt;, p &lt;dbl&gt;, and abbreviated variable
#   names 1: medianChange, 2: iqr_change

$CD4diff_96
# A tibble: 12 x 12
   cat_var group1 group2 median1    iqr2 median2 medianCha~1 iqr_c~2    n1    n2
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt; &lt;int&gt; &lt;int&gt;
 1 150_H   0      24     755107  363016  755848       70786  313404.    18    21
 2 150_L   0      24     978377  583364. 857580       65675  697720.     5     5
 3 150_H   24     48     744607  343154. 938572       75225  190301     21    20
 4 150_L   24     48     893415  203482. 984452.     131587  254005.     5     4
 5 150_H   24     96     745798  301453. 922104      109176. 206820.    21    21
 6 150_L   24     96     857580  490355  939568       33213  113397.     5     5
 7 150_H   0      48     748944  328650. 818263       73708  213384     29    32
 8 150_L   0      48     788174  147899. 935456      164482   79790      8     7
 9 150_H   0      96     755107  275205. 839854      126401  152558     34    37
10 150_L   0      96     790040. 437621. 871840.      16462. 178483.     8     8
11 150_H   48     96     850663  248154. 875026       62544  234746     32    32
12 150_L   48     96     935456  273455  939568      -93881  188273      7     8
# ... with 2 more variables: statistic &lt;dbl&gt;, p &lt;dbl&gt;, and abbreviated variable
#   names 1: medianChange, 2: iqr_change</code></pre>
<pre class="r"><code>  # write.csv(.,&quot;output/group_pvals_long.csv&quot;,quote = T,row.names = F)  




cat_vector %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(cv){
    dat &lt;-  boxplots %&gt;% 
      purrr::pluck(cv, &quot;data&quot;)
      
    comp &lt;- combn(unique(dat$cat_var),2, simplify = F)[[1]]
          
    get_group_comparisons(dat = dat,
                      link_var = &quot;link_var&quot;,
                      long_var = &quot;long_var&quot;,
                      cat_vector = &quot;cat_var&quot;,
                      num_vector = &quot;richness&quot;,
                      comps = comp,
                      type = &quot;categorical&quot;
                      ) %&gt;% 
  pluck(&quot;cat_var&quot;, &quot;richness&quot;)  
  })</code></pre>
<pre><code>$group
$group$stats
# A tibble: 8 x 6
# Groups:   cat_var [2]
  cat_var long_var median upper_ci lower_ci     iqr
  &lt;chr&gt;      &lt;int&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 bDRV           0 748385  948385   614476. 333909.
2 bDRV          24 700943  910314.  551909. 358405.
3 bDRV          48 773533  982808.  620399. 362409 
4 bDRV          96 800115 1014203.  663826  350377.
5 DTG            0 776668  923667.  641983. 281685.
6 DTG           24 938264 1011618.  739281  272338.
7 DTG           48 938572 1044632.  804957. 239675.
8 DTG           96 931461 1040301.  803331. 236970.

$group$test
# A tibble: 4 x 13
  long_~1 estim~2 .y.   group1 group2    n1    n2 stati~3      p conf.~4 conf.~5
*   &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;   &lt;dbl&gt;  &lt;dbl&gt;   &lt;int&gt;   &lt;int&gt;
1       0 -5.32e3 num_~ bDRV   DTG       35    35     599 0.879  -117800  105248
2      24 -1.54e5 num_~ bDRV   DTG       18    28     156 0.0306 -322571  -18931
3      48 -1.46e5 num_~ bDRV   DTG       29    35     339 0.0227 -253601  -19425
4      96 -9.38e4 num_~ bDRV   DTG       29    34     349 0.0473 -217298   -2110
# ... with 2 more variables: method &lt;chr&gt;, alternative &lt;chr&gt;, and abbreviated
#   variable names 1: long_var, 2: estimate, 3: statistic, 4: conf.low,
#   5: conf.high


$CD4diff_96
$CD4diff_96$stats
# A tibble: 8 x 6
# Groups:   cat_var [2]
  cat_var long_var  median upper_ci lower_ci     iqr
  &lt;chr&gt;      &lt;int&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;
1 150_H          0 752026.  926533.  656602. 269930.
2 150_H         24 746989   984616.  690515  294101.
3 150_H         48 850663  1020655.  701302. 319353 
4 150_H         96 839854  1049911   780933. 268978.
5 150_L          0 790040.  990543   763405. 227138.
6 150_L         24 857580   989167.  405803  583364.
7 150_L         48 935456  1022202.  874304. 147899.
8 150_L         96 871840. 1036385   598764. 437621.

$CD4diff_96$test
# A tibble: 4 x 13
  long_var estim~1 .y.   group1 group2    n1    n2 stati~2     p conf.~3 conf.~4
*    &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;int&gt;   &lt;int&gt;
1        0 -74646. num_~ 150_H  150_L     34     8      97 0.222 -250289   74566
2       24  61627  num_~ 150_H  150_L     21     5      59 0.705 -249202  460339
3       48 -92680. num_~ 150_H  150_L     32     7      90 0.44  -259424   88455
4       96  29065  num_~ 150_H  150_L     37     8     161 0.716 -144813  279976
# ... with 2 more variables: method &lt;chr&gt;, alternative &lt;chr&gt;, and abbreviated
#   variable names 1: estimate, 2: statistic, 3: conf.low, 4: conf.high</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nlme_3.1-148       lmerTest_3.1-3     lme4_1.1-29        Matrix_1.5-3      
 [5] rstatix_0.7.1      ggpubr_0.4.0       RColorBrewer_1.1-3 forcats_0.5.2     
 [9] stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4        readr_2.0.1       
[13] tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   
[17] metar_0.1.5        aws.s3_0.3.21     

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.4.1        blme_1.0-5            
  [4] workflowr_1.7.0        systemfonts_1.0.4      plyr_1.8.8            
  [7] igraph_1.2.9           splines_4.0.2          listenv_0.9.0         
 [10] digest_0.6.29          foreach_1.5.2          htmltools_0.5.2       
 [13] fansi_1.0.3            magrittr_2.0.3         paletteer_1.4.0       
 [16] cluster_2.1.0          tzdb_0.3.0             aws.signature_0.6.0   
 [19] globals_0.16.2         Biostrings_2.58.0      modelr_0.1.8          
 [22] svglite_2.1.0          colorspace_2.0-3       rvest_1.0.1           
 [25] textshaping_0.3.6      haven_2.3.1            xfun_0.30             
 [28] crayon_1.5.2           jsonlite_1.8.0         zeallot_0.1.0         
 [31] survival_3.1-12        iterators_1.0.14       ape_5.6-2             
 [34] glue_1.6.2             kableExtra_1.3.4       gtable_0.3.1          
 [37] emmeans_1.7.1-1        zlibbioc_1.36.0        XVector_0.30.0        
 [40] webshot_0.5.2          statsExpressions_1.3.3 phyloseq_1.34.0       
 [43] car_3.1-1              Rhdf5lib_1.12.1        BiocGenerics_0.36.1   
 [46] abind_1.4-5            scales_1.2.1           mvtnorm_1.1-3         
 [49] DBI_1.1.3              Rcpp_1.0.8.3           viridisLite_0.4.1     
 [52] performance_0.10.0     xtable_1.8-4           merTools_0.5.2        
 [55] stats4_4.0.2           datawizard_0.6.5       htmlwidgets_1.5.4     
 [58] httr_1.4.4             ellipsis_0.3.2         farver_2.1.1          
 [61] DataExplorer_0.8.2     pkgconfig_2.0.3        dbplyr_2.1.1          
 [64] here_1.0.1             utf8_1.2.2             labeling_0.4.2        
 [67] tidyselect_1.1.2       rlang_1.0.2            reshape2_1.4.4        
 [70] later_1.3.0            munsell_0.5.0          cellranger_1.1.0      
 [73] tools_4.0.2            cli_3.2.0              generics_0.1.3        
 [76] selbal_0.1.0           ade4_1.7-20            broom_1.0.2           
 [79] evaluate_0.19          biomformat_1.18.0      fastmap_1.1.0         
 [82] ragg_1.2.4             arm_1.12-2             yaml_2.3.6            
 [85] rematch2_2.1.2         knitr_1.38             fs_1.5.2              
 [88] future_1.30.0          whisker_0.4.1          mime_0.12             
 [91] xml2_1.3.2             correlation_0.8.2      compiler_4.0.2        
 [94] rstudioapi_0.13        curl_4.3.3             ggsignif_0.6.4        
 [97] reprex_2.0.1           stringi_1.7.6          highr_0.10            
[100] parameters_0.18.2      logger_0.2.2           lattice_0.20-41       
[103] nloptr_2.0.0           vegan_2.6-4            permute_0.9-7         
[106] multtest_2.46.0        vctrs_0.4.1            pillar_1.8.1          
[109] lifecycle_1.0.1        networkD3_0.4          rhdf5filters_1.2.1    
[112] furrr_0.3.1            jquerylib_0.1.4        estimability_1.4.1    
[115] cowplot_1.1.1          insight_0.18.8         data.table_1.14.6     
[118] patchwork_1.1.2        httpuv_1.6.7           R6_2.5.1              
[121] promises_1.2.0.1       gridExtra_2.3          IRanges_2.24.1        
[124] parallelly_1.33.0      codetools_0.2-16       boot_1.3-25           
[127] MASS_7.3-51.6          assertthat_0.2.1       rhdf5_2.34.0          
[130] rprojroot_2.0.3        withr_2.5.0            S4Vectors_0.28.1      
[133] broom.mixed_0.2.9.4    bayestestR_0.12.1      mgcv_1.8-31           
[136] parallel_4.0.2         hms_1.1.2              grid_4.0.2            
[139] coda_0.19-4            minqa_1.2.5            rmarkdown_2.19        
[142] carData_3.0-5          git2r_0.30.1           numDeriv_2016.8-1.1   
[145] Biobase_2.50.0         shiny_1.7.1            lubridate_1.8.0       
[148] base64enc_0.1-3        ggstatsplot_0.9.4     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
