<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />

<meta name="date" content="2023-03-06" />

<title>BMI_stratification</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Diversity overall
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="310_IGC_analysis.html">Gene richness</a>
    </li>
    <li>
      <a href="311_GR_vs_BMI.html">BMI vs diversity</a>
    </li>
    <li>
      <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
    </li>
    <li>
      <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
    </li>
    <li>
      <a href="317_GeneRichness_vs_Inflammation.html">Gene richness vs inflammation</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">beta diversity</li>
    <li>
      <a href="320_Beta_Diversity.html">Beta Diversity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Baseline Microbiome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
    </li>
    <li>
      <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Differential Abundance
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
    </li>
    <li>
      <a href="420_Taxonomy_DA.html">Differential Abundance</a>
    </li>
    <li>
      <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI_stratification</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">2023-03-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree669d6f725b6d8840cf747b6ac44d62e1510f7fcbtargetblank669d6f7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/669d6f725b6d8840cf747b6ac44d62e1510f7fcb" target="_blank">669d6f7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree669d6f725b6d8840cf747b6ac44d62e1510f7fcbtargetblank669d6f7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/669d6f725b6d8840cf747b6ac44d62e1510f7fcb" target="_blank">669d6f7</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/505_Adenovirus_Analysis_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  Metadata/MetadataManagement/.~lock.LabData.xls#
    Untracked:  output/BMI_vs_GR.pdf

Unstaged changes:
    Modified:   analysis/311_GR_vs_BMI.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/311_GR_vs_BMI.Rmd</code>) and HTML (<code>docs/311_GR_vs_BMI.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/669d6f725b6d8840cf747b6ac44d62e1510f7fcb/analysis/311_GR_vs_BMI.Rmd" target="_blank">669d6f7</a>
</td>
<td>
cblazquez
</td>
<td>
2023-05-05
</td>
<td>
first try at fixing spaghetti plot at BMI vs GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c13ab300674c26cc36a8c6fa7220ca96dff061a9/analysis/311_GR_vs_BMI.Rmd" target="_blank">c13ab30</a>
</td>
<td>
cblazquez
</td>
<td>
2023-05-04
</td>
<td>
changes to alluvial and tests in BMI test
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/209eb6d9b734ce1208bdbfa7fe349b16c46ad71d/analysis/311_GR_vs_BMI.Rmd" target="_blank">209eb6d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-28
</td>
<td>
Added all info for BMI and updated alluvial plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/47cd70d3f44763f08d1f713862fd69d38096ea88/analysis/311_GR_vs_BMI.Rmd" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
<td>
First fixes to BMI analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/47cd70d3f44763f08d1f713862fd69d38096ea88/docs/311_GR_vs_BMI.html" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
<td>
First fixes to BMI analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/4db2a47fef82222929130b3a2a36769b1106e3d3/analysis/311_GR_vs_BMI.Rmd" target="_blank">4db2a47</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-19
</td>
<td>
added spaghetti plot to BMI rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/4db2a47fef82222929130b3a2a36769b1106e3d3/docs/311_GR_vs_BMI.html" target="_blank">4db2a47</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-19
</td>
<td>
added spaghetti plot to BMI rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/72f15339f90d852482e264aa42757ad925bf6e1d/docs/311_GR_vs_BMI.html" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
<td>
Updated website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/0aef81987ff281deec8894018695691895f80a3b/docs/311_GR_vs_BMI.html" target="_blank">0aef819</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
<td>
fixed bug on GR_inflammation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/efdac31cf4332143b561f3f956761d9b8895d748/analysis/311_GR_vs_BMI.Rmd" target="_blank">efdac31</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
<td>
Reestructured website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/efdac31cf4332143b561f3f956761d9b8895d748/docs/311_GR_vs_BMI.html" target="_blank">efdac31</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
<td>
Reestructured website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/62208ccf057dfcb2d3e5ed5cf3e206eb85823d91/analysis/311_GR_vs_BMI.Rmd" target="_blank">62208cc</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-22
</td>
<td>
text changes on BMI vs GR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/62208ccf057dfcb2d3e5ed5cf3e206eb85823d91/docs/311_GR_vs_BMI.html" target="_blank">62208cc</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-22
</td>
<td>
text changes on BMI vs GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ec0749dc797b4eb1e7cb4cf5e8f4b05bf699c921/analysis/311_GR_vs_BMI.Rmd" target="_blank">ec0749d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added BMI tests for week 24
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/ec0749dc797b4eb1e7cb4cf5e8f4b05bf699c921/docs/311_GR_vs_BMI.html" target="_blank">ec0749d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added BMI tests for week 24
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1b0ddcbaff018d4f448ac865958b47049611e559/analysis/311_GR_vs_BMI.Rmd" target="_blank">1b0ddcb</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added richness vs stratification by BMI_group
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/1b0ddcbaff018d4f448ac865958b47049611e559/docs/311_GR_vs_BMI.html" target="_blank">1b0ddcb</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
<td>
Added richness vs stratification by BMI_group
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/13934937790cdaa1d7797b79257775a36849ee2a/analysis/311_GR_vs_BMI.Rmd" target="_blank">1393493</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
<td>
started stratifcation by BMI
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/13934937790cdaa1d7797b79257775a36849ee2a/docs/311_GR_vs_BMI.html" target="_blank">1393493</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
<td>
started stratifcation by BMI
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/40fda18774f1450d6a209987f551e8653f7ca40a/analysis/311_GR_vs_BMI.Rmd" target="_blank">40fda18</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
<td>
Readded delta BMI vs GR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/40fda18774f1450d6a209987f551e8653f7ca40a/docs/311_GR_vs_BMI.html" target="_blank">40fda18</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
<td>
Readded delta BMI vs GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/cbd814a4074642b4d8adb494acd1565ec8a60635/analysis/311_GR_vs_BMI.Rmd" target="_blank">cbd814a</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-09
</td>
<td>
added distribution per group
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/cbd814a4074642b4d8adb494acd1565ec8a60635/docs/311_GR_vs_BMI.html" target="_blank">cbd814a</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-09
</td>
<td>
added distribution per group
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ec3949c8abd001aa4c07563476791b0c859c652a/analysis/311_GR_vs_BMI.Rmd" target="_blank">ec3949c</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-08
</td>
<td>
updated BMI_vs_GR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/ec3949c8abd001aa4c07563476791b0c859c652a/docs/311_GR_vs_BMI.html" target="_blank">ec3949c</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-08
</td>
<td>
updated BMI_vs_GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1710d26f13b98dd1b41d35034e5f1c325cf0830/analysis/311_GR_vs_BMI.Rmd" target="_blank">e1710d2</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-07
</td>
<td>
started analysis n GR vs BMI
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<div id="stratification-by-bmi" class="section level1">
<h1>Stratification by BMI</h1>
<p>This set of tests comes from recent studies which found that INSTIs are associated with greater BMI increases, respective to other treatments in ART-receiving patients. BMI is highly related to gut health, metabolism and microbiome, and may be a possible confussor regarding the interaction between treatment and microbiote, especially gene richness, as it may be masking the actual effect of treatment. The first is to assess how big of an impact our treatments have on the patients BMIs.</p>
<p>As we are looking strictly at BMI now, we’ll use the entries for ALL participants included in the study, regardless of wether there are faecal samples in a certain timepoint or not.</p>
<pre><code>[1] &quot;group&quot;
[1] &quot;BMI&quot;</code></pre>
<p><img src="figure/311_GR_vs_BMI.Rmd/BMI_vs_group-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/311_GR_vs_BMI.Rmd/BMI_vs_group-2.png" width="672" style="display: block; margin: auto;" /></p>
<table>
<thead>
<tr class="header">
<th align="left">long_var</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="left">groups</th>
<th align="right">p</th>
<th align="left">p.adj.signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0</td>
<td align="left">bDRV</td>
<td align="left">DTG</td>
<td align="left">bDRV, DTG</td>
<td align="right">0.4270</td>
<td align="left">ns</td>
</tr>
<tr class="even">
<td align="left">24</td>
<td align="left">bDRV</td>
<td align="left">DTG</td>
<td align="left">bDRV, DTG</td>
<td align="right">0.9720</td>
<td align="left">ns</td>
</tr>
<tr class="odd">
<td align="left">48</td>
<td align="left">bDRV</td>
<td align="left">DTG</td>
<td align="left">bDRV, DTG</td>
<td align="right">0.4280</td>
<td align="left">ns</td>
</tr>
<tr class="even">
<td align="left">96</td>
<td align="left">bDRV</td>
<td align="left">DTG</td>
<td align="left">bDRV, DTG</td>
<td align="right">0.0329</td>
<td align="left">ns</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr class="header">
<th align="left">cat_var</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">p</th>
<th align="left">p.adj.signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">bDRV</td>
<td align="left">0</td>
<td align="left">24</td>
<td align="right">4.70e-06</td>
<td align="left">****</td>
</tr>
<tr class="even">
<td align="left">bDRV</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">0.00e+00</td>
<td align="left">****</td>
</tr>
<tr class="odd">
<td align="left">bDRV</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">1.71e-04</td>
<td align="left">***</td>
</tr>
<tr class="even">
<td align="left">bDRV</td>
<td align="left">24</td>
<td align="left">48</td>
<td align="right">1.31e-02</td>
<td align="left">*</td>
</tr>
<tr class="odd">
<td align="left">bDRV</td>
<td align="left">24</td>
<td align="left">96</td>
<td align="right">3.28e-01</td>
<td align="left">ns</td>
</tr>
<tr class="even">
<td align="left">bDRV</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">4.37e-01</td>
<td align="left">ns</td>
</tr>
<tr class="odd">
<td align="left">DTG</td>
<td align="left">0</td>
<td align="left">24</td>
<td align="right">4.60e-06</td>
<td align="left">****</td>
</tr>
<tr class="even">
<td align="left">DTG</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">0.00e+00</td>
<td align="left">****</td>
</tr>
<tr class="odd">
<td align="left">DTG</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">2.20e-06</td>
<td align="left">****</td>
</tr>
<tr class="even">
<td align="left">DTG</td>
<td align="left">24</td>
<td align="left">48</td>
<td align="right">6.40e-03</td>
<td align="left">**</td>
</tr>
<tr class="odd">
<td align="left">DTG</td>
<td align="left">24</td>
<td align="left">96</td>
<td align="right">1.15e-03</td>
<td align="left">**</td>
</tr>
<tr class="even">
<td align="left">DTG</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">3.86e-03</td>
<td align="left">**</td>
</tr>
</tbody>
</table>
<p>Turns out looking at BMI alone, no categorical differences can be found. While both treatment groups significantly increase in BMI, no diferences between DTG an DRV are found.</p>
<p>Anyhow, this may have something to do with BMI differences at BL. Let’s check for its distribution:</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-basal_dist_df-1">
Past versions of basal_dist_df-1.png
</button>
</p>
<div id="fig-basal_dist_df-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/47cd70d3f44763f08d1f713862fd69d38096ea88/docs/figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/efdac31cf4332143b561f3f956761d9b8895d748/docs/figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" target="_blank">efdac31</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/13934937790cdaa1d7797b79257775a36849ee2a/docs/figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" target="_blank">1393493</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ec3949c8abd001aa4c07563476791b0c859c652a/docs/figure/311_GR_vs_BMI.Rmd/basal_dist_df-1.png" target="_blank">ec3949c</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It seems there is a certain bimodality in the distribution of BMI at BL in the DTG, in which a peak of patients emerges around BMI of 28. Anyway, both groups have their peaks farily close to the overall median, and their distributions, especially in the bDRV group approaches normality. The overall median is 22.064, which is well within the healthy weight.</p>
<ul>
<li>18.5 (Underweight)</li>
<li>18.5 to 25 (healthy weight).</li>
<li>25 (overweight)</li>
</ul>
<p>The total number of patients in each group at basal is as follows</p>
<pre><code>    
    \begin{longtable}{lccc}\caption{Summary descriptives table by groups of `group&#39;}\\
    \hline  
     &amp;    bDRV    &amp;    DTG     &amp; \multirow{2}{*}{p.overall}\\ 
 &amp;    N=42    &amp;    N=44    &amp;           \\ 
  
    \hline
    \hline     
    \endfirsthead 
    \multicolumn{4}{l}{\tablename\ \thetable{} \textit{-- continued from previous page}}\\ 
    \hline
     &amp;    bDRV    &amp;    DTG     &amp; \multirow{2}{*}{p.overall}\\ 
 &amp;    N=42    &amp;    N=44    &amp;           \\ 

    \hline
    \hline  
    \endhead   
    \hline
    \multicolumn{4}{l}{\textit{continued on next page}} \\ 
    \endfoot   
    \multicolumn{4}{l}{}  \\ 
    \endlastfoot 
    group\_bmi: &amp;            &amp;            &amp;   1.000  \\ 
$\qquad$underweight &amp; 5 (11.9\%)  &amp; 5 (11.4\%)  &amp;          \\ 
$\qquad$healthy weight &amp; 29 (69.0\%) &amp; 31 (70.5\%) &amp;          \\ 
$\qquad$overweight &amp; 8 (19.0\%)  &amp; 8 (18.2\%)  &amp;           \\ 
 
    \hline
    \end{longtable} </code></pre>
<p>As we can see, the majority of the cohort is within the range of healthy weight and both groups are pretty well balanced at baseline. This sets a good statistical base from which to find signals after treatment.</p>
</div>
<div id="longitudinal-evolution-of-bmi" class="section level1">
<h1>Longitudinal evolution of BMI</h1>
<p>Before entering in the complexities of gut microbiota, we’ll assess wether DTG/bDRV have a differential effect on the chances of patients changing BMI categories. A good way to visualize this will be an alluvial plot.</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/alluvial_plot-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>There is an increase in overweight incidence in both groups, but it is more noticeable in the DTG group. In the bDRV it appears the BMI increase is more temporary, as many patients who become overweight after treatment initiation, return to a healthy weight after 96 weeks. This is not the case in the DTG group, however, where all the participants who become overweight remain so during the whole study.</p>
<p>The population of underweight seems to improve more in the DTG, as none of the participants fall into this category as soon as week 48, while sme patients remain so along the whole study in the bDRV arm.</p>
</div>
<div id="gr-stratification-by-bmi-at-baseline" class="section level1">
<h1>GR Stratification by BMI at baseline</h1>
<p>Now it is time to assess whether the BMI category may be affecting/be influenced by gene richness. The First thing we should look is whether baseline richness is influencing the bmi of our cohort</p>
<div id="gr-vs-bmi-correlation-at-baseline" class="section level3">
<h3>GR vs BMI correlation at baseline</h3>
<p>Now we found a threshold appropiate to split by BMI, we can stratify the population by treatment and bmi, vs gene richness.</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/BMI_vs_GR_BL-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>No correlation could be found between BMI and GR at baseline in either group nor overall. However, the microbiota may still influence later BMI changes, or even just BMI at later times. Now we’ll correlate GRBL to BMI at different timepoints:</p>
<pre><code># A tibble: 12 × 4
   group   time_point     cor     p
   &lt;chr&gt;        &lt;int&gt;   &lt;dbl&gt; &lt;dbl&gt;
 1 bDRV             0 -0.059  0.739
 2 bDRV            24  0.29   0.257
 3 bDRV            48 -0.16   0.406
 4 bDRV            96  0.36   0.108
 5 DTG              0  0.049  0.784
 6 DTG             24  0.098  0.635
 7 DTG             48  0.1    0.564
 8 DTG             96 -0.27   0.149
 9 overall          0 -0.0077 0.95 
10 overall         24  0.19   0.227
11 overall         48 -0.02   0.879
12 overall         96  0.11   0.437</code></pre>
<p><img src="figure/311_GR_vs_BMI.Rmd/BMI_timepoints-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Also, we can see there is a patient with a REALLY low gene count, although it passed the 2nd percentile filtering. Patient E006 at month 0 received trteatment for a latent syphlis infection, just a week before collecting the sample. This means he received a penicilin injection hammered its microbiota to oblivion, and may be wise to discard it.</p>
<p>But, does baseline richness influence the the distribution of BMI groups at a specific timepoint? This could shed some light as why we didn’t find any correlation, as the effect of baseline richness may only manifest very late or very early and dilute afterwards. the best way to look at this is via boxlots:</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/gr0_BMI_cat_boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>No differences between treatment group or bmi group could be observed in what regards to their respective GR at baseline in any timepoint. This suggests gene richness may not have an overall big impact on the BMI, regardless of group.</p>
</div>
</div>
<div id="stratification-by-bmi-changes" class="section level1">
<h1>Stratification by BMI changes</h1>
<p>Maybe the increase in gene richness may be associated not so much to absolute BMI but of changes in BMI. In this part we’ll correlate richness to changes in BMI over time.</p>
<div id="baseline-richness-vs-bmi-changes" class="section level2">
<h2>Baseline richness vs BMI changes</h2>
<p>The first hypotesis to test is whether differences in richness may be affecting BMI changes after treatment.</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/delta_cors_BL-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-delta_cors_BL-1">
Past versions of delta_cors_BL-1.png
</button>
</p>
<div id="fig-delta_cors_BL-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/47cd70d3f44763f08d1f713862fd69d38096ea88/docs/figure/311_GR_vs_BMI.Rmd/delta_cors_BL-1.png" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/4db2a47fef82222929130b3a2a36769b1106e3d3/docs/figure/311_GR_vs_BMI.Rmd/delta_cors_BL-1.png" target="_blank">4db2a47</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Again, nothing. No correlation between richness at basal and change of BMI at any timepoint could be found.</p>
</div>
<div id="bmi-vs-gr-change" class="section level2">
<h2>BMI vs GR change</h2>
<p>We already look on the effect of gene richness at baseline, but now the next question would be whether the chages in richness we reported before are linked to BMI changes as well.</p>
<p><img src="figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_distributions_96-1">
Past versions of plot_distributions_96-1.png
</button>
</p>
<div id="fig-plot_distributions_96-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/47cd70d3f44763f08d1f713862fd69d38096ea88/docs/figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/4db2a47fef82222929130b3a2a36769b1106e3d3/docs/figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" target="_blank">4db2a47</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/efdac31cf4332143b561f3f956761d9b8895d748/docs/figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" target="_blank">efdac31</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ec0749dc797b4eb1e7cb4cf5e8f4b05bf699c921/docs/figure/311_GR_vs_BMI.Rmd/plot_distributions_96-1.png" target="_blank">ec0749d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-03-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Both treatment groups appear to have a unimodal distribution regarding their BMI increase at week 96, both in absolute and relative terms. However there seems to be a shift between both, as the “valley” between both peaks in the DTG group seems to be shifted toward the left, relative to the DRV group.</p>
<pre class="r"><code>test_abs_group &lt;-
  delta_bmi_df %&gt;%
  dplyr::filter(!time_point == 0) %&gt;%
  group_by(time_point, group) %&gt;%
  rstatix::cor_test(delta_gr_abs, delta_BMI_abs, method = &quot;spearman&quot;, use = &quot;complete.obs&quot;)

test_abs &lt;-
  delta_bmi_df %&gt;%
  dplyr::filter(!time_point == 0) %&gt;%
  group_by(time_point) %&gt;%
  rstatix::cor_test(delta_gr_abs, delta_BMI_abs, method = &quot;spearman&quot;, use = &quot;complete.obs&quot;) %&gt;% 
  dplyr::mutate(group = &quot;overall&quot;, .before = 1) %&gt;% 
  rbind(test_abs_group) %&gt;% 
  mutate(x = -650000, 
         y = case_when(
           group == &quot;bDRV&quot; ~ 10,
           group == &quot;DTG&quot; ~ 9,
           group == &quot;overall&quot; ~ 8
         ))



deltas_abs &lt;-
  delta_bmi_df %&gt;%
  dplyr::filter(!time_point == 0) %&gt;%
  # unique() %&gt;%
  ggplot(aes(x = delta_gr_abs, y = delta_BMI_abs)) +
  facet_wrap(~ time_point, ) +
  geom_point() +
  geom_smooth(method = &quot;glm&quot;, aes(colour = group, lty = group), fill = NA) +
  geom_smooth(method = &quot;glm&quot;, colour = &quot;black&quot;, fill = NA) +
  geom_text(data = test_abs, 
             aes(label = paste(group, &quot;cor = &quot;,cor, &quot;p =&quot;, p), x = x, y = y, colour = group), 
             inherit.aes = F, vjust=1, hjust = 0.1, size = 3)+
  labs(title = &quot;Absolute&quot;, x = &quot;Gene richness at baseline&quot;, y = &quot;BMI increase from baseline&quot;) +
  theme_light() +
  scale_color_manual(values = c(&quot;#AAAAAA&quot;, &quot;#AAAAAA&quot;, &quot;#000000&quot;), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) +
  scale_fill_manual(values = c(&quot;#AAAAAA&quot;,&quot;#AAAAAA&quot;,&quot;#000000&quot;), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) +
  scale_linetype_manual(values = c(2,3,1), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) 


test_rel_group &lt;-
  delta_bmi_df %&gt;%
  dplyr::filter(!time_point == 0) %&gt;%
  group_by(time_point, group) %&gt;%
  rstatix::cor_test(delta_gr_rel, delta_BMI_rel, method = &quot;spearman&quot;, use = &quot;complete.obs&quot;)

test_rel &lt;-
  delta_bmi_df %&gt;%
  dplyr::filter(!time_point == 0) %&gt;%
  group_by(time_point) %&gt;%
  rstatix::cor_test(delta_gr_rel, delta_BMI_rel, method = &quot;spearman&quot;, use = &quot;complete.obs&quot;) %&gt;% 
  dplyr::mutate(group = &quot;overall&quot;, .before = 1) %&gt;% 
  rbind(test_rel_group) %&gt;% 
  mutate(x = -0.8, 
         y = case_when(
           group == &quot;bDRV&quot; ~ 40,
           group == &quot;DTG&quot; ~ 37,
           group == &quot;overall&quot; ~ 34
         ))


deltas_rel &lt;- 
  delta_bmi_df %&gt;% 
  dplyr::filter(!time_point == 0) %&gt;% 
  ggplot(aes(x = delta_gr_rel, y = delta_BMI_rel, color = group)) +
  facet_wrap(~ time_point, ) +
  geom_point() +
  geom_smooth(method = &quot;glm&quot;, aes(colour = group, lty = group), fill = NA) +
  geom_smooth(method = &quot;glm&quot;, colour = &quot;black&quot;, fill = NA) +
  geom_text(data = test_rel, 
             aes(label = paste(group, &quot;cor = &quot;,cor, &quot;p =&quot;, p), x = x, y = y, colour = group), 
             inherit.aes = F, vjust=1, hjust = 0.1, size = 3)+
  labs(title = &quot;Relative&quot;, x = &quot;Gene richness at baseline&quot;, y = &quot;BMI increase from baseline&quot;) +
  theme_light() +
  scale_color_manual(values = c(&quot;#AAAAAA&quot;, &quot;#AAAAAA&quot;, &quot;#000000&quot;), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) +
  scale_fill_manual(values = c(&quot;#AAAAAA&quot;,&quot;#AAAAAA&quot;,&quot;#000000&quot;), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) +
  scale_linetype_manual(values = c(2,3,1), limits = c(&quot;bDRV&quot;, &quot;DTG&quot;, &quot;overall&quot;)) 



ggpubr::ggarrange(deltas_abs, deltas_rel, common.legend = T, ncol = 1, nrow = 2)</code></pre>
<p><img src="figure/311_GR_vs_BMI.Rmd/lmms_deltas-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-lmms_deltas-1">
Past versions of lmms_deltas-1.png
</button>
</p>
<div id="fig-lmms_deltas-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/47cd70d3f44763f08d1f713862fd69d38096ea88/docs/figure/311_GR_vs_BMI.Rmd/lmms_deltas-1.png" target="_blank">47cd70d</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/4db2a47fef82222929130b3a2a36769b1106e3d3/docs/figure/311_GR_vs_BMI.Rmd/lmms_deltas-1.png" target="_blank">4db2a47</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>No significant differences in trends could be observed either. Interesting to note, in the absolute plot there are 2 apparent outliers. Those took antibiotics at some point of the study and it reflected in a significant loss of gene richness, but in absolute numbers they aren’t really low on richness, it’s just the change.</p>
<p>Now let’s categorize these results:</p>
<pre class="r"><code>deltas_abs &lt;- 
  delta_bmi_df %&gt;% 
  dplyr::filter(!time_point == 0) %&gt;% 
  ggplot(aes(x = group_bmi, y = delta_gr_abs, color = group)) +
  facet_wrap(~ time_point ) +
  geom_boxplot() +
  labs(title = &quot;Absolute&quot;, x = &quot;Gene richness at baseline&quot;, y = &quot;GR increase from baseline&quot;) +
  theme_light() +
  scale_color_manual(values = c(&quot;#AAAAAA&quot;, &quot;#555555&quot;, &quot;#000000&quot;)) +
  scale_fill_manual(values = c(&quot;#AAAAAA&quot;, &quot;#555555&quot;, &quot;#000000&quot;)) +
  stat_compare_means(method = &quot;kruskal.test&quot;, label = &quot;p.signif&quot;, hide.ns = T) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5))



deltas_rel &lt;- 
  delta_bmi_df %&gt;% 
  dplyr::filter(!time_point == 0) %&gt;% 
  ggplot(aes(x = group_bmi, y = delta_gr_rel, color = group)) +
  facet_wrap(~ time_point ) +
  geom_boxplot() +
  labs(title = &quot;Relative&quot;, x = &quot;Gene richness at baseline&quot;, y = &quot;GR increase from baseline&quot;) +
  theme_light() +
  scale_color_manual(values = c(&quot;#AAAAAA&quot;, &quot;#555555&quot;, &quot;#000000&quot;)) +
  scale_fill_manual(values = c(&quot;#AAAAAA&quot;, &quot;#555555&quot;, &quot;#000000&quot;)) +
  stat_compare_means(method = &quot;kruskal.test&quot;, label = &quot;p.signif&quot;, hide.ns = T) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5))




ggpubr::ggarrange(deltas_abs, deltas_rel, common.legend = T, ncol = 1, nrow = 2)</code></pre>
<p><img src="figure/311_GR_vs_BMI.Rmd/BMI_class_boxplots-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>A we can see, no</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] rstatix_0.7.0      metar_0.1.5        ggpubr_0.4.0       RColorBrewer_1.1-3
 [5] ggalluvial_0.12.3  forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9       
 [9] purrr_0.3.4        readr_2.1.2        tidyr_1.2.0        tibble_3.1.7      
[13] ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] uuid_1.1-0          readxl_1.4.0        backports_1.4.1    
  [4] blme_1.0-5          workflowr_1.7.0     systemfonts_1.0.4  
  [7] plyr_1.8.7          igraph_1.3.2        splines_4.0.5      
 [10] listenv_0.8.0       digest_0.6.29       foreach_1.5.2      
 [13] htmltools_0.5.2     lmerTest_3.1-3      fansi_1.0.3        
 [16] Rsolnp_1.16         magrittr_2.0.3      cluster_2.1.3      
 [19] tzdb_0.3.0          globals_0.15.0      Biostrings_2.58.0  
 [22] modelr_0.1.8        officer_0.4.3       svglite_2.1.0      
 [25] colorspace_2.0-3    rvest_1.0.2         haven_2.5.0        
 [28] xfun_0.31           crayon_1.5.1        jsonlite_1.8.0     
 [31] lme4_1.1-29         survival_3.3-1      iterators_1.0.14   
 [34] ape_5.6-2           glue_1.6.2          kableExtra_1.3.4   
 [37] gtable_0.3.0        zlibbioc_1.36.0     XVector_0.30.0     
 [40] webshot_0.5.3       phyloseq_1.34.0     car_3.1-0          
 [43] Rhdf5lib_1.12.1     BiocGenerics_0.36.1 abind_1.4-5        
 [46] scales_1.2.0        mvtnorm_1.1-3       DBI_1.1.3          
 [49] Rcpp_1.0.8.3        viridisLite_0.4.0   xtable_1.8-4       
 [52] merTools_0.5.2      stats4_4.0.5        truncnorm_1.0-8    
 [55] htmlwidgets_1.5.4   httr_1.4.2          ellipsis_0.3.2     
 [58] mice_3.14.0         farver_2.1.0        DataExplorer_0.8.2 
 [61] pkgconfig_2.0.3     nnet_7.3-17         sass_0.4.1         
 [64] dbplyr_2.2.0        utf8_1.2.2          here_1.0.1         
 [67] labeling_0.4.2      tidyselect_1.1.2    rlang_1.0.2        
 [70] reshape2_1.4.4      later_1.3.0         munsell_0.5.0      
 [73] cellranger_1.1.0    tools_4.0.5         cli_3.3.0          
 [76] generics_0.1.2      selbal_0.1.0        ade4_1.7-19        
 [79] broom_0.8.0         evaluate_0.15       biomformat_1.18.0  
 [82] fastmap_1.1.0       arm_1.12-2          yaml_2.3.5         
 [85] knitr_1.39          fs_1.5.2            compareGroups_4.5.1
 [88] zip_2.2.0           future_1.26.1       nlme_3.1-158       
 [91] mime_0.12           whisker_0.4         xml2_1.3.2         
 [94] compiler_4.0.5      rstudioapi_0.13     ggsignif_0.6.3.9000
 [97] reprex_2.0.1        HardyWeinberg_1.7.5 bslib_0.3.1        
[100] stringi_1.7.6       highr_0.9           gdtools_0.2.4      
[103] logger_0.2.2        lattice_0.20-45     Matrix_1.4-1       
[106] nloptr_2.0.3        vegan_2.6-2         permute_0.9-7      
[109] multtest_2.46.0     vctrs_0.4.1         pillar_1.7.0       
[112] lifecycle_1.0.1     networkD3_0.4       rhdf5filters_1.2.1 
[115] furrr_0.3.0         jquerylib_0.1.4     cowplot_1.1.1      
[118] flextable_0.7.2     data.table_1.14.2   httpuv_1.6.5       
[121] R6_2.5.1            promises_1.2.0.1    gridExtra_2.3      
[124] writexl_1.4.0       IRanges_2.24.1      parallelly_1.32.0  
[127] codetools_0.2-18    boot_1.3-28         MASS_7.3-57        
[130] assertthat_0.2.1    chron_2.3-57        rhdf5_2.34.0       
[133] rprojroot_2.0.3     withr_2.5.0         S4Vectors_0.28.1   
[136] broom.mixed_0.2.9.4 mgcv_1.8-40         parallel_4.0.5     
[139] hms_1.1.1           grid_4.0.5          coda_0.19-4        
[142] minqa_1.2.4         rmarkdown_2.14      carData_3.0-5      
[145] git2r_0.30.1        base64enc_0.1-3     numDeriv_2016.8-1.1
[148] Biobase_2.50.0      shiny_1.7.1         lubridate_1.8.0    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
