<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />

<meta name="date" content="2022-11-11" />

<title>Gene Richness vs CD4</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene Richness vs CD4</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">11/11/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef0e92e1a746d3195eda0a946ef9c97b4dd72a2f0targetblankf0e92e1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MicrobialGenomics/advanz4/tree/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0" target="_blank">f0e92e1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef0e92e1a746d3195eda0a946ef9c97b4dd72a2f0targetblankf0e92e1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MicrobialGenomics/advanz4/tree/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0" target="_blank">f0e92e1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Metadata/MetadataManagement/Old_Metadata/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/010_Basic_Report_cache/
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   advanz4.Rproj
    Modified:   analysis/315_GeneRichness_vs_CD4.Rmd
    Deleted:    analysis/figure/315_GeneRichness_vs_CD4.Rmd/basal_CD4-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/315_GeneRichness_vs_CD4.Rmd</code>) and HTML
(<code>docs/315_GeneRichness_vs_CD4.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">f0e92e1</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-16
</td>
<td>
Started with baseline distributions of CD4 and GR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0/docs/315_GeneRichness_vs_CD4.html" target="_blank">f0e92e1</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-16
</td>
<td>
Started with baseline distributions of CD4 and GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/df53555b1f5f30cb294433fd028ea0cf105d565b/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">df53555</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-11
</td>
<td>
Update 315_GeneRichness_vs_CD4.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/86d7c3aa84215f8988c88c31d59e9f23c51ac3a9/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">86d7c3a</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-11
</td>
<td>
Create 315_GeneRichness_vs_CD4.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>metadata &lt;- 
  here::here(&quot;Metadata&quot;) %&gt;% 
  file.info(list.files(., full.names = T)) %&gt;%
  rownames() %&gt;% 
  magrittr::extract(str_detect(.,&quot;clean_metadata&quot;)) %&gt;% 
  sort(decreasing = T) %&gt;% 
  magrittr::extract(1L) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;mistral-wp6-advanz4&quot;,object = &quot;metagenome/WMGS/MREObject.rds&quot;, region = &quot;eu-west-1&quot;)</code></pre>
<div id="gene-richness-vs-cd4" class="section level2">
<h2>Gene Richness vs CD4</h2>
<p>In this analysis we are interested in the relationship between
CD4/CD4 recovery and gene richness. For that, several questions are of
importance.</p>
<div id="cd4-baseline-distribution" class="section level3">
<h3>CD4 baseline distribution</h3>
<p>We already know that CD4 mean(sd) is similar for DTG/DRVr groups, but
we have not checked the distribution. Is it unimodal or bimodal? a
density plot should help.</p>
<pre class="r"><code>basal_df &lt;- 
metadata %&gt;% 
  # dplyr::mutate(delta_cd4 = CD4[time_point == 48] - CD4[time_point == 0])
  dplyr::filter(time_point == 0) %&gt;% 
  dplyr::select(SampleID, group, CD4)  
  
median_df &lt;- basal_df %&gt;% 
  dplyr::group_by(group) %&gt;% 
  summarise(med = median(CD4))
  

peak_finder &lt;-
  function(density, keep = F) {
    ### Let&#39;s define this function heere but we&#39;ll use it later
    res &lt;-
      density %&gt;%
      purrr::keep(names(.) %in% c(&quot;x&quot;, &quot;y&quot;)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::mutate(is_peak = c(F, # 1st value can&#39;t be a local maxima, as there is no prior value to compare
                                diff(y) %&gt;% # For each nth element of the vector, get (n+1) - n
                                  sign() %&gt;%  # convert to -1 (n+1 is lower than n) or +1 (n+1 is higher than n)
                                  diff() == -2, # if n is at a local maxima, both n+1 and n-1 are negative so [(n+1)-n] - [n - (n-1)] is -2
                                F)) # Last element can&#39;t be a local maxima either)
                    
                    if (!keep) {
                      res %&gt;%
                        dplyr::filter(is_peak) %&gt;%
                        dplyr::select(-is_peak)
                    } else{
                      res %&gt;%
                        dplyr::select(-is_peak)
                    }
  }


peak_df &lt;- unique(basal_df$group) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(function(arm){
   basal_df %&gt;% 
      dplyr::filter(group == arm) %&gt;% 
      pull(CD4) %&gt;% 
      density(n=nrow(.)) %&gt;% 
      peak_finder(., keep = F) %&gt;% 
      mutate(group = arm)

  })



  
  
  
plt &lt;- ggplot(basal_df, aes(x=CD4, color=group)) +

    # geom_histogram(aes(fill = group), bins = 1000) +
    geom_density( alpha = .4) +
  theme_bw() +
  scale_color_brewer(palette = &quot;Accent&quot;) +
  scale_fill_brewer(palette = &quot;Accent&quot;) +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) +
  geom_text(data = median_df, aes(label = paste(&quot;med =&quot;,med), x = med, y = c(0.012,0.015)), color = &quot;black&quot;)
  # geom_vline(data = peak_df, aes(xintercept =  x),color = &quot;black&quot;,lty = 3) +)
  labs (x = &quot;CD4 (counts)&quot;)</code></pre>
<pre><code>$x
[1] &quot;CD4 (counts)&quot;

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<pre class="r"><code>  # geom_errorbar(stat = &quot;hline&quot;, xintercept = &quot;median&quot;,
  # width=0.8,aes(xmax=..x..,xmin=..x..))

  
  
plt</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/basal_CD4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>peak_df</code></pre>
<pre><code>          x            y group
1  26.10519 0.0103359201   DTG
2 229.10519 0.0005447832   DTG
3  11.04344 0.0169875257 DRV/r
4  68.04344 0.0063346692 DRV/r</code></pre>
<p>It appears the distribution of CD4 at BL shows two peaks in kernel
density, one at 26 counts/ml and one at 229 count/ml. There is a second
peak in the DRV as well, but it seems the density function smoothed it
out.</p>
</div>
<div id="generichness-at-baseline-distribution" class="section level3">
<h3>GeneRichness at baseline distribution</h3>
<p>We know (I think) that there is no difference in generichness between
arme at baseline, but we do not know the distribution. In Guillen et al,
we found a bimodal distribution in HIV patients. Is that the case of
advanz4?</p>
<pre class="r"><code>igc&lt;-metar::get_diversity(mymre)@igc
Threshold &lt;-  igc@dataTable %&gt;%
  dplyr::select(SampleID, NumberMappedReads) %&gt;%
  unique() %&gt;%
  summarise(Q = quantile(NumberMappedReads,0.02)) %&gt;% 
  as.numeric()

TableList &lt;- &quot;group&quot; %&gt;%
    set_names() %&gt;%
    map( ~{

        
          ThresholdTable &lt;- igc@dataTable %&gt;%
                filter(NumberMappedReads &gt;= Threshold) %&gt;%
                filter(ReadCountReal&gt;=Threshold) %&gt;% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %&gt;%
                summarise(across(everything(), min)) %&gt;%
                ungroup() %&gt;%
          right_join(., metadata, by=&quot;SampleID&quot;) %&gt;%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=record_id,
                 cat_var = .x,
                 long_var = time_point,
                 NumberMappedReads)
  

        
    }) %&gt;% 
  purrr::pluck(&quot;group&quot;) 

basal_df &lt;- 
  TableList %&gt;% 
  dplyr::filter(long_var == 0)


median_df &lt;- 
  basal_df %&gt;% 
  dplyr::rename(group = cat_var) %&gt;% 
  dplyr::group_by(group) %&gt;% 
  filter(!is.na(richness)) %&gt;% 
  summarise(med = median(richness))

  
plt &lt;- 
basal_df %&gt;% 
  ggplot(aes(x = richness, fill = cat_var)) +
  # geom_histogram(aes(y = ..density..),alpha = .4) +
  geom_density(alpha = .4) +
  theme_bw() +
  scale_fill_brewer(palette=&quot;Accent&quot;) +
  scale_color_brewer(palette = &quot;Accent&quot;) +
  labs(x=&quot;gene_count&quot;, fill = &quot;group&quot;) +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) 


peak_df &lt;- 
basal_df %&gt;% 
  pull(cat_var) %&gt;% 
  unique() %&gt;% 
  map_dfr(function(arm){
    basal_df %&gt;% 
      dplyr::filter(cat_var == arm) %&gt;%
      dplyr::filter(!is.na(richness)) %&gt;% 
      dplyr::pull(richness) %&gt;% 
      density() %&gt;%
      peak_finder() %&gt;% 
      dplyr::mutate(group = arm)

  })

plt</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/GR_basal-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>peak_df</code></pre>
<pre><code>         x            y group
1 774765.3 1.863682e-06   DTG
2 731633.3 1.494479e-06 DRV/r</code></pre>
<p>It seems both groups have a similar median richness, but with
somewhat different distributions, with DTG following a slightly bimodal
distribution with a higher median peak than Drv/r. Although the left
“step” in may look like a local maxima, it actually does not decrease on
the right side.</p>
</div>
<div id="gene-richness-vs-cd4-at-baseline" class="section level3">
<h3>Gene Richness vs CD4 at baseline</h3>
<p>Is there any association between gene richness and CD4 at baseline
only, overall, and per-group(DTG and DRVr only)?</p>
<pre class="r"><code>df &lt;- 
basal_df %&gt;% 
  right_join(metadata[,c(&quot;SampleID&quot;,&quot;CD4&quot;)]) %&gt;% 
  dplyr::filter(!is.na(richness),
                !is.na(CD4))


overall &lt;- cor.test(x= df$richness, y=df$CD4, method = &quot;spearman&quot;) %&gt;% 
  keep(names(.) %in% c(&quot;statistic&quot;,&quot;estimate&quot;, &quot;p.value&quot;))


per_group &lt;- 
  unique(metadata$group) %&gt;% 
  set_names() %&gt;% 
  purrr::map(function(group){
    df_1 &lt;- 
      df %&gt;% 
      filter(cat_var == group) %&gt;% 
      select(richness, CD4) 
    cor.test(x=df_1$richness, y = df_1$CD4, method = &quot;spearman&quot;) %&gt;% 
      keep(names(.) %in% c(&quot;statistic&quot;, &quot;estimate&quot;, &quot;p.value&quot;)) %&gt;% 
    as.data.frame()
  })

overall</code></pre>
<pre><code>$statistic
       S 
47338.42 

$p.value
[1] 0.1551108

$estimate
      rho 
0.1717536 </code></pre>
<pre class="r"><code>per_group</code></pre>
<pre><code>$DTG
  statistic   p.value  estimate
S   6356.89 0.5305265 0.1096792

$`DRV/r`
  statistic   p.value  estimate
S  5778.569 0.2725564 0.1906766</code></pre>
<p>Doesn’t seem to be any signifcant association between gene richness
and CD4 counts at basal.</p>
</div>
<div id="delta-cd4-and-deltagr" class="section level3">
<h3>Delta-CD4 and DeltaGR</h3>
<pre class="r"><code>deltas &lt;- 
TableList %&gt;% 
  dplyr::right_join(., metadata[,c(&quot;SampleID&quot;,&quot;CD4&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  dplyr::group_by(link_var) %&gt;% 
  # dplyr::filter(long_var %in% c(0,96)) %&gt;% 
  dplyr::filter(any(long_var == 0)) %&gt;%
  dplyr::mutate(GR_abs = richness - richness[long_var == 0],
                GR_rel = (richness - richness[long_var == 0])/(richness + richness[long_var == 0]),
                CD4_abs = CD4- CD4[long_var == 0],
                CD4_rel = (CD4 - CD4[long_var == 0])/(CD4 + CD4[long_var == 0]))</code></pre>
<p>We need to check if the increments in CD4 are linked to increments in
GeneRichness. Let’s calculate these increments in two different
ways:</p>
<ul>
<li>Absolute: CD4(tx)-CD4(t0) or GR(tx)-GR(to)</li>
<li>Relative: (CD4(tx)-CD(t0)/(CD4(tx)+CD4(t0))) or
(GR(tx)-GR(t0)/(GR(tx)+GR(t0)))</li>
</ul>
<p>Check distribution of these values and see if there is a possibility
of reasonable categorization.</p>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/plot_dists-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/315_GeneRichness_vs_CD4.Rmd/plot_dists-2.png" width="672" style="display: block; margin: auto;" />
As we can see, the distribution of Gene Richness change vs baseline is
pretty similar at both weeks 488 and 96, albeit week 96 has a slightly
higher median. The differences get bigger when comparing between groups
though, as DTG seems significantly bigger than DRV, especially at week
96 (DTG=0.88 vs DRV/r = 0.39 median delta. Let’s test with a wilcoxon
test)</p>
<pre class="r"><code>long_deltas &lt;- 
deltas %&gt;% 
  dplyr::filter(long_var %in% c(48,96)) %&gt;% 
  group_by(link_var) %&gt;%
  dplyr::filter(!is.na(GR_rel)) %&gt;% 
  dplyr::filter(n()==2) %&gt;% 
  dplyr::group_by(cat_var, .drop= T) %&gt;% 
  rstatix::wilcox_test(., GR_rel ~ long_var, paired = T) %&gt;% 
  dplyr::select(-.y.) %&gt;% 
  setNames(c(&quot;time_point&quot;, &quot;time1&quot;, &quot;time2&quot;, &quot;n1&quot;,&quot;n2&quot;, &quot;stat&quot;,&quot;p&quot;)) %&gt;% 
  as.data.frame() %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;)


cat_deltas &lt;- 
deltas %&gt;% 
  dplyr::filter(long_var %in% c(48,96)) %&gt;% 
  group_by(long_var) %&gt;% 
  dplyr::filter(!is.na(GR_rel)) %&gt;% 
  rstatix::wilcox_test(., GR_rel ~ cat_var) %&gt;% 
  dplyr::select(-.y.) %&gt;% 
  setNames(c(&quot;time_point&quot;, &quot;group1&quot;, &quot;group2&quot;, &quot;n1&quot;,&quot;n2&quot;, &quot;stat&quot;,&quot;p&quot;)) %&gt;% 
  as.data.frame() %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;)

long_deltas</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">time_point</th>
<th align="left">time1</th>
<th align="left">time2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">stat</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DRV/r</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">20</td>
<td align="right">20</td>
<td align="right">114</td>
<td align="right">0.756</td>
</tr>
<tr class="even">
<td align="left">DTG</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">20</td>
<td align="right">20</td>
<td align="right">67</td>
<td align="right">0.165</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat_deltas</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">time_point</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">stat</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">48</td>
<td align="left">DRV/r</td>
<td align="left">DTG</td>
<td align="right">26</td>
<td align="right">25</td>
<td align="right">267</td>
<td align="right">0.2810</td>
</tr>
<tr class="even">
<td align="right">96</td>
<td align="left">DRV/r</td>
<td align="left">DTG</td>
<td align="right">25</td>
<td align="right">24</td>
<td align="right">199</td>
<td align="right">0.0437</td>
</tr>
</tbody>
</table>
<p>The difference in deltas between groups is significant at week 96 but
not at 48. However, no significant differences were found longitudinally
between groups.</p>
<p>And plot:</p>
<ul>
<li>Values vs timepoint, overall and per-group(DTG and DRVr only). Fit
ANOVA/t-test vs DTG/DRVr at each timepoint to check for differential
increases per ARM</li>
<li>AbsCD4 vs AbsGR, overall and per-group(DTG and dRVr only). Fit
LM</li>
<li>RelCD4 vs RelGR, overall and per-group(DTG and DRVr only). Fit
LM</li>
</ul>
</div>
<div id="categorized-cd4" class="section level3">
<h3>CAtegorized CD4</h3>
<p>This has partially been done. Split patients between
150&gt;CD4(t48)/(t96)&gt;150, and check if DeltaGR is different for
groups and fisher.test of these groups vs arm is significant.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.4        testthat_3.0.1     RColorBrewer_1.1-3 forcats_0.5.1     
 [5] stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4        readr_2.0.1       
 [9] tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] readxl_1.3.1        backports_1.4.1     blme_1.0-5         
  [4] workflowr_1.7.0     systemfonts_1.0.2   plyr_1.8.6         
  [7] igraph_1.2.9        splines_4.0.2       usethis_2.0.0      
 [10] digest_0.6.29       foreach_1.5.2       htmltools_0.5.2    
 [13] wesanderson_0.3.6   fansi_0.5.0         magrittr_2.0.2     
 [16] memoise_2.0.0       cluster_2.1.0       remotes_2.4.0      
 [19] tzdb_0.3.0          aws.signature_0.6.0 Biostrings_2.58.0  
 [22] modelr_0.1.8        vroom_1.5.7         svglite_2.0.0      
 [25] prettyunits_1.1.1   colorspace_2.0-2    rvest_1.0.1        
 [28] haven_2.3.1         xfun_0.30           callr_3.7.0        
 [31] crayon_1.5.1        jsonlite_1.7.2      lme4_1.1-27.1      
 [34] survival_3.1-12     iterators_1.0.14    ape_5.5            
 [37] glue_1.5.1          kableExtra_1.3.4    polyclip_1.10-0    
 [40] gtable_0.3.0        zlibbioc_1.36.0     XVector_0.30.0     
 [43] webshot_0.5.2       phyloseq_1.34.0     car_3.0-12         
 [46] pkgbuild_1.2.0      Rhdf5lib_1.12.1     BiocGenerics_0.36.1
 [49] abind_1.4-5         scales_1.2.0        mvtnorm_1.1-2      
 [52] DBI_1.1.1           rstatix_0.7.0       Rcpp_1.0.7         
 [55] viridisLite_0.4.0   xtable_1.8-4        merTools_0.5.2     
 [58] bit_4.0.4           stats4_4.0.2        htmlwidgets_1.5.4  
 [61] httr_1.4.2          ellipsis_0.3.2      farver_2.1.0       
 [64] DataExplorer_0.8.2  pkgconfig_2.0.3     sass_0.4.0         
 [67] dbplyr_2.1.1        utf8_1.2.2          here_1.0.1         
 [70] labeling_0.4.2      tidyselect_1.1.2    rlang_1.0.2        
 [73] reshape2_1.4.4      later_1.3.0         munsell_0.5.0      
 [76] cellranger_1.1.0    tools_4.0.2         cachem_1.0.1       
 [79] cli_3.0.1           generics_0.1.2      selbal_0.1.0       
 [82] ade4_1.7-18         devtools_2.3.2      broom_0.8.0        
 [85] aws.s3_0.3.21       evaluate_0.15       biomformat_1.18.0  
 [88] fastmap_1.1.0       arm_1.12-2          yaml_2.3.5         
 [91] bit64_4.0.5         processx_3.5.2      knitr_1.38         
 [94] fs_1.5.0            nlme_3.1-148        whisker_0.4        
 [97] mime_0.11           xml2_1.3.2          compiler_4.0.2     
[100] rstudioapi_0.13     curl_4.3            ggsignif_0.6.3     
[103] reprex_2.0.1        tweenr_1.0.2        bslib_0.3.1        
[106] stringi_1.7.6       highr_0.9           ps_1.5.0           
[109] desc_1.4.1          logger_0.2.2        lattice_0.20-41    
[112] Matrix_1.2-18       nloptr_1.2.2.3      vegan_2.5-7        
[115] permute_0.9-7       multtest_2.46.0     vctrs_0.3.8        
[118] pillar_1.7.0        lifecycle_1.0.1     networkD3_0.4      
[121] rhdf5filters_1.2.1  jquerylib_0.1.4     data.table_1.14.2  
[124] patchwork_1.1.1     httpuv_1.6.3        R6_2.5.1           
[127] promises_1.2.0.1    gridExtra_2.3       IRanges_2.24.1     
[130] sessioninfo_1.1.1   codetools_0.2-16    pkgload_1.2.4      
[133] boot_1.3-25         MASS_7.3-51.6       assertthat_0.2.1   
[136] rhdf5_2.34.0        rprojroot_2.0.3     withr_2.5.0        
[139] S4Vectors_0.28.1    broom.mixed_0.2.7   mgcv_1.8-31        
[142] parallel_4.0.2      hms_1.1.1           grid_4.0.2         
[145] coda_0.19-4         minqa_1.2.4         rmarkdown_2.11     
[148] carData_3.0-5       logging_0.10-108    ggpubr_0.4.0       
[151] git2r_0.28.0        ggforce_0.3.3       Biobase_2.50.0     
[154] shiny_1.7.1         lubridate_1.8.0     base64enc_0.1-3    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
