<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blazquez" />

<meta name="date" content="2022-11-11" />

<title>Gene Richness vs CD4</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene Richness vs CD4</h1>
<h4 class="author">Carlos Blazquez</h4>
<h4 class="date">11/11/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef65ff406ad7f0c33e2880f05da07a22c0a32aefetargetblankf65ff40a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MicrobialGenomics/advanz4/tree/f65ff406ad7f0c33e2880f05da07a22c0a32aefe" target="_blank">f65ff40</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef65ff406ad7f0c33e2880f05da07a22c0a32aefetargetblankf65ff40a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MicrobialGenomics/advanz4/tree/f65ff406ad7f0c33e2880f05da07a22c0a32aefe" target="_blank">f65ff40</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Metadata/MetadataManagement/Old_Metadata/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/010_Basic_Report_cache/
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   analysis/315_GeneRichness_vs_CD4.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/315_GeneRichness_vs_CD4.Rmd</code>) and HTML
(<code>docs/315_GeneRichness_vs_CD4.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f65ff406ad7f0c33e2880f05da07a22c0a32aefe/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">f65ff40</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Forgot to render html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/f65ff406ad7f0c33e2880f05da07a22c0a32aefe/docs/315_GeneRichness_vs_CD4.html" target="_blank">f65ff40</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Forgot to render html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/27df3cbd6182d522f0342d5985ff57c229c46ff4/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">27df3cb</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Added comparisons and lmms of GR per recovery
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/27df3cbd6182d522f0342d5985ff57c229c46ff4/docs/315_GeneRichness_vs_CD4.html" target="_blank">27df3cb</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Added comparisons and lmms of GR per recovery
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/6778971304fc9248a78e92037ad654e226a251d0/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">6778971</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Added comparisons and lmms of GR per recovery
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/6778971304fc9248a78e92037ad654e226a251d0/docs/315_GeneRichness_vs_CD4.html" target="_blank">6778971</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-28
</td>
<td>
Added comparisons and lmms of GR per recovery
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/a0f19e26ed38110e625ec87f05921c88274a473c/docs/315_GeneRichness_vs_CD4.html" target="_blank">a0f19e2</a>
</td>
<td>
cblazquez
</td>
<td>
2022-11-25
</td>
<td>
moved files to code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/074d657fa84a274ad41976f978b04c48c7415fa2/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">074d657</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-25
</td>
<td>
Added comparisons in GR and CD4
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/074d657fa84a274ad41976f978b04c48c7415fa2/docs/315_GeneRichness_vs_CD4.html" target="_blank">074d657</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-25
</td>
<td>
Added comparisons in GR and CD4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e39741ca8f6064ebc9c4e129538f02e92954aee9/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">e39741c</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-23
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a6ac6e99d02c48b2eaf0615efa89d6aced8c355f/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">a6ac6e9</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-23
</td>
<td>
Merge branch ‘main’ of <a
href="https://github.com/MicrobialGenomics/advanz4"
class="uri">https://github.com/MicrobialGenomics/advanz4</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/649e4c5bbb240d9e3844ebabcf07eb03d532b165/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">649e4c5</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-23
</td>
<td>
Added relative deltas in GR and CD4
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/649e4c5bbb240d9e3844ebabcf07eb03d532b165/docs/315_GeneRichness_vs_CD4.html" target="_blank">649e4c5</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-23
</td>
<td>
Added relative deltas in GR and CD4
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/042b81ea168e880658e3f0c76df657a6a9ead865/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">042b81e</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-18
</td>
<td>
Merge branch ‘main’ of <a
href="https://github.com/MicrobialGenomics/advanz4"
class="uri">https://github.com/MicrobialGenomics/advanz4</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/89172953a16040ed3d2c2ac5e65029544ec8f7d0/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">8917295</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-18
</td>
<td>
Update 315_GeneRichness_vs_CD4.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">f0e92e1</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-16
</td>
<td>
Started with baseline distributions of CD4 and GR
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/f0e92e1a746d3195eda0a946ef9c97b4dd72a2f0/docs/315_GeneRichness_vs_CD4.html" target="_blank">f0e92e1</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-11-16
</td>
<td>
Started with baseline distributions of CD4 and GR
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/df53555b1f5f30cb294433fd028ea0cf105d565b/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">df53555</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-11
</td>
<td>
Update 315_GeneRichness_vs_CD4.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/86d7c3aa84215f8988c88c31d59e9f23c51ac3a9/analysis/315_GeneRichness_vs_CD4.Rmd" target="_blank">86d7c3a</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-11-11
</td>
<td>
Create 315_GeneRichness_vs_CD4.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>metadata &lt;- 
  here::here(&quot;Metadata&quot;) %&gt;% 
  file.info(list.files(., full.names = T)) %&gt;%
  rownames() %&gt;% 
  magrittr::extract(str_detect(.,&quot;clean_metadata&quot;)) %&gt;% 
  sort(decreasing = T) %&gt;% 
  magrittr::extract(1L) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;mistral-wp6-advanz4&quot;,object = &quot;metagenome/WMGS/MREObject.rds&quot;, region = &quot;eu-west-1&quot;)</code></pre>
<div id="gene-richness-vs-cd4" class="section level2">
<h2>Gene Richness vs CD4</h2>
<p>In this analysis we are interested in the relationship between
CD4/CD4 recovery and gene richness. For that, several questions are of
importance.</p>
<div id="cd4-baseline-distribution" class="section level3">
<h3>CD4 baseline distribution</h3>
<p>We already know that CD4 mean(sd) is similar for DTG/DRVr groups, but
we have not checked the distribution. Is it unimodal or bimodal? a
density plot should help.</p>
<pre class="r"><code>basal_df &lt;- 
metadata %&gt;% 
  # dplyr::mutate(delta_cd4 = CD4[time_point == 48] - CD4[time_point == 0])
  dplyr::filter(time_point == 0) %&gt;% 
  dplyr::select(SampleID, group, CD4)  
  
median_df &lt;- basal_df %&gt;% 
  dplyr::group_by(group) %&gt;% 
  summarise(med = median(CD4))
  

peak_finder &lt;-
  function(density, goal = &quot;max&quot;,keep = F) {
    ### Let&#39;s define this function heere but we&#39;ll use it later
    stopifnot(&quot;goal can only be either local maxima (use: &#39;max&#39;) or local minima (use &#39;min)&quot; = goal %in% c(&quot;min&quot;,&quot;max&quot;))
    
    val &lt;- case_when(goal == &quot;max&quot; ~ -2,
                   goal == &quot;min&quot; ~ 2)
    
    res &lt;-
      density %&gt;%
      purrr::keep(names(.) %in% c(&quot;x&quot;, &quot;y&quot;)) %&gt;%
      as.data.frame() %&gt;%
      dplyr::mutate(is_peak = c(F, # 1st value can&#39;t be a local maxima, as there is no prior value to compare
                                diff(y) %&gt;% # For each nth element of the vector, get (n+1) - n
                                  sign() %&gt;%  # convert to -1 (n+1 is lower than n) or +1 (n+1 is higher than n)
                                  diff() == val, # if n is at a local maxima, both n+1 and n-1 are negative so [(n+1)-n] - [n - (n-1)] is -2
                                F)) # Last element can&#39;t be a local maxima either)
                    
                    if (!keep) {
                      res %&gt;%
                        dplyr::filter(is_peak) %&gt;%
                        dplyr::select(-is_peak)
                    } else{
                      res %&gt;%
                        dplyr::select(-is_peak)
                    }
  }


peak_df &lt;- unique(basal_df$group) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(function(arm){
   basal_df %&gt;% 
      dplyr::filter(group == arm) %&gt;% 
      pull(CD4) %&gt;% 
      density(n=nrow(.)) %&gt;% 
      peak_finder(., goal = &quot;max&quot;,keep = F) %&gt;% 
      mutate(group = arm)

  })



  
  
  
plt &lt;- ggplot(basal_df, aes(x=CD4, color=group)) +

    # geom_histogram(aes(fill = group), bins = 1000) +
    geom_density( alpha = .4) +
  theme_bw() +
  scale_color_brewer(palette = &quot;Accent&quot;) +
  scale_fill_brewer(palette = &quot;Accent&quot;) +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) +
  geom_text(data = median_df, aes(label = paste(&quot;med =&quot;,med), x = med, y = c(0.012,0.015)), color = &quot;black&quot;) +
  # geom_vline(data = peak_df, aes(xintercept =  x),color = &quot;black&quot;,lty = 3) +
  labs (x = &quot;CD4 (counts)&quot;)
  # geom_errorbar(stat = &quot;hline&quot;, xintercept = &quot;median&quot;,
  # width=0.8,aes(xmax=..x..,xmin=..x..))

  
  
plt</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/basal_CD4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>peak_df</code></pre>
<pre><code>          x            y group
1  26.10519 0.0103359201   DTG
2 229.10519 0.0005447832   DTG
3  11.04344 0.0169875257 DRV/r
4  68.04344 0.0063346692 DRV/r</code></pre>
<p>It appears the distribution of CD4 at BL shows two peaks in kernel
density, one at 26 counts/ml and one at 229 count/ml. There is a second
peak in the DRV as well, but it seems the density function smoothed it
out.</p>
</div>
<div id="generichness-at-baseline-distribution" class="section level3">
<h3>GeneRichness at baseline distribution</h3>
<p>We know (I think) that there is no difference in generichness between
arme at baseline, but we do not know the distribution. In Guillen et al,
we found a bimodal distribution in HIV patients. Is that the case of
advanz4?</p>
<pre class="r"><code>igc&lt;-metar::get_diversity(mymre)@igc
Threshold &lt;-  igc@dataTable %&gt;%
  dplyr::select(SampleID, NumberMappedReads) %&gt;%
  unique() %&gt;%
  summarise(Q = quantile(NumberMappedReads,0.02)) %&gt;% 
  as.numeric()

TableList &lt;- c(cat_vector) %&gt;%
    set_names() %&gt;%
    map( ~{

        
          ThresholdTable &lt;- igc@dataTable %&gt;%
                filter(NumberMappedReads &gt;= Threshold) %&gt;%
                filter(ReadCountReal&gt;=Threshold) %&gt;% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %&gt;%
                summarise(across(everything(), min)) %&gt;%
                ungroup() %&gt;%
          right_join(., metadata, by=&quot;SampleID&quot;) %&gt;%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=record_id,
                 cat_var = .x,
                 long_var = time_point,
                 NumberMappedReads)
  

        
    })

basal_df &lt;- 
  TableList %&gt;% 
  pluck(&quot;group&quot;) %&gt;% 
  dplyr::filter(long_var == 0)


median_df &lt;- 
  basal_df %&gt;% 
  dplyr::rename(group = cat_var) %&gt;% 
  dplyr::group_by(group) %&gt;% 
  filter(!is.na(richness)) %&gt;% 
  summarise(med = median(richness))

  
plt &lt;- 
basal_df %&gt;% 
  ggplot(aes(x = richness, fill = cat_var)) +
  # geom_histogram(aes(y = ..density..),alpha = .4) +
  geom_density(alpha = .4) +
  theme_bw() +
  scale_fill_brewer(palette=&quot;Accent&quot;) +
  scale_color_brewer(palette = &quot;Accent&quot;) +
  labs(x=&quot;gene_count&quot;, fill = &quot;group&quot;) +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) 


peak_df &lt;- 
basal_df %&gt;% 
  pull(cat_var) %&gt;% 
  unique() %&gt;% 
  map_dfr(function(arm){
    basal_df %&gt;% 
      dplyr::filter(cat_var == arm) %&gt;%
      dplyr::filter(!is.na(richness)) %&gt;% 
      dplyr::pull(richness) %&gt;% 
      density() %&gt;%
      peak_finder() %&gt;% 
      dplyr::mutate(group = arm)

  })

plt</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/GR_basal-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>peak_df</code></pre>
<pre><code>         x            y group
1 774765.3 1.863682e-06   DTG
2 731633.3 1.494479e-06 DRV/r</code></pre>
<p>It seems both groups have a similar median richness, but with
somewhat different distributions, with DTG following a slightly bimodal
distribution with a higher median peak than Drv/r. Although the left
“step” in may look like a local maxima, it actually does not decrease on
the right side.</p>
</div>
<div id="gene-richness-vs-cd4-at-baseline" class="section level3">
<h3>Gene Richness vs CD4 at baseline</h3>
<p>Is there any association between gene richness and CD4 at baseline
only, overall, and per-group(DTG and DRVr only)?</p>
<pre class="r"><code>df &lt;- 
basal_df %&gt;% 
  right_join(metadata[,c(&quot;SampleID&quot;,&quot;CD4&quot;)]) %&gt;% 
  dplyr::filter(!is.na(richness),
                !is.na(CD4))


overall &lt;- cor.test(x= df$richness, y=df$CD4, method = &quot;spearman&quot;) %&gt;% 
  keep(names(.) %in% c(&quot;statistic&quot;,&quot;estimate&quot;, &quot;p.value&quot;))


per_group &lt;- 
  unique(metadata$group) %&gt;% 
  set_names() %&gt;% 
  purrr::map(function(group){
    df_1 &lt;- 
      df %&gt;% 
      filter(cat_var == group) %&gt;% 
      select(richness, CD4) 
    cor.test(x=df_1$richness, y = df_1$CD4, method = &quot;spearman&quot;) %&gt;% 
      keep(names(.) %in% c(&quot;statistic&quot;, &quot;estimate&quot;, &quot;p.value&quot;)) %&gt;% 
    as.data.frame()
  })

overall</code></pre>
<pre><code>$statistic
       S 
47338.42 

$p.value
[1] 0.1551108

$estimate
      rho 
0.1717536 </code></pre>
<pre class="r"><code>per_group</code></pre>
<pre><code>$DTG
  statistic   p.value  estimate
S   6356.89 0.5305265 0.1096792

$`DRV/r`
  statistic   p.value  estimate
S  5778.569 0.2725564 0.1906766</code></pre>
<p>Doesn’t seem to be any signifcant association between gene richness
and CD4 counts at basal.</p>
</div>
<div id="delta-cd4-and-deltagr" class="section level3">
<h3>Delta-CD4 and DeltaGR</h3>
<pre class="r"><code>deltas &lt;- 
TableList %&gt;% 
  pluck(&quot;group&quot;) %&gt;% 
  dplyr::right_join(., metadata[,c(&quot;SampleID&quot;,&quot;CD4&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  dplyr::group_by(link_var) %&gt;% 
  # dplyr::filter(long_var %in% c(0,96)) %&gt;% 
  dplyr::filter(any(long_var == 0)) %&gt;%
  dplyr::mutate(GR_abs = richness - richness[long_var == 0],
                GR_rel = (richness - richness[long_var == 0])/(richness + richness[long_var == 0]),
                CD4_abs = CD4- CD4[long_var == 0],
                CD4_rel = (CD4 - CD4[long_var == 0])/(CD4 + CD4[long_var == 0]))</code></pre>
<p>We need to check if the increments in CD4 are linked to increments in
GeneRichness. Let’s calculate these increments in two different
ways:</p>
<ul>
<li>Absolute: CD4(tx)-CD4(t0) or GR(tx)-GR(to)</li>
<li>Relative: (CD4(tx)-CD(t0)/(CD4(tx)+CD4(t0))) or
(GR(tx)-GR(t0)/(GR(tx)+GR(t0)))</li>
</ul>
<p>Check distribution of these values and see if there is a possibility
of reasonable categorization.</p>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/plot_dists-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/315_GeneRichness_vs_CD4.Rmd/plot_dists-2.png" width="672" style="display: block; margin: auto;" />
As we can see, the distribution of Gene Richness change vs baseline is
pretty similar at both weeks 488 and 96, albeit week 96 has a slightly
higher median. The differences get bigger when comparing between groups
though, as DTG seems significantly bigger than DRV, especially at week
96 (DTG=0.88 vs DRV/r = 0.39 median delta. Let’s test with a wilcoxon
test)</p>
<pre class="r"><code>long_deltas &lt;- 
deltas %&gt;% 
  dplyr::filter(long_var %in% c(48,96)) %&gt;% 
  group_by(link_var) %&gt;%
  dplyr::filter(!is.na(GR_rel)) %&gt;% 
  dplyr::filter(n()==2) %&gt;% 
  dplyr::group_by(cat_var, .drop= T) %&gt;% 
  rstatix::wilcox_test(., GR_rel ~ long_var, paired = T) %&gt;% 
  dplyr::select(-.y.) %&gt;% 
  setNames(c(&quot;time_point&quot;, &quot;time1&quot;, &quot;time2&quot;, &quot;n1&quot;,&quot;n2&quot;, &quot;stat&quot;,&quot;p&quot;)) %&gt;% 
  as.data.frame() %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;)


cat_deltas &lt;- 
deltas %&gt;% 
  dplyr::filter(long_var %in% c(48,96)) %&gt;% 
  group_by(long_var) %&gt;% 
  dplyr::filter(!is.na(GR_rel)) %&gt;% 
  rstatix::wilcox_test(., GR_rel ~ cat_var) %&gt;% 
  dplyr::select(-.y.) %&gt;% 
  setNames(c(&quot;time_point&quot;, &quot;group1&quot;, &quot;group2&quot;, &quot;n1&quot;,&quot;n2&quot;, &quot;stat&quot;,&quot;p&quot;)) %&gt;% 
  as.data.frame() %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;)

long_deltas</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">time_point</th>
<th align="left">time1</th>
<th align="left">time2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">stat</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DRV/r</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">20</td>
<td align="right">20</td>
<td align="right">114</td>
<td align="right">0.756</td>
</tr>
<tr class="even">
<td align="left">DTG</td>
<td align="left">48</td>
<td align="left">96</td>
<td align="right">20</td>
<td align="right">20</td>
<td align="right">67</td>
<td align="right">0.165</td>
</tr>
</tbody>
</table>
<pre class="r"><code>cat_deltas</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">time_point</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">stat</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">48</td>
<td align="left">DRV/r</td>
<td align="left">DTG</td>
<td align="right">26</td>
<td align="right">25</td>
<td align="right">267</td>
<td align="right">0.2810</td>
</tr>
<tr class="even">
<td align="right">96</td>
<td align="left">DRV/r</td>
<td align="left">DTG</td>
<td align="right">25</td>
<td align="right">24</td>
<td align="right">199</td>
<td align="right">0.0437</td>
</tr>
</tbody>
</table>
<p>The difference in deltas between groups is significant at week 96 but
not at 48. However, no significant differences were found longitudinally
between groups.</p>
</div>
</div>
<div id="correlations-between-cd4-and-gr" class="section level1">
<h1>Correlations between CD4 and GR</h1>
<p>Let’s check if there is a correlation between both variables. First
overall:</p>
<pre class="r"><code>absolute_cors &lt;- 
deltas %&gt;% 
  dplyr::filter(!long_var == 0) %&gt;% 
  correlation::cor_test(., &quot;CD4_abs&quot;, &quot;GR_abs&quot;, method =&quot;spearman&quot;)

plot_overall_abs &lt;- deltas %&gt;% 
  ggplot(aes(x=CD4_abs, y = GR_abs)) +
  geom_point(aes(color = cat_var)) +
  geom_smooth(method = &quot;lm&quot;) +
  theme_bw() + 
  labs(color = &quot;group&quot;, title = &quot;Absolute&quot;)


relative_cors &lt;- 
deltas %&gt;% 
  dplyr::filter(!long_var == 0) %&gt;% 
  correlation::cor_test(., &quot;CD4_rel&quot;, &quot;GR_rel&quot;, method =&quot;spearman&quot;)

plot_overall_rel &lt;- deltas %&gt;% 
  ggplot(aes(x=CD4_rel, y = GR_rel)) +
  geom_point(aes(color = cat_var)) +
  geom_smooth(method = &quot;lm&quot;) +
  theme_bw() +
  labs(color = &quot;group&quot;, title = &quot;relative&quot;)


absolute_cors</code></pre>
<pre><code>Parameter1 | Parameter2 |  rho |        95% CI |        S |     p
-----------------------------------------------------------------
CD4_abs    |     GR_abs | 0.12 | [-0.06, 0.30] | 2.65e+05 | 0.177

Observations: 122</code></pre>
<pre class="r"><code>relative_cors</code></pre>
<pre><code>Parameter1 | Parameter2 |   rho |        95% CI |        S |     p
------------------------------------------------------------------
CD4_rel    |     GR_rel | -0.01 | [-0.19, 0.17] | 3.06e+05 | 0.895

Observations: 122</code></pre>
<pre class="r"><code>ggarrange(plot_overall_abs, plot_overall_rel, common.legend = T)</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/overall_corrs-1.png" width="672" style="display: block; margin: auto;" />
No overall correlations could be found. Let’s check split per
timepoint:</p>
<pre class="r"><code>c(48,96) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(tp){
  deltas %&gt;% 
  dplyr::filter(long_var == tp) %&gt;% 
  correlation::cor_test(., &quot;CD4_rel&quot;, &quot;GR_rel&quot;, method =&quot;spearman&quot;)

  })</code></pre>
<pre><code>$`48`
Parameter1 | Parameter2 |   rho |        95% CI |        S |     p
------------------------------------------------------------------
CD4_rel    |     GR_rel | -0.21 | [-0.47, 0.08] | 25144.00 | 0.148

Observations: 50

$`96`
Parameter1 | Parameter2 |  rho |        95% CI |       S |     p
----------------------------------------------------------------
CD4_rel    |     GR_rel | 0.16 | [-0.17, 0.46] | 8315.92 | 0.336

Observations: 39</code></pre>
<p>No Meaningful correlations could be found neither overall nor at
specific timepoints.</p>
</div>
<div id="interactions-between-cd4-and-gr-lmms" class="section level1">
<h1>Interactions between CD4 and GR: LMMs</h1>
<p>To have a better view of wether both variables interact, we will
perform LMMs. First on overall:</p>
<pre class="r"><code>lmm_abs &lt;- 
deltas %&gt;% 
  mutate(mock = 1) %&gt;% ## The function needs a cat_var, so we&#39;ll create a mock one
  dplyr::filter(!is.na(CD4_abs), !is.na(GR_abs), !long_var == 0) %&gt;% 
  # dplyr::rename(record_id = link_var) %&gt;% 
  create_LMM(., num_var = &quot;CD4_abs&quot;, long_var = &quot;GR_abs&quot;, link_var =  &quot;link_var&quot;, cat_var = &quot;mock&quot;)
stat_abs &lt;- lmm_abs$stats
plt_abs &lt;- lmm_abs$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = &quot;delta richness&quot;, y = &quot;delta CD4&quot;, title = &quot;Absolute&quot;) +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))



lmm_rel &lt;- 
deltas %&gt;% 
  mutate(mock = 1) %&gt;% ## The function needs a cat_var, so we&#39;ll create a mock one
  dplyr::filter(!is.na(CD4_rel), !is.na(GR_rel), !long_var == 0) %&gt;% 
  # dplyr::rename(record_id = link_var) %&gt;% 
  create_LMM(., num_var = &quot;CD4_rel&quot;, long_var = &quot;GR_rel&quot;, link_var =  &quot;link_var&quot;, cat_var = &quot;mock&quot;)
stat_rel &lt;- lmm_rel$stats
plt_rel &lt;- lmm_rel$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = &quot;delta richness&quot;, y = &quot;delta CD4&quot;, title = &quot;Relative&quot;)+
  facet_wrap(NULL) +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))


ggpubr::ggarrange(plt_abs, plt_rel)</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/LMMs_overall-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>nothing could be observed overall. Let’s check per group:</p>
<pre class="r"><code>lmm_abs &lt;- 
deltas %&gt;% 
  dplyr::filter(!is.na(CD4_abs), !is.na(GR_abs), !long_var == 0) %&gt;% 
  # dplyr::rename(record_id = link_var) %&gt;% 
  create_LMM(., num_var = &quot;CD4_abs&quot;, long_var = &quot;GR_abs&quot;, link_var =  &quot;link_var&quot;, cat_var = &quot;cat_var&quot;)
stat_abs &lt;- lmm_abs$stats
plt_abs &lt;- lmm_abs$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = &quot;delta richness&quot;, y = &quot;delta CD4&quot;, title = &quot;Absolute&quot;) +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))



lmm_rel &lt;- 
deltas %&gt;% 
  dplyr::filter(!is.na(CD4_rel), !is.na(GR_rel), !long_var == 0) %&gt;% 
  # dplyr::rename(record_id = link_var) %&gt;% 
  create_LMM(., num_var = &quot;CD4_rel&quot;, long_var = &quot;GR_rel&quot;, link_var =  &quot;link_var&quot;, cat_var = &quot;cat_var&quot;)
stat_rel &lt;- lmm_rel$stats
plt_rel &lt;- lmm_rel$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = &quot;delta richness&quot;, y = &quot;delta CD4&quot;, title = &quot;Relative&quot;)+
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))


ggpubr::ggarrange(plt_abs, plt_rel, nrow = 2, ncol = 1)</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/LMMs_group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>No quantitative difference could be found between delta GR and delta
CD4. The only thing to look at now is to check wether there is a
significant GR differences between those who recovered more or less than
150 CD4-</p>
</div>
<div id="comparisons-by-cd4-recovery" class="section level1">
<h1>Comparisons by CD4 recovery</h1>
<p>We did not find many signals correlating numerical absolute and
relative numbers of CD4 and Gene Richness. Now we’r going to check if
there is any actual difference in Gene richness in those with CD4
recovery &lt; 150. The patients have been stratified in 3 groups: those
who reached an increase of 150 counts/ml at week 48 (48), those who did
at week 96 (96) and those who didn’t surpass this threshold at all
(No)</p>
<pre class="r"><code>df &lt;- 
deltas %&gt;% 
  left_join(metadata[,c(&quot;SampleID&quot;, &quot;CD4diff_48&quot;, &quot;CD4diff_96&quot;)], by = &quot;SampleID&quot;) %&gt;% 
  # mutate(CD4diff_48 = recode(CD4diff_48, &quot;150_H&quot; = &quot;48_H&quot;, &quot;150_L&quot; = &quot;48_L&quot;),
  #        CD4diff_96 = recode(CD4diff_96, &quot;150_H&quot; = &quot;96_H&quot;, &quot;150_L&quot; = &quot;96_L&quot;)) %&gt;% 
  mutate(recovery = case_when(
    CD4diff_48 == &quot;150_H&quot; ~ &quot;48&quot;,
    CD4diff_48 == &quot;150_L&quot; &amp; CD4diff_96 == &quot;150_H&quot; ~ &quot;96&quot;,
    CD4diff_48 == &quot;150_L&quot; &amp; CD4diff_96 == &quot;150_L&quot; ~ &quot;No&quot;,
    is.na(CD4diff_48) &amp; CD4diff_96 == &quot;150_H&quot; ~ &quot;96&quot;,
    is.na(CD4diff_48) &amp; CD4diff_96 == &quot;150_L&quot; ~ &quot;No&quot;)) %&gt;% 
  filter(!is.na(recovery), !long_var == 0)

test_cat &lt;- 
df %&gt;% 
  group_by(long_var) %&gt;% 
  rstatix::wilcox_test(richness ~ recovery) %&gt;% 
  rstatix::add_xy_position(x = &quot;long_var&quot;, group = &quot;recovery&quot;)


plt &lt;- 
df %&gt;% 
  ggplot(., aes(x = as.factor(long_var), y = richness)) +
  geom_boxplot(aes(fill = recovery)) +
  labs(x = &quot;weeks&quot;, y = &quot;gene richness (absolute)&quot;, fill = &quot;recovery\n(&gt;150 increase)&quot;) +
  scale_fill_brewer(palette = &quot;Dark2&quot;) +
  theme_bw() +
  stat_pvalue_manual(test_cat,inhertit.aes = F ,hide.ns = T) +
  theme(
    axis.title = element_text(face = &quot;bold&quot;)
  )</code></pre>
<pre><code>Warning: Ignoring unknown parameters: inhertit.aes</code></pre>
<pre class="r"><code>lmm &lt;- 
df %&gt;% 
 get_lmm_effects(., cat_vector = c(&quot;recovery&quot;),num_vector = &quot;richness&quot;, link_var = &quot;link_var&quot;, long_var = &quot;long_var&quot;)</code></pre>
<pre><code>[1] &quot;recovery&quot;
[1] &quot;richness&quot;</code></pre>
<pre><code>Warning in as_lmerModLT(model, devfun): Model may not have converged with 1
eigenvalue close to zero: 2.9e-09</code></pre>
<pre><code>Warning in as_lmerModLT(model, devfun): Model may not have converged with 1
eigenvalue close to zero: 2.6e-09</code></pre>
<pre class="r"><code>plt</code></pre>
<pre><code>Warning: Removed 9 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/richness_CD4_categorical-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lmm$recovery$richness$plot</code></pre>
<pre><code>Warning: Removed 9 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 6 row(s) containing missing values (geom_path).</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/richness_CD4_categorical-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lmm$recovery$richness$summary</code></pre>
<pre><code>                      estimate      st.err        df        tval            p
(Intercept)        795788.5599  43417.3975 127.73715 18.32879458 1.433652e-37
long_var              770.1625    505.7558  80.74575  1.52279522 1.317169e-01
cat_var96            8010.1415 116519.5970 134.79827  0.06874502 9.452945e-01
cat_varNo          -29525.4107 107031.8415 126.49272 -0.27585633 7.831086e-01
long_var:cat_var96    253.5737   1334.0261  98.66064  0.19008152 8.496358e-01
long_var:cat_varNo   -252.2829   1181.9625  78.58672 -0.21344409 8.315334e-01</code></pre>
<p>No differences in gene richness could be found between those who
recovered more than &lt;150 CD4+/ml at week 48 and those who did at week
96. We’ll test for deltaGR now:</p>
<pre class="r"><code>df &lt;- 
deltas %&gt;% 
  left_join(metadata[,c(&quot;SampleID&quot;, &quot;CD4diff_48&quot;, &quot;CD4diff_96&quot;)], by = &quot;SampleID&quot;) %&gt;% 
  # mutate(CD4diff_48 = recode(CD4diff_48, &quot;150_H&quot; = &quot;48_H&quot;, &quot;150_L&quot; = &quot;48_L&quot;),
  #        CD4diff_96 = recode(CD4diff_96, &quot;150_H&quot; = &quot;96_H&quot;, &quot;150_L&quot; = &quot;96_L&quot;)) %&gt;% 
  mutate(recovery = case_when(
    CD4diff_48 == &quot;150_H&quot; ~ &quot;48&quot;,
    CD4diff_48 == &quot;150_L&quot; &amp; CD4diff_96 == &quot;150_H&quot; ~ &quot;96&quot;,
    CD4diff_48 == &quot;150_L&quot; &amp; CD4diff_96 == &quot;150_L&quot; ~ &quot;No&quot;,
    is.na(CD4diff_48) &amp; CD4diff_96 == &quot;150_H&quot; ~ &quot;96&quot;,
    is.na(CD4diff_48) &amp; CD4diff_96 == &quot;150_L&quot; ~ &quot;No&quot;)) %&gt;% 
  filter(!is.na(recovery), !long_var == 0)

test_cat &lt;- 
df %&gt;% 
  group_by(long_var) %&gt;% 
  rstatix::wilcox_test(GR_rel ~ recovery) %&gt;% 
  rstatix::add_xy_position(x = &quot;long_var&quot;, group = &quot;recovery&quot;)


plt &lt;- 
df %&gt;% 
  ggplot(., aes(x = as.factor(long_var), y = GR_rel)) +
  geom_boxplot(aes(fill = recovery)) +
  labs(x = &quot;weeks&quot;, y = &quot;deltaGR (relative)&quot;, fill = &quot;recovery\n(&gt;150 increase)&quot;) +
  scale_fill_brewer(palette = &quot;Dark2&quot;) +
  theme_bw() +
  stat_pvalue_manual(test_cat,inhertit.aes = F ,hide.ns = T) +
  theme(
    axis.title = element_text(face = &quot;bold&quot;)
  )




lmm &lt;- 
df %&gt;% 
 get_lmm_effects(., cat_vector = c(&quot;recovery&quot;),num_vector = &quot;GR_rel&quot;, link_var = &quot;link_var&quot;, long_var = &quot;long_var&quot;)</code></pre>
<pre><code>[1] &quot;recovery&quot;
[1] &quot;GR_rel&quot;</code></pre>
<pre class="r"><code>plt</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/delta_GR_CD4_cat-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lmm$recovery$GR_rel$plot</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/delta_GR_CD4_cat-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>lmm$recovery$GR_rel$summary</code></pre>
<pre><code>                        estimate       st.err        df       tval         p
(Intercept)         0.0479822750 0.0348358559 101.35262  1.3773818 0.1714284
long_var            0.0005301657 0.0003544080  70.11652  1.4959189 0.1391625
cat_var96          -0.0347647331 0.0854416581 116.43001 -0.4068827 0.6848412
cat_varNo          -0.1263813874 0.0811144307  98.18136 -1.5580629 0.1224360
long_var:cat_var96  0.0002888975 0.0008931111  79.37839  0.3234732 0.7471875
long_var:cat_varNo -0.0001419035 0.0007703357  68.69652 -0.1842100 0.8543914</code></pre>
<p>Finally, we are going to tests if overall having an increase higher
than 150 counts is tied to different deltaGR.</p>
<pre class="r"><code>dif48_abs &lt;- 
df %&gt;%
  ungroup() %&gt;% 
  dplyr::filter(long_var == 48) %&gt;% 
  rstatix::wilcox_test(GR_abs ~ CD4diff_48)


dif48_rel &lt;- 
  df %&gt;%
  ungroup() %&gt;%
  dplyr::filter(long_var == 48) %&gt;% 
  rstatix::wilcox_test(GR_rel ~ CD4diff_48)
dif_48 &lt;- rbind(dif48_abs, dif48_rel)


plt_48_abs &lt;- 
df %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(long_var %in% c(48),
                !is.na(CD4diff_48)) %&gt;% 
  ggplot(aes(y = GR_abs, fill = CD4diff_48)) +
  geom_boxplot() +
    theme_bw() +
  labs(title = &quot;Absolute&quot;)

plt_48_rel &lt;- 
df %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(long_var %in% c(48),
                !is.na(CD4diff_48)) %&gt;% 
  ggplot(aes(y = GR_rel, fill = CD4diff_48)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = &quot;Relative&quot;)

dif96_abs &lt;- 
  df %&gt;%
  ungroup() %&gt;%
  dplyr::filter(long_var == 96) %&gt;% 
  rstatix::wilcox_test(GR_abs ~ CD4diff_96)


dif96_rel &lt;- 
  df %&gt;%
  ungroup() %&gt;%   
  dplyr::filter(long_var == 96) %&gt;% 
  rstatix::wilcox_test(GR_rel ~ CD4diff_96)
dif_96 &lt;- rbind(dif96_abs, dif96_rel)


plt_96_abs &lt;- 
df %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(long_var %in% c(96),
                !is.na(CD4diff_96)) %&gt;% 
  ggplot(aes(y = GR_abs, fill = CD4diff_96)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = &quot;Absolute&quot;)


plt_96_rel &lt;- 
df %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(long_var %in% c(96),
                !is.na(CD4diff_96)) %&gt;% 
  ggplot(aes(y = GR_rel, fill = CD4diff_96)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = &quot;Relative&quot;)



dif_48</code></pre>
<pre><code># A tibble: 2 x 9
  .y.    group1 group2    n1    n2 statistic     p p.adj p.adj.signif
  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       
1 GR_abs 150_H  150_L     42    14       183 0.444 0.444 ns          
2 GR_rel 150_H  150_L     42    14       186 0.488 0.488 ns          </code></pre>
<pre class="r"><code>ggpubr::ggarrange(plt_48_abs, plt_48_rel, common.legend = T)</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/diff_CD4_test-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dif_96</code></pre>
<pre><code># A tibble: 2 x 9
  .y.    group1 group2    n1    n2 statistic     p p.adj p.adj.signif
  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       
1 GR_abs 150_H  150_L     40     8       169 0.123 0.123 ns          
2 GR_rel 150_H  150_L     40     8       174 0.085 0.085 ns          </code></pre>
<pre class="r"><code>ggpubr::ggarrange(plt_96_abs, plt_96_rel, common.legend = T)</code></pre>
<p><img src="figure/315_GeneRichness_vs_CD4.Rmd/diff_CD4_test-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>No differences could be observed between those who had &gt;150 CD4
count increase at either week 96 or week 48.</p>
<pre class="r"><code>contingency_table &lt;- 
df %&gt;% 
  ungroup() %&gt;% 
  dplyr::select(cat_var,recovery) %&gt;% 
  dplyr::group_by(cat_var, recovery) %&gt;% 
  tally() %&gt;% 
  pivot_wider(values_from = n, names_from = recovery) 


contingency_table</code></pre>
<pre><code># A tibble: 2 x 4
# Groups:   cat_var [2]
  cat_var  `48`  `96`    No
  &lt;chr&gt;   &lt;int&gt; &lt;int&gt; &lt;int&gt;
1 DRV/r      46     9    13
2 DTG        59    15     8</code></pre>
<pre class="r"><code>contingency_table %&gt;% 
  column_to_rownames(&quot;cat_var&quot;) %&gt;% 
  chisq.test()</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  .
X-squared = 3.0196, df = 2, p-value = 0.2209</code></pre>
<p>It appears, belonging to either treatment group does not affect the
odds of recovering &lt;150 CD4 counts at week 48 or 96.</p>
<p>And plot:</p>
<ul>
<li>Values vs timepoint, overall and per-group(DTG and DRVr only). Fit
ANOVA/t-test vs DTG/DRVr at each timepoint to check for differential
increases per ARM</li>
<li>AbsCD4 vs AbsGR, overall and per-group(DTG and dRVr only). Fit
LM</li>
<li>RelCD4 vs RelGR, overall and per-group(DTG and DRVr only). Fit
LM</li>
</ul>
<div id="categorized-cd4" class="section level3">
<h3>CAtegorized CD4</h3>
<p>This has partially been done. Split patients between
150&gt;CD4(t48)/(t96)&gt;150, and check if DeltaGR is different for
groups and fisher.test of these groups vs arm is significant.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nlme_3.1-148       metar_0.1.4        testthat_3.0.1     ggpubr_0.4.0      
 [5] RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0      dplyr_1.0.8       
 [9] purrr_0.3.4        readr_2.0.1        tidyr_1.2.0        tibble_3.1.6      
[13] ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2          tidyselect_1.1.2    lme4_1.1-27.1      
  [4] htmlwidgets_1.5.4   grid_4.0.2          devtools_2.3.2     
  [7] aws.signature_0.6.0 munsell_0.5.0       codetools_0.2-16   
 [10] withr_2.5.0         colorspace_2.0-2    Biobase_2.50.0     
 [13] phyloseq_1.34.0     logger_0.2.2        highr_0.9          
 [16] knitr_1.38          rstudioapi_0.13     stats4_4.0.2       
 [19] ggsignif_0.6.3      labeling_0.4.2      git2r_0.28.0       
 [22] polyclip_1.10-0     datawizard_0.6.2    bit64_4.0.5        
 [25] farver_2.1.0        rhdf5_2.34.0        rprojroot_2.0.3    
 [28] coda_0.19-4         vctrs_0.3.8         generics_0.1.2     
 [31] xfun_0.30           R6_2.5.1            arm_1.12-2         
 [34] rhdf5filters_1.2.1  cachem_1.0.1        assertthat_0.2.1   
 [37] promises_1.2.0.1    networkD3_0.4       scales_1.2.0       
 [40] vroom_1.5.7         merTools_0.5.2      gtable_0.3.0       
 [43] processx_3.5.2      workflowr_1.7.0     rlang_1.0.2        
 [46] systemfonts_1.0.2   splines_4.0.2       rstatix_0.7.0      
 [49] broom_0.8.0         yaml_2.3.5          reshape2_1.4.4     
 [52] abind_1.4-5         modelr_0.1.8        backports_1.4.1    
 [55] httpuv_1.6.3        tools_4.0.2         usethis_2.0.0      
 [58] logging_0.10-108    ellipsis_0.3.2      kableExtra_1.3.4   
 [61] jquerylib_0.1.4     biomformat_1.18.0   selbal_0.1.0       
 [64] BiocGenerics_0.36.1 sessioninfo_1.1.1   Rcpp_1.0.7         
 [67] plyr_1.8.6          base64enc_0.1-3     zlibbioc_1.36.0    
 [70] ps_1.5.0            prettyunits_1.1.1   cowplot_1.1.1      
 [73] correlation_0.8.2   S4Vectors_0.28.1    haven_2.3.1        
 [76] cluster_2.1.0       fs_1.5.0            here_1.0.1         
 [79] DataExplorer_0.8.2  magrittr_2.0.2      data.table_1.14.2  
 [82] lmerTest_3.1-3      reprex_2.0.1        mvtnorm_1.1-2      
 [85] whisker_0.4         pkgload_1.2.4       hms_1.1.1          
 [88] patchwork_1.1.1     mime_0.11           evaluate_0.15      
 [91] xtable_1.8-4        broom.mixed_0.2.7   readxl_1.3.1       
 [94] IRanges_2.24.1      gridExtra_2.3       compiler_4.0.2     
 [97] crayon_1.5.1        minqa_1.2.4         htmltools_0.5.2    
[100] mgcv_1.8-31         later_1.3.0         tzdb_0.3.0         
[103] lubridate_1.8.0     aws.s3_0.3.21       DBI_1.1.1          
[106] tweenr_1.0.2        dbplyr_2.1.1        MASS_7.3-51.6      
[109] boot_1.3-25         Matrix_1.2-18       ade4_1.7-18        
[112] car_3.0-12          wesanderson_0.3.6   permute_0.9-7      
[115] cli_3.0.1           insight_0.18.4      parallel_4.0.2     
[118] igraph_1.2.9        pkgconfig_2.0.3     numDeriv_2016.8-1.1
[121] xml2_1.3.2          foreach_1.5.2       svglite_2.0.0      
[124] bslib_0.3.1         blme_1.0-5          multtest_2.46.0    
[127] webshot_0.5.2       XVector_0.30.0      rvest_1.0.1        
[130] callr_3.7.0         digest_0.6.29       vegan_2.5-7        
[133] Biostrings_2.58.0   rmarkdown_2.11      cellranger_1.1.0   
[136] curl_4.3            shiny_1.7.1         nloptr_1.2.2.3     
[139] lifecycle_1.0.1     jsonlite_1.7.2      Rhdf5lib_1.12.1    
[142] carData_3.0-5       desc_1.4.1          viridisLite_0.4.0  
[145] fansi_0.5.0         pillar_1.7.0        lattice_0.20-41    
[148] fastmap_1.1.0       httr_1.4.2          pkgbuild_1.2.0     
[151] survival_3.1-12     glue_1.5.1          remotes_2.4.0      
[154] bayestestR_0.12.1   iterators_1.0.14    bit_4.0.4          
[157] ggforce_0.3.3       stringi_1.7.6       sass_0.4.0         
[160] memoise_2.0.0       ape_5.5            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
