<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />

<meta name="date" content="2022-11-30" />

<title>CMV vs Gene Richness</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="331_GR_BMI.html">BMI vs diversity</a>
</li>
<li>
  <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
</li>
<li>
  <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
</li>
<li>
  <a href="317_GeneRichness_vs_Inflammation.html">Gene richness vs inflammation</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CMV vs Gene Richness</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">2022-11-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1b0ddcbaff018d4f448ac865958b47049611e559targetblank1b0ddcba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/1b0ddcbaff018d4f448ac865958b47049611e559" target="_blank">1b0ddcb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1b0ddcbaff018d4f448ac865958b47049611e559targetblank1b0ddcba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/1b0ddcbaff018d4f448ac865958b47049611e559" target="_blank">1b0ddcb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/505_Adenovirus_Analysis_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    analysis/output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   analysis/304_Taxonomy_DA.Rmd
    Modified:   analysis/310_IGC_analysis.Rmd
    Modified:   analysis/311_GR_vs_BMI.Rmd
    Modified:   analysis/316_GeneRichness_vs_CMV.Rmd
    Modified:   analysis/320_Beta_Diversity.Rmd
    Modified:   analysis/_site.yml
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/316_GeneRichness_vs_CMV.Rmd</code>) and HTML (<code>docs/316_GeneRichness_vs_CMV.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/92767277ff8748b517aa47008bdc77234a80509f/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">9276727</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
<td>
printed cmv results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/92767277ff8748b517aa47008bdc77234a80509f/docs/316_GeneRichness_vs_CMV.html" target="_blank">9276727</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
<td>
printed cmv results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c1a9770c3ca2f9505837e8532b1cbd108d865a0c/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">c1a9770</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
<td>
updated website and CMV
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c1a9770c3ca2f9505837e8532b1cbd108d865a0c/docs/316_GeneRichness_vs_CMV.html" target="_blank">c1a9770</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
<td>
updated website and CMV
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">b9a23b3</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
<td>
fixed bugs on CMV analysis html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3/docs/316_GeneRichness_vs_CMV.html" target="_blank">b9a23b3</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
<td>
fixed bugs on CMV analysis html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/893bea98ae29993d8bbd277e325c55a428e25ad8/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">893bea9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
<td>
Added global comparison in cmv-richness, adonis to NMDS and within-group distances comparison
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/893bea98ae29993d8bbd277e325c55a428e25ad8/docs/316_GeneRichness_vs_CMV.html" target="_blank">893bea9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
<td>
Added global comparison in cmv-richness, adonis to NMDS and within-group distances comparison
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e95b1c9afc4be781b1e956209299a984ed46cb2a/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">e95b1c9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
<td>
compiled CMV file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e95b1c9afc4be781b1e956209299a984ed46cb2a/docs/316_GeneRichness_vs_CMV.html" target="_blank">e95b1c9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
<td>
compiled CMV file
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a87f1a4f145b6f6fddf6bb3955387f5225d3b90d/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">a87f1a4</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
<td>
Added CMV results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1343ce4175f7fe7b98b376c9652e4382e707d3b7/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">1343ce4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-12-13
</td>
<td>
expaded o CMV
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/024b1077514245ecd47e073de16b3f022d37ef4a/analysis/316_GeneRichness_vs_CMV.Rmd" target="_blank">024b107</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-12-01
</td>
<td>
Started CMV and updated metadata
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all(&quot;../WMGSPipeline/&quot;)</code></pre>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre class="r"><code>mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as_tibble(metadata)

mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()
mymre@taxa@metaphlan@phyloseq@sam_data &lt;- phyloseq::sample_data(metadata)</code></pre>
<div id="presence-of-cmv-at-bl-interaction-with-gr" class="section level2">
<h2>presence of CMV at BL interaction with GR</h2>
<p>First we’ll compare those samples at basal in gene richness as whether or not they have CMV presence.</p>
<pre class="r"><code>working_df &lt;- 
  TableList %&gt;% 
  purrr::pluck(&quot;group&quot;) %&gt;% 
  left_join(metadata[,c(&quot;SampleID&quot;,&quot;cmv_bl&quot;)])

bl_df &lt;- 
working_df %&gt;% 
  dplyr::filter(long_var == 0) %&gt;% 
  dplyr::mutate(has_cmv = case_when(
    cmv_bl == 0 ~ &quot;no&quot;,
    cmv_bl &gt; 0 ~ &quot;yes&quot;
  ))

bl_df %&gt;% 
  ggplot(aes(x = has_cmv, y = richness, fill = has_cmv)) +
  geom_violin(alpha = 1, draw_quantiles = c(.25,.5,.75), trim = F) +
  ggpubr::stat_compare_means(method=&quot;wilcox.test&quot;) +
  theme_bw() +
  labs(x = &quot;CMV presence&quot;, fill = &quot;CMV presence&quot;)</code></pre>
<p><img src="figure/316_GeneRichness_vs_CMV.Rmd/CMV_Basal_vs_GR-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-CMV_Basal_vs_GR-1">
Past versions of CMV_Basal_vs_GR-1.png
</button>
</p>
<div id="fig-CMV_Basal_vs_GR-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/893bea98ae29993d8bbd277e325c55a428e25ad8/docs/figure/316_GeneRichness_vs_CMV.Rmd/CMV_Basal_vs_GR-1.png" target="_blank">893bea9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e95b1c9afc4be781b1e956209299a984ed46cb2a/docs/figure/316_GeneRichness_vs_CMV.Rmd/CMV_Basal_vs_GR-1.png" target="_blank">e95b1c9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>bl_df %&gt;% 
  ggplot(aes(x=richness, y = cmv_bl,color=cat_var)) +
  geom_point() +
  geom_smooth(method = &quot;glm&quot;)</code></pre>
<p><img src="figure/316_GeneRichness_vs_CMV.Rmd/CMV_Basal_vs_GR-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-CMV_Basal_vs_GR-2">
Past versions of CMV_Basal_vs_GR-2.png
</button>
</p>
<div id="fig-CMV_Basal_vs_GR-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e95b1c9afc4be781b1e956209299a984ed46cb2a/docs/figure/316_GeneRichness_vs_CMV.Rmd/CMV_Basal_vs_GR-2.png" target="_blank">e95b1c9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor.test(x=bl_df$richness, y=bl_df$cmv_bl)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  bl_df$richness and bl_df$cmv_bl
t = -0.8449, df = 68, p-value = 0.4011
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.3290196  0.1363132
sample estimates:
       cor 
-0.1019255 </code></pre>
<p>Overall, doesn’t seem like there is a difference in Gene Richness between CMV+ and CMV-, although the p.val is close to 0.05.</p>
</div>
<div id="cmv-along-the-study" class="section level2">
<h2>CMV along the study</h2>
<p>We’ll perform a quick view of how CMV evolves along the timepoints and try to find if they correlate with Gene Richness</p>
<p><img src="figure/316_GeneRichness_vs_CMV.Rmd/CMV_tp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-CMV_tp-1">
Past versions of CMV_tp-1.png
</button>
</p>
<div id="fig-CMV_tp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e95b1c9afc4be781b1e956209299a984ed46cb2a/docs/figure/316_GeneRichness_vs_CMV.Rmd/CMV_tp-1.png" target="_blank">e95b1c9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Apparently, there’s barely any CMV viral load outside of baseline. It appears there is some value at week 48 but a closer look shows all entries at this timepoint have a VL of 35, which is the standard result when measurement is below LLOQ (&lt;35). We’ll stick to BL comparisons.</p>
</div>
<div id="beta-diversity-ordination-by-cmv-presence" class="section level1">
<h1>Beta diversity ordination by CMV presence</h1>
<pre class="r"><code>bl_df &lt;- 
  metadata %&gt;% 
  mutate(cmv_bl_c = case_when(cmv_bl == 0 ~ &quot;no&quot;, cmv_bl &gt; 0 ~ &quot;yes&quot;)) %&gt;% 
  dplyr::filter(time_point == 0) 

basal_mre &lt;- 
  metar::filter_samples(mre = mymre, sample_ids = bl_df$SampleID)
basal_mre@metadata@metadata_df &lt;- tibble(bl_df)
basal_mre@metadata@categorical_vals &lt;- tibble(CategoricalVariable = c(&quot;cmv_bl_c&quot;, &quot;time_point&quot;), PaletteName = c(&quot;Accent&quot;, &quot;Set1&quot;))
add_num(basal_mre, c(&quot;HIV_VL&quot;,&quot;CMV_VL&quot;), palette = c(&quot;Set1&quot;,&quot;Set1&quot;))</code></pre>
<pre><code>MetarSet-class experimental-level object 
  [MetadataSet     ]: 79 samples and 9 selected variables 
  [DiversitySet    ]: 
     ✔ igc: 3037 observations 
     ✖ virgo: 0 observations 
     ✖ dada2: 0 observations 
  [TaxaSet         ]: 
     ✔ metaphlan: phyloseq object with 354 taxa and 79 samples 
     ✖ kraken: no phyloseq object available 
     ✖ bracken: no phyloseq object available 
     ✔ motus: phyloseq object with 1176 taxa and 79 samples 
     ✖ dada2: no phyloseq object available 
  [GeneFunctionSet ]: 
     ✔ humann: 6235 kegg_id, 0 cog_id and 0 eggnog_id 
     ✔ igc: 6264 kegg_id, 0 cog_id and 0 eggnog_id 
     ✖ virgo: no functional data aviable 
     ✖ fmap: no functional data aviable </code></pre>
<pre class="r"><code>phy &lt;- get_phyloseq(basal_mre, type = &quot;metaphlan&quot;)
phy@sam_data &lt;- phyloseq::sample_data(bl_df)</code></pre>
<pre class="r"><code>getAbundances&lt;-function(phyloseq, level){
  
  collapsedPhyloseq&lt;- phyloseq::tax_glom(phyloseq, level)
  
  OtuTable&lt;- phyloseq::otu_table(collapsedPhyloseq) %&gt;%
    magrittr::set_rownames(as.character(phyloseq::tax_table(collapsedPhyloseq)[,level])) %&gt;%
    as.data.frame()
  return(OtuTable)
  
}



myAbundanceDF&lt;- getAbundances(phy, &quot;Species&quot;)

myBrayDistMat &lt;-vegdist(wisconsin(sqrt(t(myAbundanceDF))))

myNMDS &lt;- myAbundanceDF %&gt;%
  t() %&gt;% 
    as.data.frame() %&gt;%
    # vegan::vegdist(., method = &quot;bray&quot;) %&gt;%
 
    metaMDS(.,trymax=800,  distance = &quot;bray&quot;,
            k = 2) %&gt;%
    pluck(&quot;points&quot;) %&gt;%
    as.data.frame() </code></pre>
<pre class="r"><code>cat_df &lt;- tibble(CategoricalVariable = c(&quot;cmv_bl_c&quot;,&quot;time_point&quot;), PaletteName = c(&quot;Accent&quot;,&quot;Set1&quot;))


cat_var &lt;- &quot;cmv_bl_c&quot;
link_var &lt;- &quot;record_id&quot;
id_var &lt;- &quot;SampleID&quot;


myNMDS_DF &lt;- myNMDS %&gt;%
  rownames_to_column(var = id_var) %&gt;%
  dplyr::full_join(., bl_df, by =id_var) %&gt;%
  dplyr::select(SampleID = id_var,
                NMDS1 = MDS1,
                NMDS2 = MDS2,
                linkVar = !!sym(link_var),
                catVar = cmv_bl_c,
                shape = CD4diff_48)


  myXmin&lt;-min(myNMDS$MDS1)
  # +(max(dim_NMDS$points[,1])-min(dim_NMDS$points[,1]))/20
  myYmin&lt;-max(myNMDS$MDS2)-(max(myNMDS$MDS1)-min(myNMDS$MDS1))




myAdonis&lt;- adonis2(myBrayDistMat~ catVar, data = myNMDS_DF) %&gt;%
  slice(1L) %&gt;%
  as.data.frame() %&gt;%
  magrittr::set_rownames(cat_var)


myNMDSplot&lt;-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point(aes(color=catVar, shape = as.factor(shape)), size = 3) +
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour=&quot;black&quot;),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face=&quot;italic&quot;),
          legend.title=element_text(size=11))+
        stat_ellipse(geom=&quot;polygon&quot;,alpha=0.15,color=&quot;black&quot;,aes(fill=catVar),level=0.95)+
    # ggtitle(expression(atop(&quot;Species composition&quot;,atop(italic(&quot;all samples&quot;))))) +
    theme(plot.title=element_text(lineheight=1,face=&quot;bold&quot;,size=13))+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var) +
  scale_color_brewer(palette = &quot;Accent&quot;) +
  scale_fill_brewer(palette = &quot;Accent&quot;) + 
  annotation_custom(gridExtra::tableGrob(round(myAdonis,3),
                                         theme = ttheme_default(base_size = 10)),ymin = 1)




myNMDSplot</code></pre>
<p><img src="figure/316_GeneRichness_vs_CMV.Rmd/NMDS_Group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-NMDS_Group-1">
Past versions of NMDS_Group-1.png
</button>
</p>
<div id="fig-NMDS_Group-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c1a9770c3ca2f9505837e8532b1cbd108d865a0c/docs/figure/316_GeneRichness_vs_CMV.Rmd/NMDS_Group-1.png" target="_blank">c1a9770</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3/docs/figure/316_GeneRichness_vs_CMV.Rmd/NMDS_Group-1.png" target="_blank">b9a23b3</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>While it seems the samples with detectable CMV are more disprerse along the ordination, Adonis seems unable to find a significance. It may be due lo low sample count. Interestingly enough, there seems to be a correlation between CMV presence at basal and CD4 recovery. Most shockingly, presence of CMV seems to be correlated with a higher chance of recovering CD4 in more than 150 counts/ml. It could however mean that patients with CMV have a higher CD4 activation as they are trying to contain the virus.</p>
<pre class="r"><code>t_df &lt;- 
  myNMDS_DF %&gt;% 
  dplyr::select(shape, catVar) %&gt;% 
  filter(!is.na(shape)) %&gt;% 
  mutate(across(everything(), as.factor)) %&gt;% 
  group_by(shape,catVar) %&gt;% 
  summarise(n=n()) %&gt;% 
  spread(shape, n) %&gt;% 
  column_to_rownames(&quot;catVar&quot;) 

test_chi &lt;- 
  t_df %&gt;% 
  as.matrix() %&gt;%
  chisq_test(x = .)
t_df</code></pre>
<pre><code>    150_H 150_L
no     10    13
yes    32     4</code></pre>
<pre class="r"><code>test_chi</code></pre>
<pre><code># A tibble: 1 × 6
      n statistic        p    df method          p.signif
* &lt;int&gt;     &lt;dbl&gt;    &lt;dbl&gt; &lt;int&gt; &lt;chr&gt;           &lt;chr&gt;   
1    59      12.0 0.000537     1 Chi-square test ***     </code></pre>
<pre class="r"><code>cmv_pos &lt;- bl_df %&gt;% 
  dplyr::filter(cmv_bl_c == &quot;yes&quot;) %&gt;% 
  dplyr::pull(SampleID)

cmv_neg &lt;- bl_df %&gt;% 
  dplyr::filter(cmv_bl_c == &quot;no&quot;) %&gt;% 
  dplyr::pull(SampleID)


upper_distmat &lt;- as.matrix(myBrayDistMat)
upper_distmat[lower.tri(upper_distmat)] &lt;- NA


distance_df &lt;- 
upper_distmat %&gt;% 
  as.matrix() %&gt;% 
  as.data.frame() %&gt;% 
  # dplyr::filter(!rownames(.)==colnames(.)) %&gt;%
  rownames_to_column(&quot;Sample1&quot;) %&gt;% 
  pivot_longer(cols = !Sample1, names_to = &quot;Sample2&quot;, values_to = &quot;dist&quot;) %&gt;% 

    mutate(cmv = case_when(
      Sample1 %in% cmv_pos &amp; Sample2 %in% cmv_pos ~ &quot;yes&quot;,
      Sample1 %in% cmv_neg &amp; Sample2 %in% cmv_neg ~ &quot;no&quot;
    )) %&gt;% 
    dplyr::filter(Sample1 != Sample2,
                  !is.na(cmv),
                  !is.na(dist)) 


distance_df %&gt;% 

  group_by(cmv) %&gt;% 
  count()</code></pre>
<pre><code># A tibble: 2 × 2
# Groups:   cmv [2]
  cmv       n
  &lt;chr&gt; &lt;int&gt;
1 no      406
2 yes    1225</code></pre>
<pre class="r"><code>distance_df %&gt;% 
  sample_n_by(cmv, size = 406) %&gt;%
  ggstatsplot::ggbetweenstats(.,
                              x = &quot;cmv&quot;,
                              y = &quot;dist&quot;,
                              type = &quot;np&quot;) +
  labs(y = &quot;pairwise within-group distances&quot;)</code></pre>
<p><img src="figure/316_GeneRichness_vs_CMV.Rmd/intra_group_distances-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-intra_group_distances-1">
Past versions of intra_group_distances-1.png
</button>
</p>
<div id="fig-intra_group_distances-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c1a9770c3ca2f9505837e8532b1cbd108d865a0c/docs/figure/316_GeneRichness_vs_CMV.Rmd/intra_group_distances-1.png" target="_blank">c1a9770</a>
</td>
<td>
cblazquez
</td>
<td>
2023-01-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3/docs/figure/316_GeneRichness_vs_CMV.Rmd/intra_group_distances-1.png" target="_blank">b9a23b3</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/893bea98ae29993d8bbd277e325c55a428e25ad8/docs/figure/316_GeneRichness_vs_CMV.Rmd/intra_group_distances-1.png" target="_blank">893bea9</a>
</td>
<td>
cblazquez
</td>
<td>
2022-12-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>On the other hand though, A non-parametric mean comparison on the within-group distances shows a significant difference between both groups. Even when the CMV+ group is randomly subsampled to match the N of CMV-.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5        testthat_3.1.4     gridExtra_2.3      vegan_2.6-2       
 [5] lattice_0.20-45    permute_0.9-7      RColorBrewer_1.1-3 ggpubr_0.4.0      
 [9] rstatix_0.7.0      forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9       
[13] purrr_0.3.4        readr_2.1.2        tidyr_1.2.0        tibble_3.1.7      
[17] ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             tidyselect_1.1.2       lme4_1.1-29           
  [4] htmlwidgets_1.5.4      grid_4.0.5             devtools_2.4.3        
  [7] aws.signature_0.6.0    munsell_0.5.0          effectsize_0.7.0      
 [10] codetools_0.2-18       future_1.26.1          withr_2.5.0           
 [13] colorspace_2.0-3       Biobase_2.50.0         phyloseq_1.34.0       
 [16] highr_0.9              logger_0.2.2           knitr_1.39            
 [19] rstudioapi_0.13        stats4_4.0.5           ggsignif_0.6.3.9000   
 [22] listenv_0.8.0          labeling_0.4.2         emmeans_1.7.4-1       
 [25] git2r_0.30.1           polyclip_1.10-0        datawizard_0.4.1      
 [28] bit64_4.0.5            farver_2.1.0           rhdf5_2.34.0          
 [31] rprojroot_2.0.3        TH.data_1.1-1          coda_0.19-4           
 [34] parallelly_1.32.0      vctrs_0.4.1            generics_0.1.2        
 [37] xfun_0.31              R6_2.5.1               arm_1.12-2            
 [40] rhdf5filters_1.2.1     cachem_1.0.6           assertthat_0.2.1      
 [43] promises_1.2.0.1       networkD3_0.4          scales_1.2.0          
 [46] vroom_1.5.7            multcomp_1.4-19        merTools_0.5.2        
 [49] gtable_0.3.0           globals_0.15.0         processx_3.6.1        
 [52] sandwich_3.0-2         workflowr_1.7.0        rlang_1.0.2           
 [55] zeallot_0.1.0          splines_4.0.5          prismatic_1.1.0       
 [58] broom_0.8.0            yaml_2.3.5             reshape2_1.4.4        
 [61] abind_1.4-5            modelr_0.1.8           backports_1.4.1       
 [64] httpuv_1.6.5           tools_4.0.5            usethis_2.1.6         
 [67] logging_0.10-108       ellipsis_0.3.2         jquerylib_0.1.4       
 [70] biomformat_1.18.0      selbal_0.1.0           BiocGenerics_0.36.1   
 [73] sessioninfo_1.2.2      Rcpp_1.0.8.3           plyr_1.8.7            
 [76] base64enc_0.1-3        zlibbioc_1.36.0        ps_1.7.1              
 [79] prettyunits_1.1.1      correlation_0.8.1      zoo_1.8-10            
 [82] S4Vectors_0.28.1       ggrepel_0.9.1          haven_2.5.0           
 [85] cluster_2.1.3          fs_1.5.2               here_1.0.1            
 [88] furrr_0.3.0            DataExplorer_0.8.2     magrittr_2.0.3        
 [91] data.table_1.14.2      reprex_2.0.1           mvtnorm_1.1-3         
 [94] whisker_0.4            pkgload_1.2.4          hms_1.1.1             
 [97] patchwork_1.1.1        mime_0.12              evaluate_0.15         
[100] xtable_1.8-4           broom.mixed_0.2.9.4    readxl_1.4.0          
[103] IRanges_2.24.1         compiler_4.0.5         ggstatsplot_0.9.3     
[106] crayon_1.5.1           minqa_1.2.4            htmltools_0.5.2       
[109] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[112] lubridate_1.8.0        aws.s3_0.3.21          DBI_1.1.3             
[115] tweenr_1.0.2           dbplyr_2.2.0           MASS_7.3-57           
[118] boot_1.3-28            Matrix_1.4-1           ade4_1.7-19           
[121] car_3.1-0              wesanderson_0.3.6      brio_1.1.3            
[124] cli_3.3.0              insight_0.17.1         parallel_4.0.5        
[127] igraph_1.3.2           pkgconfig_2.0.3        statsExpressions_1.3.2
[130] paletteer_1.4.0        xml2_1.3.2             foreach_1.5.2         
[133] bslib_0.3.1            blme_1.0-5             multtest_2.46.0       
[136] XVector_0.30.0         estimability_1.3       rvest_1.0.2           
[139] callr_3.7.0            digest_0.6.29          parameters_0.18.1     
[142] Biostrings_2.58.0      rmarkdown_2.14         cellranger_1.1.0      
[145] curl_4.3.2             shiny_1.7.1            nloptr_2.0.3          
[148] lifecycle_1.0.1        nlme_3.1-158           jsonlite_1.8.0        
[151] Rhdf5lib_1.12.1        carData_3.0-5          desc_1.4.1            
[154] fansi_1.0.3            pillar_1.7.0           fastmap_1.1.0         
[157] httr_1.4.2             pkgbuild_1.3.1         survival_3.3-1        
[160] glue_1.6.2             remotes_2.4.2          bayestestR_0.12.1     
[163] iterators_1.0.14       bit_4.0.4              performance_0.9.1     
[166] ggforce_0.3.3          stringi_1.7.6          sass_0.4.1            
[169] rematch2_2.1.2         memoise_2.0.1          ape_5.6-2             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
