<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />

<meta name="date" content="2022-12-15" />

<title>317_GeneRichness_Inflammation</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
</li>
<li>
  <a href="317_GeneRichness_vs_Inflammation.html">Gene richness vs inflammation</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">317_GeneRichness_Inflammation</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">2022-12-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treeda33aea4103f1864427112e7cf78ca056bb0097ftargetblankda33aeaa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MicrobialGenomics/advanz4/tree/da33aea4103f1864427112e7cf78ca056bb0097f" target="_blank">da33aea</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treeda33aea4103f1864427112e7cf78ca056bb0097ftargetblankda33aeaa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MicrobialGenomics/advanz4/tree/da33aea4103f1864427112e7cf78ca056bb0097f" target="_blank">da33aea</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Metadata/MetadataManagement/Old_Metadata/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/010_Basic_Report_cache/
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    data/filt_mre.rds
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   advanz4.Rproj
    Modified:   analysis/317_GeneRichness_vs_Inflammation.Rmd
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/317_GeneRichness_vs_Inflammation.Rmd</code>) and HTML
(<code>docs/317_GeneRichness_vs_Inflammation.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ad0a04c40c748f71af516303f6d7dc754e53bd44/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">ad0a04c</a>
</td>
<td>
cblazquez
</td>
<td>
2023-02-16
</td>
<td>
Started correlation of GR vs Inflammation
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="r-markdown" class="section level2">
<h2>R Markdown</h2>
<p>This is an R Markdown document. Markdown is a simple formatting
syntax for authoring HTML, PDF, and MS Word documents. For more details
on using R Markdown see <a href="http://rmarkdown.rstudio.com"
class="uri">http://rmarkdown.rstudio.com</a>.</p>
<p>When you click the <strong>Knit</strong> button a document will be
generated that includes both content as well as the output of any
embedded R code chunks within the document. You can embed an R code
chunk like this:</p>
<pre class="r"><code># mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;, region = &quot;eu-west-1&quot;)
# metadata &lt;-get_meta(mymre)
mymre &lt;- here::here(&quot;data&quot;,&quot;filt_mre.rds&quot;) %&gt;%
  readRDS(.)
metadata &lt;- 
  here::here(&quot;Metadata&quot;) %&gt;% 
  file.info(list.files(., full.names = T)) %&gt;%
  rownames() %&gt;% 
  magrittr::extract(str_detect(.,&quot;clean_metadata&quot;)) %&gt;% 
  sort(decreasing = T) %&gt;% 
  magrittr::extract(1L) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

id_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all(&quot;../WMGSPipeline&quot;)
mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- tibble(metadata)

mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
# 
# mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
#   phyloseq::sample_data(.)


mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;)</code></pre>
</div>
<div id="basal-exploration-of-inflammation-markers"
class="section level1">
<h1>Basal exploration of inflammation markers</h1>
<p>Perform basic distributin analysis of each marker at baseline, test
for imbalances between groups which may be creating a bias.</p>
<pre class="r"><code>library(compareGroups)

metadata %&gt;% 
  dplyr::select(record_id, time_point, group, num_var) %&gt;% 
  dplyr::filter(time_point == 0) %&gt;% 
  compareGroups::compareGroups(formula(paste(&quot;group&quot;, &quot;~&quot;, paste(num_var, collapse = &quot; + &quot;), sep = &quot; &quot;)), data = .,) %&gt;% 
  compareGroups::createTable(.,show.n = T)</code></pre>
<pre><code>
--------Summary descriptives table by &#39;group&#39;---------

_____________________________________________________ 
               DRV/r          DTG        p.overall N  
               N=39           N=40                    
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
CD4         30.9 (26.5)   51.5 (43.0)      0.012   79 
CD8          606 (486)     670 (403)       0.529   79 
CD8_CD38_DR 42.3 (14.9)   46.0 (17.4)      0.311   79 
CRP         0.66 (0.97)   1.86 (4.17)      0.082   79 
IL6         25.2 (31.2)   37.6 (48.6)      0.261   55 
TNFa        17.4 (10.7)   20.7 (12.4)      0.203   79 
ADV         3249 (7784) 703266 (2325877)   0.342   23 
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<p>No significant changes could be observed in any inflammation marker,
except for CD4 counts, which is still a farily small change.</p>
</div>
<div id="correlation-between-gr-and-markers" class="section level1">
<h1>Correlation between GR and markers</h1>
<p>Create a corr table and heatmap. This will be a good exploratory
analysis to see which inflammation markers change with the microbiome.
This is a longitudinal study, which means time may be an important
variable we need to account with the models. A good exploratory first
test will be to correlate markers with gene richness at basal. Then a
correlation of deltas may give information on those who grow more
depending on gene richness.</p>
<pre class="r"><code>num_var %&gt;%
  purrr::set_names() %&gt;%
  purrr::map_dfr(function(nv) {
    print(nv)
    igc_df %&gt;%
      dplyr::filter(long_var == 0) %&gt;%
      dplyr::select(SampleID, cat_var, nv, richness) %&gt;% 
      dplyr::mutate(!!sym(nv) := as.numeric(!!sym(nv))) %&gt;% 
      cor_test(&quot;richness&quot;, nv,  method = &quot;spearman&quot;)
  })</code></pre>
<pre><code>[1] &quot;CD4&quot;
[1] &quot;CD8&quot;
[1] &quot;CD8_CD38_DR&quot;
[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;
[1] &quot;ADV&quot;</code></pre>
<pre><code># A tibble: 8 x 6
  var1     var2           cor statistic     p method  
  &lt;chr&gt;    &lt;chr&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   
1 richness CD4          0.17     47338. 0.155 Spearman
2 richness CD8          0.07     53166. 0.566 Spearman
3 richness CD8_CD38_DR -0.047    59815. 0.702 Spearman
4 richness CRP         -0.13     64681. 0.277 Spearman
5 richness IL6          0.1      16535. 0.488 Spearman
6 richness TNFa        -0.029    58827. 0.81  Spearman
7 richness sCD14       -0.11     63384  0.368 Spearman
8 richness ADV         -0.17      2072  0.448 Spearman</code></pre>
<p>Apparently, no correlation could be found between GR and inflammation
markers at basal. Let’s look for the longitudinal approach.</p>
<pre class="r"><code>num_var %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    print(nv)
    igc_df %&gt;% 
      group_by(link_var) %&gt;% 
      dplyr::filter(long_var %in% c(0,48)) %&gt;% 
      dplyr::filter(n() == 2) %&gt;% 
      dplyr::mutate(num_var = as.numeric(!!sym(nv))) %&gt;% 
      dplyr::mutate(delta_nv := num_var[long_var == 48]/num_var[long_var == 0],
                    delta_gr = richness[long_var == 48]/richness[long_var == 0]) %&gt;% 

      filter(!is.na(num_var)) %&gt;% 
      group_by(cat_var,.drop = T) %&gt;%
      # ungroup() %&gt;% 
      rstatix::wilcox_test(num_var ~ long_var)
  })</code></pre>
<pre><code>[1] &quot;CD4&quot;
[1] &quot;CD8&quot;
[1] &quot;CD8_CD38_DR&quot;
[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;
[1] &quot;ADV&quot;</code></pre>
<pre><code>$CD4
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic        p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;    &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    30        21 2.35e-10
2 DTG     num_var 0      48        30    29        15 2.01e-10

$CD8
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic       p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    30       272 0.00801
2 DTG     num_var 0      48        30    29       312 0.0629 

$CD8_CD38_DR
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic           p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;       &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    30      756. 0.0000065  
2 DTG     num_var 0      48        30    29      768  0.000000462

$CRP
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic      p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    22      380. 0.354 
2 DTG     num_var 0      48        30    22      463  0.0141

$IL6
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic          p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;      &lt;dbl&gt;
1 DRV/r   num_var 0      48        18    30       475 0.0000061 
2 DTG     num_var 0      48        22    30       569 0.00000485

$TNFa
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic           p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;       &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    30       715 0.0000866  
2 DTG     num_var 0      48        30    30       790 0.000000487

$sCD14
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic          p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;      &lt;dbl&gt;
1 DRV/r   num_var 0      48        30    30       706 0.000158  
2 DTG     num_var 0      48        30    30       768 0.00000062

$ADV
# A tibble: 2 x 8
  cat_var .y.     group1 group2    n1    n2 statistic     p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt;
1 DRV/r   num_var 0      48         9    12        64 0.508
2 DTG     num_var 0      48         8    10        46 0.633</code></pre>
<pre class="r"><code>res_list &lt;- 
num_var[!num_var %in% c(&quot;CD4&quot;, &quot;CD8&quot;,&quot;CD8_CD38_DR&quot;, &quot;ADV&quot;)] %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    print(nv)
    igc_df %&gt;% 
      dplyr::mutate(num_var := as.numeric(!!sym(nv))) %&gt;% 
      filter(!is.na(num_var)) %&gt;% 
    get_comp_boxplots(dat = .,cat_var = &quot;cat_var&quot;,num_var = &quot;num_var&quot;, long_var = &quot;long_var&quot;,link_var = &quot;link_var&quot;,y_trans = &quot;sqrt&quot;, pal = &quot;Accent&quot;) %&gt;% 
      purrr::pluck(&quot;plot&quot;) +
      labs(y = nv, fill = &quot;group&quot;)
    
  })</code></pre>
<pre><code>[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;</code></pre>
<pre class="r"><code>ggarrange(plotlist = res_list, nrow = 2, ncol = 2)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There are significant differences in both CRP and sCD14, which were
lower in the DTG group. CRP became significant at week 96 only, while
sCD14 became significant after 48 weeks, and kept doing so at week
96</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] compareGroups_4.5.1 ggpubr_0.4.0        rstatix_0.7.1      
 [4] metar_0.1.5         RColorBrewer_1.1-3  forcats_0.5.2      
 [7] stringr_1.4.0       dplyr_1.0.8         purrr_0.3.4        
[10] readr_2.0.1         tidyr_1.2.0         tibble_3.1.6       
[13] ggplot2_3.3.5       tidyverse_1.3.1    

loaded via a namespace (and not attached):
  [1] uuid_1.1-0          readxl_1.3.1        backports_1.4.1    
  [4] blme_1.0-5          workflowr_1.7.0     systemfonts_1.0.4  
  [7] plyr_1.8.8          igraph_1.2.9        splines_4.0.2      
 [10] listenv_0.9.0       digest_0.6.29       foreach_1.5.2      
 [13] htmltools_0.5.2     fansi_1.0.3         Rsolnp_1.16        
 [16] magrittr_2.0.3      cluster_2.1.0       tzdb_0.3.0         
 [19] globals_0.16.2      Biostrings_2.58.0   modelr_0.1.8       
 [22] officer_0.4.4       svglite_2.1.0       colorspace_2.0-3   
 [25] rvest_1.0.1         haven_2.3.1         xfun_0.30          
 [28] crayon_1.5.2        jsonlite_1.8.0      lme4_1.1-29        
 [31] survival_3.1-12     iterators_1.0.14    ape_5.6-2          
 [34] glue_1.6.2          kableExtra_1.3.4    gtable_0.3.1       
 [37] zlibbioc_1.36.0     XVector_0.30.0      webshot_0.5.2      
 [40] phyloseq_1.34.0     car_3.1-1           Rhdf5lib_1.12.1    
 [43] BiocGenerics_0.36.1 abind_1.4-5         scales_1.2.1       
 [46] mvtnorm_1.1-3       DBI_1.1.3           Rcpp_1.0.8.3       
 [49] viridisLite_0.4.1   xtable_1.8-4        merTools_0.5.2     
 [52] stats4_4.0.2        truncnorm_1.0-8     htmlwidgets_1.5.4  
 [55] httr_1.4.4          ellipsis_0.3.2      mice_3.13.0        
 [58] farver_2.1.1        DataExplorer_0.8.2  pkgconfig_2.0.3    
 [61] dbplyr_2.1.1        utf8_1.2.2          here_1.0.1         
 [64] labeling_0.4.2      tidyselect_1.1.2    rlang_1.0.2        
 [67] reshape2_1.4.4      later_1.3.0         munsell_0.5.0      
 [70] cellranger_1.1.0    tools_4.0.2         cli_3.2.0          
 [73] generics_0.1.3      selbal_0.1.0        ade4_1.7-20        
 [76] broom_1.0.2         evaluate_0.19       biomformat_1.18.0  
 [79] fastmap_1.1.0       arm_1.12-2          yaml_2.3.6         
 [82] knitr_1.38          fs_1.5.2            zip_2.2.2          
 [85] future_1.30.0       nlme_3.1-148        whisker_0.4.1      
 [88] mime_0.12           xml2_1.3.2          compiler_4.0.2     
 [91] rstudioapi_0.13     ggsignif_0.6.4      reprex_2.0.1       
 [94] HardyWeinberg_1.7.2 stringi_1.7.6       highr_0.10         
 [97] gdtools_0.2.4       logger_0.2.2        lattice_0.20-41    
[100] Matrix_1.5-3        nloptr_2.0.0        vegan_2.6-4        
[103] permute_0.9-7       multtest_2.46.0     vctrs_0.4.1        
[106] pillar_1.8.1        lifecycle_1.0.1     networkD3_0.4      
[109] rhdf5filters_1.2.1  furrr_0.3.1         jquerylib_0.1.4    
[112] cowplot_1.1.1       flextable_0.6.8     data.table_1.14.6  
[115] httpuv_1.6.7        R6_2.5.1            promises_1.2.0.1   
[118] gridExtra_2.3       writexl_1.4.1       IRanges_2.24.1     
[121] parallelly_1.33.0   codetools_0.2-16    boot_1.3-25        
[124] MASS_7.3-51.6       assertthat_0.2.1    chron_2.3-58       
[127] rhdf5_2.34.0        rprojroot_2.0.3     withr_2.5.0        
[130] S4Vectors_0.28.1    broom.mixed_0.2.9.4 mgcv_1.8-31        
[133] parallel_4.0.2      hms_1.1.2           grid_4.0.2         
[136] coda_0.19-4         minqa_1.2.5         rmarkdown_2.19     
[139] carData_3.0-5       git2r_0.30.1        base64enc_0.1-3    
[142] Biobase_2.50.0      shiny_1.7.1         lubridate_1.8.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
