<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />

<meta name="date" content="2022-12-15" />

<title>317_GeneRichness_Inflammation</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Diversity overall
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="310_IGC_analysis.html">Gene richness</a>
    </li>
    <li>
      <a href="311_GR_vs_BMI.html">BMI vs diversity</a>
    </li>
    <li>
      <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
    </li>
    <li>
      <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
    </li>
    <li>
      <a href="317_GeneRichness_vs_Inflammation.html">Gene richness vs inflammation</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">beta diversity</li>
    <li>
      <a href="320_Beta_Diversity.html">Beta Diversity</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Baseline Microbiome
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
    </li>
    <li>
      <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Differential Abundance
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
    </li>
    <li>
      <a href="420_Taxonomy_DA.html">Differential Abundance</a>
    </li>
    <li>
      <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">317_GeneRichness_Inflammation</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">2022-12-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree85710c1a5b2a8c22a62030b66ea702779ecda1b7targetblank85710c1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/85710c1a5b2a8c22a62030b66ea702779ecda1b7" target="_blank">85710c1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree85710c1a5b2a8c22a62030b66ea702779ecda1b7targetblank85710c1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/85710c1a5b2a8c22a62030b66ea702779ecda1b7" target="_blank">85710c1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/505_Adenovirus_Analysis_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  output/BMI_vs_GR.pdf

Unstaged changes:
    Modified:   analysis/311_GR_vs_BMI.Rmd
    Modified:   analysis/320_Beta_Diversity.Rmd
    Modified:   code/group_comparisons.R
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/317_GeneRichness_vs_Inflammation.Rmd</code>) and HTML (<code>docs/317_GeneRichness_vs_Inflammation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/209eb6d9b734ce1208bdbfa7fe349b16c46ad71d/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">209eb6d</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-28
</td>
<td>
Added all info for BMI and updated alluvial plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
<td>
Updated website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/72f15339f90d852482e264aa42757ad925bf6e1d/docs/317_GeneRichness_vs_Inflammation.html" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
<td>
Updated website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0aef81987ff281deec8894018695691895f80a3b/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">0aef819</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
<td>
fixed bug on GR_inflammation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/efdac31cf4332143b561f3f956761d9b8895d748/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">efdac31</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-04-13
</td>
<td>
Reestructured website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9e2fd42198f450579592f1e076394255d450567/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
<td>
Fixed rendering bugs on 310.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/b9e2fd42198f450579592f1e076394255d450567/docs/317_GeneRichness_vs_Inflammation.html" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
<td>
Fixed rendering bugs on 310.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/54b8abb5590d61cf23d99771d7e72b458cdaa83c/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">54b8abb</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
<td>
fixed problem with sCD14
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/54b8abb5590d61cf23d99771d7e72b458cdaa83c/docs/317_GeneRichness_vs_Inflammation.html" target="_blank">54b8abb</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
<td>
fixed problem with sCD14
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
<td>
Added evolution of inflammation markers per GR group
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/317_GeneRichness_vs_Inflammation.html" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
<td>
Added evolution of inflammation markers per GR group
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1710d26f13b98dd1b41d35034e5f1c325cf0830/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">e1710d2</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-07
</td>
<td>
started analysis n GR vs BMI
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e1710d26f13b98dd1b41d35034e5f1c325cf0830/docs/317_GeneRichness_vs_Inflammation.html" target="_blank">e1710d2</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-07
</td>
<td>
started analysis n GR vs BMI
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ad0a04c40c748f71af516303f6d7dc754e53bd44/analysis/317_GeneRichness_vs_Inflammation.Rmd" target="_blank">ad0a04c</a>
</td>
<td>
cblazquez
</td>
<td>
2023-02-16
</td>
<td>
Started correlation of GR vs Inflammation
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;, region = &quot;eu-west-1&quot;)
# metadata &lt;-get_meta(mymre)
mymre &lt;- here::here(&quot;data&quot;,&quot;filt_mre.rds&quot;) %&gt;%
  readRDS(.)
metadata &lt;- 
  here::here(&quot;Metadata&quot;) %&gt;% 
  file.info(list.files(., full.names = T)) %&gt;%
  rownames() %&gt;% 
  magrittr::extract(str_detect(.,&quot;clean_metadata&quot;)) %&gt;% 
  sort(decreasing = T) %&gt;% 
  magrittr::extract(1L) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

id_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all(&quot;../WMGSPipeline&quot;)
mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- tibble(metadata)

mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
# 
# mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
#   phyloseq::sample_data(.)


mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;)</code></pre>
<div id="basal-exploration-of-inflammation-markers" class="section level1">
<h1>Basal exploration of inflammation markers</h1>
<p>Perform basic distributin analysis of each marker at baseline, test for imbalances between groups which may be creating a bias.</p>
<pre class="r"><code>library(compareGroups)

metadata %&gt;% 
  dplyr::select(record_id, time_point, group, num_var) %&gt;% 
  mutate(sCD14 = as.numeric(sCD14)) %&gt;% 
  dplyr::filter(time_point == 0) %&gt;% 
  compareGroups::compareGroups(formula(paste(&quot;group&quot;, &quot;~&quot;, paste(num_var, collapse = &quot; + &quot;), sep = &quot; &quot;)), data = .) %&gt;% 
  compareGroups::createTable(.,show.n = T)</code></pre>
<pre><code>
--------Summary descriptives table by &#39;group&#39;---------

_____________________________________________________ 
               bDRV           DTG        p.overall N  
               N=39           N=40                    
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
CD4         30.9 (26.5)   51.5 (43.0)      0.012   79 
CD8          606 (486)     670 (403)       0.529   79 
CD4_CD38_DR 30.4 (19.4)   29.1 (17.7)      0.750   79 
CD8_CD38_DR 42.3 (14.9)   46.0 (17.4)      0.311   79 
CRP         0.66 (0.97)   1.86 (4.17)      0.082   79 
IL6         25.2 (31.2)   37.6 (48.6)      0.261   55 
TNFa        17.4 (10.7)   20.7 (12.4)      0.203   79 
sCD14       3096 (1251)   3012 (1294)      0.770   79 
ADV         3249 (7784) 703266 (2325877)   0.342   23 
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<p>No significant changes could be observed in any inflammation marker, except for CD4 counts, which is still a farily small change.</p>
</div>
<div id="gene-richness-correlations" class="section level1">
<h1>Gene richness correlations</h1>
<p>Now to the main point of this document. The first thing to do is to assess any correlation between initial GR and inflammation markers.</p>
<p>First let’s look how richness distributes across time, this will allow us to choose the best correlation and possibly stratify the population between low and high gene richness.</p>
<pre class="r"><code>peaks_df &lt;- 
c(0,24,48,96) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(function(tp){
    igc_df %&gt;% 
      dplyr::filter(!is.na(richness), long_var == tp) %&gt;% 
      dplyr::pull(richness) %&gt;% 
      density() %&gt;% 
      peak_finder(., goal = &quot;min&quot;) %&gt;% 
      dplyr::mutate(long_var = tp)
  })


igc_df %&gt;% 
  dplyr::filter(!is.na(richness)) %&gt;% 
  ggplot(aes(x = richness, color = as.factor(long_var))) +
  geom_density(lwd = 2) +
  scale_color_brewer(palette = &quot;YlOrRd&quot;) +
  geom_vline(data = peaks_df, aes(xintercept = x, color = as.factor(long_var)), lty = 2) +
  geom_text(data = peaks_df, aes(x= x, y = y*.9, color = as.factor(long_var), label = round(x,3))) +
  theme_bw() + 
  labs(color = &quot;week&quot;)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/plot_distributions-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_distributions-1">
Past versions of plot_distributions-1.png
</button>
</p>
<div id="fig-plot_distributions-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/plot_distributions-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/plot_distributions-1.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It seems richness is bimodal, with both peaks becoming increasingly separate as the study goes on. There seems to be no shift on the overall distribution. Instead, patients with really low microbiome counts tend to move towards one of the the two peaks (LGC and HGC) to check wether patients are inherently prone to move towards either peak, we could check differences in BL GR between LGC and HGC.</p>
<pre class="r"><code>peak &lt;-
  igc_df %&gt;%
  dplyr::filter(!is.na(richness), long_var == 96) %&gt;%
  dplyr::pull(richness) %&gt;%
  density() %&gt;%
  peak_finder(., goal = &quot;min&quot;) %&gt;%
  dplyr::mutate(long_var = 96) %&gt;%
  dplyr::filter(x == max(x)) %&gt;% 
  dplyr::pull(x)

igc_df %&gt;% 
  group_by(link_var) %&gt;% 
  dplyr::filter(long_var %in% c(0,96)) %&gt;% 
  dplyr::filter(n() == 2) %&gt;% 
  dplyr::mutate(gr0 = richness[long_var == 0],
                gr96 = richness[long_var == 96]) %&gt;%
  dplyr::select(SampleID,link_var, cat_var, gr0, gr96) %&gt;% 
  unique() %&gt;% 
  ungroup() %&gt;% 
  dplyr::filter(!is.na(gr0), !is.na(gr96)) %&gt;% 
  cor_test(., gr96, gr0, method = &quot;spearman&quot;)</code></pre>
<pre><code># A tibble: 1 × 6
  var1  var2    cor statistic            p method  
  &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt;     &lt;dbl&gt;        &lt;dbl&gt; &lt;chr&gt;   
1 gr96  gr0    0.53    73319. 0.0000000166 Spearman</code></pre>
<pre class="r"><code>igc_df &lt;- 
igc_df %&gt;% 
  group_by(link_var) %&gt;% 
  dplyr::filter(long_var %in% c(0,96)) %&gt;% 
  dplyr::filter(n() == 2) %&gt;% 
  dplyr::mutate(gr0 = richness[long_var == 0],
                gr96 = richness[long_var == 96]) %&gt;%
  dplyr::select(link_var, gr0, gr96) %&gt;% 
  unique() %&gt;% 
  dplyr::filter(!is.na(gr0), !is.na(gr96)) %&gt;% 
  dplyr::mutate( gr_group = case_when(gr96 &gt;= peak ~ &quot;HGC&quot;,
                                      gr96 &lt; peak ~ &quot;LGC&quot;)) %&gt;%
  right_join(igc_df, by = &quot;link_var&quot;)

igc_df %&gt;% 
  ggplot(aes(x = cat_var, y = gr0, fill = gr_group)) +
  geom_boxplot() +
  stat_compare_means(method = &quot;wilcox.test&quot;) +
  theme_bw() +
  labs( y = &quot;Gene richness at BL&quot;, x = &quot;group&quot;, fill = &quot;GR cluster 96&quot;)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/test_basal_peaks-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test_basal_peaks-1">
Past versions of test_basal_peaks-1.png
</button>
</p>
<div id="fig-test_basal_peaks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/test_basal_peaks-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/test_basal_peaks-1.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Create a corr table and heatmap. This will be a good exploratory analysis to see which inflammation markers change with the microbiome. This is a longitudinal study, which means time may be an important variable we need to account with the models. A good exploratory first test will be to correlate markers with gene richness at basal. Then a correlation of deltas may give information on those who grow more depending on gene richness.</p>
<pre class="r"><code>num_var %&gt;%
  purrr::set_names() %&gt;%
  purrr::map_dfr(function(nv) {
    print(nv)
    igc_df %&gt;%
      dplyr::filter(long_var == 0) %&gt;%
      dplyr::select(SampleID, cat_var, nv, richness) %&gt;% 
      dplyr::mutate(!!sym(nv) := as.numeric(!!sym(nv))) %&gt;% 
      ungroup() %&gt;% 
      cor_test(&quot;richness&quot;, nv,  method = &quot;spearman&quot;)
  })</code></pre>
<pre><code>[1] &quot;CD4&quot;
[1] &quot;CD8&quot;
[1] &quot;CD4_CD38_DR&quot;
[1] &quot;CD8_CD38_DR&quot;
[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;
[1] &quot;ADV&quot;</code></pre>
<pre><code># A tibble: 9 × 6
  var1     var2           cor statistic     p method  
  &lt;chr&gt;    &lt;chr&gt;        &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   
1 richness CD4          0.17     47338. 0.155 Spearman
2 richness CD8          0.07     53166. 0.566 Spearman
3 richness CD4_CD38_DR -0.079    61661. 0.516 Spearman
4 richness CD8_CD38_DR -0.047    59815. 0.702 Spearman
5 richness CRP         -0.13     64681. 0.277 Spearman
6 richness IL6          0.1      16535. 0.488 Spearman
7 richness TNFa        -0.029    58827. 0.81  Spearman
8 richness sCD14       -0.11     63384  0.368 Spearman
9 richness ADV         -0.17      2072  0.448 Spearman</code></pre>
<p>Apparently, no correlation could be found between GR and inflammation markers at basal. Let’s look for the longitudinal approach.</p>
<pre class="r"><code>num_var %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    print(nv)
    igc_df %&gt;% 
      group_by(link_var) %&gt;% 
      dplyr::filter(long_var %in% c(0,48)) %&gt;% 
      dplyr::filter(n() == 2) %&gt;% 
      dplyr::mutate(num_var = as.numeric(!!sym(nv))) %&gt;% 
      dplyr::mutate(delta_nv := num_var[long_var == 48]/num_var[long_var == 0],
                    delta_gr = richness[long_var == 48]/richness[long_var == 0]) %&gt;% 

      filter(!is.na(num_var)) %&gt;% 
      group_by(cat_var,.drop = T) %&gt;%
      # ungroup() %&gt;% 
      rstatix::wilcox_test(num_var ~ long_var)
  })</code></pre>
<pre><code>[1] &quot;CD4&quot;
[1] &quot;CD8&quot;
[1] &quot;CD4_CD38_DR&quot;
[1] &quot;CD8_CD38_DR&quot;
[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;
[1] &quot;ADV&quot;</code></pre>
<pre><code>$CD4
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic        p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;    &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30        21 2.35e-10
2 DTG     num_var 0      48        30    29        15 2.01e-10

$CD8
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic       p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30       272 0.00801
2 DTG     num_var 0      48        30    29       312 0.0629 

$CD4_CD38_DR
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic           p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;       &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30       766 0.00000309 
2 DTG     num_var 0      48        30    29       754 0.000000243

$CD8_CD38_DR
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic           p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;       &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30      756. 0.0000065  
2 DTG     num_var 0      48        30    29      768  0.000000462

$CRP
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic      p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;
1 bDRV    num_var 0      48        30    22      380. 0.354 
2 DTG     num_var 0      48        30    22      463  0.0141

$IL6
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic          p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;      &lt;dbl&gt;
1 bDRV    num_var 0      48        18    30       475 0.0000061 
2 DTG     num_var 0      48        22    30       569 0.00000485

$TNFa
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic           p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;       &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30       715 0.0000866  
2 DTG     num_var 0      48        30    30       790 0.000000487

$sCD14
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic          p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;      &lt;dbl&gt;
1 bDRV    num_var 0      48        30    30       706 0.000158  
2 DTG     num_var 0      48        30    30       768 0.00000062

$ADV
# A tibble: 2 × 8
  cat_var .y.     group1 group2    n1    n2 statistic     p
* &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt;
1 bDRV    num_var 0      48         9    12        64 0.508
2 DTG     num_var 0      48         8    10        46 0.633</code></pre>
<pre class="r"><code>res_list &lt;- 
num_var[!num_var %in% c(&quot;CD8_CD38_DR&quot;, &quot;ADV&quot;)] %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    print(nv)
    igc_df %&gt;% 
      dplyr::mutate(num_var := as.numeric(!!sym(nv))) %&gt;% 
      filter(!is.na(num_var)) %&gt;% 
    get_comp_boxplots(dat = .,cat_var = &quot;cat_var&quot;,num_var = &quot;num_var&quot;, long_var = &quot;long_var&quot;,link_var = &quot;link_var&quot;,y_trans = &quot;sqrt&quot;, pal = &quot;Accent&quot;) %&gt;% 
      purrr::pluck(&quot;plot&quot;) +
      labs(y = nv, fill = &quot;group&quot;)
    
  })</code></pre>
<pre><code>[1] &quot;CD4&quot;
[1] &quot;CD8&quot;
[1] &quot;CD4_CD38_DR&quot;
[1] &quot;CRP&quot;
[1] &quot;IL6&quot;
[1] &quot;TNFa&quot;
[1] &quot;sCD14&quot;</code></pre>
<pre class="r"><code>test_list &lt;- 
 get_group_comparisons(dat = igc_df,cat_vector = &quot;cat_var&quot;,num_vector = num_var,long_var = &quot;long_var&quot;,link_var = &quot;link_var&quot;, comps = c(&quot;bDRV&quot;,&quot;DTG&quot;), graph_coords = T) %&gt;% 
   pluck(&quot;cat_var&quot;) %&gt;% 
   purrr::map(~{pluck(.x, &quot;test&quot;)})



num_var %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(nv) {
    igc_df %&gt;%
      dplyr::filter(!is.na(!!sym(nv))) %&gt;%
      dplyr::mutate(num_var = as.numeric(!!sym(nv)),
                    sd_var = sd(num_var, na.rm = T),
                    med_var = median(num_var)) %&gt;%
      ggplot(., aes(x = factor(long_var), y = med_var)) +
      # geom_boxplot(aes(fill = cat_var)) +
      # geom_jitter(aes(colour = rec_group )) +
      geom_point(
        aes(
          group = cat_var,
          shape = cat_var,
          color = cat_var),
        size = 3,
        position = position_dodge(.8)
      ) +
      geom_line(
        aes(group = cat_var,
            color = cat_var),
        position = position_dodge(0.8),
        lwd = 1
      ) +
      # geom_line(data = df, aes(x = factor(long_var), y = richness, group = link_var, color = recovery, lty = cat_var), inherit.aes = F, alpha = .4) +
      geom_errorbar(
        aes(
          ymin = med_var - sd_var,
          ymax = med_var + sd_var,
          group = cat_var,
          color = cat_var
        ),
        width = .2,
        position = position_dodge(0.8)
      ) +
      labs(x = &quot;weeks&quot;, y = nv) +
      # scale_color_brewer(palette = &quot;Dark2&quot;) +
      scale_color_viridis_d(
        option = &quot;magma&quot;,
        begin = .1,
        end = .8,
        direction = 1
      ) +
      theme_bw() +
      stat_pvalue_manual(test_list[[nv]], inhertit.aes = F , hide.ns = T) +
      theme(axis.title = element_text(face = &quot;bold&quot;))
    
  })</code></pre>
<pre><code>$CD4</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-1">
Past versions of corrr_deltas-1.png
</button>
</p>
<div id="fig-corrr_deltas-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9e2fd42198f450579592f1e076394255d450567/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1710d26f13b98dd1b41d35034e5f1c325cf0830/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-1.png" target="_blank">e1710d2</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD8</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-2">
Past versions of corrr_deltas-2.png
</button>
</p>
<div id="fig-corrr_deltas-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-2.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9e2fd42198f450579592f1e076394255d450567/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-2.png" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD4_CD38_DR</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-3">
Past versions of corrr_deltas-3.png
</button>
</p>
<div id="fig-corrr_deltas-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-3.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD8_CD38_DR</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-4">
Past versions of corrr_deltas-4.png
</button>
</p>
<div id="fig-corrr_deltas-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-4.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CRP</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-5">
Past versions of corrr_deltas-5.png
</button>
</p>
<div id="fig-corrr_deltas-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-5.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$IL6</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-6">
Past versions of corrr_deltas-6.png
</button>
</p>
<div id="fig-corrr_deltas-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-6.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$TNFa</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-7">
Past versions of corrr_deltas-7.png
</button>
</p>
<div id="fig-corrr_deltas-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-7.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$sCD14</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-8">
Past versions of corrr_deltas-8.png
</button>
</p>
<div id="fig-corrr_deltas-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-8.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$ADV</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-9">
Past versions of corrr_deltas-9.png
</button>
</p>
<div id="fig-corrr_deltas-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-9.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggarrange(plotlist = res_list, nrow = 3, ncol = 2, common.legend = T)</code></pre>
<pre><code>$`1`</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-10">
Past versions of corrr_deltas-10.png
</button>
</p>
<div id="fig-corrr_deltas-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-10.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$`2`</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-corrr_deltas-11">
Past versions of corrr_deltas-11.png
</button>
</p>
<div id="fig-corrr_deltas-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/corrr_deltas-11.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
attr(,&quot;class&quot;)
[1] &quot;list&quot;      &quot;ggarrange&quot;</code></pre>
<p>There are significant differences in both CRP and sCD14, which were lower in the DTG group. CRP became significant at week 96 only, while sCD14 became significant after 48 weeks, and kept doing so at week 96</p>
</div>
<div id="inflammation-markers-by-gene-richness" class="section level1">
<h1>Inflammation markers by gene richness</h1>
<div id="baseline" class="section level2">
<h2>Baseline</h2>
<p>First let’s check if Gene Richness had any effect on the baseline levels of inflammation markers.</p>
<pre class="r"><code>filt_vars &lt;- num_var[!num_var %in% c(&quot;CD4&quot;, &quot;CD8&quot;,&quot;CD8_CD38_DR&quot;, &quot;ADV&quot;)]

deltas &lt;- 
igc_df %&gt;% 
  dplyr::select(SampleID, link_var, long_var, cat_var, richness,filt_vars) %&gt;% 
  dplyr::group_by(link_var) %&gt;% 
  dplyr::filter(long_var %in% c(0,24,48,96)) %&gt;% 
  dplyr::filter(min(long_var) == 0) %&gt;% 
  # dplyr::filter(n() == 2) %&gt;% 
  dplyr::mutate(delta_richness = (richness-richness[long_var == 0]) / (richness + richness[long_var == 0]),
    across(.cols = filt_vars, 
                       .fns =  ~ .x  / .x[long_var == 0], 
                       .names = &quot;delta_{.col}&quot; ))  %&gt;% 
  dplyr::ungroup() %&gt;% 
  dplyr::select(SampleID, link_var, cat_var, long_var, contains(&quot;delta&quot;))

new_numvars &lt;- paste(&quot;delta&quot;,filt_vars,sep= &quot;_&quot;)
new_numvars %&gt;% 
  purrr::set_names() %&gt;%
  purrr::map(~ {
    deltas %&gt;% 
      ggplot(., aes(x=!!sym(.x), y = delta_richness, colour = cat_var)) +
      geom_point() +
      theme_bw()
  }) %&gt;% 
  ggpubr::ggarrange(plotlist = ., common.legend = T)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_inflammation_BL-1">
Past versions of GR_inflammation_BL-1.png
</button>
</p>
<div id="fig-GR_inflammation_BL-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-1.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>lmms &lt;- 
  deltas %&gt;%  
  # dplyr::mutate(cat_var = &quot;mock&quot;) %&gt;% 
get_lmm_effects(
  .,
  cat_vector = &quot;cat_var&quot;,
  long_var = &quot;delta_richness&quot;,
  num_vector = new_numvars,
  link_var = &quot;link_var&quot;
) </code></pre>
<pre><code>[1] &quot;cat_var&quot;
[1] &quot;delta_CD4_CD38_DR&quot;
[1] &quot;delta_CRP&quot;
[1] &quot;delta_IL6&quot;
[1] &quot;delta_TNFa&quot;
[1] &quot;delta_sCD14&quot;</code></pre>
<pre class="r"><code>new_numvars %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(~ {
    max_tp &lt;- deltas %&gt;% 
      dplyr::filter(!is.na(!!sym(.x))) %&gt;% 
      dplyr::pull(long_var) %&gt;% 
      max()
    deltas %&gt;%
      dplyr::mutate(cat_var = &quot;mock&quot;) %&gt;%
      dplyr::filter(!is.na(!!sym(.x)), 
                    !is.na(delta_richness)) %&gt;% 
      
      dplyr::filter(long_var == max_tp) %&gt;%
      create_LMM(
        .,
        num_var = .x,
        long_var = &quot;delta_richness&quot;,
        cat_var = &quot;cat_var&quot;,
        link_var = &quot;link_var&quot;,
        breakpoints = NULL
      ) %&gt;% 
      purrr::pluck(&quot;plot&quot;) +
      scale_x_continuous(breaks = c(0,2,4,6,8)) +
      labs(subtitle = paste(&quot;week&quot;, max_tp)) +
      # scale_y_continuous(trans = &quot;log2&quot;) +
      theme(axis.title.x = element_text(size = 12),
            axis.title.y = element_text(size = 12, angle = 90))
  }) %&gt;%
  ggpubr::ggarrange(plotlist = .)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_inflammation_BL-2">
Past versions of GR_inflammation_BL-2.png
</button>
</p>
<div id="fig-GR_inflammation_BL-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-2.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_inflammation_BL-2.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># deltas %&gt;% 
#   dplyr::filter(long_var == max(long_var)) %&gt;% 
#   dplyr::select(delta_richness, new_numvars) %&gt;% 
#   cor_test(use = &quot;complete.obs&quot;) %&gt;% 
#   corrplot::corrplot(corr = .) 
#   dplyr::filter()</code></pre>
</div>
</div>
<div id="differences-by-groups" class="section level1">
<h1>Differences by groups</h1>
<pre class="r"><code>igc_df &lt;-
  igc_df %&gt;%
  dplyr::filter(!is.na(gr_group)) %&gt;%
  mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;))


my_list &lt;- 
num_var %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    igc_df %&gt;% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %&gt;% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;)) %&gt;% 
      # get_lmm_effects()
      get_comp_boxplots(cat_var = &quot;cat_var2&quot;, num_var = nv, long_var = &quot;long_var&quot;,link_var = &quot;link_var&quot;)
  })

myPlotList &lt;- map(my_list, pluck, &quot;plot&quot;)
ggarrange(plotlist = myPlotList, common.legend = T)</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-1">
Past versions of GR_group_boxplots-1.png
</button>
</p>
<div id="fig-GR_group_boxplots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-1.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-1.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cat_df &lt;- 
num_var %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(function(nv){
    
    obj &lt;- 
    igc_df %&gt;% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %&gt;% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;)) %&gt;% 
      get_group_comparisons(.,link_var = &quot;link_var&quot;, long_var = &quot;long_var&quot;,cat_vector = &quot;cat_var2&quot;,num_vector = nv,type = &quot;categorical&quot;, comps = unique(igc_df$cat_var2))
    
    objtest &lt;- obj %&gt;% pluck(&quot;cat_var2&quot;,nv,&quot;test&quot;)
    objstats &lt;- obj %&gt;% pluck(&quot;cat_var2&quot;,nv,&quot;stats&quot;) %&gt;% 
      mutate(stat = paste(round(median,2),&quot; [&quot;,round(iqr,2),&quot;]&quot;, sep = &quot;&quot;))
  
    objtest %&gt;% 
      pivot_longer(cols = c(&quot;group1&quot;,&quot;group2&quot;)) %&gt;% 
      dplyr::rename(cat_var = value) %&gt;%
      dplyr::left_join(objstats[,c(&quot;cat_var&quot;,&quot;long_var&quot;,&quot;stat&quot;)], by = c(&quot;cat_var&quot;, &quot;long_var&quot;)) %&gt;% 
      pivot_wider(names_from = &quot;name&quot;, values_from = c(&quot;cat_var&quot;, &quot;stat&quot;)) %&gt;% 
      dplyr::select(long_var, group1 = cat_var_group1, group2 = cat_var_group2, contains(&quot;stat&quot;), p.adj)
    
  })

cat_df %&gt;% 
  dplyr::filter(p.adj &lt; 0.05)</code></pre>
<pre><code># A tibble: 0 × 7
# … with 7 variables: long_var &lt;int&gt;, group1 &lt;chr&gt;, group2 &lt;chr&gt;,
#   statistic &lt;dbl&gt;, stat_group1 &lt;chr&gt;, stat_group2 &lt;chr&gt;, p.adj &lt;dbl&gt;</code></pre>
<pre class="r"><code>long_df &lt;- 
num_var %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(function(nv){
    
    obj &lt;- 
    igc_df %&gt;% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %&gt;% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;)) %&gt;% 
      get_group_comparisons(.,link_var = &quot;link_var&quot;, long_var = &quot;long_var&quot;,cat_vector = &quot;cat_var2&quot;,num_vector = nv,type = &quot;longitudinal&quot;, comps = c(0,48))
    
    obj %&gt;% 
      pluck(&quot;cat_var2&quot;,nv,&quot;test&quot;) %&gt;% 
      rstatix::adjust_pvalue( method = &quot;BH&quot;, p.col = &quot;p&quot;) %&gt;% 
      full_join(obj$cat_var2[[nv]]$stats, by = &quot;cat_var&quot;)%&gt;% 
      dplyr::mutate(.y. = nv) %&gt;% 
      dplyr::select(cat_var, group1, group2,num_var = .y., contains(&quot;median&quot;), contains(&quot;iqr&quot;), &quot;p.adj&quot;)
  })


long_df %&gt;%
  dplyr::filter(p.adj &lt; 0.05 )</code></pre>
<pre><code># A tibble: 18 × 11
   cat_var  group1 group2 num_var    medianChange median1 median2   iqr1    iqr2
   &lt;chr&gt;    &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;             &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
 1 HGC_bDRV 0      48     CD4               163.     63.5  206.   6.65e1 1.59e+2
 2 HGC_DTG  0      48     CD4               161      40    196    3.80e1 1.09e+2
 3 LGC_bDRV 0      48     CD4               190      18    192    4.20e1 2.09e+2
 4 LGC_DTG  0      48     CD4               228      70    370    8.55e1 1.99e+2
 5 HGC_DTG  0      48     CD4_CD38_…        -12.5    19.1    6.01 2.72e1 4.89e+0
 6 LGC_bDRV 0      48     CD4_CD38_…        -18      24      7.3  2.06e1 7.22e+0
 7 LGC_DTG  0      48     CD4_CD38_…        -22.6    31.1    5.94 3.47e1 6.48e+0
 8 HGC_bDRV 0      48     CD8_CD38_…        -29.6    43.0   15.0  1.76e1 4.07e+0
 9 HGC_DTG  0      48     CD8_CD38_…        -22.3    42.4   17.3  2.45e1 1.86e+1
10 LGC_bDRV 0      48     CD8_CD38_…        -15.9    39.3   20.3  3.10e1 2.17e+1
11 LGC_DTG  0      48     CD8_CD38_…        -36      65.7   21.9  3.62e1 1.18e+1
12 HGC_DTG  0      48     IL6               -14      16      2    1.87e1 6.67e-1
13 HGC_DTG  0      48     TNFa               -9.5    18.5    9    5.17e0 1.58e+0
14 LGC_bDRV 0      48     TNFa               -5      15     10    5.67e0 4.33e+0
15 LGC_DTG  0      48     TNFa              -11      20      7    1.9 e1 4.00e+0
16 HGC_DTG  0      48     sCD14           -1168    2800   1690    9.85e2 6.13e+2
17 LGC_bDRV 0      48     sCD14            -836    2788   1956    1.70e3 7.63e+2
18 LGC_DTG  0      48     sCD14           -1320    2876   1700    2.31e3 1.53e+3
# … with 2 more variables: iqr_change &lt;dbl&gt;, p.adj &lt;dbl&gt;</code></pre>
<pre class="r"><code>spaghetti_plot_list &lt;- 
num_var[!num_var %in% c(&quot;ADV&quot;)] %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(nv){
    
    name &lt;- case_when(nv == &quot;CD4_CD38_DR&quot; ~ &quot;CD4+ CD38+ DR+&quot;,
      nv == &quot;CD8_CD38_DR&quot; ~ &quot;CD8+ CD38+ DR+&quot;,
      TRUE ~ nv)
    un &lt;- ifelse(nv %in% c(&quot;CD4_CD38_DR&quot;,&quot;CD8_CD38_DR&quot;), &quot;units&quot;,
                 metadata %&gt;% 
      dplyr::pull(paste(&quot;units&quot;, nv, sep = &quot;_&quot;)) %&gt;% 
      discard(is.na) %&gt;% unique())

    
    summ_df &lt;- 
      igc_df %&gt;% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %&gt;% 
      group_by(cat_var, gr_group, long_var) %&gt;% 
      dplyr::summarise(mean_gr = mean(!!sym(nv)),
                       sd_gr = sd(!!sym(nv))) %&gt;% 
      dplyr::mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;),
                    long_var = as.factor(long_var))

    tests &lt;-
      igc_df %&gt;% 
      dplyr::filter(!is.na(nv)) %&gt;% 
      dplyr::rename(num_var = !!sym(nv)) %&gt;% 
      dplyr::mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;)) %&gt;% 
      dplyr::filter(!is.na(num_var)) %&gt;% 
      dplyr::group_by(long_var) %&gt;% 
      rstatix::wilcox_test( num_var ~ cat_var2) %&gt;% 
      rstatix::add_xy_position(x = &quot;long_var&quot;, group = &quot;cat_var2&quot;)

    igc_df %&gt;%
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %&gt;%
      mutate(cat_var2 = paste(gr_group, cat_var, sep = &quot;_&quot;),
             long_var = as.factor(long_var)) %&gt;%
      ggplot(aes(
        x = long_var,
        y = !!sym(nv),
        color = cat_var
      )) +
      geom_point(
        data = summ_df,
        aes(
          x = long_var,
          y = mean_gr,
          color = cat_var,
          shape = gr_group,
          group = cat_var2
        ),
        size = 3,
        position = position_dodge(.8)
      ) +
      
      geom_errorbar(
        data = summ_df,
        aes(
          x = long_var,
          ymin = mean_gr - sd_gr,
          ymax = mean_gr + sd_gr,
          color = cat_var,
          lty = gr_group,
          group = cat_var2
        ),
        inherit.aes = F,
        width = 1,
        position = position_dodge(.8)
      ) +
      geom_line(
        data = summ_df,
        aes(
          x = long_var,
          y = mean_gr,
          color = cat_var,
          lty = gr_group,
          group = cat_var2
        ),
        inherit.aes = F,
        position = position_dodge(.8)
      ) +
      # stat_smooth(method = &quot;loess&quot;, alpha = .2, aes(fill = cat_var), position = position_dodge(.8)) +
      theme_bw() +
      labs(
        x = &quot;weeks&quot;,
        y = un,
        color = &quot;group&quot;,
        shape = &quot;gene count&quot;,
        lty = &quot;gene count&quot;,
        title = name
      ) +
      stat_pvalue_manual(tests, label = &quot;p.adj.signif&quot;, hide.ns = T) +
      scale_color_brewer(palette = &quot;Dark2&quot;) +
      scale_fill_brewer(palette = &quot;Dark2&quot;) +
      theme(axis.title.x = element_blank())
    # scale_x_continuous(breaks = c(0,48,96))
  }) 
spaghetti_plot_list %&gt;%
  ggarrange(plotlist = ., common.legend = T, ncol = 2,nrow = 4) %&gt;%
  ggpubr::annotate_figure(bottom = text_grob(&quot;Weeks&quot;, size = 12, face = &quot;bold&quot;)) %&gt;% 
ggsave(plot= ., device = &quot;svg&quot;,path = here::here(&quot;output&quot;,&quot;figures&quot;), filename = &quot;inflammation.svg&quot;, dpi= 300, width = 15, height = 25, units = &quot;cm&quot;)

spaghetti_plot_list</code></pre>
<pre><code>$CD4</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-2">
Past versions of GR_group_boxplots-2.png
</button>
</p>
<div id="fig-GR_group_boxplots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-2.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/b9e2fd42198f450579592f1e076394255d450567/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-2.png" target="_blank">b9e2fd4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5ba2c0eba12ddd3d27cf39d53a4f4c0902f10524/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-2.png" target="_blank">5ba2c0e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2023-03-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD8</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-3">
Past versions of GR_group_boxplots-3.png
</button>
</p>
<div id="fig-GR_group_boxplots-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-3.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD4_CD38_DR</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-4">
Past versions of GR_group_boxplots-4.png
</button>
</p>
<div id="fig-GR_group_boxplots-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-4.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CD8_CD38_DR</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-5">
Past versions of GR_group_boxplots-5.png
</button>
</p>
<div id="fig-GR_group_boxplots-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-5.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$CRP</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-6">
Past versions of GR_group_boxplots-6.png
</button>
</p>
<div id="fig-GR_group_boxplots-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-6.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$IL6</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-7">
Past versions of GR_group_boxplots-7.png
</button>
</p>
<div id="fig-GR_group_boxplots-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-7.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$TNFa</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-8">
Past versions of GR_group_boxplots-8.png
</button>
</p>
<div id="fig-GR_group_boxplots-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-8.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$sCD14</code></pre>
<p><img src="figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GR_group_boxplots-9">
Past versions of GR_group_boxplots-9.png
</button>
</p>
<div id="fig-GR_group_boxplots-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/72f15339f90d852482e264aa42757ad925bf6e1d/docs/figure/317_GeneRichness_vs_Inflammation.Rmd/GR_group_boxplots-9.png" target="_blank">72f1533</a>
</td>
<td>
cblazquez
</td>
<td>
2023-04-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>num_var &lt;- num_var[!num_var == &quot;ADV&quot;]</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nlme_3.1-158        compareGroups_4.5.1 ggpubr_0.4.0       
 [4] rstatix_0.7.0       metar_0.1.5         RColorBrewer_1.1-3 
 [7] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.9        
[10] purrr_0.3.4         readr_2.1.2         tidyr_1.2.0        
[13] tibble_3.1.7        ggplot2_3.3.6       tidyverse_1.3.1    
[16] devtools_2.4.3      usethis_2.1.6       workflowr_1.7.0    

loaded via a namespace (and not attached):
  [1] utf8_1.2.2          tidyselect_1.1.2    lme4_1.1-29        
  [4] htmlwidgets_1.5.4   grid_4.0.5          munsell_0.5.0      
  [7] codetools_0.2-18    chron_2.3-57        future_1.26.1      
 [10] withr_2.5.0         colorspace_2.0-3    Biobase_2.50.0     
 [13] phyloseq_1.34.0     highr_0.9           logger_0.2.2       
 [16] knitr_1.39          uuid_1.1-0          rstudioapi_0.13    
 [19] stats4_4.0.5        ggsignif_0.6.3.9000 officer_0.4.3      
 [22] listenv_0.8.0       labeling_0.4.2      git2r_0.30.1       
 [25] farver_2.1.0        rhdf5_2.34.0        rprojroot_2.0.3    
 [28] coda_0.19-4         parallelly_1.32.0   vctrs_0.4.1        
 [31] generics_0.1.2      xfun_0.31           R6_2.5.1           
 [34] arm_1.12-2          rhdf5filters_1.2.1  cachem_1.0.6       
 [37] assertthat_0.2.1    promises_1.2.0.1    networkD3_0.4      
 [40] scales_1.2.0        nnet_7.3-17         merTools_0.5.2     
 [43] gtable_0.3.0        globals_0.15.0      processx_3.6.1     
 [46] rlang_1.0.2         systemfonts_1.0.4   splines_4.0.5      
 [49] broom_0.8.0         yaml_2.3.5          reshape2_1.4.4     
 [52] abind_1.4-5         modelr_0.1.8        backports_1.4.1    
 [55] httpuv_1.6.5        HardyWeinberg_1.7.5 tools_4.0.5        
 [58] ellipsis_0.3.2      kableExtra_1.3.4    jquerylib_0.1.4    
 [61] biomformat_1.18.0   selbal_0.1.0        Rsolnp_1.16        
 [64] BiocGenerics_0.36.1 sessioninfo_1.2.2   Rcpp_1.0.8.3       
 [67] plyr_1.8.7          base64enc_0.1-3     zlibbioc_1.36.0    
 [70] ps_1.7.1            prettyunits_1.1.1   cowplot_1.1.1      
 [73] S4Vectors_0.28.1    haven_2.5.0         cluster_2.1.3      
 [76] fs_1.5.2            here_1.0.1          furrr_0.3.0        
 [79] DataExplorer_0.8.2  magrittr_2.0.3      data.table_1.14.2  
 [82] lmerTest_3.1-3      flextable_0.7.2     reprex_2.0.1       
 [85] truncnorm_1.0-8     mvtnorm_1.1-3       whisker_0.4        
 [88] pkgload_1.2.4       hms_1.1.1           mime_0.12          
 [91] evaluate_0.15       xtable_1.8-4        broom.mixed_0.2.9.4
 [94] readxl_1.4.0        IRanges_2.24.1      gridExtra_2.3      
 [97] testthat_3.1.4      compiler_4.0.5      mice_3.14.0        
[100] writexl_1.4.0       crayon_1.5.1        minqa_1.2.4        
[103] htmltools_0.5.2     mgcv_1.8-40         later_1.3.0        
[106] tzdb_0.3.0          lubridate_1.8.0     DBI_1.1.3          
[109] dbplyr_2.2.0        MASS_7.3-57         boot_1.3-28        
[112] Matrix_1.4-1        ade4_1.7-19         car_3.1-0          
[115] permute_0.9-7       brio_1.1.3          cli_3.3.0          
[118] parallel_4.0.5      igraph_1.3.2        pkgconfig_2.0.3    
[121] getPass_0.2-2       numDeriv_2016.8-1.1 xml2_1.3.2         
[124] foreach_1.5.2       svglite_2.1.0       bslib_0.3.1        
[127] blme_1.0-5          webshot_0.5.3       multtest_2.46.0    
[130] XVector_0.30.0      rvest_1.0.2         callr_3.7.0        
[133] digest_0.6.29       vegan_2.6-2         Biostrings_2.58.0  
[136] rmarkdown_2.14      cellranger_1.1.0    gdtools_0.2.4      
[139] shiny_1.7.1         nloptr_2.0.3        lifecycle_1.0.1    
[142] jsonlite_1.8.0      Rhdf5lib_1.12.1     carData_3.0-5      
[145] viridisLite_0.4.0   desc_1.4.1          fansi_1.0.3        
[148] pillar_1.7.0        lattice_0.20-45     fastmap_1.1.0      
[151] httr_1.4.2          pkgbuild_1.3.1      survival_3.3-1     
[154] glue_1.6.2          remotes_2.4.2       zip_2.2.0          
[157] iterators_1.0.14    stringi_1.7.6       sass_0.4.1         
[160] memoise_2.0.1       ape_5.6-2          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
