<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Beta Diversity and NMDS</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Beta Diversity and NMDS</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treed0aa42c869510b71b66f140654c673418e0cacd5targetblankd0aa42ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/d0aa42c869510b71b66f140654c673418e0cacd5" target="_blank">d0aa42c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treed0aa42c869510b71b66f140654c673418e0cacd5targetblankd0aa42ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/d0aa42c869510b71b66f140654c673418e0cacd5" target="_blank">d0aa42c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  Metadata/MetadataManagement/Resultats Real Time.xlsx
    Untracked:  analysis/figure/

Unstaged changes:
    Modified:   advanz4.Rproj
    Modified:   analysis/200_ADVANZ4_Report.Rmd
    Modified:   analysis/310_IGC_analysis.Rmd
    Modified:   analysis/320_Beta_Diversity.Rmd
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/320_Beta_Diversity.Rmd</code>) and HTML (<code>docs/320_Beta_Diversity.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/eba3a180ec68006819926efda2b5ca478bd6f808/analysis/320_Beta_Diversity.Rmd" target="_blank">eba3a18</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-29
</td>
<td>
comments for Carlos
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e253850975c479f491140a3a60d9e54a3eca44ff/analysis/320_Beta_Diversity.Rmd" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e253850975c479f491140a3a60d9e54a3eca44ff/docs/320_Beta_Diversity.html" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/320_Beta_Diversity.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/320_Beta_Diversity.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/11c1e7d2e4c2b76ccb9e08a96305fa7f34ee48de/analysis/320_Beta_Diversity.Rmd" target="_blank">11c1e7d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
Code reorganized towards new structure
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="r-markdown" class="section level2">
<h2>R Markdown</h2>
<p>This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <a href="http://rmarkdown.rstudio.com" class="uri">http://rmarkdown.rstudio.com</a>.</p>
<p>When you click the <strong>Knit</strong> button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:</p>
<pre><code>suggested package selbal vsuggested package DataExplorer vsuggested package lme4 vsuggested package merTools v</code></pre>
</div>
<div id="including-plots" class="section level2">
<h2>Including Plots</h2>
<p>You can also embed plots, for example:</p>
<pre class="r"><code>myPath&lt;-here::here()
setwd(myPath)
mymre&lt;-readRDS(&quot;data/mre.rds&quot;)
metadata &lt;- metar::get_meta(mymre) %&gt;% as.data.frame()

myphyloseq &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;)



CategoricalVariablesDF &lt;- get_metadata(mymre)@categorical_vals
NumericalVariablesDF &lt;- get_metadata(mymre)@numeric_vals
LongitudinalVariablesDF &lt;- get_metadata(mymre)@longitudinal_vals</code></pre>
<pre class="r"><code>getAbundances&lt;-function(phyloseq, level){
  
  collapsedPhyloseq&lt;- phyloseq::tax_glom(phyloseq, level)
  
  OtuTable&lt;- phyloseq::otu_table(collapsedPhyloseq) %&gt;%
    magrittr::set_rownames(as.character(phyloseq::tax_table(collapsedPhyloseq)[,level])) %&gt;%
    as.data.frame()
  return(OtuTable)
  
}




myAbundanceDF&lt;- getAbundances(myphyloseq, &quot;Species&quot;)

Top50&lt;- myAbundanceDF %&gt;%
    mutate(sum=rowSums(.)) %&gt;%
    arrange(desc(sum)) %&gt;%
    dplyr::slice(1:50) %&gt;%
    rownames()



myNMDS &lt;- myAbundanceDF %&gt;%
    # dplyr::filter(row.names(.) %in% Top50) %&gt;%
    t() %&gt;%
    as.data.frame() %&gt;%
 
    metaMDS(.,distance=&quot;bray&quot;,trymax=100, 
            k = nrow(unique(metadata[,CategoricalVariablesDF$CategoricalVariable[1]]))) %&gt;%
    pluck(&quot;points&quot;) %&gt;%
    as.data.frame() 

  # NMDS = data.frame(NMDS1 = dim_NMDS$points[,1], NMDS2 = dim_NMDS$points[,2])

BiPlotCoords&lt;- envfit(myNMDS, t(myAbundanceDF), perm=999) 

BiPlotCoordsDF &lt;- c(&quot;MDS1&quot;,&quot;MDS2&quot;,&quot;r&quot;,&quot;pvals&quot;) %&gt;%
  set_names() %&gt;%
  
  map_dfr(., ~ {
    
    if(.x %in% c(&quot;MDS1&quot;,&quot;MDS2&quot;)){
      BiPlotCoords$vectors$arrows[,.x] %&gt;%
        as.numeric()
    } else{
      BiPlotCoords$vectors %&gt;%
        pluck(.x) %&gt;%
        as.numeric()
    } 
      
  }) %&gt;%
  as.data.frame() %&gt;%
  set_names(nm=c(&quot;NMDS1&quot;,&quot;NMDS2&quot;,&quot;r&quot;,&quot;p&quot;)) %&gt;%
  dplyr::mutate(Dim1 = NMDS1*sqrt(r),
                Dim2 = NMDS2*sqrt(r),
                species = rownames(myAbundanceDF)) %&gt;%
  dplyr::filter(species %in% Top50)
  

myBrayDistMat &lt;-vegdist(wisconsin(sqrt(t(myAbundanceDF))))</code></pre>
</div>
<div id="nmds-by-treatment-group" class="section level2">
<h2>NMDS by Treatment group</h2>
<p>One of the variables: <strong>CD4diff_96</strong> has only one level. That means all patients in the study had an increase of CD4 counts higher than 50 cells/ml between weeks 96 and 0. As this variable only has one level it will cause the following code to break (as it performs conrast analysis). Hence, it will be removed from the following analysis. However, we’ll look at the distribution of the CD4 increases as a distribution:</p>
<pre class="r"><code>metadata %&gt;% 
  dplyr::filter(time_point %in% c(0,48),
                !is.na(CD4diff_96)) %&gt;%
  dplyr::select(record_id, time_point, CD4) %&gt;%
  dplyr::group_by(record_id) %&gt;%
  dplyr::summarise(dif = diff(CD4)) %&gt;%
  ggplot(., aes(x= dif)) +
  geom_density()</code></pre>
<pre><code>`summarise()` has grouped output by &#39;record_id&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning: Removed 1 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/CD4diff_dist-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>long_var&lt;-LongitudinalVariablesDF$LongitudinalVariable[1]
CategoricalVariablesDF &lt;- CategoricalVariablesDF %&gt;%
  filter(!CategoricalVariable == &quot;CD4diff_96&quot;)
cat_var &lt;- CategoricalVariablesDF$CategoricalVariable
mymre@metadata@categorical_vals &lt;- as_tibble(CategoricalVariablesDF)



# link_var &lt;- LongitudinalVariablesDF$LinkVariable[1]
# id_var &lt;- &quot;SampleID&quot;

# myPal&lt;-RColorBrewer::brewer.pal(name =  CategoricalVariablesDF$PaletteName[1],
#                                 n=length(unique(metadata[,cat_var])))
# BiPlotCoordsDF&lt;-BiPlotCoordsDF %&gt;%
#   mutate(Dim1s = Dim1*3,
#          Dim2s = Dim2*3)
# 
# 
# myNMDS_DF &lt;- myNMDS %&gt;%
#   rownames_to_column(var = id_var) %&gt;%
#   dplyr::full_join(., metadata, by =id_var) %&gt;%
#   dplyr::select(SampleID = id_var,
#                 NMDS1 = MDS1,
#                 NMDS2 = MDS2,
#                 linkVar = !!sym(link_var),
#                 catVar = !!sym(cat_var),
#                 LongVar = long_var)
# 
# 
#   myXmin&lt;-min(myNMDS$MDS1)
#   # +(max(dim_NMDS$points[,1])-min(dim_NMDS$points[,1]))/20
#   myYmin&lt;-max(myNMDS$MDS2)-(max(myNMDS$MDS1)-min(myNMDS$MDS1))
# 
# 
# 
#   
# myAdonis&lt;- adonis(myBrayDistMat~ catVar, data = myNMDS_DF) %&gt;%
#   pluck(&quot;aov.tab&quot;) %&gt;%
#   as.data.frame() %&gt;%
#   slice(1L) %&gt;%
#   magrittr::set_rownames(cat_var)
#   
#   
# myNMDSplot&lt;-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) + 
#     geom_point() +
#     theme_bw()+
#     theme(panel.border=element_blank(),
#           panel.grid.major=element_blank(),
#           panel.grid.minor=element_blank(),
#           
#           axis.line=element_line(colour=&quot;black&quot;),
#           axis.text.x=element_text(size=11),
#           axis.text.y=element_text(size=11),
#           axis.title.x=element_text(size=11),
#           axis.title.y=element_text(size=11),
#           
#           legend.text=element_text(size=11,face=&quot;italic&quot;),
#           legend.title=element_text(size=11))+
#     
#     ggtitle(expression(atop(&quot;Species composition&quot;,atop(italic(&quot;all samples&quot;))))) +
#     theme(plot.title=element_text(lineheight=1,face=&quot;bold&quot;,size=13))+
#     coord_fixed()+
#     labs(fill = cat_var, color=cat_var, shape = long_var) 
# 
# 
# 
# 
# myNMDSBiplot&lt;-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) + 
#     geom_point(aes(colour=catVar, shape = as.character(LongVar))) +
#     scale_colour_manual(values = myPal, labels = c(&quot;DGT&quot;, &quot;DRV/r&quot;)) +
#     scale_fill_manual(values=myPal, labels = c(&quot;DGT&quot;, &quot;DRV/r&quot;))+
#     theme_bw()+
#     theme(panel.border=element_blank(),
#           panel.grid.major=element_blank(),
#           panel.grid.minor=element_blank(),
#           
#           axis.line=element_line(colour=&quot;black&quot;),
#           axis.text.x=element_text(size=11),
#           axis.text.y=element_text(size=11),
#           axis.title.x=element_text(size=11),
#           axis.title.y=element_text(size=11),
#           
#           legend.text=element_text(size=11,face=&quot;italic&quot;),
#           legend.title=element_text(size=11))+
#     
#     ggtitle(expression(atop(&quot;Species composition&quot;,atop(italic(&quot;all samples&quot;))))) +
#     theme(plot.title=element_text(lineheight=1,face=&quot;bold&quot;,size=13))+
#     stat_ellipse(geom=&quot;polygon&quot;,
#                  alpha=0.15,color=&quot;black&quot;,
#                  aes(fill=catVar),level=0.95)+
#     # scale_size_manual(values = c(1:length(unique(myNMDS_DF[,&quot;LongVar&quot;])))/1.5) +
#     geom_segment(data=BiPlotCoordsDF,aes(x=0,xend=Dim1s,y=0,yend=Dim2s),
#                  arrow = arrow(length = unit(0.2,&quot;cm&quot;)),colour=&quot;darkgrey&quot;)+
#     geom_text_repel(data=BiPlotCoordsDF,aes(Dim1s,Dim2s,label=species),
#                     size=2.5, box.padding = unit(0.25, &quot;lines&quot;),
#                     segment.colour=&quot;black&quot;,
#                     segment.alpha=0.5,
#                     segment.size=0.10)+
#     coord_fixed()+
#     labs(fill = cat_var, color=cat_var, shape = long_var) +
#   annotation_custom(gridExtra::tableGrob(round(myAdonis,3), 
#                                          theme = ttheme_default(base_size = 8)),ymin = 3.5, xmin=2.4)
# </code></pre>
<p>When we look at the most simple ordination before colouring by categorical variables, we see no clear separated clusters between samples. They all appear to be mostly concentrated in one big cluster that randomly spreads in the positive direction for both NMDS components.</p>
<pre class="r"><code>mymre &lt;- metar::metaphlan_nmds(mymre, tax_level = &quot;Species&quot;,top_n = 50, save_files = F)

mymre@taxa@metaphlan@nmds$Species$top_50$categorical %&gt;%
  discard(., names(.) == &quot;record_id&quot;)</code></pre>
<pre><code>$group</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$risk_group</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$center</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$gender</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$ethnic_group</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4diff_48</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8diff_48</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4after_48</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8after_48</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8diff_96</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD4after_96</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8after_96</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$time_point</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$cluster</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/plot_basic_NMDS-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>In a more formal analysis, the silhouette coefficient was tested for different number of clusters. This analysis found the biggest coefficient was found for n=1 clustrs, with massive drops between 2-3 and 5-6.</p>
<pre class="r"><code>myBrayDistMat &lt;- phyloseq::otu_table(myphyloseq) %&gt;%
  t() %&gt;%
  sqrt() %&gt;%
  vegan::wisconsin() %&gt;%
  vegan::vegdist()



c(1:11) %&gt;%
  set_names() %&gt;%
  map_df(., ~{
    cluster::pam(myBrayDistMat, k = .x) %&gt;%
      pluck(&quot;silinfo&quot;) %&gt;%
      pluck(&quot;avg.width&quot;)
  }) %&gt;%
  t() %&gt;%
  as_data_frame() %&gt;%
  rownames_to_column(var=&quot;clusters&quot;) %&gt;%
  set_names(c(&quot;clusters&quot;, &quot;coefficient&quot;)) %&gt;%
  ggplot(., aes(x=as.numeric(clusters), y=coefficient, group=1)) +
  geom_point() + 
  geom_line() + 
  scale_x_continuous(breaks=c(1:10)) +
  theme_bw() +
  theme(panel.grid.minor = element_blank()) + 
  labs(title = &quot;Silhouette coefficient by number of clusters&quot;, 
       x=&quot;nº Clusters&quot;, y = &quot;Silhouette coefficient&quot;)</code></pre>
<p><img src="figure/320_Beta_Diversity.Rmd/silhouette_analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>In the same fashion, when we start identifying samples by group and Timepoint, we see no clear separation, although samples from group 1 (+DGT) group appear to be more clustered together than group 2 (DRV+RLT), which show a more diffuse “central” cluster and are more dispersed. When looking at weight of the variables, the concentric spread of the arrows can be split in two sides, one more enriched in Prevotella and another in bacteroides, which spread mostly around the 2nd component of the NMDS.</p>
<p>However, the Adonis test found the variable group to have a significant effect over sample distances (p&lt;0.001), although it only accounted for a small portion of the variance</p>
<pre class="r"><code>myNMDSBiplot</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gridExtra_2.3   cluster_2.1.0   ggrepel_0.9.1   vegan_2.5-7    
 [5] lattice_0.20-41 permute_0.9-7   forcats_0.5.1   stringr_1.4.0  
 [9] dplyr_1.0.8     purrr_0.3.4     readr_2.0.1     tidyr_1.2.0    
[13] tibble_3.1.6    ggplot2_3.3.5   tidyverse_1.3.1 phyloseq_1.34.0
[17] metar_0.1.5    

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                tidyselect_1.1.2         
  [3] lme4_1.1-27.1             htmlwidgets_1.5.4        
  [5] grid_4.0.2                gmp_0.6-2                
  [7] munsell_0.5.0             codetools_0.2-16         
  [9] effectsize_0.4.5          miniUI_0.1.1.1           
 [11] withr_2.5.0               colorspace_2.0-2         
 [13] Biobase_2.50.0            logger_0.2.2             
 [15] highr_0.9                 knitr_1.38               
 [17] rstudioapi_0.13           stats4_4.0.2             
 [19] ipmisc_6.0.2              ggsignif_0.6.3           
 [21] labeling_0.4.2            emmeans_1.7.1-1          
 [23] git2r_0.28.0              polyclip_1.10-0          
 [25] farver_2.1.0              datawizard_0.2.0.1       
 [27] rhdf5_2.34.0              rprojroot_2.0.2          
 [29] coda_0.19-4               vctrs_0.3.8              
 [31] generics_0.1.2            xfun_0.30                
 [33] BWStest_0.2.2             R6_2.5.1                 
 [35] arm_1.12-2                BayesFactor_0.9.12-4.2   
 [37] rhdf5filters_1.2.1        cachem_1.0.1             
 [39] reshape_0.8.8             assertthat_0.2.1         
 [41] promises_1.2.0.1          networkD3_0.4            
 [43] scales_1.1.1              ggExtra_0.10.0           
 [45] merTools_0.5.2            gtable_0.3.0             
 [47] multcompView_0.1-8        workflowr_1.7.0          
 [49] rlang_1.0.2               MatrixModels_0.5-0       
 [51] zeallot_0.1.0             PMCMRplus_1.9.0          
 [53] splines_4.0.2             rstatix_0.7.0            
 [55] broom_0.7.12              yaml_2.3.5               
 [57] reshape2_1.4.4            abind_1.4-5              
 [59] modelr_0.1.8              backports_1.4.1          
 [61] httpuv_1.6.3              tools_4.0.2              
 [63] ellipsis_0.3.2            jquerylib_0.1.4          
 [65] biomformat_1.18.0         selbal_0.1.0             
 [67] RColorBrewer_1.1-2        BiocGenerics_0.36.1      
 [69] WRS2_1.1-3                Rcpp_1.0.7               
 [71] plyr_1.8.6                zlibbioc_1.36.0          
 [73] ggpubr_0.4.0              pbapply_1.5-0            
 [75] correlation_0.7.0         S4Vectors_0.28.1         
 [77] haven_2.3.1               fs_1.5.0                 
 [79] here_1.0.1                DataExplorer_0.8.2       
 [81] magrittr_2.0.2            data.table_1.14.2        
 [83] reprex_2.0.1              mvtnorm_1.1-2            
 [85] whisker_0.4               hms_1.1.1                
 [87] patchwork_1.1.1           mime_0.11                
 [89] evaluate_0.15             xtable_1.8-4             
 [91] pairwiseComparisons_3.1.6 broom.mixed_0.2.7        
 [93] readxl_1.3.1              IRanges_2.24.1           
 [95] compiler_4.0.2            ggstatsplot_0.8.0        
 [97] crayon_1.5.1              minqa_1.2.4              
 [99] htmltools_0.5.2           mgcv_1.8-31              
[101] mc2d_0.1-21               later_1.3.0              
[103] tzdb_0.3.0                lubridate_1.8.0          
[105] DBI_1.1.1                 SuppDists_1.1-9.5        
[107] kSamples_1.2-9            tweenr_1.0.2             
[109] dbplyr_2.1.1              MASS_7.3-51.6            
[111] boot_1.3-25               Matrix_1.2-18            
[113] ade4_1.7-18               car_3.0-12               
[115] wesanderson_0.3.6         cli_3.0.1                
[117] parallel_4.0.2            insight_0.14.4           
[119] igraph_1.2.9              pkgconfig_2.0.3          
[121] statsExpressions_1.1.0    xml2_1.3.2               
[123] paletteer_1.4.0           foreach_1.5.2            
[125] bslib_0.3.1               blme_1.0-5               
[127] multtest_2.46.0           XVector_0.30.0           
[129] estimability_1.3          rvest_1.0.1              
[131] snakecase_0.11.0          digest_0.6.29            
[133] parameters_0.14.0         janitor_2.1.0            
[135] Biostrings_2.58.0         rmarkdown_2.11           
[137] cellranger_1.1.0          shiny_1.7.1              
[139] gtools_3.9.2              nloptr_1.2.2.3           
[141] lifecycle_1.0.1           nlme_3.1-148             
[143] jsonlite_1.7.2            Rhdf5lib_1.12.1          
[145] carData_3.0-5             fansi_0.5.0              
[147] pillar_1.7.0              fastmap_1.1.0            
[149] httr_1.4.2                survival_3.1-12          
[151] glue_1.5.1                bayestestR_0.11.0        
[153] iterators_1.0.14          ggforce_0.3.3            
[155] stringi_1.7.6             sass_0.4.0               
[157] performance_0.7.3         rematch2_2.1.2           
[159] memoise_2.0.0             Rmpfr_0.8-4              
[161] ape_5.5                  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
