<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>LMM for taxa</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">LMM for taxa</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">3/29/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1930f16a52ac738ba1a12980b405691720e9a062targetblank1930f16a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/1930f16a52ac738ba1a12980b405691720e9a062" target="_blank">1930f16</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1930f16a52ac738ba1a12980b405691720e9a062targetblank1930f16a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/1930f16a52ac738ba1a12980b405691720e9a062" target="_blank">1930f16</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   analysis/330_TaxaLevel_LMM.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/330_TaxaLevel_LMM.Rmd</code>) and HTML (<code>docs/330_TaxaLevel_LMM.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/532dfdf5668213cc738532604fa0c429e355eab4/analysis/330_TaxaLevel_LMM.Rmd" target="_blank">532dfdf</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-29
</td>
<td>
change ordinal
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = here::here())</code></pre>
<div id="taxa-level-lmm" class="section level2">
<h2>Taxa level LMM</h2>
<p>The main purpose of this script is to extract signals from Linear-Mixed-Models(LMM) from longitudinal designs of data. For each Taxa, at a tax_level, we’ll run a LMM through the longitudinal dimension and either, rank taxa according to their longitudinal increase/decrease or to the highest differences between a group variable in cat_var.</p>
<p>Try different data transformation approaches:</p>
<ul>
<li>Relative abudnaces(RA)</li>
<li>raw counts, with previous rarefaction at a certain number of reads, using metaphlan_counts object</li>
<li>clr transformed data for counts, using microbiome::transform()</li>
</ul>
<pre class="r"><code>require(phyloseq)
require(ggstatsplot)
require(ggplot2)
require(here)
library(metar)</code></pre>
<pre><code>suggested package selbal ✓suggested package DataExplorer ✓suggested package lme4 ✓suggested package merTools ✓</code></pre>
<pre class="r"><code>library(tidyverse)

bucket&lt;-&quot;s3://mistral-wp6-advanz4/&quot;
bucketDir&lt;-&quot;metagenome/WMGS&quot;

# Load package from local copy

devtools::load_all(here::here(&quot;..&quot;,&quot;WMGSPipeline&quot;))### Change to require/library/load local package</code></pre>
<pre><code>suggested package selbal ✓suggested package DataExplorer ✓suggested package lme4 ✓suggested package merTools ✓</code></pre>
<pre class="r"><code>mymre &lt;- readRDS(here::here(&quot;data&quot;, &quot;mre.rds&quot;))
# mymre&lt;-aws.s3::s3readRDS(bucket=bucket,object = paste0(bucketDir,&quot;/MREObject.rds&quot;)) ## Change as needed
# mymre@metadata@categorical_vals &lt;-
#   here::here(&quot;Metadata&quot;,&quot;CategoricalVariables.txt&quot;) %&gt;%
#   read.table(sep=&quot;\t&quot;) %&gt;%
#   setNames(c(&quot;CategoricalVariable&quot;,&quot;PaletteName&quot;)) %&gt;%
#   as_tibble()
phy &lt;- metar::get_phyloseq(mymre, type=&quot;metaphlan_counts&quot;)
id_var &lt;- &quot;record_id&quot;
long_var &lt;- &quot;time_point&quot;
cat_df &lt;- metar::get_cat(mymre) %&gt;%
  filter(!CategoricalVariable == &quot;CD4diff_96&quot;)
cat_vector &lt;- cat_df$CategoricalVariable

metadata &lt;-
  metar::get_meta(mymre) %&gt;%
  mutate(rnames = SampleID) %&gt;%
  tibble::column_to_rownames(&quot;rnames&quot;)

abundance &lt;-   
  metar:::get_merged_df(phy, tax_level=&quot;Species&quot;,sample_data_df = F) %&gt;%
  full_join(
    metadata[,c(&quot;SampleID&quot;,id_var,long_var, cat_vector)], by=&quot;SampleID&quot;) %&gt;%
  dplyr::rename(id_var = !!sym(id_var),
                long_var = !!sym(long_var))</code></pre>
</div>
<div id="including-plots" class="section level2">
<h2>Including Plots</h2>
<pre class="r"><code>num_vector &lt;- abundance %&gt;% 
  dplyr::select(-id_var, -cat_vector, -long_var, -SampleID) %&gt;%
  colnames()

get_multiLMM_table &lt;- function(df,num_vector, sam_var, cat_vector, long_var, id_var){
  
  myUnifiedModelList &lt;- cat_vector %&gt;%
  set_names() %&gt;%
  
  map(function(cat_var) {
    # print(cat_var)
    num_vector %&gt;%
      set_names() %&gt;%
      
      map_dfr(function(num_var) {
        
        dat &lt;-
          df %&gt;%
          select(sam_var, id_var, long_var, cat_var, num_var) %&gt;%
          dplyr::rename(num_var = !!sym(num_var),
                        cat_var = !!sym(cat_var))
        
        
        cat_levels &lt;- #Count nº of levels in cat_var
          dat %&gt;% 
          dplyr::select(id_var, cat_var) %&gt;%
          unique() %&gt;%
          dplyr::group_by(cat_var) %&gt;%
          dplyr::summarize(n=n()) %&gt;%
          dplyr::filter(n&gt;1,
                        !is.na(cat_var)) %&gt;%
          dplyr::pull(cat_var) 
        

        dat &lt;- # filter NAs and bad entries
          dat %&gt;%
          dplyr::filter(cat_var %in% cat_levels,
                 !is.na(cat_var))
        
        
        if (length(!is.na(cat_levels)) &gt;= 2){ # Check if cat_var has at least 2 levels
        
        myUnifiedModel &lt;- 
          lmerTest::lmer(formula = &quot;num_var ~ long_var * cat_var + (1|id_var)&quot;, data = dat)
        
        finalTable &lt;-
          anova(myUnifiedModel) %&gt;%

          # magrittr::extract2(&quot;coefficients&quot;) %&gt;%
          as.data.frame() %&gt;%
          tibble::rownames_to_column(var = &quot;term&quot;) %&gt;%
          dplyr::filter(term == &quot;long_var:cat_var&quot;) %&gt;%
          magrittr::set_names(c(&quot;term&quot;,&quot;SumSq&quot;, &quot;MeanSq&quot;, &quot;df&quot;, &quot;DenDF&quot;, &quot;Fval&quot;,&quot;pval&quot;)) %&gt;%
          dplyr::mutate(pval = round(pval, 3)) %&gt;%
          mutate(term = num_var) %&gt;%
          rename(Species = term)
        

        return(finalTable)
        
        } else {return(NA)}
      })
    
  })

}


testTable &lt;- get_multiLMM_table(df = abundance,
                                num_vector = num_vector,
                                sam_var = &quot;SampleID&quot;,
                                cat_vector = cat_vector, 
                                long_var = long_var,
                                id_var = id_var)


testTable$group %&gt;% 
  arrange(desc(Fval)) %&gt;%
  mutate(padj = p.adjust(pval, method= &quot;fdr&quot;)) %&gt;%
  filter(pval &lt; 0.05)</code></pre>
<pre><code>                            Species        SumSq       MeanSq df    DenDF
1        Parabacteroides_sp_CAG_409    101419099    101419099  1 211.8447
2                Bacteroides_caccae 265245999380 265245999380  1 199.6204
3               Eubacterium_siraeum 116538338716 116538338716  1 190.0998
4                 Dialister_invisus    474806330    474806330  1 267.0000
5             Klebsiella_pneumoniae  67153138463  67153138463  1 267.0000
6              Klebsiella_variicola   4949424170   4949424170  1 267.0000
7            Brachyspira_sp_CAG_700    559603398    559603398  1 218.9961
8               Eubacterium_eligens  50024301350  50024301350  1 207.2899
9  Parasutterella_excrementihominis    243040406    243040406  1 213.0428
10                 Blautia_hansenii   2272655130   2272655130  1 211.2817
11        Parabacteroides_johnsonii   2386360576   2386360576  1 185.0711
12            Sutterella_parvirubra      1572281      1572281  1 210.1913
13             Clostridium_scindens    299107779    299107779  1 223.3821
14            Megamonas_hypermegale    311998087    311998087  1 174.1880
15             Megasphaera_elsdenii   2698127550   2698127550  1 197.6824
16       Streptococcus_pasteurianus     83631262     83631262  1 209.6904
17             Bacteroides_plebeius   1921701886   1921701886  1 207.0133
18       Methanobrevibacter_smithii 108156809036 108156809036  1 170.6128
19         Haemophilus_sp_HMSC71H05      1247713      1247713  1 196.7644
       Fval  pval  padj
1  8.542042 0.004 0.708
2  8.498982 0.004 0.708
3  7.869238 0.006 0.708
4  7.092457 0.008 0.708
5  6.384543 0.012 0.708
6  6.364770 0.012 0.708
7  6.181924 0.014 0.708
8  5.411539 0.021 0.767
9  4.591718 0.033 0.767
10 4.526867 0.035 0.767
11 4.449466 0.036 0.767
12 4.358263 0.038 0.767
13 4.341019 0.038 0.767
14 4.140342 0.043 0.767
15 4.088398 0.045 0.767
16 4.063380 0.045 0.767
17 4.016833 0.046 0.767
18 3.920464 0.049 0.767
19 3.919029 0.049 0.767</code></pre>
<p>As this was a proper LMM using cat_var as fixed effect, we can’t get an overall “slope”, as the model computes an estimate for each cat_var level. Hence, as a proxy for effect size we’ll ANOVA F-statistic which is calculated as the division betwen inter-group mean variance and intra-group variance. From the significant species we’ll be able to greatly reduce the table so we can comfortably estimate slopes for each one.</p>
<p>By looking at group test statistic, we see a few species with a significant time:group interaction, which becomes unsignificant even with a permisive fdr correction.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5       testthat_3.1.2    forcats_0.5.1     stringr_1.4.0    
 [5] dplyr_1.0.8       purrr_0.3.4       readr_2.1.0       tidyr_1.2.0      
 [9] tibble_3.1.6      tidyverse_1.3.1   here_1.0.1        ggplot2_3.3.5    
[13] ggstatsplot_0.9.0 phyloseq_1.34.0  

loaded via a namespace (and not attached):
  [1] estimability_1.3            coda_0.19-4                
  [3] bit64_4.0.5                 knitr_1.37                 
  [5] DelayedArray_0.16.3         multcomp_1.4-17            
  [7] wesanderson_0.3.6           data.table_1.14.2          
  [9] RCurl_1.98-1.5              DataExplorer_0.8.2         
 [11] generics_0.1.2              BiocGenerics_0.36.1        
 [13] callr_3.7.0                 TH.data_1.1-0              
 [15] usethis_2.1.3               mixOmics_6.14.1            
 [17] future_1.23.0               correlation_0.7.1.1        
 [19] bit_4.0.4                   tzdb_0.2.0                 
 [21] bayesm_3.1-4                xml2_1.3.2                 
 [23] lubridate_1.8.0             httpuv_1.6.3               
 [25] SummarizedExperiment_1.20.0 assertthat_0.2.1           
 [27] WRS2_1.1-3                  xfun_0.30                  
 [29] hms_1.1.1                   jquerylib_0.1.4            
 [31] evaluate_0.15               promises_1.2.0.1           
 [33] DEoptimR_1.0-9              fansi_1.0.2                
 [35] dbplyr_2.1.1                readxl_1.3.1               
 [37] igraph_1.2.11               DBI_1.1.1                  
 [39] htmlwidgets_1.5.4           reshape_0.8.8              
 [41] tensorA_0.36.2              rARPACK_0.11-0             
 [43] kSamples_1.2-9              stats4_4.0.5               
 [45] Rmpfr_0.8-7                 paletteer_1.4.0            
 [47] ellipsis_0.3.2              RSpectra_0.16-0            
 [49] ggpubr_0.4.0                backports_1.4.1            
 [51] insight_0.14.5.1            permute_0.9-7              
 [53] compositions_2.0-2          libcoin_1.0-9              
 [55] MatrixGenerics_1.2.1        vctrs_0.3.8                
 [57] Biobase_2.50.0              remotes_2.4.1              
 [59] Cairo_1.5-12.2              abind_1.4-5                
 [61] cachem_1.0.6                withr_2.5.0                
 [63] ggforce_0.3.3               robustbase_0.93-9          
 [65] vroom_1.5.7                 emmeans_1.7.0              
 [67] vegan_2.5-7                 prettyunits_1.1.1          
 [69] cluster_2.1.2               ape_5.6-2                  
 [71] crayon_1.5.0                ellipse_0.4.2              
 [73] pkgconfig_2.0.3             SuppDists_1.1-9.5          
 [75] GenomeInfoDb_1.26.7         tweenr_1.0.2               
 [77] nlme_3.1-153                statsExpressions_1.2.0     
 [79] pkgload_1.2.4               blme_1.0-5                 
 [81] devtools_2.4.2              rlang_1.0.2                
 [83] globals_0.14.0              lifecycle_1.0.1            
 [85] sandwich_3.0-1              selbal_0.1.0               
 [87] modelr_0.1.8                cellranger_1.1.0           
 [89] rprojroot_2.0.2             polyclip_1.10-0            
 [91] matrixStats_0.61.0          datawizard_0.2.1.9000      
 [93] Matrix_1.3-4                mc2d_0.1-21                
 [95] carData_3.0-5               Rhdf5lib_1.12.1            
 [97] boot_1.3-28                 zoo_1.8-9                  
 [99] reprex_2.0.1                whisker_0.4                
[101] GlobalOptions_0.1.2         processx_3.5.2             
[103] png_0.1-7                   rjson_0.2.20               
[105] PMCMRplus_1.9.3             bitops_1.0-7               
[107] parameters_0.15.0.1         pROC_1.18.0                
[109] rhdf5filters_1.2.1          Biostrings_2.58.0          
[111] workflowr_1.7.0             shape_1.4.6                
[113] multcompView_0.1-8          coin_1.4-2                 
[115] arm_1.12-2                  parallelly_1.28.1          
[117] lefser_1.0.0                rstatix_0.7.0              
[119] S4Vectors_0.28.1            ggsignif_0.6.3             
[121] scales_1.1.1                memoise_2.0.0              
[123] magrittr_2.0.2              plyr_1.8.6                 
[125] zlibbioc_1.36.0             compiler_4.0.5             
[127] RColorBrewer_1.1-2          clue_0.3-60                
[129] lme4_1.1-28                 cli_3.2.0                  
[131] ade4_1.7-18                 XVector_0.30.0             
[133] lmerTest_3.1-3              listenv_0.8.0              
[135] patchwork_1.1.1             ps_1.6.0                   
[137] ggside_0.1.3                MASS_7.3-54                
[139] mgcv_1.8-38                 tidyselect_1.1.2           
[141] stringi_1.7.6               yaml_2.3.5                 
[143] ggrepel_0.9.1               grid_4.0.5                 
[145] sass_0.4.0                  tools_4.0.5                
[147] parallel_4.0.5              circlize_0.4.13            
[149] rstudioapi_0.13             modules_0.10.0             
[151] foreach_1.5.2               logging_0.10-108           
[153] git2r_0.29.0                gridExtra_2.3              
[155] farver_2.1.0                digest_0.6.29              
[157] shiny_1.7.1                 networkD3_0.4              
[159] BWStest_0.2.2               Rcpp_1.0.8                 
[161] GenomicRanges_1.42.0        car_3.0-12                 
[163] broom_0.7.12                performance_0.8.0.1        
[165] later_1.3.0                 httr_1.4.2                 
[167] ComplexHeatmap_2.6.2        colorspace_2.0-3           
[169] rvest_1.0.2                 brio_1.1.3                 
[171] fs_1.5.0                    IRanges_2.24.1             
[173] splines_4.0.5               rematch2_2.1.2             
[175] multtest_2.46.0             sessioninfo_1.2.1          
[177] xtable_1.8-4                gmp_0.6-2.1                
[179] jsonlite_1.8.0              nloptr_1.2.2.3             
[181] corpcor_1.6.10              zeallot_0.1.0              
[183] modeltools_0.2-23           R6_2.5.1                   
[185] broom.mixed_0.2.7           pillar_1.7.0               
[187] htmltools_0.5.2             mime_0.12                  
[189] BiocParallel_1.24.1         glue_1.6.2                 
[191] fastmap_1.1.0               minqa_1.2.4                
[193] codetools_0.2-18            pkgbuild_1.2.0             
[195] mvtnorm_1.1-3               utf8_1.2.2                 
[197] lattice_0.20-45             bslib_0.3.1                
[199] logger_0.2.2                numDeriv_2016.8-1.1        
[201] survival_3.2-13             rmarkdown_2.11             
[203] desc_1.4.1                  biomformat_1.18.0          
[205] merTools_0.5.2              munsell_0.5.0              
[207] GenomeInfoDbData_1.2.4      GetoptLong_1.0.5           
[209] rhdf5_2.34.0                iterators_1.0.14           
[211] haven_2.4.3                 reshape2_1.4.4             
[213] gtable_0.3.0                bayestestR_0.11.5.1        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
