<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>Baseline microbiome vs Immune response</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Baseline microbiome vs Immune response</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">3/29/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree9beba4fea9875a042a02bc7b625c64366d94fd2ctargetblank9beba4fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/9beba4fea9875a042a02bc7b625c64366d94fd2c" target="_blank">9beba4f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree9beba4fea9875a042a02bc7b625c64366d94fd2ctargetblank9beba4fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/9beba4fea9875a042a02bc7b625c64366d94fd2c" target="_blank">9beba4f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/sample_cluster_labels.csv
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/silhouette_coefs.csv
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/silhouette_plt.pdf

Unstaged changes:
    Modified:   advanz4.Rproj
    Modified:   analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/bray_stressplot.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_cluster.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD4.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD8.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD8_CD38_DR.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CRP.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_IL6.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_TNFa.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_sCD14.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/nmds_adonis_results.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/nmds_coordinates.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD4.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD8.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD8_CD38_DR.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CRP.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_IL6.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_TNFa.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_sCD14.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd</code>) and HTML (<code>docs/335_BLMicrobiome_vs_ImmuneResponse.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/9beba4fea9875a042a02bc7b625c64366d94fd2c/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">9beba4f</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
conflict solved on BL_Microbiome.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f213c10f0f5907550448cf2733d862a4929fa1cd/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">f213c10</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
reworked basal analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0527e8849e3d4382a8cca0c0cf7365901189897f/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">0527e88</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-19
</td>
<td>
Added buld_metadata.Rmd to analysis. Added MetadataManagement to .gitignore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/8eaf17d0113a268e4cbf0467f54ce48400f4f3ac/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">8eaf17d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
improved deltaCD4/CD8/ratio plots and checked missing samples in 335_BLMicrobiome.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/8eaf17d0113a268e4cbf0467f54ce48400f4f3ac/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">8eaf17d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
improved deltaCD4/CD8/ratio plots and checked missing samples in 335_BLMicrobiome.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a3e13ccbf07d4452eea311fc1e448e0dee16971c/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">a3e13cc</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-04
</td>
<td>
diversity comparisons between CD4delta and microbiome at basal (335_BLMicrobiome…RMD)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c20090b3a3c8e714d8be91e4ea67aa5d3ff7c522/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">c20090b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-01
</td>
<td>
Compiled taxa ranking html + several bug fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1930f16a52ac738ba1a12980b405691720e9a062/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">1930f16</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-29
</td>
<td>
Create 335_BLMicrobiome_vs_ImmuneResponse.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># mymre &lt;- here::here(&quot;data&quot;,&quot;mre.rds&quot;) %&gt;%
#   readRDS(.) 

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;)
metadata &lt;-get_meta(mymre)

cat_vector &lt;- get_cat(mymre) %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- get_lon(mymre) %&gt;%
  pull(LongitudinalVariable)

id_var &lt;- get_lon(mymre) %&gt;%
  pull(LinkVariable)</code></pre>
<div id="baseline-differences-in-microbiome" class="section level1">
<h1>Baseline differences in microbiome</h1>
<div id="alpha-diversity" class="section level2">
<h2>Alpha diversity</h2>
<pre class="r"><code>GR &lt;- get_diversity(mymre, type = &quot;igc&quot;, res_slot = &quot;dataTable&quot;) 



Threshold &lt;- GR %&gt;%
  dplyr::select(SampleID, NumberMappedReads) %&gt;%
  unique() %&gt;%
  summarise(Q = quantile(NumberMappedReads,0.02)) %&gt;% ### Again, you are using the full data table to calculate percentiles. This is wrong.
  as.numeric()


richness_df &lt;-
  GR %&gt;%
  filter(ReadCountReal &gt;= Threshold) %&gt;% ### Get the Mapped Read count (per-sample) just above the threshold
  group_by(SampleID) %&gt;%
  summarise(across(everything(), min)) %&gt;%
  ungroup() %&gt;%
  select(SampleID, GeneNumber) %&gt;%
  left_join(., metadata, by=&quot;SampleID&quot;) %&gt;%
  select(id_var = !!sym(id_var),
         long_var = !!sym(long_var),
         cat_vector,
         GeneNumber) %&gt;%
  filter(long_var == 0) 


c(&quot;CD4diff_48&quot;, &quot;CD8diff_48&quot;) %&gt;%
  set_names() %&gt;% 
  map(function(cat_var){
    richness_df %&gt;%
      dplyr::select(id_var,
                    cat_var = !!sym(cat_var),
                    GeneNumber) %&gt;%
            filter(!is.na(cat_var)) %&gt;%
      ggplot(., aes(x=cat_var, y=GeneNumber, fill = cat_var))+
      geom_violin(draw_quantiles = T) +
      # geom_boxplot(fill=&quot;lightgrey&quot;, width=0.5, na.rm=T) +
      geom_point() +
      scale_fill_brewer(palette=&quot;Set1&quot;) +
      ggpubr::stat_compare_means(method=&quot;wilcox&quot;,label = &quot;p.signif&quot;, hide.ns = T) +
      theme_bw() + 
      labs(fill = cat_var)
    
  })</code></pre>
<pre><code>$CD4diff_48</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/baseline_richness-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8diff_48</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/baseline_richness-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>No differences in basal Gene richness were found between patients with an increase of CD4 higher than 50 counts/ul and whose with lower increase. The same could be observed in CD8 counts.</p>
<pre class="r"><code>ids &lt;- metadata %&gt;%
  dplyr::group_by(record_id) %&gt;%
  dplyr::filter(time_point %in% c(0,48)) %&gt;%
  dplyr::filter(!is.na(CD4)) %&gt;%
  dplyr::filter(!is.na(CD8)) %&gt;% 
  dplyr::arrange(time_point) %&gt;%
  dplyr::summarise(n=n()) %&gt;%
  dplyr::filter(n==2) %&gt;%
  dplyr::pull(record_id)


deltas &lt;- metadata %&gt;% 
  # dplyr::filter(record_id %in% ids) %&gt;%
  dplyr::filter(time_point %in% c(0,48,96)) %&gt;% 
  dplyr::select(record_id, time_point,CD4,CD8) %&gt;%
  dplyr::mutate(ratio = CD4/CD8) %&gt;% 
  dplyr::group_by(record_id) %&gt;% 
  dplyr::arrange(record_id,time_point) %&gt;%
  tidyr::pivot_wider(id_cols = record_id, names_from= time_point, values_from = c(CD4,CD8, ratio)) %&gt;%
  dplyr::mutate(
    deltaCD4_48 = CD4_48 - CD4_0, 
    deltaCD8_48 = CD8_48 - CD8_0,
    deltaCD4_96 = CD4_96 - CD4_0, 
    deltaCD8_96 = CD8_96 - CD8_0,
    deltaratio_48 = ratio_48 - ratio_0,
    deltaratio_96 = ratio_96 - ratio_0) %&gt;%
  dplyr::select(record_id, contains(&quot;delta&quot;))



deltas %&gt;%
  pivot_longer(cols=c(deltaCD4_48,deltaCD8_48,deltaCD4_96,deltaCD8_96, deltaratio_48, deltaratio_96)) %&gt;%
  mutate(term = str_remove_all(name, &quot;delta|_48|_96&quot;),
         tp = str_remove_all(name, &quot;.*(?&lt;=_)&quot;)) %&gt;%
  ggplot(., aes(color=tp)) +
  geom_density(aes(x=value), alpha=0.4) +
  facet_wrap(~ term, scales=&quot;free&quot;, ncol = 2) +
  # geom_density(data=pivot_longer(cols=c(deltaCD4,deltaCD8)),aes(x=value), alpha=0.4) +
  theme_bw() +
  scale_color_brewer(palette = &quot;Set1&quot;)+

  labs(color=&quot;week&quot;)</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/Immune_delta_Distributions-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>deltas %&gt;% 
  select(contains(&quot;delta&quot;)) %&gt;% 
  colnames() %&gt;% 
  set_names() %&gt;% 
  map(~{
    summary(deltas[,.x])
  })</code></pre>
<pre><code>$record_id
  record_id        
 Length:93         
 Class :character  
 Mode  :character  

$deltaCD4_48
  deltaCD4_48   
 Min.   : 17.0  
 1st Qu.:133.6  
 Median :189.0  
 Mean   :219.9  
 3rd Qu.:300.0  
 Max.   :683.0  
 NA&#39;s   :34     

$deltaCD8_48
  deltaCD8_48    
 Min.   :-743.6  
 1st Qu.: -19.5  
 Median : 311.0  
 Mean   : 282.5  
 3rd Qu.: 521.5  
 Max.   :1975.0  
 NA&#39;s   :34      

$deltaCD4_96
  deltaCD4_96 
 Min.   : NA  
 1st Qu.: NA  
 Median : NA  
 Mean   :NaN  
 3rd Qu.: NA  
 Max.   : NA  
 NA&#39;s   :93   

$deltaCD8_96
  deltaCD8_96 
 Min.   : NA  
 1st Qu.: NA  
 Median : NA  
 Mean   :NaN  
 3rd Qu.: NA  
 Max.   : NA  
 NA&#39;s   :93   

$deltaratio_48
 deltaratio_48    
 Min.   :0.00066  
 1st Qu.:0.12393  
 Median :0.18854  
 Mean   :0.38909  
 3rd Qu.:0.28660  
 Max.   :8.83744  
 NA&#39;s   :34       

$deltaratio_96
 deltaratio_96
 Min.   : NA  
 1st Qu.: NA  
 Median : NA  
 Mean   :NaN  
 3rd Qu.: NA  
 Max.   : NA  
 NA&#39;s   :93   </code></pre>
<p>The last piece of output (summaries) tell there is a great ammount of missing data. This is due to a combination of many patients missing a sample and a few not having entries for analytical data:</p>
<pre class="r"><code>no0 &lt;- metadata %&gt;% 
  filter(!record_id %in% metadata[metadata$time_point == 0, ]$record_id) %&gt;%
  pull(record_id)

##patients without lymphocite data but with microbiome sample
no48_CD4 &lt;- metadata %&gt;%
  filter(is.na(CD4) ) %&gt;%
  filter(time_point == 48) %&gt;% 
  pull(record_id)  

no48_CD8 &lt;- metadata %&gt;%
  filter(is.na(CD8) &amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 48) %&gt;% 
  pull(record_id)  


no96_CD4 &lt;- metadata %&gt;%
  filter(is.na(CD4)&amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 96) %&gt;% 
  pull(record_id)

no96_CD8 &lt;- metadata %&gt;%
  filter(is.na(CD8)&amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 96) %&gt;% 
  pull(record_id)


# patients without sample
noS48 &lt;- metadata %&gt;%
  filter(!record_id %in% metadata[metadata$time_point == 48, ]$record_id) %&gt;% 
  pull(record_id)

noS96 &lt;- metadata %&gt;%
  filter(!record_id %in% metadata[metadata$time_point == 96, ]$record_id) %&gt;% 
  pull(record_id)



totalMissing_CD4_48 &lt;- length(unique(c(no0, no48_CD4, noS48)))
totalMissing_CD4_96 &lt;- length(unique(c(no0,no96_CD4, noS96)))
totalMissing_CD8_48 &lt;- length(unique(c(no0, no48_CD8, noS48)))
totalMissing_CD8_96 &lt;- length(unique(c(no0,no96_CD8, noS96)))

totalMissing_CD4_48</code></pre>
<pre><code>[1] 36</code></pre>
<pre class="r"><code>totalMissing_CD4_96</code></pre>
<pre><code>[1] 95</code></pre>
<pre class="r"><code>totalMissing_CD8_96</code></pre>
<pre><code>[1] 95</code></pre>
<pre class="r"><code>totalMissing_CD8_96</code></pre>
<pre><code>[1] 95</code></pre>
<p>Adding the different kind of missing data checks out with the summaries. The biggest issue came with patients not having no sample whatsoever for such timepoints. Of which: 25 didn’t have a basal sample, 22 didnt have sample for week 48 and 30 for week 96.</p>
<p>delta CD4 threshold was too low. To keep the study clinically interpretable We’ll do a cutoff as a round number around the mean. In this case &lt;150/&gt;150 according to the mITT design results of the parent study.</p>
<p>Basal gene richness vs delta CD4_48 as continuous variables</p>
<pre class="r"><code>basal_samples &lt;- metadata %&gt;%
  filter(time_point == 0) %&gt;% 
  pull(SampleID) %&gt;% 
  as.character()

basal_mre &lt;- filter_samples(mymre, basal_samples)

  
  
basal_mre@metadata@categorical_vals &lt;- get_cat(mymre) %&gt;% 
  filter(CategoricalVariable %in% &quot;group&quot;)


basal_mre &lt;- metar::metaphlan_nmds(basal_mre, tax_level=&quot;Species&quot;)

mynmds &lt;- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$group

library(patchwork)


mynmds &amp; 
    scale_color_brewer(palette=&quot;Set1&quot;) &amp;
    scale_fill_brewer(palette = &quot;Set1&quot;)</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_numeric</code></pre>
<pre><code>$CD4</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CD8_CD38_DR</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$CRP</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$IL6</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$TNFa</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$sCD14</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical </code></pre>
<pre><code>$group</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="baseline-microbiome-vs-immune-response" class="section level2">
<h2>Baseline microbiome vs Immune response</h2>
<p>The goal of this analysis is to describe the patients baseline microbiome vs immune response either using:</p>
<ul>
<li>Increases in CD4, categorized in a useful way (Delta_week48 for instance)</li>
<li>Decreased in CD8, categorized in a useful way</li>
</ul>
<p>It may be useful to try to classify patients solely on CD4/CD8/immune markers and the contrast this classification with microbiome composition</p>
<p>Also it is relevant to show that the baseline microbiom was NOT different between ARMs (DTG/DRVr)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252   
[3] LC_MONETARY=Spanish_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.1.1    metar_0.1.5        RColorBrewer_1.1-3 forcats_0.5.1     
 [5] stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4        readr_2.0.1       
 [9] tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                tidyselect_1.1.2         
  [3] lme4_1.1-27.1             htmlwidgets_1.5.4        
  [5] grid_4.0.2                gmp_0.6-2                
  [7] aws.signature_0.6.0       munsell_0.5.0            
  [9] effectsize_0.4.5          codetools_0.2-16         
 [11] miniUI_0.1.1.1            withr_2.5.0              
 [13] colorspace_2.0-2          Biobase_2.50.0           
 [15] phyloseq_1.34.0           logger_0.2.2             
 [17] highr_0.9                 knitr_1.38               
 [19] rstudioapi_0.13           stats4_4.0.2             
 [21] ipmisc_6.0.2              ggsignif_0.6.3           
 [23] labeling_0.4.2            emmeans_1.7.1-1          
 [25] git2r_0.28.0              polyclip_1.10-0          
 [27] bit64_4.0.5               farver_2.1.0             
 [29] datawizard_0.2.0.1        rhdf5_2.34.0             
 [31] rprojroot_2.0.3           coda_0.19-4              
 [33] vctrs_0.3.8               generics_0.1.2           
 [35] xfun_0.30                 BWStest_0.2.2            
 [37] R6_2.5.1                  arm_1.12-2               
 [39] BayesFactor_0.9.12-4.2    rhdf5filters_1.2.1       
 [41] cachem_1.0.1              reshape_0.8.8            
 [43] assertthat_0.2.1          vroom_1.5.7              
 [45] promises_1.2.0.1          networkD3_0.4            
 [47] scales_1.2.0              ggExtra_0.10.0           
 [49] merTools_0.5.2            gtable_0.3.0             
 [51] multcompView_0.1-8        workflowr_1.7.0          
 [53] rlang_1.0.2               MatrixModels_0.5-0       
 [55] zeallot_0.1.0             PMCMRplus_1.9.0          
 [57] splines_4.0.2             rstatix_0.7.0            
 [59] prismatic_1.0.0           broom_0.8.0              
 [61] yaml_2.3.5                reshape2_1.4.4           
 [63] abind_1.4-5               modelr_0.1.8             
 [65] backports_1.4.1           httpuv_1.6.3             
 [67] tools_4.0.2               ellipsis_0.3.2           
 [69] jquerylib_0.1.4           biomformat_1.18.0        
 [71] selbal_0.1.0              BiocGenerics_0.36.1      
 [73] WRS2_1.1-3                Rcpp_1.0.7               
 [75] plyr_1.8.6                base64enc_0.1-3          
 [77] zlibbioc_1.36.0           ggpubr_0.4.0             
 [79] pbapply_1.5-0             correlation_0.7.0        
 [81] S4Vectors_0.28.1          haven_2.3.1              
 [83] ggrepel_0.9.1             cluster_2.1.0            
 [85] fs_1.5.0                  DataExplorer_0.8.2       
 [87] magrittr_2.0.2            data.table_1.14.2        
 [89] reprex_2.0.1              mvtnorm_1.1-2            
 [91] whisker_0.4               hms_1.1.1                
 [93] mime_0.11                 evaluate_0.15            
 [95] xtable_1.8-4              pairwiseComparisons_3.1.6
 [97] broom.mixed_0.2.7         readxl_1.3.1             
 [99] IRanges_2.24.1            gridExtra_2.3            
[101] compiler_4.0.2            ggstatsplot_0.8.0        
[103] crayon_1.5.1              minqa_1.2.4              
[105] htmltools_0.5.2           mgcv_1.8-31              
[107] mc2d_0.1-21               later_1.3.0              
[109] tzdb_0.3.0                lubridate_1.8.0          
[111] aws.s3_0.3.21             DBI_1.1.1                
[113] SuppDists_1.1-9.5         kSamples_1.2-9           
[115] tweenr_1.0.2              dbplyr_2.1.1             
[117] MASS_7.3-51.6             boot_1.3-25              
[119] wesanderson_0.3.6         Matrix_1.2-18            
[121] ade4_1.7-18               car_3.0-12               
[123] permute_0.9-7             cli_3.0.1                
[125] parallel_4.0.2            insight_0.14.4           
[127] igraph_1.2.9              pkgconfig_2.0.3          
[129] statsExpressions_1.1.0    xml2_1.3.2               
[131] paletteer_1.4.0           foreach_1.5.2            
[133] bslib_0.3.1               blme_1.0-5               
[135] multtest_2.46.0           XVector_0.30.0           
[137] estimability_1.3          rvest_1.0.1              
[139] snakecase_0.11.0          digest_0.6.29            
[141] parameters_0.14.0         janitor_2.1.0            
[143] vegan_2.5-7               Biostrings_2.58.0        
[145] rmarkdown_2.11            cellranger_1.1.0         
[147] curl_4.3                  shiny_1.7.1              
[149] gtools_3.9.2              nloptr_1.2.2.3           
[151] lifecycle_1.0.1           nlme_3.1-148             
[153] jsonlite_1.7.2            Rhdf5lib_1.12.1          
[155] carData_3.0-5             fansi_0.5.0              
[157] pillar_1.7.0              lattice_0.20-41          
[159] fastmap_1.1.0             httr_1.4.2               
[161] survival_3.1-12           glue_1.5.1               
[163] bayestestR_0.11.0         iterators_1.0.14         
[165] bit_4.0.4                 ggforce_0.3.3            
[167] performance_0.7.3         stringi_1.7.6            
[169] sass_0.4.0                rematch2_2.1.2           
[171] memoise_2.0.0             Rmpfr_0.8-4              
[173] ape_5.5                  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
