<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>Baseline microbiome vs Immune response</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Baseline microbiome vs Immune response</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">3/29/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1badc06452e7e1c9a1b53a7cfae2350810fa594atargetblank1badc06a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/1badc06452e7e1c9a1b53a7cfae2350810fa594a" target="_blank">1badc06</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree1badc06452e7e1c9a1b53a7cfae2350810fa594atargetblank1badc06a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/1badc06452e7e1c9a1b53a7cfae2350810fa594a" target="_blank">1badc06</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  Metadata/2022_04_21_clean_metadata_LIMS.csv
    Untracked:  analysis/figure/
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD4after_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD4diff_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD8after_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD8diff_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD4after_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD4diff_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD8after_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD8diff_96.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_ADV.pdf
    Untracked:  output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_ADV.pdf

Unstaged changes:
    Deleted:    Metadata/.~lock.2022_04_01_clean_metadata_LIMS.csv#
    Modified:   Metadata/CategoricalVariables.txt
    Modified:   Metadata/LongitudinalVariables.txt
    Deleted:    Metadata/MetadataManagement/metadata.csv
    Modified:   Metadata/NumericalVariables.txt
    Modified:   analysis/04_build_metadata.Rmd
    Modified:   analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd
    Modified:   code/flowchart_sample_count.Rmd
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/bray_stressplot.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD4after_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD4diff_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD8after_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_CD8diff_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_center.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_cluster.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_ethnic_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_gender.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/categorical_nmds_risk_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD4after_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD4diff_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD8after_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_CD8diff_48.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_center.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_ethnic_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_gender.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_categorical_nmds_risk_group.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD4.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD8.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CD8_CD38_DR.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/cluster_numeric_nmds_CRP.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/nmds_adonis_results.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/nmds_coordinates.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD4.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD8.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CD8_CD38_DR.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/numeric_nmds_CRP.pdf
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/sample_cluster_labels.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/silhouette_coefs.csv
    Modified:   output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/Species/silhouette_plt.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd</code>) and HTML (<code>docs/335_BLMicrobiome_vs_ImmuneResponse.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/8bc62caa93c57e033ff55d41024cd8dbe0c0957b/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">8bc62ca</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-20
</td>
<td>
Fixed problem that arose when incorporating Adenovirus counts to metadata.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1f702e5d58b41215d4eaad8fd8796f88e0bb616/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">e1f702e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
added NMDS section to basal study
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e1f702e5d58b41215d4eaad8fd8796f88e0bb616/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">e1f702e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
added NMDS section to basal study
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/9beba4fea9875a042a02bc7b625c64366d94fd2c/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">9beba4f</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
conflict solved on BL_Microbiome.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f213c10f0f5907550448cf2733d862a4929fa1cd/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">f213c10</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
<td>
reworked basal analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0527e8849e3d4382a8cca0c0cf7365901189897f/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">0527e88</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-19
</td>
<td>
Added buld_metadata.Rmd to analysis. Added MetadataManagement to .gitignore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/8eaf17d0113a268e4cbf0467f54ce48400f4f3ac/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">8eaf17d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
improved deltaCD4/CD8/ratio plots and checked missing samples in 335_BLMicrobiome.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/8eaf17d0113a268e4cbf0467f54ce48400f4f3ac/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">8eaf17d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
improved deltaCD4/CD8/ratio plots and checked missing samples in 335_BLMicrobiome.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/93ba24c7c7e2213da3d5f2411ecff5580c1cad53/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">93ba24c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
<td>
added delta CD4/CD8 density plot to 335_BLMicrobiome.RMd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a3e13ccbf07d4452eea311fc1e448e0dee16971c/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">a3e13cc</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-04
</td>
<td>
diversity comparisons between CD4delta and microbiome at basal (335_BLMicrobiome…RMD)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c20090b3a3c8e714d8be91e4ea67aa5d3ff7c522/docs/335_BLMicrobiome_vs_ImmuneResponse.html" target="_blank">c20090b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-01
</td>
<td>
Compiled taxa ranking html + several bug fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1930f16a52ac738ba1a12980b405691720e9a062/analysis/335_BLMicrobiome_vs_ImmuneResponse.Rmd" target="_blank">1930f16</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-29
</td>
<td>
Create 335_BLMicrobiome_vs_ImmuneResponse.Rmd
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># This chunk is only for executing a load_all on the reoot directory of the metar repository.
# This is a workaround for a bug present in metar::filter_samples in which fails to find an 
# inherited method for fun_input(), a step in filtering samples in the Gene Function Slot. This 
# chunk will be removed as soon as this issue is solved.

devtools::load_all(here::here(&quot;../&quot;,&quot;WMGSPipeline&quot;))</code></pre>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre class="r"><code># Load all function arguments should be the path to where metar is installed. Change it
# accordingly.</code></pre>
<pre class="r"><code># mymre &lt;- here::here(&quot;data&quot;,&quot;mre.rds&quot;) %&gt;%
#   readRDS(.)

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;)
# metadata &lt;-get_meta(mymre)
metadata &lt;- here::here(&quot;Metadata&quot;, &quot;2022_04_27_clean_metadata_LIMS.csv&quot;) %&gt;% 
  read.csv() 



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

id_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- tibble(metadata)
mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.), .before=1) %&gt;% 
  phyloseq::sample_data(.)

mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()</code></pre>
<pre class="r"><code>basal_meta &lt;- metadata %&gt;%
  filter(time_point == 0) %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.), .before=1)

basal_mre &lt;- filter_samples(mre = mymre, sample_ids =  basal_meta$SampleID)
# basal_mre@metadata@categorical_vals &lt;-remove_cat(basal_mre, &quot;CD4diff_96&quot;)


basal_mre@metadata@categorical_vals &lt;- get_cat(mymre) %&gt;% 
  filter(!CategoricalVariable %in% c(&quot;ethnic_group&quot;))


phyloseq::sample_data(basal_mre@taxa@metaphlan@phyloseq) &lt;- basal_meta
phyloseq::sample_data(basal_mre@taxa@metaphlan@phyloseq_sec) &lt;- basal_meta




# cat_vector %&gt;% 
#   set_names() %&gt;% 
#   map_chr(~{
#     metadata %&gt;% 
#       filter(time_point == 0) %&gt;% 
#       pull(!!sym(.x)) %&gt;% 
#       unique() %&gt;% 
#       length()
#   })</code></pre>
<div id="baseline-differences-in-microbiome" class="section level1">
<h1>Baseline differences in microbiome</h1>
<div id="alpha-diversity" class="section level2">
<h2>Alpha diversity</h2>
<pre class="r"><code>GR &lt;- get_diversity(mymre, type = &quot;igc&quot;, res_slot = &quot;dataTable&quot;) 



Threshold &lt;- GR %&gt;%
  dplyr::select(SampleID, NumberMappedReads) %&gt;%
  unique() %&gt;%
  summarise(Q = quantile(NumberMappedReads,0.02)) %&gt;% ### Again, you are using the full data table to calculate percentiles. This is wrong.
  as.numeric()


richness_df &lt;-
  GR %&gt;%
  filter(ReadCountReal &gt;= Threshold) %&gt;% ### Get the Mapped Read count (per-sample) just above the threshold
  group_by(SampleID) %&gt;%
  summarise(across(everything(), min)) %&gt;%
  ungroup() %&gt;%
  select(SampleID, GeneNumber) %&gt;%
  left_join(., metadata, by=&quot;SampleID&quot;) %&gt;%
  select(id_var = !!sym(id_var),
         long_var = !!sym(long_var),
         cat_vector,
         GeneNumber) %&gt;%
  filter(long_var == 0) 

# cat_vector %&gt;% 
#   set_names() %&gt;% 
#   map(function(cat_var){
#     richness_df %&gt;%
#       dplyr::select(id_var,
#                     cat_var = !!sym(cat_var),
#                     GeneNumber) %&gt;%
#             filter(!is.na(cat_var)) %&gt;%
#       ggplot(., aes(x=cat_var, y=GeneNumber, fill = cat_var))+
#       geom_violin(draw_quantiles = T) +
#       geom_boxplot(fill=&quot;lightgrey&quot;, width=0.5, na.rm=T) +
#       geom_point() +
#       scale_fill_brewer(palette=&quot;Set1&quot;) +
#       ggpubr::stat_compare_means(method=&quot;wilcox&quot;,label = &quot;p.signif&quot;, hide.ns = T) +
#       theme_bw() + 
#       labs(fill = cat_var)
#     
#   })

  
plotlist &lt;- cat_vector %&gt;% 
  set_names() %&gt;% 
  map(function(cat_var){
    richness_df %&gt;%
      dplyr::select(id_var,
                    cat_var = !!sym(cat_var),
                    GeneNumber) %&gt;%
            filter(!is.na(cat_var)) %&gt;%
            ggstatsplot::ggbetweenstats(
              data = .,
              x = cat_var,
              y = GeneNumber,
              bf.message = FALSE,
              type = &quot;nonparametric&quot;,
              pairwise.comparisons = T,
              xlab = cat_var, 
              ggtheme = ggplot2::theme_bw()
              
            )
  })</code></pre>
<p>No differences in basal Gene richness were found between patients with an increase of CD4 higher than 50 counts/ul and whose with lower increase. The same could be observed in CD8 counts.</p>
<pre class="r"><code>ids &lt;- metadata %&gt;%
  dplyr::group_by(record_id) %&gt;%
  dplyr::filter(time_point %in% c(0,48)) %&gt;%
  dplyr::filter(!is.na(CD4)) %&gt;%
  dplyr::filter(!is.na(CD8)) %&gt;% 
  dplyr::arrange(time_point) %&gt;%
  dplyr::summarise(n=n()) %&gt;%
  dplyr::filter(n==2) %&gt;%
  dplyr::pull(record_id)


deltas &lt;- metadata %&gt;% 
  # dplyr::filter(record_id %in% ids) %&gt;%
  dplyr::filter(time_point %in% c(0,48,96)) %&gt;% 
  dplyr::select(record_id, time_point,CD4,CD8) %&gt;%
  dplyr::mutate(ratio = CD4/CD8) %&gt;% 
  dplyr::group_by(record_id) %&gt;% 
  dplyr::arrange(record_id,time_point) %&gt;%
  tidyr::pivot_wider(id_cols = record_id, names_from= time_point, values_from = c(CD4,CD8, ratio)) %&gt;%
  dplyr::mutate(
    deltaCD4_48 = CD4_48 - CD4_0, 
    deltaCD8_48 = CD8_48 - CD8_0,
    deltaCD4_96 = CD4_96 - CD4_0, 
    deltaCD8_96 = CD8_96 - CD8_0,
    deltaratio_48 = ratio_48 - ratio_0,
    deltaratio_96 = ratio_96 - ratio_0) %&gt;%
  dplyr::select(record_id, contains(&quot;delta&quot;))



deltas %&gt;%
  pivot_longer(cols=c(deltaCD4_48,deltaCD8_48,deltaCD4_96,deltaCD8_96, deltaratio_48, deltaratio_96)) %&gt;%
  mutate(term = str_remove_all(name, &quot;delta|_48|_96&quot;),
         tp = str_remove_all(name, &quot;.*(?&lt;=_)&quot;)) %&gt;%
  ggplot(., aes(color=tp)) +
  geom_density(aes(x=value), alpha=0.4) +
  facet_wrap(~ term, scales=&quot;free&quot;, ncol = 2) +
  # geom_density(data=pivot_longer(cols=c(deltaCD4,deltaCD8)),aes(x=value), alpha=0.4) +
  theme_bw() +
  scale_color_brewer(palette = &quot;Set1&quot;)+

  labs(color=&quot;week&quot;)</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/Immune_delta_Distributions-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Immune_delta_Distributions-1">
Past versions of Immune_delta_Distributions-1.png
</button>
</p>
<div id="fig-Immune_delta_Distributions-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1f702e5d58b41215d4eaad8fd8796f88e0bb616/docs/figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/Immune_delta_Distributions-1.png" target="_blank">e1f702e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/8eaf17d0113a268e4cbf0467f54ce48400f4f3ac/docs/figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/Immune_delta_Distributions-1.png" target="_blank">8eaf17d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>deltas %&gt;% 
  select(contains(&quot;delta&quot;)) %&gt;% 
  colnames() %&gt;% 
  set_names() %&gt;% 
  map(~{
    summary(deltas[,.x])
  })</code></pre>
<pre><code>$record_id
  record_id        
 Length:79         
 Class :character  
 Mode  :character  

$deltaCD4_48
  deltaCD4_48   
 Min.   : 17.0  
 1st Qu.:133.6  
 Median :189.0  
 Mean   :219.9  
 3rd Qu.:300.0  
 Max.   :683.0  
 NA&#39;s   :20     

$deltaCD8_48
  deltaCD8_48    
 Min.   :-743.6  
 1st Qu.: -19.5  
 Median : 311.0  
 Mean   : 282.5  
 3rd Qu.: 521.5  
 Max.   :1975.0  
 NA&#39;s   :20      

$deltaCD4_96
  deltaCD4_96    
 Min.   :  90.0  
 1st Qu.: 177.5  
 Median : 250.7  
 Mean   : 304.7  
 3rd Qu.: 340.5  
 Max.   :1210.0  
 NA&#39;s   :31      

$deltaCD8_96
  deltaCD8_96    
 Min.   :-748.0  
 1st Qu.:-110.0  
 Median : 203.6  
 Mean   : 244.3  
 3rd Qu.: 484.8  
 Max.   :2963.0  
 NA&#39;s   :31      

$deltaratio_48
 deltaratio_48     
 Min.   :0.000655  
 1st Qu.:0.123930  
 Median :0.188536  
 Mean   :0.389095  
 3rd Qu.:0.286597  
 Max.   :8.837443  
 NA&#39;s   :20        

$deltaratio_96
 deltaratio_96   
 Min.   :0.0176  
 1st Qu.:0.2120  
 Median :0.2724  
 Mean   :0.3537  
 3rd Qu.:0.4544  
 Max.   :1.2042  
 NA&#39;s   :31      </code></pre>
<p>The last piece of output (summaries) tell there is a great ammount of missing data. This is due to a combination of many patients missing a sample and a few not having entries for analytical data:</p>
<pre class="r"><code>no0 &lt;- metadata %&gt;% 
  filter(!record_id %in% metadata[metadata$time_point == 0, ]$record_id) %&gt;%
  pull(record_id)

##patients without lymphocite data but with microbiome sample
no48_CD4 &lt;- metadata %&gt;%
  filter(is.na(CD4) ) %&gt;%
  filter(time_point == 48) %&gt;% 
  pull(record_id)  

no48_CD8 &lt;- metadata %&gt;%
  filter(is.na(CD8) &amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 48) %&gt;% 
  pull(record_id)  


no96_CD4 &lt;- metadata %&gt;%
  filter(is.na(CD4)&amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 96) %&gt;% 
  pull(record_id)

no96_CD8 &lt;- metadata %&gt;%
  filter(is.na(CD8)&amp; !is.na(SampleID)) %&gt;%
  filter(time_point == 96) %&gt;% 
  pull(record_id)


# patients without sample
noS48 &lt;- metadata %&gt;%
  filter(!record_id %in% metadata[metadata$time_point == 48, ]$record_id) %&gt;% 
  pull(record_id)

noS96 &lt;- metadata %&gt;%
  filter(!record_id %in% metadata[metadata$time_point == 96, ]$record_id) %&gt;% 
  pull(record_id)



totalMissing_CD4_48 &lt;- length(unique(c(no0, no48_CD4, noS48)))
totalMissing_CD4_96 &lt;- length(unique(c(no0,no96_CD4, noS96)))
totalMissing_CD8_48 &lt;- length(unique(c(no0, no48_CD8, noS48)))
totalMissing_CD8_96 &lt;- length(unique(c(no0,no96_CD8, noS96)))

totalMissing_CD4_48</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>totalMissing_CD4_96</code></pre>
<pre><code>[1] 31</code></pre>
<pre class="r"><code>totalMissing_CD8_96</code></pre>
<pre><code>[1] 31</code></pre>
<pre class="r"><code>totalMissing_CD8_96</code></pre>
<pre><code>[1] 31</code></pre>
<p>Adding the different kind of missing data checks out with the summaries. The biggest issue came with patients not having no sample whatsoever for such timepoints. Of which: 25 didn’t have a basal sample, 22 didnt have sample for week 48 and 30 for week 96.</p>
<p>delta CD4 threshold was too low. To keep the study clinically interpretable We’ll do a cutoff as a round number around the mean. In this case &lt;150/&gt;150 according to the mITT design results of the parent study.</p>
<p>Basal gene richness vs delta CD4_48 as continuous variables</p>
</div>
</div>
<div id="microbiome-structure-at-basal" class="section level1">
<h1>Microbiome structure at basal</h1>
<pre class="r"><code>basal_mre &lt;- metar::metaphlan_nmds(basal_mre, tax_level=&quot;Species&quot;)</code></pre>
<pre class="r"><code>nmdscor &lt;- basal_mre@taxa@metaphlan@nmds$Species$top_50$numeric 

ggpubr::ggarrange(plotlist = nmdscor, ncol=2)</code></pre>
<pre><code>$`1`</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/numeric_nmds_cors-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$`2`</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/numeric_nmds_cors-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
$`3`</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/numeric_nmds_cors-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
attr(,&quot;class&quot;)
[1] &quot;list&quot;      &quot;ggarrange&quot;</code></pre>
<pre class="r"><code>group &lt;- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$group &amp; 
    scale_color_brewer(palette=&quot;Set1&quot;) &amp;
    scale_fill_brewer(palette = &quot;Set1&quot;) 
  



cluster &lt;- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$cluster


ggpubr::ggarrange(group,cluster, labels = c(&quot;group&quot;, &quot;cluster&quot;))</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-basal_NMDS-1">
Past versions of basal_NMDS-1.png
</button>
</p>
<div id="fig-basal_NMDS-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e1f702e5d58b41215d4eaad8fd8796f88e0bb616/docs/figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/basal_NMDS-1.png" target="_blank">e1f702e</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-04-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>NMDS shows no basal significant differences per group for either com,ponent. An unsupervised clustering created 2 well differentiated groups. While ADONIS per cluster was significant, we’ll look whether the numerical variables have something to do with them.</p>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="right">Df</th>
<th align="right">SumsOfSqs</th>
<th align="right">MeanSqs</th>
<th align="right">F.Model</th>
<th align="right">R2</th>
<th align="right">Pr..F.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">group</td>
<td align="right">1</td>
<td align="right">0.2320561</td>
<td align="right">0.2320561</td>
<td align="right">0.7569076</td>
<td align="right">0.0110085</td>
<td align="right">0.797</td>
</tr>
<tr class="even">
<td align="left">risk_group</td>
<td align="right">2</td>
<td align="right">0.8593023</td>
<td align="right">0.4296512</td>
<td align="right">1.4219349</td>
<td align="right">0.0419179</td>
<td align="right">0.053</td>
</tr>
<tr class="odd">
<td align="left">center</td>
<td align="right">5</td>
<td align="right">1.6094885</td>
<td align="right">0.3218977</td>
<td align="right">1.0580959</td>
<td align="right">0.0763522</td>
<td align="right">0.302</td>
</tr>
<tr class="even">
<td align="left">gender</td>
<td align="right">1</td>
<td align="right">0.8309639</td>
<td align="right">0.8309639</td>
<td align="right">2.7905580</td>
<td align="right">0.0394199</td>
<td align="right">0.002</td>
</tr>
<tr class="odd">
<td align="left">CD4diff_48</td>
<td align="right">1</td>
<td align="right">0.3718115</td>
<td align="right">0.3718115</td>
<td align="right">1.1770855</td>
<td align="right">0.0202328</td>
<td align="right">0.229</td>
</tr>
<tr class="even">
<td align="left">CD8diff_48</td>
<td align="right">1</td>
<td align="right">0.3253300</td>
<td align="right">0.3253300</td>
<td align="right">1.0272818</td>
<td align="right">0.0177034</td>
<td align="right">0.407</td>
</tr>
<tr class="odd">
<td align="left">CD4after_48</td>
<td align="right">2</td>
<td align="right">0.8412615</td>
<td align="right">0.4206307</td>
<td align="right">1.3433005</td>
<td align="right">0.0457788</td>
<td align="right">0.076</td>
</tr>
<tr class="even">
<td align="left">CD8after_48</td>
<td align="right">2</td>
<td align="right">0.9081250</td>
<td align="right">0.4540625</td>
<td align="right">1.4556165</td>
<td align="right">0.0494173</td>
<td align="right">0.076</td>
</tr>
<tr class="odd">
<td align="left">CD4diff_96</td>
<td align="right">1</td>
<td align="right">0.3135775</td>
<td align="right">0.3135775</td>
<td align="right">0.9763171</td>
<td align="right">0.0207832</td>
<td align="right">0.456</td>
</tr>
<tr class="even">
<td align="left">CD8diff_96</td>
<td align="right">1</td>
<td align="right">0.1747594</td>
<td align="right">0.1747594</td>
<td align="right">0.5390451</td>
<td align="right">0.0115826</td>
<td align="right">0.981</td>
</tr>
<tr class="odd">
<td align="left">CD4after_96</td>
<td align="right">2</td>
<td align="right">0.7449666</td>
<td align="right">0.3724833</td>
<td align="right">1.1686298</td>
<td align="right">0.0493746</td>
<td align="right">0.228</td>
</tr>
<tr class="even">
<td align="left">CD8after_96</td>
<td align="right">1</td>
<td align="right">0.4311689</td>
<td align="right">0.4311689</td>
<td align="right">1.3532056</td>
<td align="right">0.0285769</td>
<td align="right">0.138</td>
</tr>
<tr class="odd">
<td align="left">cluster</td>
<td align="right">1</td>
<td align="right">2.6024052</td>
<td align="right">2.6024052</td>
<td align="right">9.5773020</td>
<td align="right">0.1234550</td>
<td align="right">0.001</td>
</tr>
</tbody>
</table>
</div>
<div id="nmds-correlation-with-numerical-variables" class="section level1">
<h1>NMDS correlation with numerical variables</h1>
<pre class="r"><code>basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_numeric %&gt;% 
  ggpubr::ggarrange(plotlist=.)</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/NMDS_numeric_correlations-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="categorical-variables-vs-unsupervised-clustering" class="section level2">
<h2>Categorical variables vs unsupervised clustering</h2>
<pre class="r"><code>basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical %&gt;% 
  names() %&gt;% 
  set_names() %&gt;% 
  map(~{
    df &lt;- 
    basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical %&gt;% 
      pluck(.x) %&gt;% 
      pluck(&quot;data&quot;) %&gt;%
      filter(!is.na(!!sym(.x)))
    
    pal &lt;- cat_df %&gt;% 
      filter(CategoricalVariable == .x) %&gt;% 
      pull(PaletteName)
    
    
    test &lt;- df %&gt;%
      dplyr::select(!!sym(.x), cluster) %&gt;%
      table() %&gt;% 
      as.matrix() %&gt;% 
      fisher.test(df) %&gt;% 
      pluck(&quot;p.value&quot;) %&gt;% 
      as.numeric() %&gt;% 
      round(.,2)
      
    plt &lt;-   
      df %&gt;% 
      ggplot(., aes(x = as.factor(cluster),fill=!!sym(.x)))+
      geom_bar(stat=&quot;count&quot;) +
      theme_bw() +
      labs(x=&quot;cluster&quot;,
           title=.x,
            subtitle = glue::glue(&quot;P.value = {test} (Fisher test)&quot;)) +
      scale_fill_brewer(palette = pal)
  }) </code></pre>
<pre><code>$group</code></pre>
<pre><code>
$risk_group</code></pre>
<pre><code>
$center</code></pre>
<pre><code>
$gender</code></pre>
<pre><code>
$CD4diff_48</code></pre>
<pre><code>
$CD8diff_48</code></pre>
<pre><code>
$CD4after_48</code></pre>
<pre><code>
$CD8after_48</code></pre>
<pre><code>
$CD4diff_96</code></pre>
<pre><code>
$CD8diff_96</code></pre>
<pre><code>
$CD4after_96</code></pre>
<pre><code>
$CD8after_96</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-1.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-2.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-3.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-4.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-5.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-6.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-7.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-8.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-9.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-10.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-11.png" width="40%" style="display: block; margin: auto;" /><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/categorical_vs_cluster-12.png" width="40%" style="display: block; margin: auto;" /></p>
<p>The variabe correspoinding to the change in CD4 counts after 96 weeks <strong>CD4diff_96</strong> was dropped due to having only one value. This means all patients had an increase of CD4&gt;50, which breaks any comparison we try to make.</p>
</div>
</div>
<div id="differential-abundance" class="section level1">
<h1>DIfferential Abundance</h1>
<p>We’ll look into which genera are more correlated per group at baseline. We’ll perform both classic and compositional approaches.</p>
<pre class="r"><code>phy_basal &lt;- get_phyloseq(basal_mre, type = &quot;metaphlan&quot;) %&gt;% 
    phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


otu_df &lt;- 
  phyloseq::otu_table(phy_basal) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  setNames(phyloseq::tax_table(phy_basal)[,&quot;Species&quot;]) %&gt;% 
  rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  setNames(.,str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  setNames(.,str_remove_all(colnames(.), &quot;\\]&quot;))

basal_meta &lt;- get_meta(basal_mre)

stat_list &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    otu_df %&gt;%
      dplyr::select(-SampleID) %&gt;%
      colnames() %&gt;%
      purrr::set_names() %&gt;%
      purrr::map_dfr(function(sp) {
        df &lt;-
          otu_df %&gt;%
          dplyr::select(SampleID, !!sym(sp)) %&gt;%
          dplyr::left_join(basal_meta[, c(&quot;SampleID&quot;, cv), drop = F], by = &quot;SampleID&quot;)
        
        if (length(unique(df[, cv])) == 2) {
          test &lt;- df %&gt;%
            rstatix::wilcox_test(., formula(paste(sp, &quot;~&quot;, cv)), paired = F) %&gt;%
            dplyr::mutate(method = &quot;Wilcoxon&quot;)
          
        } else{
          test &lt;- df %&gt;%
            rstatix::kruskal_test(., formula(paste(sp, &quot;~&quot;, cv))) %&gt;%
            dplyr::mutate(method = &quot;Kruskal-Wallis&quot;)
        }
        
        
      }) %&gt;%
      dplyr::mutate(p.adj = p.adjust(.$p, &quot;BH&quot;)) %&gt;%
      dplyr::filter(p.adj &lt; 0.05)
  }) %&gt;% 
  purrr::keep(., ~nrow(.) &gt; 0)



stat_list %&gt;% 
  map(~{
    kableExtra::kable(.x, format = &quot;markdown&quot;) 
  })</code></pre>
<pre><code>$risk_group


|.y.                        |  n| statistic| df|        p|method         |     p.adj|
|:--------------------------|--:|---------:|--:|--------:|:--------------|---------:|
|Parabacteroides_distasonis | 70|  15.62025|  2| 0.000406|Kruskal-Wallis | 0.0224280|
|Roseburia_sp_CAG_471       | 70|  13.01657|  2| 0.001490|Kruskal-Wallis | 0.0331525|
|Anaerotruncus_colihominis  | 70|  14.37569|  2| 0.000756|Kruskal-Wallis | 0.0224280|
|Clostridium_innocuum       | 70|  14.59802|  2| 0.000676|Kruskal-Wallis | 0.0224280|

$gender


|.y.                                 |group1 |group2 | n1| n2| statistic|        p|method   |     p.adj|
|:-----------------------------------|:------|:------|--:|--:|---------:|--------:|:--------|---------:|
|Eggerthella_lenta                   |female |male   | 10| 60|     480.0| 0.001280|Wilcoxon | 0.0150311|
|Gordonibacter_pamelaeae             |female |male   | 10| 60|     481.5| 0.000919|Wilcoxon | 0.0150311|
|Slackia_isoflavoniconvertens        |female |male   | 10| 60|     119.5| 0.002040|Wilcoxon | 0.0181560|
|Prevotella_copri                    |female |male   | 10| 60|     113.5| 0.001520|Wilcoxon | 0.0150311|
|Prevotella_sp_CAG_279               |female |male   | 10| 60|     154.0| 0.008170|Wilcoxon | 0.0484753|
|Prevotella_sp_CAG_520               |female |male   | 10| 60|     140.0| 0.003490|Wilcoxon | 0.0282373|
|Prevotella_stercorea                |female |male   | 10| 60|     120.0| 0.001370|Wilcoxon | 0.0150311|
|Parabacteroides_distasonis          |female |male   | 10| 60|     466.0| 0.004470|Wilcoxon | 0.0331525|
|Ruminococcus_gnavus                 |female |male   | 10| 60|     466.5| 0.001310|Wilcoxon | 0.0150311|
|Dorea_formicigenerans               |female |male   | 10| 60|     135.0| 0.005420|Wilcoxon | 0.0371062|
|Dorea_longicatena                   |female |male   | 10| 60|     137.5| 0.006230|Wilcoxon | 0.0396050|
|Anaerotruncus_colihominis           |female |male   | 10| 60|     474.0| 0.000679|Wilcoxon | 0.0150311|
|Clostridium_innocuum                |female |male   | 10| 60|     504.0| 0.000216|Wilcoxon | 0.0150311|
|Erysipelatoclostridium_ramosum      |female |male   | 10| 60|     468.0| 0.000638|Wilcoxon | 0.0150311|
|Phascolarctobacterium_succinatutens |female |male   | 10| 60|     120.0| 0.001370|Wilcoxon | 0.0150311|</code></pre>
<p>The only categoricalvariables with Differential abundance at basal are Gender and Risk group. This is hardly surprising as microbiome differences between gender and sexual orientations have been widely reported. But that also means no significant signal could be detected that is significant to this study. More specifically, there is no bacterial signal that may predict increase of CD4/CD8 after 48-96 weeks of treatment.</p>
<pre class="r"><code>phy_c&lt;- basal_mre@taxa@metaphlan@phyloseq_sec %&gt;% 
    phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


basal_mre@taxa@metaphlan@phyloseq_sec &lt;- phy_c

basal_mre &lt;- metar::metaphlan_ancom(mre = basal_mre, tax_level = &quot;Species&quot;, save_files = F)


res_list &lt;- basal_mre@taxa@metaphlan@ancom$Species %&gt;% 
  map(~{
    .x %&gt;% 
      pluck(&quot;stats&quot;) %&gt;% 
      filter(detected_0.7 == TRUE)
  }) %&gt;% 
  purrr::keep(., ~nrow(.) &gt; 0)
basal_mre@taxa@metaphlan@ancom$Species[[names(res_list)]]$plot</code></pre>
<p><img src="figure/335_BLMicrobiome_vs_ImmuneResponse.Rmd/ANCOM-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>res_list</code></pre>
<pre><code>$risk_group
                     taxa_id  W detected_0.9 detected_0.8 detected_0.7
1 Parabacteroides_distasonis 32         TRUE         TRUE         TRUE
2     Flavonifractor_plautii 24        FALSE        FALSE         TRUE
  detected_0.6  mean_qval median_qval  fisher_qval clr_mean_diff
1         TRUE 0.03617902 0.001680889 4.772824e-49    -1.0275269
2         TRUE 0.08000107 0.023328846 1.363693e-20    -0.3682635
  structural_zero   x_label y_label comparison    cat_var
1              No -1.027527    22.4    hts_msm risk_group
2              No -1.027527    22.4    hts_msm risk_group</code></pre>
<p>ANCOM was even more astringent and ony found 2 significant species between hts and msm.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5        testthat_3.1.4     patchwork_1.1.1    RColorBrewer_1.1-3
 [5] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4       
 [9] readr_2.1.2        tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5     
[13] tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] estimability_1.3       coda_0.19-4            bit64_4.0.5           
  [4] knitr_1.39             multcomp_1.4-19        wesanderson_0.3.6     
  [7] data.table_1.14.2      DataExplorer_0.8.2     generics_0.1.2        
 [10] BiocGenerics_0.36.1    cowplot_1.1.1          callr_3.7.0           
 [13] TH.data_1.1-1          usethis_2.1.5          future_1.25.0         
 [16] correlation_0.8.0      bit_4.0.4              tzdb_0.3.0            
 [19] bayesm_3.1-4           webshot_0.5.3          xml2_1.3.2            
 [22] lubridate_1.8.0        httpuv_1.6.5           assertthat_0.2.1      
 [25] WRS2_1.1-3             xfun_0.30              hms_1.1.1             
 [28] jquerylib_0.1.4        evaluate_0.15          promises_1.2.0.1      
 [31] DEoptimR_1.0-11        fansi_1.0.3            dbplyr_2.1.1          
 [34] readxl_1.4.0           igraph_1.3.1           DBI_1.1.2             
 [37] htmlwidgets_1.5.4      reshape_0.8.9          tensorA_0.36.2        
 [40] kSamples_1.2-9         stats4_4.0.5           Rmpfr_0.8-7           
 [43] paletteer_1.4.0        ellipsis_0.3.2         survcomp_1.40.0       
 [46] ggpubr_0.4.0           backports_1.4.1        insight_0.17.0        
 [49] permute_0.9-7          prismatic_1.1.0        compositions_2.0-4    
 [52] vctrs_0.4.1            Biobase_2.50.0         remotes_2.4.2         
 [55] here_1.0.1             abind_1.4-5            cachem_1.0.6          
 [58] withr_2.5.0            ggforce_0.3.3          aws.signature_0.6.0   
 [61] robustbase_0.95-0      vroom_1.5.7            emmeans_1.7.3         
 [64] vegan_2.6-2            prettyunits_1.1.1      svglite_2.1.0         
 [67] cluster_2.1.3          ape_5.6-2              crayon_1.5.1          
 [70] pkgconfig_2.0.3        SuppDists_1.1-9.7      labeling_0.4.2        
 [73] tweenr_1.0.2           nlme_3.1-157           statsExpressions_1.3.1
 [76] pkgload_1.2.4          blme_1.0-5             devtools_2.4.3        
 [79] rlang_1.0.2            globals_0.14.0         lifecycle_1.0.1       
 [82] sandwich_3.0-1         survivalROC_1.0.3      selbal_0.1.0          
 [85] phyloseq_1.34.0        modelr_0.1.8           cellranger_1.1.0      
 [88] rprojroot_2.0.3        polyclip_1.10-0        datawizard_0.4.0      
 [91] Matrix_1.4-1           mc2d_0.1-21            carData_3.0-5         
 [94] Rhdf5lib_1.12.1        boot_1.3-28            zoo_1.8-10            
 [97] reprex_2.0.1           base64enc_0.1-3        whisker_0.4           
[100] processx_3.5.3         viridisLite_0.4.0      PMCMRplus_1.9.4       
[103] bootstrap_2019.6       parameters_0.17.0      KernSmooth_2.23-20    
[106] rhdf5filters_1.2.1     Biostrings_2.58.0      workflowr_1.7.0       
[109] multcompView_0.1-8     arm_1.12-2             parallelly_1.31.1     
[112] rstatix_0.7.0          S4Vectors_0.28.1       ggsignif_0.6.3.9000   
[115] rmeta_3.0              aws.s3_0.3.21          scales_1.2.0          
[118] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.7            
[121] zlibbioc_1.36.0        compiler_4.0.5         kableExtra_1.3.4      
[124] lme4_1.1-29            snakecase_0.11.0       cli_3.3.0             
[127] ade4_1.7-19            XVector_0.30.0         listenv_0.8.0         
[130] ps_1.7.0               MASS_7.3-57            mgcv_1.8-40           
[133] ggside_0.2.0           tidyselect_1.1.2       stringi_1.7.6         
[136] highr_0.9              yaml_2.3.5             ANCOM_2.2             
[139] ggrepel_0.9.1          grid_4.0.5             sass_0.4.1            
[142] tools_4.0.5            future.apply_1.9.0     parallel_4.0.5        
[145] rstudioapi_0.13        foreach_1.5.2          logging_0.10-108      
[148] git2r_0.30.1           janitor_2.1.0          gridExtra_2.3         
[151] prodlim_2019.11.13     farver_2.1.0           digest_0.6.29         
[154] lava_1.6.10            shiny_1.7.1            networkD3_0.4         
[157] BWStest_0.2.2          Rcpp_1.0.8.3           car_3.0-12            
[160] broom_0.8.0            performance_0.9.0      later_1.3.0           
[163] httr_1.4.2             effectsize_0.6.0.1     colorspace_2.0-3      
[166] rvest_1.0.2            brio_1.1.3             fs_1.5.2              
[169] IRanges_2.24.1         splines_4.0.5          rematch2_2.1.2        
[172] multtest_2.46.0        systemfonts_1.0.4      sessioninfo_1.2.2     
[175] xtable_1.8-4           gmp_0.6-5              jsonlite_1.8.0        
[178] nloptr_2.0.0           zeallot_0.1.0          R6_2.5.1              
[181] broom.mixed_0.2.9.4    pillar_1.7.0           htmltools_0.5.2       
[184] mime_0.12              glue_1.6.2             fastmap_1.1.0         
[187] minqa_1.2.4            codetools_0.2-18       pkgbuild_1.3.1        
[190] mvtnorm_1.1-3          furrr_0.2.3            utf8_1.2.2            
[193] lattice_0.20-45        bslib_0.3.1            logger_0.2.2          
[196] curl_4.3.2             survival_3.3-1         rmarkdown_2.14        
[199] desc_1.4.1             biomformat_1.18.0      merTools_0.5.2        
[202] munsell_0.5.0          rhdf5_2.34.0           iterators_1.0.14      
[205] ggstatsplot_0.9.1      haven_2.5.0            reshape2_1.4.4        
[208] gtable_0.3.0           bayestestR_0.11.5.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
