<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />


<title>Taxonomy: Differential Abundance</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.5.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="315_GeneRichness_vs_CD4.html">CD4 recovery vs Gene richness</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="316_GeneRichness_vs_CMV.html">CMV_analysis</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
<li>
  <a href="430_Species_v_Inflammation.html">DA species vs inflammation markers</a>
</li>
<li>
  <a href="500_Gene_function_exploratory.html">Functional annotation, exploratory</a>
</li>
<li>
  <a href="510_Function_DA.html">Gene function DA</a>
</li>
<li>
  <a href="505_Adenovirus_Analysis.html">Adenovirus and Enterovirus analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Taxonomy: Differential Abundance</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">3/23/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
wilcox_across_group
</li>
<li>
wilcox_within_group
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>420_Taxonomy_DA_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treeb9a23b33647f184fb1b27b48f6ec9b349cbb0ed3targetblankb9a23b3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3" target="_blank">b9a23b3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treeb9a23b33647f184fb1b27b48f6ec9b349cbb0ed3targetblankb9a23b3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/b9a23b33647f184fb1b27b48f6ec9b349cbb0ed3" target="_blank">b9a23b3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/420_Taxonomy_DA_cache/
    Ignored:    analysis/505_Adenovirus_Analysis_cache/
    Ignored:    analysis/510_Function_DA_cache/
    Ignored:    analysis/output/mre_data/SpeciesQuantification/Metaphlan3/NMDS/
    Ignored:    data/mre.rds
    Ignored:    data/signif_paths.rds

Unstaged changes:
    Modified:   analysis/316_GeneRichness_vs_CMV.Rmd
    Modified:   output/mre_data/CatalogMapping/IGC/lmm/lmm_time_point_by_group.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/420_Taxonomy_DA.Rmd</code>) and HTML (<code>docs/420_Taxonomy_DA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/analysis/420_Taxonomy_DA.Rmd" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
<td>
Rendered new workflowr site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/docs/420_Taxonomy_DA.html" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
<td>
Rendered new workflowr site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e660a2852bdbb9838f9fa3423447ab5659a5aff5/analysis/420_Taxonomy_DA.Rmd" target="_blank">e660a28</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-10-27
</td>
<td>
updates on various scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/19b15b838b9f3f9a6e8e23c1913d8d8adae38763/analysis/420_Taxonomy_DA.Rmd" target="_blank">19b15b8</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-10-14
</td>
<td>
Added Functional DA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c208fe48af079c31bc6b5026a111abfefbaf9434/analysis/420_Taxonomy_DA.Rmd" target="_blank">c208fe4</a>
</td>
<td>
cblazquez-irsi
</td>
<td>
2022-09-16
</td>
<td>
created basic report Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/176b5e371b9a60cf682a73778688b1e061a47b5c/analysis/420_Taxonomy_DA.Rmd" target="_blank">176b5e3</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-14
</td>
<td>
minor fixes on LMMs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/0aa7922e8a20e42a68c244562eec53d699c4efdd/docs/420_Taxonomy_DA.html" target="_blank">0aa7922</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-07
</td>
<td>
Improved S.parasanguinis interaction plots with inflammation markers
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/analysis/420_Taxonomy_DA.Rmd" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
<td>
Added species v inflammation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/420_Taxonomy_DA.html" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
<td>
Added species v inflammation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1743951bf18aab3b40cb5ba194de76e9401f18dc/analysis/420_Taxonomy_DA.Rmd" target="_blank">1743951</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-30
</td>
<td>
Specific analysis for S.parasanguinis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/1743951bf18aab3b40cb5ba194de76e9401f18dc/docs/420_Taxonomy_DA.html" target="_blank">1743951</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-30
</td>
<td>
Specific analysis for S.parasanguinis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/9713823b00b303b0cb85954d9a9ca081ad4a2c8d/analysis/420_Taxonomy_DA.Rmd" target="_blank">9713823</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-27
</td>
<td>
Added longitudinal DA with ANCOM in 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/9713823b00b303b0cb85954d9a9ca081ad4a2c8d/docs/420_Taxonomy_DA.html" target="_blank">9713823</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-27
</td>
<td>
Added longitudinal DA with ANCOM in 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/e7644ce88f78db3d5c661c679f6f2f88f836ef01/analysis/420_Taxonomy_DA.Rmd" target="_blank">e7644ce</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-12
</td>
<td>
Fixed bug where placebo samples were not removed for Prob+/- analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/analysis/420_Taxonomy_DA.Rmd" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
<td>
Added long_boxplots for species discovered by Venn Diagram
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/420_Taxonomy_DA.html" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
<td>
Added long_boxplots for species discovered by Venn Diagram
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/d6ec767ea503b994a4300cafe2da5ead4eeed0f5/analysis/420_Taxonomy_DA.Rmd" target="_blank">d6ec767</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-06
</td>
<td>
Added new comment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bab1e4c5a5716e45a0a8250683552016b08e2035/analysis/420_Taxonomy_DA.Rmd" target="_blank">bab1e4c</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-06
</td>
<td>
STarted filterng DA species based on Venn’s diagrams
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/38a6169b1c3cc45c9f96d45f6f5b8c1e441f34a6/analysis/420_Taxonomy_DA.Rmd" target="_blank">38a6169</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-05
</td>
<td>
Added longitudinalVenn diagrams. reworking long boxplots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/38a6169b1c3cc45c9f96d45f6f5b8c1e441f34a6/docs/420_Taxonomy_DA.html" target="_blank">38a6169</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-05
</td>
<td>
Added longitudinalVenn diagrams. reworking long boxplots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/18eec16ece3f65f151379894f49075e8550ca4ff/analysis/420_Taxonomy_DA.Rmd" target="_blank">18eec16</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-04
</td>
<td>
Added first plots of longitudinal DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/18eec16ece3f65f151379894f49075e8550ca4ff/docs/420_Taxonomy_DA.html" target="_blank">18eec16</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-04
</td>
<td>
Added first plots of longitudinal DA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/ef074ac89e3edca9e7522c1c33052646f40ccd05/analysis/420_Taxonomy_DA.Rmd" target="_blank">ef074ac</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-04
</td>
<td>
added mean comparisons to longitudinal DA changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/ef074ac89e3edca9e7522c1c33052646f40ccd05/docs/420_Taxonomy_DA.html" target="_blank">ef074ac</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-04
</td>
<td>
added mean comparisons to longitudinal DA changes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/2c9bb8128cd2450ad7f01b0bb0af1e1259f00cf8/analysis/420_Taxonomy_DA.Rmd" target="_blank">2c9bb81</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-05-03
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f51189b8f584a80a001462d968318e82da32fc05/analysis/420_Taxonomy_DA.Rmd" target="_blank">f51189b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
started longitudinal DA section
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/analysis/420_Taxonomy_DA.Rmd" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
Added Venn diagram and plotted stats of spps with DA in all TPs (Differential Abundance)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/420_Taxonomy_DA.html" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
Added Venn diagram and plotted stats of spps with DA in all TPs (Differential Abundance)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/74908f2328b99c6d1324c42c33659bf2b41a4abc/analysis/420_Taxonomy_DA.Rmd" target="_blank">74908f2</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a98995620eb26b2dc78546dc4003a944581d2d21/analysis/420_Taxonomy_DA.Rmd" target="_blank">a989956</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3f2dedbb7ecf47b42abf0d5de16eb89c999523e0/analysis/420_Taxonomy_DA.Rmd" target="_blank">3f2dedb</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0ad10f57885cf1df5eb471c896caca7e216ac3e4/analysis/420_Taxonomy_DA.Rmd" target="_blank">0ad10f5</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/analysis/420_Taxonomy_DA.Rmd" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Advanced on longitudinal DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/docs/420_Taxonomy_DA.html" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Advanced on longitudinal DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c20090b3a3c8e714d8be91e4ea67aa5d3ff7c522/docs/420_Taxonomy_DA.html" target="_blank">c20090b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-01
</td>
<td>
Compiled taxa ranking html + several bug fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5917250ea4ce78208d3a41d2c612085024734455/analysis/420_Taxonomy_DA.Rmd" target="_blank">5917250</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-31
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e253850975c479f491140a3a60d9e54a3eca44ff/docs/420_Taxonomy_DA.html" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/420_Taxonomy_DA.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/420_Taxonomy_DA.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/11c1e7d2e4c2b76ccb9e08a96305fa7f34ee48de/analysis/420_Taxonomy_DA.Rmd" target="_blank">11c1e7d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
Code reorganized towards new structure
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import-and-exploration" class="section level2">
<h2>Data import and exploration</h2>
<p>This section should explain anyithing that may have happened during the first fase of importing and formatting data, such as any modification or additional filtering step.</p>
<pre class="r"><code># mymre &lt;- here::here(&quot;data&quot;,&quot;mre.rds&quot;) %&gt;%
#   readRDS(.)

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;mistral-wp6-advanz4&quot;, object =&quot;/metagenome/WMGS/MREObject.rds&quot;, region = &quot;eu-west-1&quot;) 
# metadata &lt;-get_meta(mymre)
metadata &lt;- here::here(&quot;Metadata&quot;, &quot;2022_09_29_clean_metadata_LIMS.csv&quot;) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

link_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here(&quot;../WMGSPipeline&quot;))</code></pre>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre class="r"><code>mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
# mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
#   phyloseq::sample_data(.)




mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@taxa@metaphlan@phyloseq@sam_data &lt;- phyloseq::sample_data(metadata)
mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- phyloseq::sample_data(metadata)</code></pre>
<pre class="r"><code># Any homebrew support function that may be written will be placed here</code></pre>
<p>First we’ll start with a non-compositional approach which may provide potential candidate taxa for DA to be confirmed with more astringent and specific compositional approaches.</p>
</div>
<div id="non-compositional-analysis" class="section level1">
<h1>Non-compositional analysis</h1>
<p>In this case, we’ll test for group as categorical variables. We may extend the analysis towards other cat_vars, but those were more important for basal comparisons, which already shown no great differences were found among other potential confusor variables.</p>
<div id="cross-group-differential-abundance" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.</p>
<pre class="r"><code>phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;) %&gt;% 
      phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


otu_df &lt;- 
  phyloseq::otu_table(phy) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy)[,&quot;Species&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;))


stat_list &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    
    c(0,24,48,96) %&gt;% 
      set_names() %&gt;% 
      purrr::map_dfr(function(tp){
        otu_df %&gt;%
        dplyr::select(-SampleID) %&gt;%
        colnames() %&gt;%
        purrr::set_names() %&gt;%
        purrr::map_dfr(function(sp) {
          df &lt;-
            otu_df %&gt;%
            dplyr::select(SampleID, !!sym(sp)) %&gt;%
            dplyr::left_join(metadata[, c(&quot;SampleID&quot;, link_var, cv, long_var), drop = F], 
                             by = &quot;SampleID&quot;) %&gt;% 
            dplyr::filter(!!sym(long_var) %in% tp) 

          
          if (length(unique(dplyr::pull(.data = df, !!sym(cv)))) == 2) {
            test &lt;- df %&gt;%
              rstatix::wilcox_test(., formula(paste(sp, &quot;~&quot;, cv)), paired = F) %&gt;%
              dplyr::mutate(method = &quot;Wilcoxon&quot;) 
            
          } else{
            test &lt;- df %&gt;%
              rstatix::kruskal_test(., formula(paste(sp, &quot;~&quot;, cv))) %&gt;%
              dplyr::mutate(method = &quot;Kruskal-Wallis&quot;)
          }
          
        }) %&gt;% 
      rename(species = .y.) %&gt;% 
      dplyr::mutate(long_var = tp) %&gt;% 
      group_by(long_var) %&gt;% 
      mutate(p.adj = p.adjust(p, &quot;BH&quot;)) %&gt;% 
      # filter(p.adj &lt; .05 ) %&gt;% 
      group_by(species) 
      }) 
}) 


DAsp &lt;- stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species)


## --- init MNJ
## It is interesting that Bifidobacterium species (adolescentis, longum) are consistently found as DA in all (most) timepoints
## Produce a table showing 
stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species) %&gt;% 
  tally() %&gt;% 
  filter(n&gt;=2) ## Species being significant (Wilcoxon unadjusted &lt;0.05)</code></pre>
<pre><code># A tibble: 15 × 2
   species                          n
   &lt;chr&gt;                        &lt;int&gt;
 1 Bifidobacterium_adolescentis     3
 2 Bifidobacterium_longum           3
 3 Clostridium_citroniae            2
 4 Coprococcus_catus                2
 5 Eubacterium_eligens              2
 6 Eubacterium_siraeum              2
 7 Eubacterium_sp_CAG_180           2
 8 Firmicutes_bacterium_CAG_110     2
 9 Flavonifractor_plautii           2
10 Lachnospira_pectinoschiza        2
11 Olsenella_scatoligenes           3
12 Oscillibacter_sp_CAG_241         2
13 Parabacteroides_distasonis       2
14 Roseburia_faecis                 2
15 Roseburia_hominis                2</code></pre>
<pre class="r"><code>otu_df %&gt;%
  select(SampleID,Bifidobacterium_adolescentis) %&gt;%
  # select(SampleID, !!sym(DAsp)) %&gt;%
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;%
  group_by(group) %&gt;%
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Bifidobacterium_adolescentis,
                              pairwise.comparisons = T,
                              type=&quot;np&quot;,bf.message=NULL,
                              ylab = &quot;Bifidobacterium_adolescentis(%)&quot;,
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-wilcox_across_group-1">
Past versions of wilcox_across_group-1.png
</button>
</p>
<div id="fig-wilcox_across_group-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>otu_df %&gt;%
  select(SampleID,Bifidobacterium_longum) %&gt;%
  # select(SampleID, !!sym(DAsp)) %&gt;%
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;%
  group_by(group) %&gt;%
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Bifidobacterium_longum,
                              pairwise.comparisons = F,
                              type=&quot;np&quot;,bf.message=NULL,
                              ylab = &quot;Bifidobacterium_longum(%)&quot;,
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,
                              plotgrid.args = list(&quot;ncol&quot; = 4, &quot;nrow&quot; = 1))</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-wilcox_across_group-2">
Past versions of wilcox_across_group-2.png
</button>
</p>
<div id="fig-wilcox_across_group-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-2.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 

 ### There is a nice signal, consistent longitudinally, in Bifidobacterium Genus
phyloseq::otu_table(phy %&gt;% phyloseq::tax_glom(.,taxrank=&quot;Genus&quot;)) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy %&gt;% phyloseq::tax_glom(.,taxrank=&quot;Genus&quot;))[,&quot;Genus&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;)) %&gt;% 
  select(SampleID,Bifidobacterium) %&gt;% 
  # select(SampleID, !!sym(DAsp)) %&gt;% 
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  group_by(group) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Bifidobacterium,
                              pairwise.comparisons = F,
                              type=&quot;np&quot;, bf.message=NULL,
                              ylab = &quot;Bifidobacterium(%)&quot;, 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL) </code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># %&gt;% 
  # ggsave(.,filename = &quot;~/Downloads/plot4Carlos_Bifidobacterium.pdf&quot;)

otu_df %&gt;% 
  select(SampleID,Olsenella_scatoligenes) %&gt;% 
  # select(SampleID, !!sym(DAsp)) %&gt;% 
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  group_by(group) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Olsenella_scatoligenes,
                              pairwise.comparisons = F,
                              type=&quot;np&quot;,bf.message=NULL,
                              ylab = &quot;Olsenella_scatoligenes(%)&quot;, 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### Show how many species are differential in at least two of the timepoints in each cat var. Risk_group should result in many species showing that kins of signal. See Venn section below.
### --- end MNJ</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Bifidobacterium adolescentis, Bifidobacterium longum and Olsenella scatoligenes were significant in all timepoints. Let’s check the actual abundances per group and timepoint:</p>
<pre class="r"><code>source(&quot;code/group_comparisons.R&quot;)
sp_vec &lt;- c(&quot;Bifidobacterium_adolescentis&quot;,&quot;Bifidobacterium_longum&quot;,&quot;Olsenella_scatoligenes&quot;)


sp_res &lt;- 
otu_df %&gt;% 
  left_join(metadata, by = &quot;SampleID&quot;) %&gt;% 
  # select(SampleID, record_id, time_point, group, abundance = !!sym(sp)) %&gt;% 
  get_group_comparisons(., 
                        link_var = &quot;record_id&quot;,
                        long_var = &quot;time_point&quot;,
                        cat_vector = &quot;group&quot;,
                        num_vector = c(&quot;Bifidobacterium_adolescentis&quot;,&quot;Bifidobacterium_longum&quot;,&quot;Olsenella_scatoligenes&quot;),
                        comps = c(&quot;DRV/r&quot;, &quot;DTG&quot;),
                        type = &quot;categorical&quot;)


sp_res$group$Bifidobacterium_adolescentis %&gt;%  pluck(&quot;stats&quot;) %&gt;% as.data.frame()</code></pre>
<pre><code>  cat_var long_var   median  upper_ci lower_ci       iqr
1   DRV/r        0 0.000000 0.5821767        0 0.5821767
2   DRV/r       24 0.000000 0.0272200        0 0.0272200
3   DRV/r       48 0.000000 0.3525667        0 0.3525667
4   DRV/r       96 0.000000 0.2700283        0 0.2700283
5     DTG        0 0.000000 0.8171858        0 0.8171858
6     DTG       24 0.235300 2.2562800        0 2.2562800
7     DTG       48 1.942740 5.1234025        0 5.1234025
8     DTG       96 0.744225 5.3758367        0 5.3758367</code></pre>
<pre class="r"><code>sp_res$group$Bifidobacterium_longum %&gt;%  pluck(&quot;stats&quot;) %&gt;% as.data.frame()</code></pre>
<pre><code>  cat_var long_var   median  upper_ci lower_ci       iqr
1   DRV/r        0 0.000000 1.2301150        0 1.2301150
2   DRV/r       24 0.000000 0.3188525        0 0.3188525
3   DRV/r       48 0.005150 0.5420533        0 0.5420533
4   DRV/r       96 0.000000 0.2459417        0 0.2459417
5     DTG        0 0.375410 2.8079758        0 2.8079758
6     DTG       24 0.405140 2.1701767        0 2.1701767
7     DTG       48 0.578600 3.0563892        0 3.0563892
8     DTG       96 0.220755 2.7178458        0 2.7178458</code></pre>
<pre class="r"><code>sp_res$group$Olsenella_scatoligenes %&gt;%  pluck(&quot;stats&quot;) %&gt;% as.data.frame()</code></pre>
<pre><code>  cat_var long_var   median    upper_ci lower_ci         iqr
1   DRV/r        0 0.000000 0.011148333        0 0.011148333
2   DRV/r       24 0.000000 0.009085000        0 0.009085000
3   DRV/r       48 0.000000 0.006866667        0 0.006866667
4   DRV/r       96 0.000000 0.002895000        0 0.002895000
5     DTG        0 0.000000 0.007582500        0 0.007582500
6     DTG       24 0.002460 0.021300000        0 0.021300000
7     DTG       48 0.007115 0.035651667        0 0.035651667
8     DTG       96 0.005570 0.022184167        0 0.022184167</code></pre>
<div id="comparison-of-significant-species" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
<pre class="r"><code>### MNJ: Provide a table of species theat appear in the Venn&#39;s diagram.

maxn &lt;- DAsp %&gt;% 
  select(species,long_var) %&gt;% 
  group_by(long_var) %&gt;% 
  count() %&gt;% 
  pull(n) %&gt;% 
  max


unique(DAsp$long_var) %&gt;%
  set_names() %&gt;%
  map_dfc(function(tp) {
    vec &lt;- DAsp %&gt;%
      dplyr::select(species, long_var) %&gt;%
      dplyr::filter(long_var == tp) %&gt;%
      dplyr::pull(species) 
    
    if (length(vec) &lt; maxn) {
      vec &lt;- vctrs::vec_c(vec, rep(NA, maxn - length(vec)))
      return(vec)
    } else{
      return(vec)
    }
    
  }) %&gt;%
  ggVennDiagram::ggVennDiagram(., show_intersect = T)</code></pre>
<div id="htmlwidget-2b1930ff60158694b1f6" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-2b1930ff60158694b1f6">{"x":{"data":[{"x":[200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,null,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,null,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,null,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940],"y":[103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,null,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,null,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,null,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800],"type":"scatter","mode":"lines","line":{"width":null,"color":"transparent","dash":null},"hoveron":"points","hoverinfo":"none","showlegend":false,"_isGraticule":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[368.395,361.896,355.51,349.244,343.105,337.099,331.234,325.516,319.951,314.547,309.311,304.248,299.366,294.673,290.173,285.875,281.785,277.909,274.256,270.83,267.64,264.691,261.991,259.547,257.365,255.451,253.814,252.459,251.393,250.624,250.157,250,257.251,264.598,272.037,279.56,287.162,294.837,302.579,310.381,318.238,326.143,334.091,342.074,350.088,358.126,366.182,374.25,382.324,390.397,398.465,406.52,414.556,422.568,430.549,438.494,446.396,454.249,462.047,469.784,477.454,485.05,492.568,500,492.821,485.819,478.997,472.357,465.905,459.643,453.574,447.703,442.033,436.566,431.307,426.26,421.427,416.812,412.418,408.25,404.31,400.602,397.13,393.896,390.905,388.16,385.665,383.422,381.435,379.709,378.245,377.049,376.123,375.47,375.095,375,368.395,368.395],"y":[395.41,399.546,403.901,408.474,413.259,418.253,423.452,428.852,434.448,440.238,446.218,452.382,458.727,465.25,471.947,478.812,485.844,493.037,500.387,507.892,515.546,523.346,531.287,539.367,547.581,555.925,564.395,572.987,581.698,590.523,599.458,608.5,612.526,616.296,619.81,623.069,626.07,628.815,631.302,633.531,635.502,637.214,638.666,639.859,640.792,641.464,641.875,642.025,641.913,641.538,640.901,640,638.836,637.407,635.714,633.756,631.533,629.044,626.288,623.266,619.976,616.419,612.594,608.5,604.167,599.609,594.832,589.845,584.652,579.262,573.68,567.914,561.97,555.854,549.574,543.136,536.546,529.812,522.941,515.938,508.81,501.565,494.208,486.747,479.188,471.538,463.804,455.992,448.109,440.162,432.157,424.101,416,407.862,399.693,391.5,395.41,395.41],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(19,43,67,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[250.081,250.323,250.723,251.281,251.993,252.858,253.875,255.041,256.354,257.813,259.416,261.16,265.066,269.516,274.494,279.987,285.977,292.451,299.393,306.788,314.62,322.875,331.537,340.591,350.022,359.814,369.953,380.424,391.21,402.297,407.948,413.67,419.458,425.313,431.231,437.211,443.251,449.35,455.504,461.713,467.974,474.286,480.646,487.053,493.505,500,506.495,512.947,519.354,525.714,532.026,538.287,544.496,550.65,556.749,562.789,568.769,574.687,580.542,586.33,592.052,597.703,608.79,619.576,630.047,640.186,649.978,659.409,668.463,677.125,685.38,693.212,700.607,707.549,714.023,720.013,725.506,730.484,734.934,738.84,740.584,742.187,743.646,744.959,746.125,747.142,748.007,748.719,749.277,749.677,749.919,750,742.749,735.402,727.963,720.44,712.838,705.163,697.421,689.619,681.762,673.857,665.909,657.926,649.912,641.874,633.818,625.75,617.676,609.603,601.535,593.48,585.444,577.432,569.451,561.506,553.604,545.751,537.953,530.216,522.546,514.95,507.432,500,492.568,485.05,477.454,469.784,462.047,454.249,446.396,438.494,430.549,422.568,414.556,406.52,398.465,390.397,382.324,374.25,366.182,358.126,350.088,342.074,334.091,326.143,318.238,310.381,302.579,294.837,287.162,279.56,272.037,264.598,257.251,250,250.081,250.081,null,507.161,514.147,520.955,527.581,534.022,540.274,546.334,552.198,557.863,563.325,568.581,573.626,578.458,583.073,587.467,591.638,595.58,599.292,602.768,606.007,609.004,611.755,614.258,616.508,618.502,620.237,621.709,622.914,623.85,624.511,624.896,625,631.605,638.104,644.49,650.756,656.895,662.901,668.766,674.484,680.049,685.453,690.689,695.752,700.634,705.327,709.827,714.125,718.215,722.091,725.744,729.17,732.36,735.309,738.009,740.453,742.635,744.549,746.186,747.541,748.607,749.376,749.843,750,757.051,764.02,770.897,777.67,784.33,790.864,797.262,803.513,809.605,815.528,821.271,826.823,832.173,837.31,842.223,846.9,850.167,853.251,856.149,858.858,861.372,863.689,865.804,867.712,869.412,870.897,872.165,873.211,874.031,874.622,874.98,875.1,875.018,874.774,874.371,873.808,873.09,872.218,871.193,870.018,868.694,867.225,865.61,863.854,861.957,859.921,857.749,855.442,850.433,844.909,838.886,832.38,825.406,817.981,810.119,801.837,793.151,784.076,774.627,764.822,754.674,744.201,733.417,722.339,716.694,710.982,705.204,699.362,693.459,687.495,681.474,675.396,669.265,663.082,656.848,650.567,644.239,637.868,631.454,625,617.032,609.067,601.11,593.164,585.233,577.319,569.428,561.562,553.726,545.923,538.156,530.43,522.747,515.112,507.529,500,507.161,507.161,null,375.095,375.47,376.123,377.049,378.245,379.709,381.435,383.422,385.665,388.16,390.905,393.896,397.13,400.602,404.31,408.25,412.418,416.812,421.427,426.26,431.307,436.566,442.033,447.703,453.574,459.643,465.905,472.357,478.997,485.819,492.821,500,507.179,514.181,521.003,527.643,534.095,540.357,546.426,552.297,557.967,563.434,568.693,573.74,578.573,583.188,587.582,591.75,595.69,599.398,602.87,606.104,609.095,611.84,614.335,616.578,618.565,620.291,621.755,622.951,623.877,624.53,624.905,625,617.854,610.579,603.182,595.674,588.061,580.354,572.56,564.689,556.749,548.748,540.696,532.601,524.471,516.316,508.143,499.962,491.782,483.611,475.457,467.329,459.236,451.187,443.191,435.255,427.388,419.6,411.899,404.294,396.792,389.404,382.137,375,375.095,375.095],"y":[614.835,621.137,627.403,633.632,639.82,645.966,652.068,658.124,664.131,670.088,675.992,681.841,693.367,704.648,715.668,726.41,736.856,746.989,756.794,766.252,775.347,784.062,792.381,800.285,807.758,814.784,821.345,827.424,833.005,838.07,840.404,842.603,844.664,846.586,848.367,850.003,851.494,852.837,854.03,855.07,855.957,856.687,857.258,857.669,857.917,858,857.917,857.669,857.258,856.687,855.957,855.07,854.03,852.837,851.494,850.003,848.367,846.586,844.664,842.603,840.404,838.07,833.005,827.424,821.345,814.784,807.758,800.285,792.381,784.062,775.347,766.252,756.794,746.989,736.856,726.41,715.668,704.648,693.367,681.841,675.992,670.088,664.131,658.124,652.068,645.966,639.82,633.632,627.403,621.137,614.835,608.5,612.526,616.296,619.81,623.069,626.07,628.815,631.302,633.531,635.502,637.214,638.666,639.859,640.792,641.464,641.875,642.025,641.913,641.538,640.901,640,638.836,637.407,635.714,633.756,631.533,629.044,626.288,623.266,619.976,616.419,612.594,608.5,612.594,616.419,619.976,623.266,626.288,629.044,631.533,633.756,635.714,637.407,638.836,640,640.901,641.538,641.913,642.025,641.875,641.464,640.792,639.859,638.666,637.214,635.502,633.531,631.302,628.815,626.07,623.069,619.81,616.296,612.526,608.5,614.835,614.835,null,179.752,184.234,188.94,193.862,198.994,204.329,209.859,215.578,221.479,227.555,233.8,240.205,246.765,253.472,260.319,267.3,274.407,281.635,288.974,296.42,303.964,311.601,319.322,327.122,334.993,342.928,350.92,358.963,367.049,375.172,383.325,391.5,395.41,399.546,403.901,408.474,413.259,418.253,423.452,428.852,434.448,440.238,446.218,452.382,458.727,465.25,471.947,478.812,485.844,493.037,500.387,507.892,515.546,523.346,531.287,539.367,547.581,555.925,564.395,572.987,581.698,590.523,599.458,608.5,604.194,599.635,594.82,589.744,584.403,578.793,572.91,566.75,560.309,553.582,546.566,539.256,531.649,523.74,515.525,507,500.488,493.855,487.107,480.252,473.293,466.238,459.093,451.862,444.554,437.173,429.725,422.217,414.655,407.044,399.39,391.7,385.255,378.851,372.489,366.171,359.898,353.674,347.5,341.378,335.31,329.298,323.344,317.449,311.616,305.847,300.144,294.508,283.447,272.681,262.224,252.093,242.303,232.87,223.81,215.138,206.869,199.021,191.608,184.646,178.151,172.139,166.624,161.623,159.321,157.152,155.12,153.226,151.473,149.862,148.394,147.073,145.9,144.877,144.006,143.289,142.728,142.325,142.082,142,142.124,142.496,143.121,144,145.137,146.535,148.197,150.125,152.323,154.793,157.539,160.562,163.868,167.457,171.333,175.5,179.752,179.752,null,399.693,407.862,416,424.101,432.157,440.162,448.109,455.992,463.804,471.538,479.188,486.747,494.208,501.565,508.81,515.938,522.941,529.812,536.546,543.136,549.574,555.854,561.97,567.914,573.68,579.262,584.652,589.845,594.832,599.609,604.167,608.5,604.167,599.609,594.832,589.845,584.652,579.262,573.68,567.914,561.97,555.854,549.574,543.136,536.546,529.812,522.941,515.938,508.81,501.565,494.208,486.747,479.188,471.538,463.804,455.992,448.109,440.162,432.157,424.101,416,407.862,399.693,391.5,387.458,383.678,380.158,376.898,373.9,371.162,368.685,366.469,364.513,362.818,361.384,360.211,359.298,358.646,358.255,358.125,358.255,358.646,359.298,360.211,361.384,362.818,364.513,366.469,368.685,371.162,373.9,376.898,380.158,383.678,387.458,391.5,399.693,399.693],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(29,63,94,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[485.853,479.045,472.419,465.978,459.726,453.666,447.802,442.137,436.675,431.419,426.374,421.542,416.927,412.533,408.363,404.42,400.708,397.232,393.993,390.996,388.245,385.742,383.492,381.498,379.763,378.291,377.086,376.15,375.489,375.104,375,382.137,389.404,396.792,404.294,411.899,419.6,427.388,435.255,443.191,451.187,459.236,467.329,475.457,483.611,491.782,499.962,508.143,516.316,524.471,532.601,540.696,548.748,556.749,564.689,572.56,580.354,588.061,595.674,603.182,610.579,617.854,625,624.896,624.511,623.85,622.914,621.709,620.237,618.502,616.508,614.258,611.755,609.004,606.007,602.768,599.292,595.58,591.638,587.467,583.073,578.458,573.626,568.581,563.325,557.863,552.198,546.334,540.274,534.022,527.581,520.955,514.147,507.161,500,492.839,485.853,485.853],"y":[184.234,188.94,193.862,198.994,204.329,209.859,215.578,221.479,227.555,233.8,240.205,246.765,253.472,260.319,267.3,274.407,281.635,288.974,296.42,303.964,311.601,319.322,327.122,334.993,342.928,350.92,358.963,367.049,375.172,383.325,391.5,387.458,383.678,380.158,376.898,373.9,371.162,368.685,366.469,364.513,362.818,361.384,360.211,359.298,358.646,358.255,358.125,358.255,358.646,359.298,360.211,361.384,362.818,364.513,366.469,368.685,371.162,373.9,376.898,380.158,383.678,387.458,391.5,383.325,375.172,367.049,358.963,350.92,342.928,334.993,327.122,319.322,311.601,303.964,296.42,288.974,281.635,274.407,267.3,260.319,253.472,246.765,240.205,233.8,227.555,221.479,215.578,209.859,204.329,198.994,193.862,188.94,184.234,179.752,175.5,179.752,184.234,184.234],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(40,84,122,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[624.905,624.53,623.877,622.951,621.755,620.291,618.565,616.578,614.335,611.84,609.095,606.104,602.87,599.398,595.69,591.75,587.582,583.188,578.573,573.74,568.693,563.434,557.967,552.297,546.426,540.357,534.095,527.643,521.003,514.181,507.179,500,507.432,514.95,522.546,530.216,537.953,545.751,553.604,561.506,569.451,577.432,585.444,593.48,601.535,609.603,617.676,625.75,633.818,641.874,649.912,657.926,665.909,673.857,681.762,689.619,697.421,705.163,712.838,720.44,727.963,735.402,742.749,750,749.843,749.376,748.607,747.541,746.186,744.549,742.635,740.453,738.009,735.309,732.36,729.17,725.744,722.091,718.215,714.125,709.827,705.327,700.634,695.752,690.689,685.453,680.049,674.484,668.766,662.901,656.895,650.756,644.49,638.104,631.605,625,624.905,624.905],"y":[399.693,407.862,416,424.101,432.157,440.162,448.109,455.992,463.804,471.538,479.188,486.747,494.208,501.565,508.81,515.938,522.941,529.812,536.546,543.136,549.574,555.854,561.97,567.914,573.68,579.262,584.652,589.845,594.832,599.609,604.167,608.5,612.594,616.419,619.976,623.266,626.288,629.044,631.533,633.756,635.714,637.407,638.836,640,640.901,641.538,641.913,642.025,641.875,641.464,640.792,639.859,638.666,637.214,635.502,633.531,631.302,628.815,626.07,623.069,619.81,616.296,612.526,608.5,599.458,590.523,581.698,572.987,564.395,555.925,547.581,539.367,531.287,523.346,515.546,507.892,500.387,493.037,485.844,478.812,471.947,465.25,458.727,452.382,446.218,440.238,434.448,428.852,423.452,418.253,413.259,408.474,403.901,399.546,395.41,391.5,399.693,399.693],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(74,153,214,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[124.9,125.02,125.378,125.969,126.789,127.835,129.103,130.588,132.287,134.196,136.311,138.628,141.142,143.851,146.749,149.833,153.1,157.777,162.69,167.827,173.177,178.729,184.472,190.395,196.488,202.738,209.136,215.67,222.33,229.103,235.98,242.949,250,250.157,250.624,251.393,252.459,253.814,255.451,257.365,259.547,261.991,264.691,267.64,270.83,274.256,277.909,281.785,285.875,290.173,294.673,299.366,304.248,309.311,314.547,319.951,325.516,331.234,337.099,343.105,349.244,355.51,361.896,368.395,375,375.104,375.489,376.15,377.086,378.291,379.763,381.498,383.492,385.742,388.245,390.996,393.993,397.232,400.708,404.42,408.363,412.533,416.927,421.542,426.374,431.419,436.675,442.137,447.802,453.666,459.726,465.978,472.419,479.045,485.853,492.839,500,492.471,484.888,477.253,469.57,461.844,454.077,446.274,438.438,430.572,422.681,414.767,406.836,398.89,390.933,382.968,375,368.546,362.132,355.761,349.433,343.152,336.918,330.735,324.604,318.526,312.505,306.541,300.638,294.796,289.018,283.306,277.661,266.583,255.799,245.326,235.178,225.373,215.924,206.849,198.162,189.881,182.019,174.594,167.62,161.114,155.091,149.567,144.558,142.251,140.079,138.043,136.146,134.39,132.775,131.306,129.982,128.807,127.782,126.91,126.192,125.629,125.226,124.982,124.9,124.9],"y":[391.7,399.39,407.044,414.655,422.217,429.725,437.173,444.554,451.862,459.093,466.238,473.293,480.252,487.107,493.855,500.488,507,515.525,523.74,531.649,539.256,546.566,553.582,560.309,566.75,572.91,578.793,584.403,589.744,594.82,599.635,604.194,608.5,599.458,590.523,581.698,572.987,564.395,555.925,547.581,539.367,531.287,523.346,515.546,507.892,500.387,493.037,485.844,478.812,471.947,465.25,458.727,452.382,446.218,440.238,434.448,428.852,423.452,418.253,413.259,408.474,403.901,399.546,395.41,391.5,383.325,375.172,367.049,358.963,350.92,342.928,334.993,327.122,319.322,311.601,303.964,296.42,288.974,281.635,274.407,267.3,260.319,253.472,246.765,240.205,233.8,227.555,221.479,215.578,209.859,204.329,198.994,193.862,188.94,184.234,179.752,175.5,171.333,167.457,163.868,160.562,157.539,154.793,152.323,150.125,148.197,146.535,145.137,144,143.121,142.496,142.124,142,142.082,142.325,142.728,143.289,144.006,144.877,145.9,147.073,148.394,149.862,151.473,153.226,155.12,157.152,159.321,161.623,166.624,172.139,178.151,184.646,191.608,199.021,206.869,215.138,223.81,232.87,242.303,252.093,262.224,272.681,283.447,294.508,300.144,305.847,311.616,317.449,323.344,329.298,335.31,341.378,347.5,353.674,359.898,366.171,372.489,378.851,385.255,391.7,391.7],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(86,177,247,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[100,500,900],"y":[560,910,560],"text":["24","48","96"],"hovertext":["","",""],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[272.600353678248,500,727.399643854318,353.366830257892,500.002640315342,646.633169742108,499.998061434311],"y":[332.713058584906,726.385139817866,332.713082359295,548.571186362059,294.575969186258,548.571186362059,463.758078927749],"text":[8,3,3,2,4,7,3],"hovertext":["Prevotella_sp_AM42_24<br />Prevotella_stercorea Clostridium_bolteae<br />Roseburia_inulinivorans<br />Oscillibacter_sp_57_20<br />Erysipelatoclostridium_ramosum<br />Firmicutes_bacterium_CAG_791<br />Desulfovibrio_piger","Collinsella_stercoris<br />Bacteroides_vulgatus<br />Fusicatenibacter_saccharivorans","Collinsella_massiliensis<br />Blautia_wexlerae Ruminococcus_bromii","Parabacteroides_distasonis<br />Oscillibacter_sp_CAG_241","Eubacterium_eligens Coprococcus_catus<br />Lachnospira_pectinoschiza<br />Eubacterium_siraeum","Eubacterium_sp_CAG_180<br />Clostridium_citroniae Roseburia_faecis<br />Roseburia_hominis Flavonifractor_plautii<br />Firmicutes_bacterium_CAG_110 NA","Bifidobacterium_adolescentis<br />Bifidobacterium_longum<br />Olsenella_scatoligenes"],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[200],"y":[200],"name":"99_4fd2bcaee5e464335ae6ff05378d2c88","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#132B43"],[0.00334448160535117,"#132B44"],[0.00668896321070234,"#132C44"],[0.0100334448160535,"#142C45"],[0.0133779264214047,"#142D45"],[0.0167224080267558,"#142D46"],[0.020066889632107,"#142D46"],[0.0234113712374582,"#142E47"],[0.0267558528428093,"#152E47"],[0.0301003344481605,"#152F48"],[0.0334448160535117,"#152F48"],[0.0367892976588629,"#152F49"],[0.040133779264214,"#153049"],[0.0434782608695652,"#16304A"],[0.0468227424749164,"#16304A"],[0.0501672240802676,"#16314B"],[0.0535117056856187,"#16314B"],[0.0568561872909699,"#16324C"],[0.060200668896321,"#17324D"],[0.0635451505016723,"#17324D"],[0.0668896321070234,"#17334E"],[0.0702341137123746,"#17334E"],[0.0735785953177258,"#17344F"],[0.0769230769230769,"#18344F"],[0.0802675585284281,"#183450"],[0.0836120401337793,"#183550"],[0.0869565217391304,"#183551"],[0.0903010033444816,"#183651"],[0.0936454849498328,"#193652"],[0.096989966555184,"#193652"],[0.100334448160535,"#193753"],[0.103678929765886,"#193754"],[0.107023411371237,"#193854"],[0.110367892976589,"#1A3855"],[0.11371237458194,"#1A3955"],[0.117056856187291,"#1A3956"],[0.120401337792642,"#1A3956"],[0.123745819397993,"#1A3A57"],[0.127090301003344,"#1B3A57"],[0.130434782608696,"#1B3B58"],[0.133779264214047,"#1B3B59"],[0.137123745819398,"#1B3B59"],[0.140468227424749,"#1C3C5A"],[0.1438127090301,"#1C3C5A"],[0.147157190635452,"#1C3D5B"],[0.150501672240803,"#1C3D5B"],[0.153846153846154,"#1C3D5C"],[0.157190635451505,"#1D3E5C"],[0.160535117056856,"#1D3E5D"],[0.163879598662207,"#1D3F5D"],[0.167224080267559,"#1D3F5E"],[0.17056856187291,"#1D3F5F"],[0.173913043478261,"#1E405F"],[0.177257525083612,"#1E4060"],[0.180602006688963,"#1E4160"],[0.183946488294314,"#1E4161"],[0.187290969899666,"#1E4261"],[0.190635451505017,"#1F4262"],[0.193979933110368,"#1F4263"],[0.197324414715719,"#1F4363"],[0.20066889632107,"#1F4364"],[0.204013377926421,"#1F4464"],[0.207357859531773,"#204465"],[0.210702341137124,"#204465"],[0.214046822742475,"#204566"],[0.217391304347826,"#204566"],[0.220735785953177,"#214667"],[0.224080267558528,"#214668"],[0.22742474916388,"#214768"],[0.230769230769231,"#214769"],[0.234113712374582,"#214769"],[0.237458193979933,"#22486A"],[0.240802675585284,"#22486A"],[0.244147157190636,"#22496B"],[0.247491638795987,"#22496C"],[0.250836120401338,"#224A6C"],[0.254180602006689,"#234A6D"],[0.25752508361204,"#234A6D"],[0.260869565217391,"#234B6E"],[0.264214046822743,"#234B6E"],[0.267558528428094,"#244C6F"],[0.270903010033445,"#244C70"],[0.274247491638796,"#244C70"],[0.277591973244147,"#244D71"],[0.280936454849498,"#244D71"],[0.28428093645485,"#254E72"],[0.287625418060201,"#254E72"],[0.290969899665552,"#254F73"],[0.294314381270903,"#254F74"],[0.297658862876254,"#254F74"],[0.301003344481605,"#265075"],[0.304347826086957,"#265075"],[0.307692307692308,"#265176"],[0.311036789297659,"#265176"],[0.31438127090301,"#275277"],[0.317725752508361,"#275278"],[0.321070234113712,"#275278"],[0.324414715719064,"#275379"],[0.327759197324415,"#275379"],[0.331103678929766,"#28547A"],[0.334448160535117,"#28547B"],[0.337792642140468,"#28557B"],[0.341137123745819,"#28557C"],[0.344481605351171,"#28567C"],[0.347826086956522,"#29567D"],[0.351170568561873,"#29567D"],[0.354515050167224,"#29577E"],[0.357859531772575,"#29577F"],[0.361204013377926,"#2A587F"],[0.364548494983278,"#2A5880"],[0.367892976588629,"#2A5980"],[0.37123745819398,"#2A5981"],[0.374581939799331,"#2A5982"],[0.377926421404682,"#2B5A82"],[0.381270903010033,"#2B5A83"],[0.384615384615385,"#2B5B83"],[0.387959866220736,"#2B5B84"],[0.391304347826087,"#2C5C85"],[0.394648829431438,"#2C5C85"],[0.397993311036789,"#2C5D86"],[0.40133779264214,"#2C5D86"],[0.404682274247492,"#2C5D87"],[0.408026755852843,"#2D5E87"],[0.411371237458194,"#2D5E88"],[0.414715719063545,"#2D5F89"],[0.418060200668896,"#2D5F89"],[0.421404682274247,"#2E608A"],[0.424749163879599,"#2E608A"],[0.42809364548495,"#2E618B"],[0.431438127090301,"#2E618C"],[0.434782608695652,"#2E618C"],[0.438127090301003,"#2F628D"],[0.441471571906354,"#2F628D"],[0.444816053511706,"#2F638E"],[0.448160535117057,"#2F638F"],[0.451505016722408,"#30648F"],[0.454849498327759,"#306490"],[0.45819397993311,"#306590"],[0.461538461538462,"#306591"],[0.464882943143813,"#306592"],[0.468227424749164,"#316692"],[0.471571906354515,"#316693"],[0.474916387959866,"#316793"],[0.478260869565217,"#316794"],[0.481605351170569,"#326895"],[0.48494983277592,"#326895"],[0.488294314381271,"#326996"],[0.491638795986622,"#326996"],[0.494983277591973,"#326997"],[0.498327759197324,"#336A98"],[0.501672240802676,"#336A98"],[0.505016722408027,"#336B99"],[0.508361204013378,"#336B99"],[0.511705685618729,"#346C9A"],[0.51505016722408,"#346C9B"],[0.518394648829431,"#346D9B"],[0.521739130434783,"#346D9C"],[0.525083612040134,"#346E9D"],[0.528428093645485,"#356E9D"],[0.531772575250836,"#356E9E"],[0.535117056856187,"#356F9E"],[0.538461538461538,"#356F9F"],[0.54180602006689,"#3670A0"],[0.545150501672241,"#3670A0"],[0.548494983277592,"#3671A1"],[0.551839464882943,"#3671A1"],[0.555183946488294,"#3772A2"],[0.558528428093645,"#3772A3"],[0.561872909698997,"#3773A3"],[0.565217391304348,"#3773A4"],[0.568561872909699,"#3773A4"],[0.57190635451505,"#3874A5"],[0.575250836120401,"#3874A6"],[0.578595317725752,"#3875A6"],[0.581939799331104,"#3875A7"],[0.585284280936455,"#3976A8"],[0.588628762541806,"#3976A8"],[0.591973244147157,"#3977A9"],[0.595317725752508,"#3977A9"],[0.598662207357859,"#3978AA"],[0.602006688963211,"#3A78AB"],[0.605351170568562,"#3A79AB"],[0.608695652173913,"#3A79AC"],[0.612040133779264,"#3A79AC"],[0.615384615384615,"#3B7AAD"],[0.618729096989967,"#3B7AAE"],[0.622073578595318,"#3B7BAE"],[0.625418060200669,"#3B7BAF"],[0.62876254180602,"#3C7CB0"],[0.632107023411371,"#3C7CB0"],[0.635451505016722,"#3C7DB1"],[0.638795986622073,"#3C7DB1"],[0.642140468227425,"#3C7EB2"],[0.645484949832776,"#3D7EB3"],[0.648829431438127,"#3D7FB3"],[0.652173913043478,"#3D7FB4"],[0.655518394648829,"#3D7FB5"],[0.658862876254181,"#3E80B5"],[0.662207357859532,"#3E80B6"],[0.665551839464883,"#3E81B6"],[0.668896321070234,"#3E81B7"],[0.672240802675585,"#3F82B8"],[0.675585284280936,"#3F82B8"],[0.678929765886288,"#3F83B9"],[0.682274247491639,"#3F83BA"],[0.68561872909699,"#4084BA"],[0.688963210702341,"#4084BB"],[0.692307692307692,"#4085BB"],[0.695652173913043,"#4085BC"],[0.698996655518395,"#4086BD"],[0.702341137123746,"#4186BD"],[0.705685618729097,"#4186BE"],[0.709030100334448,"#4187BF"],[0.712374581939799,"#4187BF"],[0.71571906354515,"#4288C0"],[0.719063545150502,"#4288C1"],[0.722408026755853,"#4289C1"],[0.725752508361204,"#4289C2"],[0.729096989966555,"#438AC2"],[0.732441471571906,"#438AC3"],[0.735785953177257,"#438BC4"],[0.739130434782609,"#438BC4"],[0.74247491638796,"#438CC5"],[0.745819397993311,"#448CC6"],[0.749163879598662,"#448DC6"],[0.752508361204013,"#448DC7"],[0.755852842809365,"#448EC8"],[0.759197324414716,"#458EC8"],[0.762541806020067,"#458FC9"],[0.765886287625418,"#458FC9"],[0.769230769230769,"#458FCA"],[0.77257525083612,"#4690CB"],[0.775919732441472,"#4690CB"],[0.779264214046823,"#4691CC"],[0.782608695652174,"#4691CD"],[0.785953177257525,"#4792CD"],[0.789297658862876,"#4792CE"],[0.792642140468227,"#4793CF"],[0.795986622073579,"#4793CF"],[0.79933110367893,"#4894D0"],[0.802675585284281,"#4894D0"],[0.806020066889632,"#4895D1"],[0.809364548494983,"#4895D2"],[0.812709030100334,"#4896D2"],[0.816053511705686,"#4996D3"],[0.819397993311037,"#4997D4"],[0.822742474916388,"#4997D4"],[0.826086956521739,"#4998D5"],[0.82943143812709,"#4A98D6"],[0.832775919732441,"#4A99D6"],[0.836120401337793,"#4A99D7"],[0.839464882943144,"#4A9AD8"],[0.842809364548495,"#4B9AD8"],[0.846153846153846,"#4B9BD9"],[0.849498327759197,"#4B9BDA"],[0.852842809364548,"#4B9BDA"],[0.8561872909699,"#4C9CDB"],[0.859531772575251,"#4C9CDB"],[0.862876254180602,"#4C9DDC"],[0.866220735785953,"#4C9DDD"],[0.869565217391304,"#4D9EDD"],[0.872909698996656,"#4D9EDE"],[0.876254180602007,"#4D9FDF"],[0.879598662207358,"#4D9FDF"],[0.882943143812709,"#4DA0E0"],[0.88628762541806,"#4EA0E1"],[0.889632107023411,"#4EA1E1"],[0.892976588628763,"#4EA1E2"],[0.896321070234114,"#4EA2E3"],[0.899665551839465,"#4FA2E3"],[0.903010033444816,"#4FA3E4"],[0.906354515050167,"#4FA3E5"],[0.909698996655518,"#4FA4E5"],[0.91304347826087,"#50A4E6"],[0.916387959866221,"#50A5E7"],[0.919732441471572,"#50A5E7"],[0.923076923076923,"#50A6E8"],[0.926421404682274,"#51A6E8"],[0.929765886287625,"#51A7E9"],[0.933110367892977,"#51A7EA"],[0.936454849498328,"#51A8EA"],[0.939799331103679,"#52A8EB"],[0.94314381270903,"#52A9EC"],[0.946488294314381,"#52A9EC"],[0.949832775919732,"#52AAED"],[0.953177257525084,"#53AAEE"],[0.956521739130435,"#53ABEE"],[0.959866220735786,"#53ABEF"],[0.963210702341137,"#53ACF0"],[0.966555183946488,"#54ACF0"],[0.96989966555184,"#54ADF1"],[0.973244147157191,"#54ADF2"],[0.976588628762542,"#54AEF2"],[0.979933110367893,"#55AEF3"],[0.983277591973244,"#55AFF4"],[0.986622073578595,"#55AFF4"],[0.989966555183947,"#55B0F5"],[0.993311036789298,"#56B0F6"],[0.996655518394649,"#56B1F6"],[1,"#56B1F7"]],"colorbar":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"thickness":23.04,"title":"count","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["2","3","4","5","6","7","8"],"tickvals":[0,0.166666666666667,0.333333333333333,0.5,0.666666666666667,0.833333333333333,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"margin":{"t":16,"r":0,"b":0,"l":0},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[60,940],"tickmode":"array","ticktext":[" 200"," 400"," 600"," 800"],"tickvals":[200,400,600,800],"categoryorder":"array","categoryarray":[" 200"," 400"," 600"," 800"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"y","scaleratio":1,"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[103.6,948.4],"tickmode":"array","ticktext":[" 200"," 400"," 600"," 800"],"tickvals":[200,400,600,800],"categoryorder":"array","categoryarray":[" 200"," 400"," 600"," 800"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"x","scaleratio":1,"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3bcc53dfa647b":{"fill":{},"type":"scatter"},"3bcc5526a667":{"label":{}},"3bcc51f8564a9":{"label":{},"text":{}}},"cur_data":"3bcc53dfa647b","visdat":{"3bcc53dfa647b":["function (y) ","x"],"3bcc5526a667":["function (y) ","x"],"3bcc51f8564a9":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>We can observe there are 3 species which are consistently differentially abundant between treatment groups at every point during treatment, but not at basal. It’s interesting to note that of those 3, 2 are from the bifidobacterium genera (<em>B. adolescentis</em> and <em>B.longum</em>) plus another actinobacteria: <em>Olsenella scatoligenes</em>, all of the increasing more in the DTG group although the later has little literature regarding their role on the microbiome. (hover mouse over the figure numbers to display the names of the species)</p>
<pre class="r"><code>shared_sp &lt;- 
  DAsp %&gt;% 
  select(species, long_var) %&gt;% 
  group_by(species) %&gt;% 
  count() %&gt;% 
  filter(n==3) %&gt;% 
  pull(species)


shared_sp %&gt;% 
  set_names() %&gt;% 
  purrr::map(function(sp){
    otu_df %&gt;% 
      select(SampleID, sp) %&gt;% 
      left_join(metadata[,c(&quot;SampleID&quot;, &quot;group&quot;, long_var)], by=&quot;SampleID&quot;) %&gt;% 
      group_by(!!sym(long_var)) %&gt;% 
      ggstatsplot::grouped_ggbetweenstats(., 
                                          type = &quot;np&quot;,
                                          y = !!sym(sp), 
                                          x=&quot;group&quot;, 
                                          grouping.var = !!sym(long_var))
      
  })</code></pre>
<pre><code>$Bifidobacterium_adolescentis</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-1">
Past versions of “plot_shared species-1.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-1.png" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-1.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-1.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Bifidobacterium_longum</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-2">
Past versions of “plot_shared species-2.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-2.png" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-2.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-2.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Olsenella_scatoligenes</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-3">
Past versions of “plot_shared species-3.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/26a797d9ff2c12ed7faaf5728a2393e85b2e149f/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-3.png" target="_blank">26a797d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-10-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-3.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-3.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>shared_sp</code></pre>
<pre><code>[1] &quot;Bifidobacterium_adolescentis&quot; &quot;Bifidobacterium_longum&quot;      
[3] &quot;Olsenella_scatoligenes&quot;      </code></pre>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<pre class="r"><code>## This code may still be clunky and not work

endpoints &lt;- c(48,96)

future::plan(&quot;multicore&quot;)

stat_list_long &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  furrr::future_map(function(cv) {

    otu_df %&gt;%
      dplyr::select(-SampleID) %&gt;%
      colnames() %&gt;%
      purrr::set_names() %&gt;%
      purrr::map_dfr(function(sp) {
        metadata %&gt;%
          dplyr::filter(!is.na(!!sym(cv))) %&gt;%
          dplyr::pull(!!sym(cv)) %&gt;%
          unique() %&gt;%
          purrr::set_names() %&gt;%
          purrr::map_dfr(function(lv) {
            endpoints %&gt;%
              purrr::map_dfr(function(ep) {
                ids &lt;- metadata %&gt;%
                  dplyr::select(link_var, long_var, cv) %&gt;%
                  dplyr::filter(!!sym(long_var) %in% c(0, ep) &amp;
                                  !!sym(cv) == lv) %&gt;%
                  dplyr::group_by(!!sym(link_var)) %&gt;%
                  dplyr::count() %&gt;%
                  dplyr::filter(n &gt; 1) %&gt;%
                  dplyr::pull(!!sym(link_var))
                
# At the end it sounded better to just go and perform the pairwise comparisons.                
                test_df &lt;-
                  otu_df %&gt;%
                  dplyr::select(SampleID, !!sym(sp)) %&gt;%
                  dplyr::left_join(metadata[, c(&quot;SampleID&quot;,
                                                link_var,
                                                cv,
                                                long_var), drop = F],
                                   by = &quot;SampleID&quot;) %&gt;%
                  dplyr::filter(!!sym(cv) == lv &amp;
                                  time_point %in% c(0, ep) &amp;
                                  !!sym(link_var) %in% ids) %&gt;%
                  dplyr::arrange(!!sym(link_var), !!sym(long_var))
                
                
                mean_0 &lt;- mean(test_df %&gt;%
                                 dplyr::filter(!!sym(long_var) == 0) %&gt;%
                                 dplyr::pull(!!sym(sp)))
                mean_ep &lt;- mean(test_df %&gt;%
                                  dplyr::filter(!!sym(long_var) == ep) %&gt;%
                                  dplyr::pull(!!sym(sp)))
                max &lt;- max(test_df %&gt;%
                             dplyr::filter(!!sym(long_var) == ep) %&gt;%
                             dplyr::pull(!!sym(sp)))
                
                test &lt;-
                  rstatix::wilcox_test(test_df , formula(paste(sp, &quot;~&quot;, long_var)), paired = T) %&gt;%
                  mutate(
                    level = lv,
                    mean_0 = mean_0,
                    mean_ep = mean_ep,
                    mean_fc = mean_ep / mean_0,
                    max = max
                  )
                
                return(test)
              })
            
            
            
          }) %&gt;%
          filter(p &lt; 0.05)
      }) %&gt;%
      rename(species = .y.)
    
  })



names(stat_list_long) %&gt;% 
  set_names() %&gt;% 
  map(~{ stat_list_long %&gt;% 
      pluck(.x) %&gt;% 
      dplyr::filter(group2 == 48) %&gt;% 
  dplyr::arrange(level) %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;, caption = paste0(&quot;DA weeks 0 v 48 by &quot;, .x))})</code></pre>
<pre><code>$group


Table: DA weeks 0 v 48 by group

|species                        |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|    mean_ep|   mean_fc|      max|
|:------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|----------:|---------:|--------:|
|Actinomyces_oris               |0      |48     | 30| 30|       147| 0.038200|DRV/r | 0.0439317|  0.0108217| 0.2463295|  0.11888|
|Collinsella_stercoris          |0      |48     | 30| 30|        78| 0.023800|DRV/r | 0.0303690|  0.0496427| 1.6346494|  0.16594|
|Bacteroides_vulgatus           |0      |48     | 30| 30|        55| 0.003990|DRV/r | 1.5288983|  3.2104613| 2.0998527| 20.30201|
|Prevotella_sp_AM42_24          |0      |48     | 30| 30|       120| 0.041800|DRV/r | 0.3863270|  0.2056333| 0.5322779|  1.99877|
|Alistipes_finegoldii           |0      |48     | 30| 30|        19| 0.021400|DRV/r | 0.0734967|  0.3323200| 4.5215656|  5.58823|
|Lawsonibacter_asaccharolyticus |0      |48     | 30| 30|        40| 0.028300|DRV/r | 0.0178207|  0.0486013| 2.7272455|  0.25965|
|Eubacterium_ramulus            |0      |48     | 30| 30|        48| 0.011300|DRV/r | 0.1079797|  0.3098027| 2.8690834|  3.22699|
|Dorea_longicatena              |0      |48     | 30| 30|        79| 0.004920|DRV/r | 1.8978717|  3.1514560| 1.6605211|  8.11395|
|Desulfovibrio_piger            |0      |48     | 30| 30|         2| 0.010800|DRV/r | 0.0565480|  0.1624327| 2.8724741|  0.85597|
|Actinomyces_oris               |0      |48     | 30| 30|       182| 0.021800|DTG   | 0.0501333|  0.0129063| 0.2574402|  0.09711|
|Bifidobacterium_adolescentis   |0      |48     | 30| 30|        59| 0.017000|DTG   | 2.0788850|  5.4859353| 2.6388835| 36.53198|
|Olsenella_scatoligenes         |0      |48     | 30| 30|        22| 0.003530|DTG   | 0.0063480|  0.0161520| 2.5444234|  0.07012|
|Collinsella_aerofaciens        |0      |48     | 30| 30|        60| 0.001170|DTG   | 5.5624500| 10.0073290| 1.7990866| 29.78416|
|Collinsella_stercoris          |0      |48     | 30| 30|        60| 0.000687|DTG   | 0.0419423|  0.0745533| 1.7775199|  0.22048|
|Prevotella_sp_CAG_279          |0      |48     | 30| 30|        50| 0.041900|DTG   | 0.2394550|  0.4684700| 1.9564010|  3.57576|
|Prevotella_stercorea           |0      |48     | 30| 30|       191| 0.037700|DTG   | 1.5882923|  1.0135223| 0.6381208| 17.66690|
|Alistipes_putredinis           |0      |48     | 30| 30|        29| 0.014700|DTG   | 0.0851407|  0.5119567| 6.0130686|  9.42400|
|Streptococcus_parasanguinis    |0      |48     | 30| 30|       231| 0.021400|DTG   | 0.4277057|  0.1767477| 0.4132460|  2.28742|
|Streptococcus_salivarius       |0      |48     | 30| 30|       256| 0.042200|DTG   | 1.1966783|  0.8150510| 0.6810945| 17.14400|
|Oscillibacter_sp_57_20         |0      |48     | 30| 30|        81| 0.017000|DTG   | 0.3043163|  0.6130057| 2.0143699|  2.08000|
|Oscillibacter_sp_CAG_241       |0      |48     | 30| 30|        61| 0.020000|DTG   | 0.4518493|  1.4654737| 3.2432795| 19.36226|
|Clostridium_leptum             |0      |48     | 30| 30|        27| 0.011500|DTG   | 0.0136740|  0.0458170| 3.3506655|  0.65338|
|Catenibacterium_mitsuokai      |0      |48     | 30| 30|        49| 0.038300|DTG   | 1.4430797|  2.7269180| 1.8896517| 13.97066|
|Clostridium_innocuum           |0      |48     | 30| 30|       123| 0.029400|DTG   | 1.4544320|  0.0349487| 0.0240291|  0.34783|
|Erysipelatoclostridium_ramosum |0      |48     | 30| 30|        77| 0.030300|DTG   | 0.1679987|  0.0438080| 0.2607640|  0.59202|
|Firmicutes_bacterium_CAG_110   |0      |48     | 30| 30|        37| 0.036600|DTG   | 0.0446687|  0.1863687| 4.1722460|  1.84291|
|Firmicutes_bacterium_CAG_83    |0      |48     | 30| 30|        80| 0.047100|DTG   | 0.1838993|  0.2759920| 1.5007776|  1.17639|

$risk_group


Table: DA weeks 0 v 48 by risk_group

|species                        |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|   mean_ep|    mean_fc|      max|
|:------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|---------:|----------:|--------:|
|Actinomyces_oris               |0      |48     | 27| 27|       129| 0.013800|hts   | 0.0573352| 0.0136878|  0.2387326|  0.09711|
|Bacteroides_vulgatus           |0      |48     | 27| 27|        25| 0.001760|hts   | 0.8714770| 2.9948304|  3.4364995| 20.30201|
|Prevotella_stercorea           |0      |48     | 27| 27|        52| 0.014400|hts   | 0.8056959| 0.1244170|  0.1544218|  1.49790|
|Lawsonibacter_asaccharolyticus |0      |48     | 27| 27|        52| 0.028500|hts   | 0.0210444| 0.0460000|  2.1858501|  0.25965|
|Dorea_longicatena              |0      |48     | 27| 27|        81| 0.029300|hts   | 1.4381885| 2.2353404|  1.5542749|  6.05477|
|Oscillibacter_sp_CAG_241       |0      |48     | 27| 27|         9| 0.006950|hts   | 0.2471148| 1.5033522|  6.0836184| 19.36226|
|Bifidobacterium_adolescentis   |0      |48     | 30| 30|        37| 0.036600|msm   | 0.7617650| 2.1569693|  2.8315417| 17.23829|
|Olsenella_scatoligenes         |0      |48     | 30| 30|        62| 0.007130|msm   | 0.0084210| 0.0176763|  2.0990777|  0.07012|
|Collinsella_aerofaciens        |0      |48     | 30| 30|        85| 0.007460|msm   | 5.7444580| 9.7201823|  1.6920974| 29.78416|
|Collinsella_stercoris          |0      |48     | 30| 30|        32| 0.000103|msm   | 0.0349823| 0.0745733|  2.1317427|  0.20568|
|Collinsella_massiliensis       |0      |48     | 30| 30|        70| 0.004410|msm   | 0.0199720| 0.0072420|  0.3626077|  0.02275|
|Bacteroides_sp_CAG_530         |0      |48     | 30| 30|        39| 0.045100|msm   | 0.0635053| 0.1676333|  2.6396733|  1.68154|
|Alistipes_putredinis           |0      |48     | 30| 30|        21| 0.028800|msm   | 0.0045313| 0.0561943| 12.4012800|  1.14177|
|Eubacterium_hallii             |0      |48     | 30| 30|       129| 0.032700|msm   | 0.9245843| 1.4614497|  1.5806559|  5.27714|
|Eubacterium_ramulus            |0      |48     | 30| 30|        67| 0.003510|msm   | 0.1043493| 0.3687677|  3.5339724|  3.22699|
|Dorea_longicatena              |0      |48     | 30| 30|        90| 0.006030|msm   | 2.0998977| 3.2730557|  1.5586739|  8.11395|
|Oscillibacter_sp_57_20         |0      |48     | 30| 30|        93| 0.037300|msm   | 0.3615313| 0.6550140|  1.8117766|  2.02907|
|Gemmiger_formicilis            |0      |48     | 30| 30|       114| 0.043900|msm   | 0.1852200| 0.2543883|  1.3734388|  0.70821|
|Catenibacterium_mitsuokai      |0      |48     | 30| 30|        58| 0.005140|msm   | 1.4741540| 3.4299457|  2.3267214| 13.97066|
|Mitsuokella_jalaludinii        |0      |48     | 30| 30|        36| 0.018600|msm   | 0.0250413| 0.0849303|  3.3916059|  0.48797|
|Megasphaera_elsdenii           |0      |48     | 30| 30|        38| 0.013000|msm   | 0.0934290| 0.2806840|  3.0042492|  2.74434|
|Escherichia_coli               |0      |48     | 30| 30|       255| 0.013300|msm   | 3.8261203| 0.4037610|  0.1055275|  7.70176|

$center


Table: DA weeks 0 v 48 by center

|species                         |group1 |group2 | n1| n2| statistic|       p|level       |    mean_0|    mean_ep|   mean_fc|      max|
|:-------------------------------|:------|:------|--:|--:|---------:|-------:|:-----------|---------:|----------:|---------:|--------:|
|Slackia_isoflavoniconvertens    |0      |48     | 12| 12|         6| 0.03230|bellvitge   | 0.2759617|  0.4125658|  1.495011|  1.03564|
|Bacteroides_caccae              |0      |48     | 12| 12|         0| 0.03600|bellvitge   | 0.1194742|  1.2072775| 10.104925|  9.28240|
|Ruminococcus_torques            |0      |48     | 12| 12|         2| 0.03000|bellvitge   | 0.5894275|  1.4690925|  2.492406|  5.33843|
|Dorea_longicatena               |0      |48     | 12| 12|        10| 0.04540|bellvitge   | 1.4585783|  2.4487992|  1.678894|  6.05477|
|Flavonifractor_plautii          |0      |48     | 12| 12|         5| 0.02490|bellvitge   | 0.3142900|  0.8396983|  2.671731|  5.01285|
|Clostridium_leptum              |0      |48     | 12| 12|         0| 0.03600|bellvitge   | 0.0042758|  0.0235342|  5.503995|  0.08918|
|Ruminococcus_bicirculans        |0      |48     | 12| 12|         1| 0.02090|bellvitge   | 0.0360600|  0.3363617|  9.327833|  1.35126|
|Ruthenibacterium_lactatiformans |0      |48     | 12| 12|         3| 0.00872|bellvitge   | 0.1706033|  0.2695792|  1.580152|  1.85450|
|Firmicutes_bacterium_CAG_110    |0      |48     | 12| 12|         0| 0.03600|bellvitge   | 0.0214417|  0.0462983|  2.159269|  0.40845|
|Bifidobacterium_adolescentis    |0      |48     | 26| 26|         9| 0.01190|clinic      | 0.2992450|  2.2969212|  7.675721| 17.23829|
|Collinsella_stercoris           |0      |48     | 26| 26|        49| 0.02180|clinic      | 0.0315654|  0.0523508|  1.658487|  0.20568|
|Alistipes_putredinis            |0      |48     | 26| 26|        20| 0.02490|clinic      | 0.0171558|  0.2537500| 14.790943|  2.45873|
|Lawsonibacter_asaccharolyticus  |0      |48     | 26| 26|        23| 0.03820|clinic      | 0.0184735|  0.0683804|  3.701547|  0.78457|
|Roseburia_hominis               |0      |48     | 26| 26|       174| 0.04380|clinic      | 0.3675242|  0.0874208|  0.237864|  0.51665|
|Clostridium_leptum              |0      |48     | 26| 26|        26| 0.03190|clinic      | 0.0291700|  0.0742242|  2.544540|  0.65338|
|Catenibacterium_mitsuokai       |0      |48     | 26| 26|        29| 0.04650|clinic      | 1.1409969|  2.3450923|  2.055301| 13.97066|
|Bilophila_wadsworthia           |0      |48     | 26| 26|        27| 0.03620|clinic      | 0.0414892|  0.0816404|  1.967749|  0.47095|
|Desulfovibrio_piger             |0      |48     | 26| 26|        11| 0.03100|clinic      | 0.0501219|  0.1950712|  3.891933|  1.84978|
|Bifidobacterium_adolescentis    |0      |48     | 11| 11|         0| 0.02250|vall_hebron | 0.5455718|  8.3561627| 15.316339| 36.53198|
|Collinsella_aerofaciens         |0      |48     | 11| 11|         0| 0.00915|vall_hebron | 5.3178527| 10.3750527|  1.950985| 19.55574|
|Bacteroides_vulgatus            |0      |48     | 11| 11|         4| 0.03300|vall_hebron | 0.3223482|  1.1683373|  3.624458|  5.94228|
|Dorea_longicatena               |0      |48     | 11| 11|         6| 0.03230|vall_hebron | 1.0599845|  2.7496118|  2.594011|  6.75009|
|Oscillibacter_sp_57_20          |0      |48     | 11| 11|         3| 0.01440|vall_hebron | 0.2144364|  0.6305918|  2.940694|  1.71392|

$gender


Table: DA weeks 0 v 48 by gender

|species                        |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|    mean_ep|   mean_fc|      max|
|:------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|----------:|---------:|--------:|
|Actinomyces_oris               |0      |48     | 52| 52|       448| 1.03e-02|male  | 0.0355627|  0.0104146| 0.2928523|  0.11888|
|Bifidobacterium_adolescentis   |0      |48     | 52| 52|       113| 8.40e-03|male  | 1.1117535|  3.2089792| 2.8864126| 28.49392|
|Olsenella_scatoligenes         |0      |48     | 52| 52|       165| 2.40e-02|male  | 0.0070198|  0.0120783| 1.7205983|  0.07012|
|Collinsella_aerofaciens        |0      |48     | 52| 52|       232| 2.66e-04|male  | 5.9005775| 10.1446588| 1.7192654| 41.72475|
|Collinsella_stercoris          |0      |48     | 52| 52|       176| 2.44e-05|male  | 0.0339752|  0.0628356| 1.8494546|  0.20568|
|Collinsella_massiliensis       |0      |48     | 52| 52|       301| 1.48e-02|male  | 0.0169360|  0.0074885| 0.4421634|  0.06159|
|Bacteroides_vulgatus           |0      |48     | 52| 52|       263| 1.87e-02|male  | 0.9543829|  1.6850417| 1.7655825| 14.42645|
|Odoribacter_splanchnicus       |0      |48     | 52| 52|       281| 3.35e-02|male  | 0.1163235|  0.1787419| 1.5365939|  1.89901|
|Prevotella_sp_CAG_279          |0      |48     | 52| 52|       168| 4.54e-02|male  | 0.2968394|  0.5408294| 1.8219596|  6.83686|
|Alistipes_putredinis           |0      |48     | 52| 52|        56| 2.95e-04|male  | 0.1877860|  0.4025973| 2.1439159|  6.07415|
|Streptococcus_parasanguinis    |0      |48     | 52| 52|       720| 2.92e-03|male  | 0.3978448|  0.4686344| 1.1779327| 18.32477|
|Lawsonibacter_asaccharolyticus |0      |48     | 52| 52|       152| 2.22e-02|male  | 0.0158644|  0.0454408| 2.8643190|  0.78457|
|Dorea_longicatena              |0      |48     | 52| 52|       333| 2.01e-03|male  | 1.9763548|  2.9041137| 1.4694293|  8.11395|
|Oscillibacter_sp_57_20         |0      |48     | 52| 52|       269| 3.70e-02|male  | 0.3338527|  0.5411652| 1.6209700|  2.08000|
|Catenibacterium_mitsuokai      |0      |48     | 52| 52|       151| 2.55e-03|male  | 1.2832354|  2.6329910| 2.0518379| 13.97066|
|Bilophila_wadsworthia          |0      |48     | 52| 52|       145| 9.36e-03|male  | 0.0380790|  0.0739335| 1.9415790|  0.47095|
|Desulfovibrio_piger            |0      |48     | 52| 52|        64| 1.46e-02|male  | 0.1278337|  0.2101879| 1.6442297|  1.84978|

$ethnic_group


Table: DA weeks 0 v 48 by ethnic_group

|species                      |group1 |group2 | n1| n2| statistic|       p|level      |    mean_0|   mean_ep|    mean_fc|      max|
|:----------------------------|:------|:------|--:|--:|---------:|-------:|:----------|---------:|---------:|----------:|--------:|
|Olsenella_scatoligenes       |0      |48     | 24| 24|        14| 0.03030|caucassian | 0.0043271| 0.0102354|  2.3654309|  0.05315|
|Collinsella_aerofaciens      |0      |48     | 24| 24|        31| 0.00120|caucassian | 4.7417121| 9.9169288|  2.0914236| 29.78416|
|Collinsella_stercoris        |0      |48     | 24| 24|        36| 0.00202|caucassian | 0.0390554| 0.0729712|  1.8684028|  0.22048|
|Eggerthella_lenta            |0      |48     | 24| 24|        89| 0.02380|caucassian | 0.6680967| 0.1190417|  0.1781803|  1.43938|
|Prevotella_stercorea         |0      |48     | 24| 24|       115| 0.01620|caucassian | 1.3982821| 0.4911087|  0.3512229|  4.05957|
|Alistipes_shahii             |0      |48     | 24| 24|       104| 0.01350|caucassian | 0.1470925| 0.0302471|  0.2056331|  0.23046|
|Dorea_longicatena            |0      |48     | 24| 24|        69| 0.03720|caucassian | 2.1143717| 3.1099692|  1.4708716|  8.11395|
|Ruminococcus_bicirculans     |0      |48     | 24| 24|        12| 0.03760|caucassian | 0.1262971| 1.0609663|  8.4005602|  9.39081|
|Actinomyces_oris             |0      |48     | 22| 22|       104| 0.01350|hispanic   | 0.0554991| 0.0081282|  0.1464561|  0.07622|
|Bifidobacterium_adolescentis |0      |48     | 22| 22|        24| 0.04380|hispanic   | 2.9305886| 6.2031309|  2.1166843| 36.53198|
|Bacteroides_vulgatus         |0      |48     | 22| 22|        36| 0.01050|hispanic   | 0.1916559| 2.1718805| 11.3321862| 14.42645|
|Butyricimonas_virosa         |0      |48     | 22| 22|         9| 0.03670|hispanic   | 0.0105936| 0.0328818|  3.1039217|  0.27583|
|Odoribacter_splanchnicus     |0      |48     | 22| 22|        27| 0.00660|hispanic   | 0.0881950| 0.2783118|  3.1556417|  1.89901|
|Alistipes_putredinis         |0      |48     | 22| 22|        20| 0.04460|hispanic   | 0.4480373| 1.0177550|  2.2715856|  9.42400|
|Alistipes_shahii             |0      |48     | 22| 22|        19| 0.00697|hispanic   | 0.0351045| 0.3154423|  8.9857957|  3.66522|
|Streptococcus_parasanguinis  |0      |48     | 22| 22|       164| 0.02900|hispanic   | 0.4757064| 0.2094505|  0.4402936|  2.28742|
|Dorea_longicatena            |0      |48     | 22| 22|        44| 0.00593|hispanic   | 1.7335577| 2.9145355|  1.6812451|  6.75009|
|Clostridium_bolteae          |0      |48     | 22| 22|        67| 0.03100|hispanic   | 0.7362182| 0.0192045|  0.0260854|  0.21388|
|Oscillibacter_sp_57_20       |0      |48     | 22| 22|        38| 0.04070|hispanic   | 0.3127736| 0.6232209|  1.9925621|  2.02907|
|Catenibacterium_mitsuokai    |0      |48     | 22| 22|        16| 0.00451|hispanic   | 0.7536659| 2.6100232|  3.4631037| 11.94837|
|Mitsuokella_jalaludinii      |0      |48     | 22| 22|        10| 0.04540|hispanic   | 0.0161200| 0.0552655|  3.4283781|  0.38756|
|Mitsuokella_multacida        |0      |48     | 22| 22|         9| 0.03670|hispanic   | 0.0041086| 0.0308427|  7.5068038|  0.38657|
|Bilophila_wadsworthia        |0      |48     | 22| 22|        16| 0.01350|hispanic   | 0.0283968| 0.0777827|  2.7391353|  0.47095|
|Desulfovibrio_piger          |0      |48     | 22| 22|         3| 0.02440|hispanic   | 0.0882114| 0.1718218|  1.9478422|  0.93650|

$CD4diff_48


Table: DA weeks 0 v 48 by CD4diff_48

|species                         |group1 |group2 | n1| n2| statistic|        p|level |     mean_0|    mean_ep|   mean_fc|      max|
|:-------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|----------:|----------:|---------:|--------:|
|Actinomyces_oris                |0      |48     | 42| 42|       396| 1.39e-02|150_H |  0.0545260|  0.0127836| 0.2344493|  0.11888|
|Bifidobacterium_adolescentis    |0      |48     | 42| 42|        83| 1.95e-02|150_H |  0.9342010|  3.0194945| 3.2321681| 28.49392|
|Collinsella_aerofaciens         |0      |48     | 42| 42|       134| 1.06e-03|150_H |  5.7206495| 10.2330436| 1.7887905| 41.72475|
|Collinsella_stercoris           |0      |48     | 42| 42|        89| 7.73e-05|150_H |  0.0321264|  0.0649381| 2.0213294|  0.20568|
|Collinsella_massiliensis        |0      |48     | 42| 42|       173| 3.40e-02|150_H |  0.0202426|  0.0077507| 0.3828909|  0.06159|
|Paraprevotella_xylaniphila      |0      |48     | 42| 42|       146| 4.21e-02|150_H |  0.0622538|  0.0112914| 0.1813773|  0.12627|
|Alistipes_putredinis            |0      |48     | 42| 42|        55| 6.93e-03|150_H |  0.1988021|  0.3302526| 1.6612126|  6.07415|
|Streptococcus_parasanguinis     |0      |48     | 42| 42|       492| 1.28e-02|150_H |  0.4689638|  0.6477729| 1.3812854| 18.32477|
|Eubacterium_ramulus             |0      |48     | 42| 42|       138| 1.12e-02|150_H |  0.1025671|  0.2618621| 2.5530802|  3.22699|
|Coprococcus_catus               |0      |48     | 42| 42|       172| 3.26e-02|150_H |  0.2373440|  0.3720631| 1.5676108|  2.65820|
|Dorea_longicatena               |0      |48     | 42| 42|       226| 2.25e-02|150_H |  1.9283833|  2.6086171| 1.3527482|  6.59686|
|Catenibacterium_mitsuokai       |0      |48     | 42| 42|        92| 2.04e-02|150_H |  1.2430588|  2.1588007| 1.7366843|  9.47942|
|Clostridium_innocuum            |0      |48     | 42| 42|       239| 4.09e-02|150_H |  1.2295276|  0.0678464| 0.0551809|  0.64613|
|Firmicutes_bacterium_CAG_110    |0      |48     | 42| 42|        48| 3.49e-02|150_H |  0.0307069|  0.0859295| 2.7983779|  1.18582|
|Desulfovibrio_piger             |0      |48     | 42| 42|        34| 4.68e-02|150_H |  0.1188400|  0.1581645| 1.3309031|  0.93650|
|Bacteroides_thetaiotaomicron    |0      |48     | 17| 17|         3| 2.44e-02|150_L |  0.1354876|  0.2654200| 1.9589978|  1.39334|
|Bacteroides_vulgatus            |0      |48     | 17| 17|         0| 2.53e-03|150_L |  0.7123300|  3.1639965| 4.4417566| 20.30201|
|Prevotella_copri                |0      |48     | 17| 17|        57| 3.67e-02|150_L | 14.3589871|  7.4943229| 0.5219256| 34.68962|
|Prevotella_sp_CAG_279           |0      |48     | 17| 17|         1| 2.09e-02|150_L |  0.2648012|  0.8752476| 3.3053012|  6.83686|
|Prevotella_stercorea            |0      |48     | 17| 17|        40| 4.40e-02|150_L |  1.4005847|  0.4607347| 0.3289588|  3.71059|
|Parabacteroides_distasonis      |0      |48     | 17| 17|        23| 3.82e-02|150_L |  0.8908965|  1.4528765| 1.6308028|  8.28685|
|Dorea_longicatena               |0      |48     | 17| 17|        26| 1.50e-02|150_L |  1.6301206|  3.0095782| 1.8462304|  8.11395|
|Roseburia_faecis                |0      |48     | 17| 17|       102| 1.84e-02|150_L |  3.7909771|  0.9892076| 0.2609374|  3.53274|
|Oscillibacter_sp_CAG_241        |0      |48     | 17| 17|        20| 4.46e-02|150_L |  0.3597129|  1.7932576| 4.9852464| 19.36226|
|Ruthenibacterium_lactatiformans |0      |48     | 17| 17|        34| 4.48e-02|150_L |  0.1602665|  0.2784265| 1.7372721|  2.35945|
|Bilophila_wadsworthia           |0      |48     | 17| 17|        16| 4.27e-02|150_L |  0.0212753|  0.0893324| 4.1988775|  0.47095|

$CD8diff_48


Table: DA weeks 0 v 48 by CD8diff_48

|species                        |group1 |group2 | n1| n2| statistic|       p|level |    mean_0|   mean_ep|   mean_fc|      max|
|:------------------------------|:------|:------|--:|--:|---------:|-------:|:-----|---------:|---------:|---------:|--------:|
|Collinsella_aerofaciens        |0      |48     | 39| 39|       173| 0.02050|50_H  | 6.3291249| 9.6694936| 1.5277773| 41.72475|
|Collinsella_stercoris          |0      |48     | 39| 39|       119| 0.00234|50_H  | 0.0342164| 0.0588746| 1.7206544|  0.20568|
|Odoribacter_splanchnicus       |0      |48     | 39| 39|       145| 0.04460|50_H  | 0.1169167| 0.2164951| 1.8517046|  1.89901|
|Alistipes_putredinis           |0      |48     | 39| 39|        56| 0.00434|50_H  | 0.2474279| 0.4994010| 2.0183695|  6.07415|
|Streptococcus_parasanguinis    |0      |48     | 39| 39|       449| 0.00984|50_H  | 0.3945764| 0.6168169| 1.5632382| 18.32477|
|Lawsonibacter_asaccharolyticus |0      |48     | 39| 39|        95| 0.04220|50_H  | 0.0204087| 0.0635515| 3.1139407|  0.78457|
|Dorea_longicatena              |0      |48     | 39| 39|       223| 0.03300|50_H  | 2.0456121| 2.6963554| 1.3181167|  6.59686|
|Gemmiger_formicilis            |0      |48     | 39| 39|       181| 0.02880|50_H  | 0.2844908| 0.4540954| 1.5961691|  7.81153|
|Catenibacterium_mitsuokai      |0      |48     | 39| 39|        65| 0.00301|50_H  | 1.1480890| 2.3277169| 2.0274708|  9.65802|
|Firmicutes_bacterium_CAG_110   |0      |48     | 39| 39|        37| 0.02070|50_H  | 0.0369105| 0.1005244| 2.7234615|  1.18582|
|Desulfovibrio_piger            |0      |48     | 39| 39|        28| 0.04110|50_H  | 0.1241554| 0.1680454| 1.3535086|  0.93650|
|Actinomyces_odontolyticus      |0      |48     | 20| 20|       138| 0.02350|50_L  | 0.0672975| 0.0160805| 0.2389465|  0.14560|
|Actinomyces_oris               |0      |48     | 20| 20|        91| 0.01710|50_L  | 0.0849990| 0.0129230| 0.1520371|  0.09711|
|Collinsella_aerofaciens        |0      |48     | 20| 20|        26| 0.01790|50_L  | 5.0196300| 9.1717395| 1.8271744| 27.19569|
|Collinsella_stercoris          |0      |48     | 20| 20|        38| 0.02300|50_L  | 0.0414535| 0.0702005| 1.6934758|  0.22048|
|Eggerthella_lenta              |0      |48     | 20| 20|        68| 0.02540|50_L  | 0.7094230| 0.2064170| 0.2909646|  1.43938|
|Bacteroides_vulgatus           |0      |48     | 20| 20|        30| 0.02940|50_L  | 1.4766515| 2.7226300| 1.8437864| 20.30201|
|Ruminococcus_torques           |0      |48     | 20| 20|        24| 0.04380|50_L  | 1.1628095| 2.5481930| 2.1914105| 17.58464|
|Coprococcus_catus              |0      |48     | 20| 20|        21| 0.00532|50_L  | 0.1982025| 0.4571485| 2.3064719|  1.53044|
|Dorea_longicatena              |0      |48     | 20| 20|        32| 0.02100|50_L  | 1.4462640| 2.7783445| 1.9210493|  8.11395|
|Oscillibacter_sp_57_20         |0      |48     | 20| 20|        25| 0.02800|50_L  | 0.2655430| 0.5069835| 1.9092332|  2.08000|
|Clostridium_innocuum           |0      |48     | 20| 20|        97| 0.03820|50_L  | 1.4948370| 0.0536140| 0.0358661|  0.50642|
|Erysipelatoclostridium_ramosum |0      |48     | 20| 20|        51| 0.01910|50_L  | 0.4558435| 0.0605085| 0.1327396|  1.00701|

$CD4after_48


Table: DA weeks 0 v 48 by CD4after_48

|species                         |group1 |group2 | n1| n2| statistic|       p|level |    mean_0|    mean_ep|   mean_fc|      max|
|:-------------------------------|:------|:------|--:|--:|---------:|-------:|:-----|---------:|----------:|---------:|--------:|
|Bacteroides_caccae              |0      |48     | 25| 25|        10| 0.02540|low   | 0.0982256|  0.6300816| 6.4146373|  9.28240|
|Bacteroides_thetaiotaomicron    |0      |48     | 25| 25|        12| 0.02110|low   | 0.1187332|  0.2775004| 2.3371761|  1.86609|
|Bacteroides_vulgatus            |0      |48     | 25| 25|        20| 0.00161|low   | 0.6099176|  2.5378140| 4.1609129| 20.30201|
|Prevotella_sp_CAG_279           |0      |48     | 25| 25|         8| 0.00344|low   | 0.2565588|  0.8415620| 3.2801915|  6.83686|
|Alistipes_putredinis            |0      |48     | 25| 25|        33| 0.04180|low   | 0.2761556|  0.7540964| 2.7306939|  9.42400|
|Streptococcus_mitis             |0      |48     | 25| 25|        77| 0.03030|low   | 0.0115628|  0.0020396| 0.1763933|  0.02690|
|Lawsonibacter_asaccharolyticus  |0      |48     | 25| 25|        23| 0.00401|low   | 0.0136080|  0.0772044| 5.6734568|  0.78457|
|Dorea_longicatena               |0      |48     | 25| 25|        48| 0.00130|low   | 1.4356360|  2.9607636| 2.0623359|  8.11395|
|Roseburia_faecis                |0      |48     | 25| 25|       196| 0.02510|low   | 2.9789360|  0.9795824| 0.3288363|  3.53274|
|Clostridium_leptum              |0      |48     | 25| 25|        15| 0.02020|low   | 0.0084756|  0.0321668| 3.7952239|  0.47838|
|Ruthenibacterium_lactatiformans |0      |48     | 25| 25|        61| 0.01150|low   | 0.1328484|  0.2561032| 1.9277854|  2.35945|
|Catenibacterium_mitsuokai       |0      |48     | 25| 25|        21| 0.02880|low   | 0.7941172|  2.4938624| 3.1404211| 13.97066|
|Actinomyces_oris                |0      |48     | 29| 29|       192| 0.00825|mid   | 0.0505141|  0.0093776| 0.1856428|  0.07622|
|Collinsella_aerofaciens         |0      |48     | 29| 29|        66| 0.00979|mid   | 5.9729562| 10.5837072| 1.7719379| 41.72475|
|Collinsella_stercoris           |0      |48     | 29| 29|        61| 0.00379|mid   | 0.0331600|  0.0562997| 1.6978183|  0.15620|
|Collinsella_massiliensis        |0      |48     | 29| 29|        73| 0.02880|mid   | 0.0205731|  0.0066931| 0.3253327|  0.03063|
|Alistipes_putredinis            |0      |48     | 29| 29|        10| 0.00295|mid   | 0.1941476|  0.4544224| 2.3406030|  6.07415|
|Streptococcus_parasanguinis     |0      |48     | 29| 29|       263| 0.00131|mid   | 0.3678538|  0.1824303| 0.4959317|  2.28742|
|Clostridium_innocuum            |0      |48     | 29| 29|       131| 0.01060|mid   | 1.6609600|  0.0571148| 0.0343866|  0.50642|
|Erysipelatoclostridium_ramosum  |0      |48     | 29| 29|        66| 0.03760|mid   | 0.3403145|  0.0831552| 0.2443480|  1.00701|

$CD8after_48


Table: DA weeks 0 v 48 by CD8after_48

|species                         |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|   mean_ep|    mean_fc|      max|
|:-------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|---------:|----------:|--------:|
|Actinomyces_oris                |0      |48     | 49| 49|       416| 0.015900|high  | 0.0391880| 0.0119333|  0.3045136|  0.11888|
|Bifidobacterium_adolescentis    |0      |48     | 49| 49|        87| 0.004940|high  | 0.8948951| 3.8068945|  4.2540120| 36.53198|
|Olsenella_scatoligenes          |0      |48     | 49| 49|       133| 0.041700|high  | 0.0068384| 0.0116020|  1.6966098|  0.07012|
|Collinsella_aerofaciens         |0      |48     | 49| 49|       214| 0.001800|high  | 5.3155373| 9.0087857|  1.6948024| 41.72475|
|Collinsella_stercoris           |0      |48     | 49| 49|       196| 0.000842|high  | 0.0359041| 0.0589753|  1.6425794|  0.22048|
|Collinsella_massiliensis        |0      |48     | 49| 49|       248| 0.029900|high  | 0.0176545| 0.0069204|  0.3919914|  0.06159|
|Alistipes_putredinis            |0      |48     | 49| 49|        75| 0.002140|high  | 0.2395529| 0.4588376|  1.9153917|  6.07415|
|Streptococcus_parasanguinis     |0      |48     | 49| 49|       665| 0.007740|high  | 0.3656798| 0.5491973|  1.5018531| 18.32477|
|Lawsonibacter_asaccharolyticus  |0      |48     | 49| 49|       135| 0.046000|high  | 0.0155855| 0.0488373|  3.1335097|  0.78457|
|Dorea_longicatena               |0      |48     | 49| 49|       295| 0.004490|high  | 1.8632524| 2.6228818|  1.4076900|  6.59686|
|Clostridium_leptum              |0      |48     | 49| 49|       115| 0.046300|high  | 0.0229959| 0.0488322|  2.1235179|  0.65338|
|Ruminococcus_bicirculans        |0      |48     | 49| 49|        61| 0.020000|high  | 0.1914078| 0.6965673|  3.6391804|  9.39081|
|Catenibacterium_mitsuokai       |0      |48     | 49| 49|        98| 0.001970|high  | 1.1559049| 2.4088390|  2.0839422| 13.97066|
|Firmicutes_bacterium_CAG_110    |0      |48     | 49| 49|        75| 0.033300|high  | 0.0336600| 0.0873551|  2.5952199|  1.18582|
|Mitsuokella_jalaludinii         |0      |48     | 49| 49|        58| 0.047600|high  | 0.0295784| 0.0600437|  2.0299861|  0.48797|
|Bilophila_wadsworthia           |0      |48     | 49| 49|       131| 0.022400|high  | 0.0299718| 0.0673608|  2.2474704|  0.47095|
|Desulfovibrio_piger             |0      |48     | 49| 49|        51| 0.045800|high  | 0.1178280| 0.1614076|  1.3698578|  0.93650|
|Methanobrevibacter_smithii      |0      |48     |  9|  9|        28| 0.022500|mid   | 1.1983367| 0.5120489|  0.4272997|  2.41617|
|Bifidobacterium_longum          |0      |48     |  9|  9|        42| 0.019500|mid   | 9.5339833| 2.4422256|  0.2561600|  7.98647|
|Bacteroides_vulgatus            |0      |48     |  9|  9|         1| 0.007810|mid   | 2.1979356| 5.4527444|  2.4808482| 20.30201|
|Bacteroides_xylanisolvens       |0      |48     |  9|  9|         0| 0.036000|mid   | 0.0062633| 0.1008567| 16.1027142|  0.34337|
|Parabacteroides_merdae          |0      |48     |  9|  9|         0| 0.036000|mid   | 0.1481133| 0.8324300|  5.6202233|  3.27753|
|Lawsonibacter_asaccharolyticus  |0      |48     |  9|  9|         2| 0.030000|mid   | 0.0117667| 0.0348933|  2.9654391|  0.14448|
|Ruminococcus_bromii             |0      |48     |  9|  9|         1| 0.034600|mid   | 0.7550656| 4.4528367|  5.8972849| 12.14850|
|Ruthenibacterium_lactatiformans |0      |48     |  9|  9|         3| 0.019500|mid   | 0.0950622| 0.4557600|  4.7943335|  2.35945|

$CD4diff_96


Table: DA weeks 0 v 48 by CD4diff_96

|species                      |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|   mean_ep|    mean_fc|      max|
|:----------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|---------:|----------:|--------:|
|Actinomyces_oris             |0      |48     | 34| 34|       269| 0.004340|150_H | 0.0605147| 0.0082232|  0.1358882|  0.07622|
|Bifidobacterium_adolescentis |0      |48     | 34| 34|        38| 0.004280|150_H | 0.9276288| 5.2753476|  5.6869165| 36.53198|
|Collinsella_aerofaciens      |0      |48     | 34| 34|        98| 0.010100|150_H | 5.5434129| 9.8485812|  1.7766277| 41.72475|
|Collinsella_stercoris        |0      |48     | 34| 34|        52| 0.000214|150_H | 0.0260288| 0.0619150|  2.3787091|  0.20568|
|Collinsella_massiliensis     |0      |48     | 34| 34|        97| 0.016300|150_H | 0.0236750| 0.0083482|  0.3526182|  0.06159|
|Streptococcus_mitis          |0      |48     | 34| 34|       160| 0.041900|150_H | 0.0242400| 0.0095838|  0.3953723|  0.09767|
|Streptococcus_parasanguinis  |0      |48     | 34| 34|       343| 0.006870|150_H | 0.5144082| 0.2470206|  0.4802034|  2.71727|
|Eubacterium_eligens          |0      |48     | 34| 34|       251| 0.017900|150_H | 0.8818553| 0.2670750|  0.3028558|  1.48089|
|Ruminococcus_torques         |0      |48     | 34| 34|        75| 0.019200|150_H | 1.0180315| 2.1744791|  2.1359645| 17.58464|
|Dorea_longicatena            |0      |48     | 34| 34|        85| 0.001450|150_H | 1.2951391| 2.5274115|  1.9514595|  6.75009|
|Roseburia_sp_CAG_471         |0      |48     | 34| 34|        84| 0.035800|150_H | 0.1343818| 0.3339353|  2.4849748|  3.20162|
|Gemmiger_formicilis          |0      |48     | 34| 34|       103| 0.007970|150_H | 0.3123332| 0.5076238|  1.6252636|  7.81153|
|Catenibacterium_mitsuokai    |0      |48     | 34| 34|        27| 0.002220|150_H | 0.5646171| 2.4453638|  4.3310130| 13.97066|
|Firmicutes_bacterium_CAG_110 |0      |48     | 34| 34|        30| 0.016600|150_H | 0.0169538| 0.1234274|  7.2802075|  1.84291|
|Firmicutes_bacterium_CAG_791 |0      |48     | 34| 34|        12| 0.037600|150_H | 0.1280853| 0.2458971|  1.9197915|  1.68220|
|Mitsuokella_jalaludinii      |0      |48     | 34| 34|         5| 0.005180|150_H | 0.0040776| 0.0547685| 13.4314051|  0.48797|
|Mitsuokella_multacida        |0      |48     | 34| 34|         5| 0.005180|150_H | 0.0017156| 0.0244906| 14.2753300|  0.38657|
|Megasphaera_elsdenii         |0      |48     | 34| 34|        38| 0.040700|150_H | 0.1641094| 0.2850700|  1.7370728|  2.74434|
|Bacteroides_vulgatus         |0      |48     |  8|  8|         0| 0.036000|150_L | 1.4315950| 3.8107937|  2.6619217| 20.30201|
|Dorea_longicatena            |0      |48     |  8|  8|         2| 0.023400|150_L | 1.1686050| 3.0337088|  2.5960087|  8.11395|

$CD8diff_96


Table: DA weeks 0 v 48 by CD8diff_96

|species                      |group1 |group2 | n1| n2| statistic|       p|level |    mean_0|   mean_ep|   mean_fc|      max|
|:----------------------------|:------|:------|--:|--:|---------:|-------:|:-----|---------:|---------:|---------:|--------:|
|Olsenella_scatoligenes       |0      |48     | 25| 25|        29| 0.04650|50_H  | 0.0068336| 0.0137676| 2.0146921|  0.07012|
|Collinsella_stercoris        |0      |48     | 25| 25|        43| 0.00405|50_H  | 0.0278508| 0.0665660| 2.3900929|  0.20568|
|Slackia_isoflavoniconvertens |0      |48     | 25| 25|        58| 0.04760|50_H  | 0.3605812| 0.3799544| 1.0537277|  1.14807|
|Eubacterium_eligens          |0      |48     | 25| 25|       158| 0.01190|50_H  | 0.8451688| 0.3054896| 0.3614539|  1.48089|
|Eubacterium_ramulus          |0      |48     | 25| 25|        42| 0.01960|50_H  | 0.0861140| 0.2460664| 2.8574494|  1.78309|
|Ruminococcus_torques         |0      |48     | 25| 25|        43| 0.03820|50_H  | 1.0042948| 2.4328612| 2.4224572| 17.58464|
|Dorea_longicatena            |0      |48     | 25| 25|        42| 0.00368|50_H  | 1.3583344| 2.5397488| 1.8697522|  6.59686|
|Gemmiger_formicilis          |0      |48     | 25| 25|        31| 0.00120|50_H  | 0.2526200| 0.5856572| 2.3183327|  7.81153|
|Catenibacterium_mitsuokai    |0      |48     | 25| 25|        31| 0.01870|50_H  | 0.9876744| 2.2187936| 2.2464828|  9.65802|
|Firmicutes_bacterium_CAG_110 |0      |48     | 25| 25|        22| 0.01860|50_H  | 0.0222084| 0.1810340| 8.1516003|  1.84291|
|Firmicutes_bacterium_CAG_791 |0      |48     | 25| 25|        13| 0.04550|50_H  | 0.1405380| 0.2560748| 1.8221036|  1.50557|
|Mitsuokella_jalaludinii      |0      |48     | 25| 25|        10| 0.02540|50_H  | 0.0136516| 0.0579664| 4.2461250|  0.38756|
|Mitsuokella_multacida        |0      |48     | 25| 25|         9| 0.02070|50_H  | 0.0044052| 0.0295900| 6.7170617|  0.38657|
|Megasphaera_elsdenii         |0      |48     | 25| 25|        20| 0.02490|50_H  | 0.0819924| 0.2935976| 3.5807904|  2.74434|
|Escherichia_coli             |0      |48     | 25| 25|       181| 0.02390|50_H  | 3.8744224| 0.3724676| 0.0961350|  6.29753|
|Actinomyces_oris             |0      |48     | 17| 17|        44| 0.01290|50_L  | 0.0493429| 0.0126971| 0.2573227|  0.09711|
|Collinsella_massiliensis     |0      |48     | 17| 17|        20| 0.02490|50_L  | 0.0020141| 0.0063312| 3.1433995|  0.04881|
|Bacteroides_vulgatus         |0      |48     | 17| 17|        14| 0.03030|50_L  | 1.5361241| 3.3161224| 2.1587594| 20.30201|
|Prevotella_sp_CAG_279        |0      |48     | 17| 17|         2| 0.03000|50_L  | 0.2289700| 0.8239559| 3.5985320|  6.83686|
|Prevotella_stercorea         |0      |48     | 17| 17|        28| 0.02250|50_L  | 0.8334747| 0.1295447| 0.1554273|  1.15358|
|Alistipes_putredinis         |0      |48     | 17| 17|         6| 0.01080|50_L  | 0.3030506| 1.3343594| 4.4030913|  9.42400|
|Streptococcus_mitis          |0      |48     | 17| 17|        33| 0.04230|50_L  | 0.0395306| 0.0043712| 0.1105771|  0.06448|
|Dorea_longicatena            |0      |48     | 17| 17|        17| 0.00902|50_L  | 1.1426594| 2.7475259| 2.4045012|  8.11395|
|Oscillibacter_sp_CAG_241     |0      |48     | 17| 17|         8| 0.02940|50_L  | 0.3566812| 2.0710688| 5.8064988| 19.36226|
|Clostridium_leptum           |0      |48     | 17| 17|         7| 0.04150|50_L  | 0.0056735| 0.0348618| 6.1446345|  0.24215|
|Catenibacterium_mitsuokai    |0      |48     | 17| 17|         1| 0.03460|50_L  | 0.3831682| 2.4599788| 6.4201011| 13.97066|
|Bilophila_wadsworthia        |0      |48     | 17| 17|        17| 0.01580|50_L  | 0.0371112| 0.1153153| 3.1072929|  0.44136|

$CD4after_96


Table: DA weeks 0 v 48 by CD4after_96

|species                      |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|    mean_ep|    mean_fc|      max|
|:----------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|----------:|----------:|--------:|
|Collinsella_stercoris        |0      |48     |  7|  7|         0| 0.036000|high  | 0.0108171|  0.0634014|  5.8611992|  0.15169|
|Eubacterium_hallii           |0      |48     |  7|  7|         0| 0.036000|high  | 0.1466100|  2.2537629| 15.3725043|  4.52322|
|Anaerostipes_hadrus          |0      |48     |  7|  7|         2| 0.046900|high  | 0.1219071|  3.7887786| 31.0792172| 19.83995|
|Coprococcus_comes            |0      |48     |  7|  7|         0| 0.036000|high  | 0.1688814|  1.3186071|  7.8078872|  2.70080|
|Gemmiger_formicilis          |0      |48     |  7|  7|         0| 0.036000|high  | 0.2649157|  0.4629886|  1.7476826|  1.15365|
|Bacteroides_vulgatus         |0      |48     | 10| 10|         0| 0.022500|low   | 1.1735330|  3.7714760|  3.2137792| 20.30201|
|Dorea_longicatena            |0      |48     | 10| 10|         6| 0.027300|low   | 1.4361450|  3.0434110|  2.1191530|  8.11395|
|Actinomyces_oris             |0      |48     | 25| 25|       143| 0.013100|mid   | 0.0391052|  0.0081200|  0.2076450|  0.07622|
|Bifidobacterium_adolescentis |0      |48     | 25| 25|        35| 0.016600|mid   | 1.1980912|  5.7378100|  4.7891262| 36.53198|
|Collinsella_aerofaciens      |0      |48     | 25| 25|        56| 0.023100|mid   | 6.5594988| 11.2790892|  1.7195047| 41.72475|
|Collinsella_stercoris        |0      |48     | 25| 25|        39| 0.002740|mid   | 0.0313244|  0.0655016|  2.0910728|  0.20568|
|Collinsella_massiliensis     |0      |48     | 25| 25|        46| 0.009400|mid   | 0.0226936|  0.0095116|  0.4191314|  0.04881|
|Streptococcus_parasanguinis  |0      |48     | 25| 25|       212| 0.000848|mid   | 0.5825516|  0.1209072|  0.2075476|  0.92384|
|Streptococcus_salivarius     |0      |48     | 25| 25|       236| 0.014600|mid   | 1.4804216|  0.3457528|  0.2335502|  2.20528|
|Eubacterium_eligens          |0      |48     | 25| 25|       151| 0.025500|mid   | 1.0504052|  0.3056616|  0.2909940|  1.48089|
|Dorea_longicatena            |0      |48     | 25| 25|        47| 0.003410|mid   | 1.3821704|  2.7884456|  2.0174398|  6.75009|
|Roseburia_sp_CAG_471         |0      |48     | 25| 25|        31| 0.018700|mid   | 0.1314276|  0.3322524|  2.5280261|  3.20162|
|Oscillibacter_sp_57_20       |0      |48     | 25| 25|        42| 0.034600|mid   | 0.2262412|  0.4976764|  2.1997603|  1.76419|
|Catenibacterium_mitsuokai    |0      |48     | 25| 25|         9| 0.002490|mid   | 0.6191700|  2.6016456|  4.2018276| 11.94837|
|Firmicutes_bacterium_CAG_791 |0      |48     | 25| 25|         2| 0.017800|mid   | 0.1330860|  0.3000936|  2.2548848|  1.68220|
|Mitsuokella_jalaludinii      |0      |48     | 25| 25|         6| 0.032300|mid   | 0.0146316|  0.0598164|  4.0881653|  0.38756|
|Mitsuokella_multacida        |0      |48     | 25| 25|         2| 0.010800|mid   | 0.0018408|  0.0303348| 16.4791395|  0.38657|
|Megasphaera_elsdenii         |0      |48     | 25| 25|        20| 0.024900|mid   | 0.1549544|  0.3418592|  2.2061923|  2.74434|
|Desulfovibrio_piger          |0      |48     | 25| 25|         3| 0.014400|mid   | 0.1219708|  0.2058580|  1.6877646|  0.93650|

$CD8after_96


Table: DA weeks 0 v 48 by CD8after_96

|species                      |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|   mean_ep|    mean_fc|      max|
|:----------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|---------:|----------:|--------:|
|Actinomyces_oris             |0      |48     | 34| 34|       247| 0.005830|high  | 0.0616188| 0.0084503|  0.1371382|  0.07622|
|Bifidobacterium_adolescentis |0      |48     | 34| 34|        32| 0.006800|high  | 0.8474618| 5.0595353|  5.9702225| 36.53198|
|Collinsella_stercoris        |0      |48     | 34| 34|        79| 0.001650|high  | 0.0263956| 0.0583912|  2.2121567|  0.20568|
|Streptococcus_mitis          |0      |48     | 34| 34|       135| 0.032800|high  | 0.0266126| 0.0091476|  0.3437331|  0.09767|
|Eubacterium_eligens          |0      |48     | 34| 34|       261| 0.008370|high  | 1.0241000| 0.2682194|  0.2619074|  1.48089|
|Blautia_wexlerae             |0      |48     | 34| 34|       389| 0.019900|high  | 1.6550547| 0.6081000|  0.3674199|  4.37388|
|Ruminococcus_torques         |0      |48     | 34| 34|        81| 0.029300|high  | 0.9329188| 2.1815241|  2.3383858| 17.58464|
|Dorea_longicatena            |0      |48     | 34| 34|        78| 0.000895|high  | 1.1151559| 2.3476121|  2.1051874|  6.75009|
|Roseburia_hominis            |0      |48     | 34| 34|       277| 0.035500|high  | 0.3473674| 0.0949776|  0.2734213|  0.60125|
|Gemmiger_formicilis          |0      |48     | 34| 34|       109| 0.019500|high  | 0.3030229| 0.4837197|  1.5963138|  7.81153|
|Ruminococcus_bicirculans     |0      |48     | 34| 34|        33| 0.023500|high  | 0.1590579| 0.5360550|  3.3701870|  5.81899|
|Catenibacterium_mitsuokai    |0      |48     | 34| 34|        24| 0.002650|high  | 0.6300947| 2.4595512|  3.9034627| 13.97066|
|Mitsuokella_jalaludinii      |0      |48     | 34| 34|        11| 0.010100|high  | 0.0105144| 0.0598806|  5.6950964|  0.48797|
|Mitsuokella_multacida        |0      |48     | 34| 34|        11| 0.010100|high  | 0.0038115| 0.0258444|  6.7806930|  0.38657|
|Bacteroides_uniformis        |0      |48     |  8|  8|         3| 0.039100|mid   | 0.6747713| 2.1012850|  3.1140701|  8.05205|
|Alistipes_putredinis         |0      |48     |  8|  8|         0| 0.022500|mid   | 0.3726437| 1.8513562|  4.9681666|  9.42400|
|Anaerostipes_hadrus          |0      |48     |  8|  8|         1| 0.015600|mid   | 0.3935163| 1.3825550|  3.5133365|  3.83627|
|Oscillibacter_sp_CAG_241     |0      |48     |  8|  8|         0| 0.007810|mid   | 0.3352750| 3.7774238| 11.2666431| 19.36226|</code></pre>
<pre class="r"><code> &quot;group&quot; %&gt;%
  set_names() %&gt;%
  map( ~ {
    stat_list_long %&gt;% 
      pluck(.x) %&gt;% 
      dplyr::filter(group2 == 96) %&gt;%
      dplyr::arrange(level) %&gt;%
      kableExtra::kable(format = &quot;markdown&quot;, caption = paste0(&quot;DA weeks 0 v 96 by delta CD4 at &quot;, .x))
    
  })</code></pre>
<pre><code>$group


Table: DA weeks 0 v 96 by delta CD4 at group

|species                        |group1 |group2 | n1| n2| statistic|        p|level |    mean_0|   mean_ep|   mean_fc|      max|
|:------------------------------|:------|:------|--:|--:|---------:|--------:|:-----|---------:|---------:|---------:|--------:|
|Actinomyces_odontolyticus      |0      |96     | 27| 27|       143| 0.013100|DRV/r | 0.0506296| 0.0058900| 0.1163350|  0.04284|
|Actinomyces_oris               |0      |96     | 27| 27|       146| 0.008970|DRV/r | 0.0441715| 0.0070885| 0.1604773|  0.06163|
|Bacteroides_caccae             |0      |96     | 27| 27|        16| 0.013500|DRV/r | 0.1481763| 1.3140519| 8.8681650| 21.74016|
|Alistipes_finegoldii           |0      |96     | 27| 27|        23| 0.021400|DRV/r | 0.0536481| 0.1707048| 3.1819330|  1.30839|
|Streptococcus_parasanguinis    |0      |96     | 27| 27|       182| 0.021800|DRV/r | 0.4049241| 0.2572248| 0.6352421|  4.82574|
|Eubacterium_eligens            |0      |96     | 27| 27|       151| 0.025500|DRV/r | 0.9559100| 0.3814815| 0.3990768|  2.10420|
|Eubacterium_ramulus            |0      |96     | 27| 27|        41| 0.017800|DRV/r | 0.1167400| 0.2591670| 2.2200363|  1.48146|
|Dorea_longicatena              |0      |96     | 27| 27|        81| 0.029300|DRV/r | 2.1041381| 3.4391130| 1.6344521| 11.00520|
|Clostridium_bolteae            |0      |96     | 27| 27|        88| 0.028000|DRV/r | 0.3247070| 0.0684578| 0.2108294|  0.33631|
|Catenibacterium_mitsuokai      |0      |96     | 27| 27|        26| 0.031900|DRV/r | 0.7841678| 2.5614896| 3.2665071| 12.37330|
|Actinomyces_oris               |0      |96     | 31| 31|       201| 0.000363|DTG   | 0.0539329| 0.0060735| 0.1126130|  0.05885|
|Bifidobacterium_adolescentis   |0      |96     | 31| 31|        98| 0.029700|DTG   | 2.0197058| 4.8260681| 2.3894906| 33.82719|
|Olsenella_scatoligenes         |0      |96     | 31| 31|        56| 0.023100|DTG   | 0.0070642| 0.0194526| 2.7536874|  0.09340|
|Collinsella_massiliensis       |0      |96     | 31| 31|        92| 0.006870|DTG   | 0.0192313| 0.0064742| 0.3366489|  0.02494|
|Eggerthella_lenta              |0      |96     | 31| 31|       167| 0.004010|DTG   | 0.3163145| 0.0704648| 0.2227683|  0.91083|
|Bacteroides_ovatus             |0      |96     | 31| 31|        30| 0.009440|DTG   | 0.0524877| 0.1412745| 2.6915716|  1.07031|
|Odoribacter_splanchnicus       |0      |96     | 31| 31|        85| 0.038300|DTG   | 0.1186216| 0.2659723| 2.2421905|  1.42066|
|Prevotella_sp_CAG_279          |0      |96     | 31| 31|        36| 0.010500|DTG   | 0.2522426| 0.9848410| 3.9043407|  7.81139|
|Alistipes_putredinis           |0      |96     | 31| 31|        33| 0.013300|DTG   | 0.1904435| 0.4062832| 2.1333525|  6.08368|
|Parabacteroides_merdae         |0      |96     | 31| 31|        58| 0.047600|DTG   | 0.2944135| 0.6378426| 2.1664852|  3.66971|
|Streptococcus_parasanguinis    |0      |96     | 31| 31|       348| 0.000140|DTG   | 0.4403284| 0.0541581| 0.1229947|  0.53352|
|Streptococcus_salivarius       |0      |96     | 31| 31|       300| 0.028000|DTG   | 1.1754465| 0.1726203| 0.1468551|  1.25691|
|Blautia_wexlerae               |0      |96     | 31| 31|       377| 0.010400|DTG   | 1.9982074| 0.4799548| 0.2401927|  3.11529|
|Ruminococcus_gnavus            |0      |96     | 31| 31|        90| 0.020200|DTG   | 0.9487039| 0.0326713| 0.0344378|  0.43938|
|Coprococcus_catus              |0      |96     | 31| 31|        95| 0.008340|DTG   | 0.2409219| 0.4502735| 1.8689604|  1.48461|
|Dorea_longicatena              |0      |96     | 31| 31|       134| 0.043800|DTG   | 1.5306684| 2.5346994| 1.6559428|  5.95676|
|Clostridium_bolteae            |0      |96     | 31| 31|        99| 0.028800|DTG   | 0.5283210| 0.0071223| 0.0134809|  0.05162|
|Clostridium_citroniae          |0      |96     | 31| 31|        66| 0.003860|DTG   | 0.0186823| 0.0029452| 0.1576448|  0.03015|
|Lachnospira_pectinoschiza      |0      |96     | 31| 31|        70| 0.004410|DTG   | 0.2956155| 0.8556181| 2.8943615|  5.52976|
|Roseburia_sp_CAG_471           |0      |96     | 31| 31|        94| 0.023200|DTG   | 0.1377200| 0.4315071| 3.1332203|  5.16193|
|Oscillibacter_sp_57_20         |0      |96     | 31| 31|        73| 0.009580|DTG   | 0.2627984| 0.5732429| 2.1813030|  3.57907|
|Flavonifractor_plautii         |0      |96     | 31| 31|       314| 0.011900|DTG   | 0.6224474| 0.1420926| 0.2282805|  0.67236|
|Eubacterium_siraeum            |0      |96     | 31| 31|        63| 0.040800|DTG   | 0.3730577| 1.0144165| 2.7191942| 14.11633|
|Ruminococcus_bicirculans       |0      |96     | 31| 31|        53| 0.031200|DTG   | 0.2112955| 0.5240797| 2.4803165|  2.55314|
|Clostridium_innocuum           |0      |96     | 31| 31|       140| 0.018700|DTG   | 1.4354503| 0.0136055| 0.0094782|  0.11360|
|Erysipelatoclostridium_ramosum |0      |96     | 31| 31|       135| 0.006040|DTG   | 0.1747377| 0.0025755| 0.0147391|  0.01993|
|Firmicutes_bacterium_CAG_110   |0      |96     | 31| 31|        35| 0.029400|DTG   | 0.0316145| 0.2697748| 8.5332585|  3.96070|
|Mitsuokella_jalaludinii        |0      |96     | 31| 31|        33| 0.007600|DTG   | 0.0333926| 0.1823055| 5.4594608|  2.81767|
|Megasphaera_elsdenii           |0      |96     | 31| 31|        51| 0.045800|DTG   | 0.1834529| 0.3959106| 2.1581051|  5.28667|</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
</div>
<div id="constant-da-species-venn-diagrams" class="section level1">
<h1>Constant DA species: Venn diagrams</h1>
<p>As we can see, under exploratory conditions there are many DA species across all groups. Plotting them all would be quite computationally expensive and plotting so many longitudinal plots may be even confusing. We’ll do additional filtering. Instead of a more astringent p-value, we’ll try to select DA species more functionally. First will be Venn diagramns for DA species for each group level and time point, trying to pinpoint those who consistently changed across the study:</p>
<pre class="r"><code>### MNJ: Provide a table of species theat appear in the Venn&#39;s diagram.

comps_group &lt;- list(&quot;DTG&quot;=c(&quot;DTG_48&quot;,&quot;DTG_96&quot;),
              &quot;DRV/r&quot; = c(&quot;DRV/r_48&quot;, &quot;DRV/r_96&quot;))  # Those are the group intersections I considered most interesting

Venn_Group &lt;- stat_list_long$group %&gt;% # Construct list for Venn diagram
      dplyr::select(species, group2, level) %&gt;%
      dplyr::mutate(row=row_number()) %&gt;% 
      tidyr::pivot_wider(., values_from = &quot;species&quot;, names_from = c(&quot;level&quot;, &quot;group2&quot;)) %&gt;% 
      dplyr::select(-row) %&gt;%
      map(~{
        .x[!is.na(.x)]
      }) 

group_diff &lt;- names(comps_group) %&gt;% # DA species common between 48 and 96 for each group
  set_names() %&gt;% 
  map(~ {
    Venn_Group %&gt;% 
      ggVennDiagram::Venn(.) %&gt;% 
      ggVennDiagram::discern_overlap(., slice = comps_group[[.x]])
  }) 



Venn_Group %&gt;% # Plot Venn
  ggVennDiagram::ggVennDiagram(., show_intersect = T)</code></pre>
<div id="htmlwidget-2aecd4da706770a6acca" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-2aecd4da706770a6acca">{"x":{"data":[{"x":[0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,null,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,null,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,null,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512,null,0.0214943938344883,0.0303030303030303,0.0404040404040404,0.0505050505050505,0.0606060606060606,0.0707070707070707,0.0808080808080808,0.0909090909090909,0.101010101010101,0.111111111111111,0.121212121212121,0.131313131313131,0.141414141414141,0.151515151515151,0.161616161616162,0.171717171717172,0.181818181818182,0.191919191919192,0.202020202020202,0.212121212121212,0.222222222222222,0.232323232323232,0.242424242424242,0.252525252525253,0.262626262626263,0.272727272727273,0.282828282828283,0.292929292929293,0.303030303030303,0.313131313131313,0.323232323232323,0.333333333333333,0.343434343434343,0.353535353535354,0.363636363636364,0.373737373737374,0.383838383838384,0.393939393939394,0.404040404040404,0.414141414141414,0.424242424242424,0.434343434343434,0.444444444444444,0.454545454545455,0.464646464646465,0.474747474747475,0.484848484848485,0.494949494949495,0.505050505050505,0.515151515151515,0.525252525252525,0.535353535353535,0.545454545454546,0.555555555555556,0.565656565656566,0.575757575757576,0.585858585858586,0.595959595959596,0.606060606060606,0.616161616161616,0.626262626262626,0.636363636363636,0.646464646464647,0.656565656565656,0.666666666666667,0.676767676767677,0.686868686868687,0.696969696969697,0.707070707070707,0.717171717171717,0.727272727272727,0.737373737373737,0.747474747474748,0.757575757575758,0.767676767676768,0.777777777777778,0.787878787878788,0.797979797979798,0.808080808080808,0.818181818181818,0.828282828282828,0.838383838383838,0.848484848484848,0.858585858585859,0.868686868686869,0.878787878787879,0.888888888888889,0.898989898989899,0.909090909090909,0.919191919191919,0.929292929292929,0.939393939393939,0.94949494949495,0.95959595959596,0.96969696969697,0.978505606165512],"y":[0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.151244648660193,0.156565656565657,0.164646464646465,0.172727272727273,0.180808080808081,0.188888888888889,0.196969696969697,0.205050505050505,0.213131313131313,0.221212121212121,0.229292929292929,0.237373737373737,0.245454545454545,0.253535353535354,0.261616161616162,0.26969696969697,0.277777777777778,0.285858585858586,0.293939393939394,0.302020202020202,0.31010101010101,0.318181818181818,0.326262626262626,0.334343434343434,0.342424242424242,0.350505050505051,0.358585858585859,0.366666666666667,0.374747474747475,0.382828282828283,0.390909090909091,0.398989898989899,0.407070707070707,0.415151515151515,0.423232323232323,0.431313131313131,0.439393939393939,0.447474747474747,0.455555555555556,0.463636363636364,0.471717171717172,0.47979797979798,0.487878787878788,0.495959595959596,0.504040404040404,0.512121212121212,0.52020202020202,0.528282828282828,0.536363636363636,0.544444444444444,0.552525252525253,0.560606060606061,0.568686868686869,0.576767676767677,0.584848484848485,0.592929292929293,0.601010101010101,0.609090909090909,0.617171717171717,0.625252525252525,0.633333333333333,0.641414141414141,0.649494949494949,0.657575757575757,0.665656565656566,0.673737373737374,0.681818181818182,0.68989898989899,0.697979797979798,0.706060606060606,0.714141414141414,0.722222222222222,0.73030303030303,0.738383838383838,0.746464646464646,0.754545454545454,0.762626262626263,0.770707070707071,0.778787878787879,0.786868686868687,0.794949494949495,0.803030303030303,0.811111111111111,0.819191919191919,0.827272727272727,0.835353535353535,0.843434343434343,0.851515151515151,0.859595959595959,0.867676767676768,0.875757575757576,0.883838383838384,0.891919191919192,0.893750254825705,null,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,0.2,null,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,0.3,null,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,0.4,null,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,null,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,null,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,0.7,null,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8,0.8],"type":"scatter","mode":"lines","line":{"width":null,"color":"transparent","dash":null},"hoveron":"points","hoverinfo":"none","showlegend":false,"_isGraticule":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[0.689723867357517,0.677352581073137,0.664281365175949,0.650561805783427,0.636248047728625,0.62139658087525,0.606066017177982,0.597370908578246,0.609615927001734,0.627413129604112,0.645299472264983,0.663204365761357,0.681057147658572,0.69878736118264,0.716325033280945,0.733600950773915,0.750546933507822,0.755576593155291,0.756004420321286,0.75629779737109,0.755579684105997,0.753852914590988,0.751124303596454,0.747404619703406,0.742708542804807,0.737054606170744,0.730465123306073,0.722966099889212,0.714587131139611,0.705361285018936,0.695324971726938,0.69395919532696,0.689723867357517,0.689723867357517,null,0.602016366428415,0.609881708785646,0.616721416766958,0.622508497169717,0.627220111031436,0.630837663764735,0.63334687854168,0.634737852637896,0.635005096514101,0.634273105328381,0.623083274789937,0.607803345806105,0.592097966267041,0.576029118130679,0.559660217799456,0.543055865844739,0.526281592057887,0.5094035968341,0.498652466931534,0.512717577440789,0.528767132760538,0.544703157349734,0.560462758999801,0.575983741762902,0.591204851410935,0.600946179932138,0.602016366428415,0.602016366428415,null,0.593156430608217,0.583336867444015,0.572596430265486,0.560977506671335,0.548525951244807,0.535290904586595,0.521324599379342,0.506682154249102,0.5,0.50857864376269,0.52439756725828,0.540712186173608,0.557458114161256,0.574569262693944,0.591978101885752,0.597370908578246,0.606066017177982,0.62139658087525,0.636248047728625,0.650561805783427,0.664281365175949,0.677352581073137,0.689723867357517,0.69395919532696,0.684517800007025,0.672982420828649,0.660764359063413,0.647911833819203,0.634475568141398,0.620508588832182,0.606066017177982,0.600946179932138,0.593156430608217,0.593156430608217],"y":[0.577511510090297,0.59439697251605,0.611186151237198,0.627812787035576,0.644211262175675,0.660316859367867,0.676066017177982,0.684530020005949,0.692596430265486,0.703336867444015,0.713156430608217,0.722016366428415,0.729881708785646,0.736721416766958,0.742508497169717,0.747220111031436,0.750837663764735,0.751600269766683,0.747913410063906,0.73597637006475,0.72338429716034,0.710186886509658,0.696436222263218,0.682186572010721,0.667494172611667,0.652417008254165,0.637014581617834,0.621347679043907,0.605478130639317,0.58946856626151,0.57338216834701,0.571347533068466,0.577511510090297,0.577511510090297,null,0.456795634238643,0.438942852341428,0.42121263881736,0.403674966719055,0.386399049226085,0.369453066492178,0.352903896569301,0.336816851470545,0.321255419413441,0.308473369158742,0.311698677517379,0.316982284000754,0.32326443581638,0.330520340181081,0.338721361355239,0.347835133654967,0.357825689184687,0.368653599785973,0.37604080467304,0.385482199992975,0.397017579171351,0.409235640936587,0.422088166180797,0.435524431858602,0.449491411167818,0.458958355913603,0.456795634238643,0.456795634238643,null,0.474700527735017,0.492586870395888,0.510384072998266,0.528021898114248,0.545430737306056,0.562541885838744,0.579287813826392,0.59560243274172,0.602528970054552,0.611421356237309,0.626682154249102,0.641324599379342,0.655290904586595,0.668525951244807,0.680977506671335,0.684530020005949,0.676066017177982,0.660316859367867,0.644211262175675,0.627812787035576,0.611186151237198,0.59439697251605,0.577511510090297,0.571347533068466,0.557282422559211,0.541232867239462,0.525296842650266,0.509537241000199,0.494016258237098,0.478795148589065,0.463933982822018,0.458958355913603,0.474700527735017,0.474700527735017],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(19,43,67,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.393933982822018,0.37860341912475,0.363751952271375,0.349438194216573,0.335718634824051,0.322647418926863,0.310276132642483,0.30604080467304,0.315482199992975,0.327017579171351,0.339235640936587,0.352088166180797,0.365524431858602,0.379491411167818,0.393933982822018,0.402637560262697,0.408021898114248,0.425430737306056,0.442541885838744,0.459287813826392,0.47560243274172,0.49142135623731,0.5,0.493317845750898,0.478675400620658,0.464709095413405,0.451474048755193,0.439022493328665,0.427403569734514,0.416663132555985,0.406843569391783,0.399051104552776,0.393933982822018,0.393933982822018,null,0.49661783165299,0.48053143373849,0.464521869360683,0.448652320956093,0.432985418382166,0.417582991745835,0.402505827388333,0.387813427989279,0.373563777736782,0.365225932900163,0.364994903485898,0.365262147362104,0.36665312145832,0.369162336235264,0.372779888968564,0.377491502830283,0.383278583233042,0.390118291214355,0.397983633571585,0.399051104552776,0.409683140632133,0.425788737824325,0.442187212964424,0.458813848762802,0.47560302748395,0.492488489909703,0.498652466931534,0.49661783165299,0.49661783165299,null,0.406843569391783,0.416663132555985,0.427403569734514,0.439022493328665,0.451474048755193,0.464709095413405,0.478675400620658,0.493317845750898,0.5,0.506682154249102,0.521324599379342,0.535290904586595,0.548525951244807,0.560977506671335,0.572596430265486,0.583336867444015,0.593156430608217,0.600946179932138,0.591204851410935,0.575983741762902,0.560462758999801,0.544703157349734,0.528767132760538,0.512717577440789,0.498652466931534,0.492488489909703,0.47560302748395,0.458813848762802,0.442187212964424,0.425788737824325,0.409683140632133,0.399051104552776,0.406843569391783,0.406843569391783,null,0.298653599785973,0.287825689184687,0.277835133654967,0.268721361355239,0.260520340181081,0.25326443581638,0.246982284000754,0.241698677517379,0.237434468347193,0.234206485375659,0.232027467976841,0.230906015736915,0.230883038725603,0.232903896569301,0.249453066492178,0.266399049226085,0.283674966719055,0.30121263881736,0.318942852341428,0.336795634238643,0.354700527735017,0.372586870395888,0.390384072998266,0.402637560262697,0.393933982822018,0.379491411167818,0.365524431858602,0.352088166180797,0.339235640936587,0.327017579171351,0.315482199992975,0.30604080467304,0.298653599785973,0.298653599785973,null,0.359813113490343,0.34661570283966,0.33402362993525,0.322086589936094,0.310851692885556,0.300363277789556,0.290662737630778,0.281788356009473,0.273775156055582,0.266654762208439,0.260455275409561,0.255201162201066,0.250913158167413,0.24760818610152,0.245299289218232,0.243995579678714,0.24370220262891,0.244420315894003,0.246147085409012,0.248875696403546,0.252595380296594,0.257291457195193,0.262945393829256,0.269534876693927,0.277033900110788,0.285412868860389,0.294638714981064,0.304675028273062,0.30604080467304,0.310276132642483,0.322647418926863,0.335718634824051,0.349438194216573,0.363751952271375,0.37860341912475,0.393933982822018,0.399051104552776,0.397983633571585,0.390118291214355,0.383278583233042,0.377491502830283,0.372779888968564,0.369162336235264,0.36665312145832,0.365262147362104,0.364994903485898,0.365225932900163,0.359813113490343,0.359813113490343,null,0.635005096514101,0.634737852637896,0.63334687854168,0.630837663764735,0.627220111031436,0.622508497169717,0.616721416766958,0.609881708785646,0.602016366428415,0.600946179932138,0.606066017177982,0.620508588832182,0.634475568141398,0.647911833819203,0.660764359063413,0.672982420828649,0.684517800007025,0.69395919532696,0.701346400214027,0.712174310815313,0.722164866345033,0.731278638644761,0.739479659818919,0.74673556418362,0.753017715999246,0.758301322482621,0.762565531652807,0.765793514624341,0.767972532023159,0.769093984263086,0.769153445484473,0.768150681021047,0.766089648326023,0.762978481353853,0.75882945845922,0.75365895393998,0.747487373415292,0.740339073293959,0.732242264650813,0.723228901890489,0.713334556637987,0.702598277353723,0.691062435227093,0.678772556956745,0.665777145077502,0.65212748654301,0.637877450319571,0.634273105328381,0.635005096514101,0.635005096514101],"y":[0.463933982822018,0.479683140632133,0.495788737824325,0.512187212964423,0.528813848762801,0.54560302748395,0.562488489909703,0.568652466931534,0.582717577440789,0.598767132760538,0.614703157349734,0.630462758999801,0.645983741762902,0.661204851410935,0.676066017177982,0.684524441153954,0.680977506671335,0.668525951244807,0.655290904586595,0.641324599379341,0.626682154249103,0.611421356237309,0.602528970054552,0.59560243274172,0.579287813826392,0.562541885838744,0.545430737306056,0.528021898114248,0.510384072998266,0.492586870395888,0.474700527735017,0.458952868175387,0.463933982822018,0.463933982822018,null,0.374675028273062,0.364638714981064,0.355412868860389,0.347033900110788,0.339534876693927,0.332945393829256,0.327291457195193,0.322595380296594,0.318875696403546,0.31722117741565,0.321255419413441,0.336816851470545,0.352903896569301,0.369453066492178,0.386399049226085,0.403674966719055,0.42121263881736,0.438942852341428,0.456795634238643,0.458952868175387,0.44860341912475,0.433751952271375,0.419438194216573,0.405718634824051,0.392647418926863,0.380276132642483,0.37604080467304,0.374675028273062,0.374675028273062,null,0.474700527735017,0.492586870395888,0.510384072998266,0.528021898114248,0.545430737306056,0.562541885838744,0.579287813826392,0.59560243274172,0.602528970054552,0.59560243274172,0.579287813826392,0.562541885838744,0.545430737306056,0.528021898114248,0.510384072998266,0.492586870395888,0.474700527735017,0.458958355913603,0.449491411167818,0.435524431858602,0.422088166180797,0.409235640936587,0.397017579171351,0.385482199992975,0.37604080467304,0.380276132642483,0.392647418926863,0.405718634824051,0.419438194216573,0.433751952271375,0.44860341912475,0.458952868175387,0.474700527735017,0.474700527735017,null,0.5794035968341,0.596281592057887,0.613055865844739,0.629660217799456,0.646029118130679,0.662097966267041,0.677803345806105,0.693083274789937,0.707877450319571,0.72212748654301,0.735777145077502,0.748772556956745,0.753521612987656,0.75334687854168,0.750837663764736,0.747220111031436,0.742508497169717,0.736721416766958,0.729881708785646,0.722016366428415,0.713156430608217,0.703336867444015,0.692596430265486,0.684524441153954,0.676066017177982,0.661204851410935,0.645983741762902,0.630462758999801,0.614703157349734,0.598767132760538,0.582717577440789,0.568652466931534,0.5794035968341,0.5794035968341,null,0.316147085409012,0.314420315894003,0.31370220262891,0.313995579678714,0.315299289218232,0.31760818610152,0.320913158167413,0.325201162201066,0.330455275409561,0.336654762208439,0.343775156055582,0.351788356009473,0.360662737630778,0.370363277789556,0.380851692885556,0.392086589936094,0.40402362993525,0.41661570283966,0.429813113490343,0.443563777736782,0.457813427989279,0.472505827388333,0.487582991745835,0.502985418382166,0.518652320956093,0.534521869360683,0.55053143373849,0.56661783165299,0.568652466931534,0.562488489909703,0.54560302748395,0.528813848762801,0.512187212964423,0.495788737824325,0.479683140632133,0.463933982822018,0.458952868175387,0.456795634238643,0.438942852341428,0.42121263881736,0.403674966719055,0.386399049226085,0.369453066492178,0.352903896569301,0.336816851470545,0.321255419413441,0.31722117741565,0.316147085409012,0.316147085409012,null,0.321255419413441,0.336816851470545,0.352903896569301,0.369453066492178,0.386399049226085,0.403674966719055,0.42121263881736,0.438942852341428,0.456795634238643,0.458958355913603,0.463933982822018,0.478795148589065,0.494016258237098,0.509537241000199,0.525296842650266,0.541232867239462,0.557282422559211,0.571347533068466,0.5605964031659,0.543718407942113,0.526944134155261,0.510339782200544,0.493970881869321,0.477902033732959,0.462196654193895,0.446916725210063,0.432122549680429,0.41787251345699,0.404222854922498,0.391227443043255,0.378937564772907,0.367401722646276,0.356665443362013,0.346771098109511,0.337757735349187,0.329660926706041,0.322512626584708,0.31634104606002,0.31117054154078,0.307021518646146,0.303910351673977,0.301849318978953,0.300846554515527,0.300906015736914,0.302027467976841,0.304206485375659,0.307434468347192,0.308473369158742,0.321255419413441,0.321255419413441],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(23,50,77,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.471952733149137,0.458327123257497,0.445457958643352,0.433396028020072,0.422188934317255,0.411880906813711,0.402512626584708,0.39412106595235,0.386739342572712,0.380396588735586,0.375117836392638,0.370923918367733,0.367831386139297,0.365852444519191,0.365225932900163,0.373563777736782,0.387813427989279,0.402505827388333,0.417582991745835,0.432985418382166,0.448652320956093,0.464521869360683,0.48053143373849,0.49661783165299,0.498652466931534,0.5094035968341,0.526281592057887,0.543055865844739,0.559660217799456,0.576029118130679,0.592097966267041,0.607803345806105,0.623083274789937,0.634273105328381,0.634147555480809,0.632168613860703,0.629076081632267,0.624882163607362,0.619603411264414,0.613260657427288,0.60587893404765,0.597487373415292,0.588119093186289,0.577811065682745,0.566603971979928,0.554542041356648,0.541672876742503,0.528047266850863,0.513718985739022,0.5,0.486281014260978,0.471952733149137,0.471952733149137],"y":[0.187831386139297,0.190923918367733,0.195117836392638,0.200396588735586,0.206739342572712,0.21412106595235,0.222512626584708,0.231880906813711,0.242188934317255,0.253396028020072,0.265457958643352,0.278327123257497,0.291952733149137,0.306281014260978,0.31722117741565,0.318875696403546,0.322595380296594,0.327291457195193,0.332945393829256,0.339534876693927,0.347033900110788,0.355412868860389,0.364638714981064,0.374675028273062,0.37604080467304,0.368653599785973,0.357825689184687,0.347835133654967,0.338721361355239,0.330520340181081,0.32326443581638,0.316982284000754,0.311698677517379,0.308473369158742,0.306281014260978,0.291952733149137,0.278327123257497,0.265457958643352,0.253396028020072,0.242188934317255,0.231880906813711,0.222512626584708,0.21412106595235,0.206739342572712,0.200396588735586,0.195117836392638,0.190923918367733,0.187831386139297,0.185852444519191,0.185066797738128,0.185852444519191,0.187831386139297,0.187831386139297],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(27,58,87,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.0649949034858985,0.0658524445191911,0.0678313861392967,0.0709239183677329,0.0751178363926377,0.0803965887355857,0.0867393425727119,0.0941210659523497,0.102512626584708,0.111880906813711,0.122188934317255,0.133396028020072,0.145457958643352,0.158327123257497,0.171952733149137,0.186281014260978,0.201255419413441,0.216816851470545,0.230883038725603,0.230906015736915,0.232027467976841,0.234206485375659,0.237434468347193,0.241698677517379,0.246982284000754,0.25326443581638,0.260520340181081,0.268721361355239,0.277835133654967,0.287825689184687,0.298653599785973,0.30604080467304,0.304675028273062,0.294638714981064,0.285412868860389,0.277033900110788,0.269534876693927,0.262945393829256,0.257291457195193,0.252595380296594,0.248875696403546,0.246147085409012,0.244420315894003,0.24370220262891,0.243995579678714,0.245299289218232,0.24760818610152,0.250913158167413,0.255201162201066,0.260455275409561,0.266654762208439,0.273775156055582,0.281788356009473,0.290662737630778,0.300363277789556,0.310851692885556,0.322086589936094,0.33402362993525,0.34661570283966,0.359813113490343,0.365225932900163,0.365852444519191,0.367831386139297,0.370923918367733,0.375117836392638,0.380396588735586,0.386739342572712,0.39412106595235,0.402512626584708,0.411880906813711,0.422188934317255,0.433396028020072,0.445457958643352,0.458327123257497,0.471952733149137,0.486281014260978,0.5,0.498744580586559,0.483183148529455,0.467096103430699,0.450546933507822,0.433600950773915,0.416325033280945,0.39878736118264,0.381057147658572,0.363204365761357,0.345299472264983,0.327413129604112,0.309615927001734,0.291978101885752,0.274569262693944,0.257458114161256,0.240712186173608,0.22439756725828,0.208578643762691,0.193317845750898,0.178675400620658,0.164709095413405,0.151474048755193,0.139022493328665,0.127403569734514,0.116663132555985,0.106843569391783,0.0979836335715852,0.0901182912143544,0.0832785832330416,0.0774915028302834,0.0727798889685642,0.0691623362352644,0.0666531214583198,0.0652621473621038,0.0649949034858985,0.0649949034858985],"y":[0.618744580586559,0.633718985739022,0.648047266850863,0.661672876742503,0.674542041356648,0.686603971979928,0.697811065682745,0.708119093186289,0.717487373415292,0.72587893404765,0.733260657427288,0.739603411264414,0.744882163607362,0.749076081632267,0.752168613860703,0.754147555480809,0.755005096514102,0.754737852637896,0.753521612987656,0.748772556956745,0.735777145077502,0.72212748654301,0.707877450319571,0.693083274789937,0.677803345806105,0.662097966267041,0.646029118130679,0.629660217799456,0.613055865844739,0.596281592057887,0.5794035968341,0.568652466931534,0.56661783165299,0.55053143373849,0.534521869360683,0.518652320956093,0.502985418382166,0.487582991745835,0.472505827388333,0.457813427989279,0.443563777736782,0.429813113490343,0.41661570283966,0.40402362993525,0.392086589936094,0.380851692885556,0.370363277789556,0.360662737630778,0.351788356009473,0.343775156055582,0.336654762208439,0.330455275409561,0.325201162201066,0.320913158167413,0.31760818610152,0.315299289218232,0.313995579678714,0.31370220262891,0.314420315894003,0.316147085409012,0.31722117741565,0.306281014260978,0.291952733149137,0.278327123257497,0.265457958643352,0.253396028020072,0.242188934317255,0.231880906813711,0.222512626584708,0.21412106595235,0.206739342572712,0.200396588735586,0.195117836392638,0.190923918367733,0.187831386139297,0.185852444519191,0.185066797738128,0.184994903485898,0.185262147362104,0.18665312145832,0.189162336235264,0.192779888968564,0.197491502830283,0.203278583233042,0.210118291214354,0.217983633571585,0.226843569391783,0.236663132555985,0.247403569734514,0.259022493328665,0.271474048755193,0.284709095413405,0.298675400620658,0.313317845750897,0.32857864376269,0.34439756725828,0.360712186173608,0.377458114161256,0.394569262693944,0.411978101885752,0.429615927001734,0.447413129604112,0.465299472264983,0.483204365761357,0.501057147658572,0.518787361182639,0.536325033280944,0.553600950773915,0.570546933507822,0.587096103430699,0.603183148529455,0.618744580586559,0.618744580586559],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(30,66,97,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.500000000000001,0.513718985739022,0.528047266850863,0.541672876742503,0.554542041356648,0.566603971979928,0.577811065682745,0.588119093186289,0.597487373415292,0.60587893404765,0.613260657427288,0.619603411264414,0.624882163607362,0.629076081632267,0.632168613860703,0.634147555480809,0.634273105328381,0.637877450319571,0.65212748654301,0.665777145077502,0.678772556956745,0.691062435227093,0.702598277353723,0.713334556637987,0.723228901890489,0.732242264650813,0.740339073293959,0.747487373415292,0.75365895393998,0.75882945845922,0.762978481353853,0.766089648326023,0.768150681021047,0.769153445484473,0.769093984263086,0.767972532023159,0.765793514624341,0.762565531652807,0.758301322482621,0.753017715999246,0.74673556418362,0.739479659818919,0.731278638644761,0.722164866345033,0.712174310815313,0.701346400214027,0.69395919532696,0.695324971726938,0.705361285018936,0.714587131139611,0.722966099889212,0.730465123306073,0.737054606170744,0.742708542804807,0.747404619703406,0.751124303596454,0.753852914590988,0.755579684105997,0.75629779737109,0.756004420321286,0.755576593155291,0.767096103430699,0.783183148529455,0.798744580586559,0.813718985739022,0.828047266850863,0.841672876742503,0.854542041356648,0.866603971979928,0.877811065682745,0.888119093186289,0.897487373415292,0.90587893404765,0.913260657427288,0.919603411264414,0.924882163607362,0.929076081632267,0.932168613860703,0.934147555480809,0.935005096514102,0.934737852637896,0.93334687854168,0.930837663764736,0.927220111031436,0.922508497169717,0.916721416766959,0.909881708785646,0.902016366428415,0.893156430608217,0.883336867444015,0.872596430265486,0.860977506671335,0.848525951244807,0.835290904586595,0.821324599379342,0.806682154249103,0.79142135623731,0.77560243274172,0.759287813826392,0.742541885838744,0.725430737306056,0.708021898114248,0.690384072998266,0.672586870395888,0.654700527735017,0.636795634238643,0.618942852341428,0.60121263881736,0.583674966719055,0.566399049226085,0.549453066492178,0.532903896569301,0.516816851470545,0.501255419413441,0.500000000000001,0.500000000000001],"y":[0.185066797738128,0.185852444519191,0.187831386139297,0.190923918367733,0.195117836392638,0.200396588735586,0.206739342572712,0.21412106595235,0.222512626584708,0.231880906813711,0.242188934317255,0.253396028020072,0.265457958643352,0.278327123257497,0.291952733149137,0.306281014260978,0.308473369158742,0.307434468347192,0.304206485375659,0.302027467976841,0.300906015736914,0.300846554515527,0.301849318978953,0.303910351673977,0.307021518646146,0.31117054154078,0.31634104606002,0.322512626584708,0.329660926706041,0.337757735349187,0.346771098109511,0.356665443362013,0.367401722646276,0.378937564772907,0.391227443043255,0.404222854922498,0.41787251345699,0.432122549680429,0.446916725210063,0.462196654193895,0.477902033732959,0.493970881869321,0.510339782200544,0.526944134155261,0.543718407942113,0.5605964031659,0.571347533068466,0.57338216834701,0.58946856626151,0.605478130639317,0.621347679043907,0.637014581617834,0.652417008254165,0.667494172611667,0.682186572010721,0.696436222263218,0.710186886509658,0.72338429716034,0.73597637006475,0.747913410063906,0.751600269766683,0.75334687854168,0.754737852637896,0.755005096514101,0.754147555480809,0.752168613860703,0.749076081632267,0.744882163607362,0.739603411264414,0.733260657427288,0.72587893404765,0.717487373415292,0.708119093186289,0.697811065682745,0.686603971979928,0.674542041356648,0.661672876742503,0.648047266850863,0.633718985739022,0.618744580586559,0.603183148529455,0.587096103430699,0.570546933507822,0.553600950773915,0.536325033280944,0.51878736118264,0.501057147658572,0.483204365761357,0.465299472264983,0.447413129604112,0.429615927001734,0.411978101885752,0.394569262693944,0.377458114161256,0.360712186173608,0.34439756725828,0.32857864376269,0.313317845750897,0.298675400620658,0.284709095413405,0.271474048755193,0.259022493328665,0.247403569734514,0.236663132555985,0.226843569391783,0.217983633571585,0.210118291214354,0.203278583233042,0.197491502830283,0.192779888968564,0.189162336235264,0.18665312145832,0.185262147362104,0.184994903485898,0.185066797738128,0.185066797738128],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(34,73,108,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.487282422559211,0.471232867239462,0.455296842650266,0.439537241000199,0.424016258237098,0.408795148589065,0.402637560262697,0.390384072998266,0.372586870395888,0.354700527735017,0.336795634238643,0.318942852341428,0.30121263881736,0.283674966719055,0.266399049226085,0.249453066492178,0.232903896569301,0.230883038725603,0.230846554515527,0.231849318978953,0.233910351673977,0.237021518646147,0.24117054154078,0.24634104606002,0.252512626584708,0.259660926706041,0.267757735349187,0.276771098109511,0.286665443362013,0.297401722646276,0.308937564772907,0.321227443043255,0.334222854922498,0.34787251345699,0.362122549680428,0.376916725210063,0.392196654193895,0.407902033732959,0.423970881869321,0.440339782200544,0.456944134155261,0.473718407942113,0.4905964031659,0.501347533068466,0.487282422559211,0.487282422559211],"y":[0.754517800007025,0.742982420828649,0.730764359063413,0.717911833819203,0.704475568141398,0.690508588832182,0.684524441153954,0.692596430265486,0.703336867444015,0.713156430608217,0.722016366428415,0.729881708785646,0.736721416766958,0.742508497169717,0.747220111031436,0.750837663764736,0.75334687854168,0.753521612987656,0.761062435227093,0.772598277353723,0.783334556637987,0.793228901890489,0.802242264650813,0.810339073293959,0.817487373415292,0.82365895393998,0.82882945845922,0.832978481353853,0.836089648326023,0.838150681021047,0.839153445484473,0.839093984263085,0.837972532023159,0.835793514624341,0.832565531652807,0.828301322482621,0.823017715999246,0.81673556418362,0.809479659818919,0.801278638644761,0.792164866345032,0.782174310815313,0.771346400214027,0.76395919532696,0.754517800007025,0.754517800007025],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(38,81,119,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.591978101885752,0.574569262693944,0.557458114161256,0.540712186173608,0.52439756725828,0.50857864376269,0.5,0.49142135623731,0.47560243274172,0.459287813826392,0.442541885838744,0.425430737306056,0.408021898114248,0.402637560262697,0.408795148589065,0.424016258237098,0.439537241000199,0.455296842650266,0.471232867239462,0.487282422559211,0.501347533068466,0.507511510090297,0.52439697251605,0.541186151237198,0.557812787035576,0.574211262175675,0.590316859367867,0.597370908578246,0.591978101885752,0.591978101885752],"y":[0.680977506671335,0.668525951244807,0.655290904586595,0.641324599379342,0.626682154249102,0.611421356237309,0.602528970054552,0.611421356237309,0.626682154249103,0.641324599379341,0.655290904586595,0.668525951244807,0.680977506671335,0.684524441153954,0.690508588832182,0.704475568141398,0.717911833819203,0.730764359063413,0.742982420828649,0.754517800007025,0.76395919532696,0.759723867357517,0.747352581073137,0.734281365175949,0.720561805783428,0.706248047728625,0.69139658087525,0.684530020005949,0.680977506671335,0.680977506671335],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(55,115,163,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.50338216834701,0.51946856626151,0.535478130639317,0.551347679043907,0.567014581617834,0.582417008254165,0.597494172611667,0.612186572010721,0.626436222263218,0.640186886509658,0.65338429716034,0.66597637006475,0.677913410063906,0.689148307114444,0.699636722210444,0.709337262369222,0.718211643990527,0.726224843944418,0.733345237791561,0.739544724590439,0.744798837798934,0.749086841832587,0.75239181389848,0.754700710781767,0.755576593155291,0.750546933507822,0.733600950773915,0.716325033280945,0.69878736118264,0.681057147658572,0.663204365761357,0.645299472264983,0.627413129604112,0.609615927001734,0.597370908578246,0.590316859367867,0.574211262175675,0.557812787035576,0.541186151237198,0.52439697251605,0.507511510090297,0.501347533068466,0.50338216834701,0.50338216834701],"y":[0.765324971726938,0.775361285018936,0.78458713113961,0.792966099889212,0.800465123306073,0.807054606170744,0.812708542804807,0.817404619703405,0.821124303596454,0.823852914590988,0.825579684105997,0.82629779737109,0.826004420321286,0.824700710781767,0.82239181389848,0.819086841832587,0.814798837798933,0.809544724590439,0.803345237791561,0.796224843944418,0.788211643990526,0.779337262369222,0.769636722210444,0.759148307114444,0.751600269766683,0.750837663764735,0.747220111031436,0.742508497169717,0.736721416766958,0.729881708785646,0.722016366428415,0.713156430608217,0.703336867444015,0.692596430265486,0.684530020005949,0.69139658087525,0.706248047728625,0.720561805783428,0.734281365175949,0.747352581073137,0.759723867357517,0.76395919532696,0.765324971726938,0.765324971726938],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(86,177,247,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.08,0.26,0.71,0.93],"y":[0.78,0.86,0.85,0.78],"text":["DRV/r_96","DTG_48","DTG_96","DRV/r_48"],"hovertext":["","","",""],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.20045299544466,0.358381732758067,0.633396116298188,0.802485690816114,0.306472730239411,0.315123952707465,0.498421428685342,0.500121592161113,0.690655684959883,0.689238063246841,0.39953102161938,0.582377337297941,0.417284378929032,0.600468975765232,0.499924063415392],"y":[0.519090468430361,0.771384232274224,0.765202418425565,0.529878786855597,0.674019656966952,0.425613540671737,0.273819561432775,0.6863829572591,0.418131940967582,0.673276962924277,0.579085938438095,0.377764551702095,0.380690624900884,0.579259049632578,0.481316460773137],"text":[3,5,16,4,1,1,2,9,1,0,1,0,1,0,1],"hovertext":["Actinomyces_odontolyticus<br />Bacteroides_caccae Eubacterium_eligens","Collinsella_aerofaciens<br />Prevotella_stercorea<br />Oscillibacter_sp_CAG_241<br />Clostridium_leptum<br />Firmicutes_bacterium_CAG_83","Collinsella_massiliensis<br />Eggerthella_lenta Bacteroides_ovatus<br />Odoribacter_splanchnicus<br />Parabacteroides_merdae<br />Blautia_wexlerae Ruminococcus_gnavus<br />Coprococcus_catus Clostridium_citroniae<br />Lachnospira_pectinoschiza<br />Roseburia_sp_CAG_471<br />Flavonifractor_plautii<br />Eubacterium_siraeum<br />Ruminococcus_bicirculans<br />Mitsuokella_jalaludinii<br />Megasphaera_elsdenii","Bacteroides_vulgatus<br />Prevotella_sp_AM42_24<br />Lawsonibacter_asaccharolyticus<br />Desulfovibrio_piger","Catenibacterium_mitsuokai","Clostridium_bolteae","Alistipes_finegoldii Eubacterium_ramulus","Bifidobacterium_adolescentis<br />Olsenella_scatoligenes<br />Prevotella_sp_CAG_279<br />Alistipes_putredinis<br />Streptococcus_salivarius<br />Oscillibacter_sp_57_20<br />Clostridium_innocuum<br />Erysipelatoclostridium_ramosum<br />Firmicutes_bacterium_CAG_110","Collinsella_stercoris","","Streptococcus_parasanguinis","","Dorea_longicatena","","Actinomyces_oris"],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.2],"y":[0.2],"name":"99_3eb420ef6b918a9d1e43d12e2a257d53","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#132B43"],[0.00334448160535117,"#132B44"],[0.00668896321070234,"#132C44"],[0.0100334448160535,"#142C45"],[0.0133779264214047,"#142D45"],[0.0167224080267559,"#142D46"],[0.020066889632107,"#142D46"],[0.0234113712374582,"#142E47"],[0.0267558528428094,"#152E47"],[0.0301003344481605,"#152F48"],[0.0334448160535117,"#152F48"],[0.0367892976588629,"#152F49"],[0.040133779264214,"#153049"],[0.0434782608695652,"#16304A"],[0.0468227424749164,"#16304A"],[0.0501672240802676,"#16314B"],[0.0535117056856187,"#16314B"],[0.0568561872909699,"#16324C"],[0.0602006688963211,"#17324D"],[0.0635451505016722,"#17324D"],[0.0668896321070234,"#17334E"],[0.0702341137123746,"#17334E"],[0.0735785953177257,"#17344F"],[0.0769230769230769,"#18344F"],[0.0802675585284281,"#183450"],[0.0836120401337793,"#183550"],[0.0869565217391304,"#183551"],[0.0903010033444816,"#183651"],[0.0936454849498328,"#193652"],[0.096989966555184,"#193652"],[0.100334448160535,"#193753"],[0.103678929765886,"#193754"],[0.107023411371237,"#193854"],[0.110367892976589,"#1A3855"],[0.11371237458194,"#1A3955"],[0.117056856187291,"#1A3956"],[0.120401337792642,"#1A3956"],[0.123745819397993,"#1A3A57"],[0.127090301003344,"#1B3A57"],[0.130434782608696,"#1B3B58"],[0.133779264214047,"#1B3B59"],[0.137123745819398,"#1B3B59"],[0.140468227424749,"#1C3C5A"],[0.1438127090301,"#1C3C5A"],[0.147157190635451,"#1C3D5B"],[0.150501672240803,"#1C3D5B"],[0.153846153846154,"#1C3D5C"],[0.157190635451505,"#1D3E5C"],[0.160535117056856,"#1D3E5D"],[0.163879598662207,"#1D3F5D"],[0.167224080267559,"#1D3F5E"],[0.17056856187291,"#1D3F5F"],[0.173913043478261,"#1E405F"],[0.177257525083612,"#1E4060"],[0.180602006688963,"#1E4160"],[0.183946488294314,"#1E4161"],[0.187290969899666,"#1E4261"],[0.190635451505017,"#1F4262"],[0.193979933110368,"#1F4263"],[0.197324414715719,"#1F4363"],[0.20066889632107,"#1F4364"],[0.204013377926421,"#1F4464"],[0.207357859531773,"#204465"],[0.210702341137124,"#204465"],[0.214046822742475,"#204566"],[0.217391304347826,"#204566"],[0.220735785953177,"#214667"],[0.224080267558528,"#214668"],[0.22742474916388,"#214768"],[0.230769230769231,"#214769"],[0.234113712374582,"#214769"],[0.237458193979933,"#22486A"],[0.240802675585284,"#22486A"],[0.244147157190635,"#22496B"],[0.247491638795987,"#22496C"],[0.250836120401338,"#224A6C"],[0.254180602006689,"#234A6D"],[0.25752508361204,"#234A6D"],[0.260869565217391,"#234B6E"],[0.264214046822742,"#234B6E"],[0.267558528428094,"#244C6F"],[0.270903010033445,"#244C70"],[0.274247491638796,"#244C70"],[0.277591973244147,"#244D71"],[0.280936454849498,"#244D71"],[0.284280936454849,"#254E72"],[0.287625418060201,"#254E72"],[0.290969899665552,"#254F73"],[0.294314381270903,"#254F74"],[0.297658862876254,"#254F74"],[0.301003344481605,"#265075"],[0.304347826086957,"#265075"],[0.307692307692308,"#265176"],[0.311036789297659,"#265176"],[0.31438127090301,"#275277"],[0.317725752508361,"#275278"],[0.321070234113712,"#275278"],[0.324414715719064,"#275379"],[0.327759197324415,"#275379"],[0.331103678929766,"#28547A"],[0.334448160535117,"#28547B"],[0.337792642140468,"#28557B"],[0.341137123745819,"#28557C"],[0.344481605351171,"#28567C"],[0.347826086956522,"#29567D"],[0.351170568561873,"#29567D"],[0.354515050167224,"#29577E"],[0.357859531772575,"#29577F"],[0.361204013377926,"#2A587F"],[0.364548494983278,"#2A5880"],[0.367892976588629,"#2A5980"],[0.37123745819398,"#2A5981"],[0.374581939799331,"#2A5982"],[0.377926421404682,"#2B5A82"],[0.381270903010033,"#2B5A83"],[0.384615384615385,"#2B5B83"],[0.387959866220736,"#2B5B84"],[0.391304347826087,"#2C5C85"],[0.394648829431438,"#2C5C85"],[0.397993311036789,"#2C5D86"],[0.40133779264214,"#2C5D86"],[0.404682274247492,"#2C5D87"],[0.408026755852843,"#2D5E87"],[0.411371237458194,"#2D5E88"],[0.414715719063545,"#2D5F89"],[0.418060200668896,"#2D5F89"],[0.421404682274247,"#2E608A"],[0.424749163879599,"#2E608A"],[0.42809364548495,"#2E618B"],[0.431438127090301,"#2E618C"],[0.434782608695652,"#2E618C"],[0.438127090301003,"#2F628D"],[0.441471571906354,"#2F628D"],[0.444816053511706,"#2F638E"],[0.448160535117057,"#2F638F"],[0.451505016722408,"#30648F"],[0.454849498327759,"#306490"],[0.45819397993311,"#306590"],[0.461538461538462,"#306591"],[0.464882943143813,"#306592"],[0.468227424749164,"#316692"],[0.471571906354515,"#316693"],[0.474916387959866,"#316793"],[0.478260869565217,"#316794"],[0.481605351170569,"#326895"],[0.48494983277592,"#326895"],[0.488294314381271,"#326996"],[0.491638795986622,"#326996"],[0.494983277591973,"#326997"],[0.498327759197324,"#336A98"],[0.501672240802676,"#336A98"],[0.505016722408027,"#336B99"],[0.508361204013378,"#336B99"],[0.511705685618729,"#346C9A"],[0.51505016722408,"#346C9B"],[0.518394648829431,"#346D9B"],[0.521739130434783,"#346D9C"],[0.525083612040134,"#346E9D"],[0.528428093645485,"#356E9D"],[0.531772575250836,"#356E9E"],[0.535117056856187,"#356F9E"],[0.538461538461538,"#356F9F"],[0.54180602006689,"#3670A0"],[0.545150501672241,"#3670A0"],[0.548494983277592,"#3671A1"],[0.551839464882943,"#3671A1"],[0.555183946488294,"#3772A2"],[0.558528428093645,"#3772A3"],[0.561872909698997,"#3773A3"],[0.565217391304348,"#3773A4"],[0.568561872909699,"#3773A4"],[0.57190635451505,"#3874A5"],[0.575250836120401,"#3874A6"],[0.578595317725752,"#3875A6"],[0.581939799331104,"#3875A7"],[0.585284280936455,"#3976A8"],[0.588628762541806,"#3976A8"],[0.591973244147157,"#3977A9"],[0.595317725752508,"#3977A9"],[0.598662207357859,"#3978AA"],[0.602006688963211,"#3A78AB"],[0.605351170568562,"#3A79AB"],[0.608695652173913,"#3A79AC"],[0.612040133779264,"#3A79AC"],[0.615384615384615,"#3B7AAD"],[0.618729096989967,"#3B7AAE"],[0.622073578595318,"#3B7BAE"],[0.625418060200669,"#3B7BAF"],[0.62876254180602,"#3C7CB0"],[0.632107023411371,"#3C7CB0"],[0.635451505016722,"#3C7DB1"],[0.638795986622074,"#3C7DB1"],[0.642140468227425,"#3C7EB2"],[0.645484949832776,"#3D7EB3"],[0.648829431438127,"#3D7FB3"],[0.652173913043478,"#3D7FB4"],[0.655518394648829,"#3D7FB5"],[0.658862876254181,"#3E80B5"],[0.662207357859532,"#3E80B6"],[0.665551839464883,"#3E81B6"],[0.668896321070234,"#3E81B7"],[0.672240802675585,"#3F82B8"],[0.675585284280936,"#3F82B8"],[0.678929765886288,"#3F83B9"],[0.682274247491639,"#3F83BA"],[0.68561872909699,"#4084BA"],[0.688963210702341,"#4084BB"],[0.692307692307692,"#4085BB"],[0.695652173913043,"#4085BC"],[0.698996655518395,"#4086BD"],[0.702341137123746,"#4186BD"],[0.705685618729097,"#4186BE"],[0.709030100334448,"#4187BF"],[0.712374581939799,"#4187BF"],[0.71571906354515,"#4288C0"],[0.719063545150502,"#4288C1"],[0.722408026755853,"#4289C1"],[0.725752508361204,"#4289C2"],[0.729096989966555,"#438AC2"],[0.732441471571906,"#438AC3"],[0.735785953177257,"#438BC4"],[0.739130434782609,"#438BC4"],[0.74247491638796,"#438CC5"],[0.745819397993311,"#448CC6"],[0.749163879598662,"#448DC6"],[0.752508361204013,"#448DC7"],[0.755852842809364,"#448EC8"],[0.759197324414716,"#458EC8"],[0.762541806020067,"#458FC9"],[0.765886287625418,"#458FC9"],[0.769230769230769,"#458FCA"],[0.77257525083612,"#4690CB"],[0.775919732441472,"#4690CB"],[0.779264214046823,"#4691CC"],[0.782608695652174,"#4691CD"],[0.785953177257525,"#4792CD"],[0.789297658862876,"#4792CE"],[0.792642140468227,"#4793CF"],[0.795986622073579,"#4793CF"],[0.79933110367893,"#4894D0"],[0.802675585284281,"#4894D0"],[0.806020066889632,"#4895D1"],[0.809364548494983,"#4895D2"],[0.812709030100334,"#4896D2"],[0.816053511705686,"#4996D3"],[0.819397993311037,"#4997D4"],[0.822742474916388,"#4997D4"],[0.826086956521739,"#4998D5"],[0.82943143812709,"#4A98D6"],[0.832775919732441,"#4A99D6"],[0.836120401337793,"#4A99D7"],[0.839464882943144,"#4A9AD8"],[0.842809364548495,"#4B9AD8"],[0.846153846153846,"#4B9BD9"],[0.849498327759197,"#4B9BDA"],[0.852842809364548,"#4B9BDA"],[0.8561872909699,"#4C9CDB"],[0.859531772575251,"#4C9CDB"],[0.862876254180602,"#4C9DDC"],[0.866220735785953,"#4C9DDD"],[0.869565217391304,"#4D9EDD"],[0.872909698996655,"#4D9EDE"],[0.876254180602007,"#4D9FDF"],[0.879598662207358,"#4D9FDF"],[0.882943143812709,"#4DA0E0"],[0.88628762541806,"#4EA0E1"],[0.889632107023411,"#4EA1E1"],[0.892976588628763,"#4EA1E2"],[0.896321070234114,"#4EA2E3"],[0.899665551839465,"#4FA2E3"],[0.903010033444816,"#4FA3E4"],[0.906354515050167,"#4FA3E5"],[0.909698996655518,"#4FA4E5"],[0.91304347826087,"#50A4E6"],[0.916387959866221,"#50A5E7"],[0.919732441471572,"#50A5E7"],[0.923076923076923,"#50A6E8"],[0.926421404682274,"#51A6E8"],[0.929765886287625,"#51A7E9"],[0.933110367892977,"#51A7EA"],[0.936454849498328,"#51A8EA"],[0.939799331103679,"#52A8EB"],[0.94314381270903,"#52A9EC"],[0.946488294314381,"#52A9EC"],[0.949832775919732,"#52AAED"],[0.953177257525084,"#53AAEE"],[0.956521739130435,"#53ABEE"],[0.959866220735786,"#53ABEF"],[0.963210702341137,"#53ACF0"],[0.966555183946488,"#54ACF0"],[0.969899665551839,"#54ADF1"],[0.973244147157191,"#54ADF2"],[0.976588628762542,"#54AEF2"],[0.979933110367893,"#55AEF3"],[0.983277591973244,"#55AFF4"],[0.986622073578595,"#55AFF4"],[0.989966555183946,"#55B0F5"],[0.993311036789298,"#56B0F6"],[0.996655518394649,"#56B1F6"],[1,"#56B1F7"]],"colorbar":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"thickness":23.04,"title":"count","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["0","4","8","12","16"],"tickvals":[0,0.25,0.5,0.75,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"margin":{"t":16,"r":0,"b":0,"l":0},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.0214943938344883,0.978505606165512],"tickmode":"array","ticktext":["0.2","0.4","0.6","0.8"],"tickvals":[0.2,0.4,0.6,0.8],"categoryorder":"array","categoryarray":["0.2","0.4","0.6","0.8"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"y","scaleratio":1,"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.151244648660193,0.893750254825705],"tickmode":"array","ticktext":["0.2","0.3","0.4","0.5","0.6","0.7","0.8"],"tickvals":[0.2,0.3,0.4,0.5,0.6,0.7,0.8],"categoryorder":"array","categoryarray":["0.2","0.3","0.4","0.5","0.6","0.7","0.8"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"x","scaleratio":1,"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"3bcc5ebbcfe4":{"fill":{},"type":"scatter"},"3bcc51df23957":{"label":{}},"3bcc56ec044c0":{"label":{},"text":{}}},"cur_data":"3bcc5ebbcfe4","visdat":{"3bcc5ebbcfe4":["function (y) ","x"],"3bcc51df23957":["function (y) ","x"],"3bcc56ec044c0":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>## The following is the same for CD4 deltas at 48 and 96 weeks

# comps_delta &lt;- list(&quot;150_H&quot;=c(&quot;150_H_48&quot;,&quot;150_H_96&quot;),
#               &quot;150_L&quot; = c(&quot;150_L_48&quot;, &quot;150_L_96&quot;))
# 
# 
# Venn_delta48 &lt;- stat_list_long$CD4diff_48 %&gt;%
#       dplyr::select(species, group2, level) %&gt;%
#       dplyr::mutate(row=row_number()) %&gt;% 
#       tidyr::pivot_wider(., values_from = &quot;species&quot;, names_from = c(&quot;level&quot;, &quot;group2&quot;)) %&gt;% 
#       dplyr::select(-row) %&gt;%
# 
#       map(~{
#         .x[!is.na(.x)]
#         
#       }) 
# delta48_diff &lt;- names(comps_delta) %&gt;% 
#   set_names() %&gt;% 
#   map(~ {
#     Venn_delta48 %&gt;% 
#       ggVennDiagram::Venn(.) %&gt;% 
#       ggVennDiagram::discern_overlap(., slice = comps_delta[[.x]])
#   }) 
# 
#     
# Venn_delta96 &lt;- stat_list_long$CD4diff_96 %&gt;%
#       dplyr::select(species, group2, level) %&gt;%
#       dplyr::mutate(row=row_number()) %&gt;% 
#       tidyr::pivot_wider(., values_from = &quot;species&quot;, names_from = c(&quot;level&quot;, &quot;group2&quot;)) %&gt;% 
#       dplyr::select(-row) %&gt;%
# 
#       map(~{
#         .x[!is.na(.x)]
#         
#       }) 
# delta96_diff &lt;- names(comps_delta) %&gt;% 
#   set_names() %&gt;% 
#   map(~ {
#     Venn_delta96 %&gt;% 
#       ggVennDiagram::Venn(.) %&gt;% 
#       ggVennDiagram::discern_overlap(., slice = comps_delta[[.x]])
#   }) </code></pre>
<p>The species we found are interesting as these are the ones that may be differential between both treatment grous. We’ll plot the longitudinal analysis for these selected subsets (DA at both TP 48 and 96 only for DTG, and the same for DRV/r)</p>
<pre class="r"><code>#Code is functional bu we still need to filter the input tables.



# prepare_venn_results &lt;- function(res, cat_var, otu_df, metadata, long_var){} # Should i write it as a function?
cat_var = &quot;group&quot;

venn_group_res &lt;- group_diff %&gt;% 
  names() %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(lv){
    group_diff[[lv]] %&gt;% 
      purrr::set_names() %&gt;% 
      purrr::map(function(sp){
        otu_df %&gt;% 
          dplyr::select(SampleID,!!sym(sp)) %&gt;% 
          dplyr::left_join(., metadata[,c(&quot;SampleID&quot;, cat_var, long_var, link_var)], by=&quot;SampleID&quot;) %&gt;% 
          dplyr::filter(!!sym(cat_var) == lv)
      })
  })

cv &lt;- &quot;group&quot;
pal &lt;- RColorBrewer::brewer.pal(2,&quot;Set1&quot;)[c(1,2)] %&gt;% 
  setNames(c(&quot;DRV/r&quot;,&quot;DTG&quot;))
long_plots &lt;-
  venn_group_res %&gt;%
  names() %&gt;%
  set_names() %&gt;%
  map(function(lv)
  {
    names(venn_group_res[[lv]]) %&gt;%
      set_names() %&gt;%
      map(function(sp)
      {
        groupmax &lt;-
          venn_group_res %&gt;%
          purrr::pluck(lv, sp) %&gt;%
          dplyr::summarise(max = max(!!sym(sp)))
        
        stat_params &lt;-
          stat_list_long %&gt;%
          purrr::pluck(cv) %&gt;%
          dplyr::filter(species == sp) %&gt;%
          dplyr::mutate(group2 = as.factor(group2)) %&gt;%
          rstatix::add_xy_position() %&gt;%
          dplyr::rename(cat_var = level) %&gt;%
          dplyr::mutate(xmax = case_when(group2 == 48 ~ 2, group2 == 96 ~ 3),
                        y.position = 1.1 * max) %&gt;% 
          dplyr::mutate(p = round(p,3))
        
        venn_group_res %&gt;%
          purrr::pluck(lv, sp) %&gt;%
          dplyr::arrange(!!sym(link_var)) %&gt;%
          dplyr::rename(long_var = !!sym(long_var),
                        sp = !!sym(sp)) %&gt;%
          dplyr::filter(long_var %in% c(0, stat_params$group2)) %&gt;%
          dplyr::mutate(long_var = as.factor(long_var)) %&gt;%
          ggplot(., aes(x = long_var, y = sp)) +
          geom_violin(width = 1.5,
                      trim = T,
                      fill = pal[lv]) +
          geom_boxplot(width = 0.1, fill = &quot;white&quot;) +
          geom_line(aes(group = !!sym(link_var)), color = &quot;grey&quot;) +
          scale_fill_brewer(palette = &quot;Set1&quot;) +
          ggpubr::stat_pvalue_manual(
            stat_params,
            label = &quot;p = {p}&quot;,
            xmin = &quot;xmin&quot;,
            xmax = &quot;xmax&quot;,
            y = &quot;y.position&quot;
          ) +
          theme_bw() +
          labs(x = &quot;week&quot;, y = &quot;RA (%)&quot;, title = sp)
        
        
      })
    
  })</code></pre>
<div id="da-dtg-group" class="section level2">
<h2>DA (DTG group)</h2>
<pre><code>$`1`</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/DTG_longPlots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DTG_longPlots-1">
Past versions of DTG_longPlots-1.png
</button>
</p>
<div id="fig-DTG_longPlots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DTG_longPlots-1.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$`2`</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/DTG_longPlots-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DTG_longPlots-2">
Past versions of DTG_longPlots-2.png
</button>
</p>
<div id="fig-DTG_longPlots-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DTG_longPlots-2.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$`3`</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/DTG_longPlots-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DTG_longPlots-3">
Past versions of DTG_longPlots-3.png
</button>
</p>
<div id="fig-DTG_longPlots-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DTG_longPlots-3.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$`4`</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/DTG_longPlots-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DTG_longPlots-4">
Past versions of DTG_longPlots-4.png
</button>
</p>
<div id="fig-DTG_longPlots-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DTG_longPlots-4.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$`5`</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/DTG_longPlots-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DTG_longPlots-5">
Past versions of DTG_longPlots-5.png
</button>
</p>
<div id="fig-DTG_longPlots-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DTG_longPlots-5.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
attr(,&quot;class&quot;)
[1] &quot;list&quot;      &quot;ggarrange&quot;</code></pre>
</div>
<div id="da-dtg-group-1" class="section level2">
<h2>DA (DTG group)</h2>
<p><img src="figure/420_Taxonomy_DA.Rmd/DRV_longPlots-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-DRV_longPlots-1">
Past versions of DRV_longPlots-1.png
</button>
</p>
<div id="fig-DRV_longPlots-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/bd6a5229da72953a0e9021a8ea59e78a5d88cac9/docs/figure/420_Taxonomy_DA.Rmd/DRV_longPlots-1.png" target="_blank">bd6a522</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compositional-approach-ancom" class="section level1">
<h1>Compositional approach (ANCOM)</h1>
<div id="cross-group-differential-abundance-1" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 24, week 48 and 96.</p>
<pre class="r"><code>mymre &lt;- metar::metaphlan_ancom(mymre, tax_level = &quot;Species&quot;, save_files = F)

ancom_group &lt;- get_taxa(mymre, type = &quot;metaphlan&quot;,result_slot = &quot;ancom&quot;) %&gt;% 
  purrr::pluck(&quot;Species&quot;, &quot;group&quot;,&quot;plot&quot;)

cat_ancom_list &lt;- 
c(0,48,96) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(tp){
    keep_ids &lt;- 
    metadata %&gt;% 
      dplyr::filter(time_point == tp) %&gt;% 
      dplyr::pull(SampleID)
    
    
    my_ancom &lt;-
      metar::filter_samples(mymre, sample_ids = keep_ids) %&gt;%
      metar::metaphlan_ancom(., tax_level = &quot;Species&quot;, save_files = F) %&gt;%
      metar::get_taxa(., type = &quot;metaphlan&quot;,result_slot = &quot;ancom&quot;) %&gt;% 
      purrr::pluck(&quot;Species&quot;, &quot;group&quot;,&quot;plot&quot;)

})


names(cat_ancom_list) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(tp){
    cat_ancom_list %&gt;% 
      purrr::pluck(tp, &quot;data&quot;) %&gt;% 
      dplyr::filter(detected_0.6 == T)
  })</code></pre>
<pre><code>$`0`
 [1] taxa_id         W               detected_0.9    detected_0.8   
 [5] detected_0.7    detected_0.6    mean_qval       median_qval    
 [9] fisher_qval     clr_mean_diff   structural_zero x_label        
[13] y_label         comparison      cat_var        
&lt;0 rows&gt; (or 0-length row.names)

$`48`
 [1] taxa_id         W               detected_0.9    detected_0.8   
 [5] detected_0.7    detected_0.6    mean_qval       median_qval    
 [9] fisher_qval     clr_mean_diff   structural_zero x_label        
[13] y_label         comparison      cat_var        
&lt;0 rows&gt; (or 0-length row.names)

$`96`
 [1] taxa_id         W               detected_0.9    detected_0.8   
 [5] detected_0.7    detected_0.6    mean_qval       median_qval    
 [9] fisher_qval     clr_mean_diff   structural_zero x_label        
[13] y_label         comparison      cat_var        
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<p>Interestingly tough, no species were found significant at any timepoint between both groups from a compositional point of view.</p>
<div id="comparison-of-significant-species-1" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance-1" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<pre class="r"><code>cat_vars &lt;- data.frame(CategoricalVariable = &quot;time_point&quot;, PaketteName = &quot;Set1&quot;)
cat_vec &lt;- c(&quot;group&quot;)
ancom_longlist &lt;- 
cat_vec %&gt;% 
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    metadata %&gt;%
      dplyr::pull(!!sym(cv)) %&gt;%
      unique() %&gt;%
      purrr::set_names() %&gt;%
      purrr::map(function(lv) {
        
        c(48, 96) %&gt;%
          set_names() %&gt;%
          purrr::map(function(tp) {
            
            ids &lt;-
              metadata %&gt;%
              dplyr::filter(!!sym(cv) == lv &amp;
                            !!sym(long_var) %in% c(0,tp)) %&gt;%
              dplyr::pull(SampleID)
            
            my_ancom &lt;- 
              metar::filter_samples(mymre, sample_ids = ids) %&gt;%
              metar::get_phyloseq(object = ., type = &quot;metaphlan_counts&quot;) %&gt;% 
              metar::run_ancom(phy = ., categorical_vars = cat_vars[1,], tax_level = &quot;Species&quot;,save_files = F)
            
            plt &lt;- 
              my_ancom$time_point$plot +
              labs(title=paste(&quot;ANCOM &quot;, lv, &quot; group&quot;, sep=&quot;&quot;))
            stats &lt;- 
              my_ancom$time_point$stats %&gt;% 
              dplyr::filter(detected_0.7 == T) %&gt;% 
              mutate(comp = paste(cv, &quot;_0v&quot;,tp, sep=&quot;&quot;))
            return(list(plot = plt, stats = stats))
          })
      })
  })

# pull significan

DTG_plotlist &lt;- list(ancom_longlist$group$DTG$`48`$plot, 
                     ancom_longlist$group$DTG$`96`$plot)

ggpubr::ggarrange(plotlist = DTG_plotlist)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/ANCOM_within_group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ANCOM_within_group-1">
Past versions of ANCOM_within_group-1.png
</button>
</p>
<div id="fig-ANCOM_within_group-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/9713823b00b303b0cb85954d9a9ca081ad4a2c8d/docs/figure/420_Taxonomy_DA.Rmd/ANCOM_within_group-1.png" target="_blank">9713823</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>RTV_plotlist &lt;- list(ancom_longlist$group$`DRV/r`$`48`$plot,
                     ancom_longlist$group$`DRV/r`$`96`$plot)

ggpubr::ggarrange(plotlist = RTV_plotlist)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/ANCOM_within_group-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-ANCOM_within_group-2">
Past versions of ANCOM_within_group-2.png
</button>
</p>
<div id="fig-ANCOM_within_group-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/9713823b00b303b0cb85954d9a9ca081ad4a2c8d/docs/figure/420_Taxonomy_DA.Rmd/ANCOM_within_group-2.png" target="_blank">9713823</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>statlist &lt;- rbind(ancom_longlist$group$DTG$`48`$stats, 
                     ancom_longlist$group$DTG$`96`$stats,
                      ancom_longlist$group$`DRV/r`$`48`$stats,
                     ancom_longlist$group$`DRV/r`$`96`$stats)

kableExtra::kable(statlist, format = &quot;markdown&quot;)</code></pre>
<table>
<colgroup>
<col width="14%" />
<col width="1%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="6%" />
<col width="5%" />
<col width="6%" />
<col width="6%" />
<col width="7%" />
<col width="8%" />
<col width="5%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">taxa_id</th>
<th align="right">W</th>
<th align="left">detected_0.9</th>
<th align="left">detected_0.8</th>
<th align="left">detected_0.7</th>
<th align="left">detected_0.6</th>
<th align="right">mean_qval</th>
<th align="right">median_qval</th>
<th align="right">fisher_qval</th>
<th align="right">clr_mean_diff</th>
<th align="left">structural_zero</th>
<th align="right">x_label</th>
<th align="right">y_label</th>
<th align="left">comparison</th>
<th align="left">cat_var</th>
<th align="left">comp</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Streptococcus_parasanguinis</td>
<td align="right">31</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
<td align="right">0.0576419</td>
<td align="right">0.0144763</td>
<td align="right">0</td>
<td align="right">-0.1700753</td>
<td align="left">No</td>
<td align="right">-1.037554</td>
<td align="right">25.2</td>
<td align="left">0_96</td>
<td align="left">time_point</td>
<td align="left">group_0v96</td>
</tr>
</tbody>
</table>
<p>Interestingly though, only one species (Streptococcus parasanguinis) appeared as statistically significant DA, only after 96 weeks in the DTG group. This species was also identified in the non-compositional analysis, being in the intersection between DTG(0-48), DTG(0-96) and DRV/r(0-96).</p>
<pre class="r"><code>phy &lt;- metar::get_phyloseq(mymre, type = &quot;metaphlan&quot;)



phyloseq::otu_table(phy) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy)[,&quot;Species&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  dplyr::select(SampleID,SampleID,statlist$taxa_id) %&gt;% 
  dplyr::inner_join(metadata[,c(&quot;SampleID&quot;,&quot;record_id&quot; ,&quot;time_point&quot;,&quot;group&quot;)], by = &quot;SampleID&quot;) %&gt;% 
  create_LMM(data = .,
             breakpoints = NULL, 
             cat_var = &quot;group&quot;,
             num_var= statlist$taxa_id, 
             long_var = &quot;time_point&quot;, 
             link_var = &quot;record_id&quot;) </code></pre>
<pre><code>$plot</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_strep-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plot_strep-1">
Past versions of plot_strep-1.png
</button>
</p>
<div id="fig-plot_strep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/1743951bf18aab3b40cb5ba194de76e9401f18dc/docs/figure/420_Taxonomy_DA.Rmd/plot_strep-1.png" target="_blank">1743951</a>
</td>
<td>
cblazquez
</td>
<td>
2022-05-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$stats
       slope  pval cat_var
DRV/r -0.002 0.726   DRV/r
DTG   -0.004 0.030     DTG</code></pre>
<p>This is backed by LMMs which present a really small albeit significant reduction in this strain in the DTG group.</p>
<pre class="r"><code>oral_db &lt;- here::here(&quot;Metadata&quot;,&quot;HOMD_oral_microbiome.txt&quot;) %&gt;% 
  read.delim(., header = T, skip = 1) %&gt;% 
  dplyr::select(Genus,Species) %&gt;% 
  mutate(name = paste(Genus, Species, sep=&quot;_&quot;)) %&gt;% 
  dplyr::pull(name)
  

stat_list_oral &lt;- 
names(stat_list_long) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(cv){
    stat_list_long %&gt;% 
      purrr::pluck(cv) %&gt;% 
      dplyr::filter(species %in% oral_db) %&gt;% 
      dplyr::pull(species) %&gt;% 
      unique()
  })


stat_list_oral$group %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map(function(sp){
    otu_df %&gt;% 
      dplyr::select(SampleID, sp) %&gt;% 
      inner_join(metadata[,c(&quot;SampleID&quot;,&quot;record_id&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(., grouping.var = &quot;group&quot;, x = &quot;time_point&quot;, y = !!sym(sp),type = &quot;np&quot;, pairwise.comparisons = T)

  })</code></pre>
<pre><code>$Actinomyces_oris</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/oral_microbiome-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oral_microbiome-1">
Past versions of oral_microbiome-1.png
</button>
</p>
<div id="fig-oral_microbiome-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/oral_microbiome-1.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Eggerthella_lenta</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/oral_microbiome-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oral_microbiome-2">
Past versions of oral_microbiome-2.png
</button>
</p>
<div id="fig-oral_microbiome-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/oral_microbiome-2.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Streptococcus_parasanguinis</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/oral_microbiome-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oral_microbiome-3">
Past versions of oral_microbiome-3.png
</button>
</p>
<div id="fig-oral_microbiome-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/oral_microbiome-3.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Streptococcus_salivarius</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/oral_microbiome-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oral_microbiome-4">
Past versions of oral_microbiome-4.png
</button>
</p>
<div id="fig-oral_microbiome-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3fa20ba400c57e7465b5ff4468c303398b28ff60/docs/figure/420_Taxonomy_DA.Rmd/oral_microbiome-4.png" target="_blank">3fa20ba</a>
</td>
<td>
cblazquez
</td>
<td>
2022-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nlme_3.1-158       metar_0.1.5        testthat_3.1.4     ggpubr_0.4.0      
 [5] rstatix_0.7.0      RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0     
 [9] dplyr_1.0.9        purrr_0.3.4        readr_2.1.2        tidyr_1.2.0       
[13] tibble_3.1.7       ggplot2_3.3.6      tidyverse_1.3.1    devtools_2.4.3    
[17] usethis_2.1.6      workflowr_1.7.0   

loaded via a namespace (and not attached):
  [1] estimability_1.3       coda_0.19-4            bit64_4.0.5           
  [4] knitr_1.39             multcomp_1.4-19        wesanderson_0.3.6     
  [7] data.table_1.14.2      DataExplorer_0.8.2     generics_0.1.2        
 [10] BiocGenerics_0.36.1    cowplot_1.1.1          callr_3.7.0           
 [13] TH.data_1.1-1          proxy_0.4-27           future_1.26.1         
 [16] correlation_0.8.1      bit_4.0.4              tzdb_0.3.0            
 [19] bayesm_3.1-4           webshot_0.5.3          xml2_1.3.2            
 [22] lubridate_1.8.0        httpuv_1.6.5           assertthat_0.2.1      
 [25] xfun_0.31              hms_1.1.1              jquerylib_0.1.4       
 [28] evaluate_0.15          promises_1.2.0.1       DEoptimR_1.0-11       
 [31] fansi_1.0.3            dbplyr_2.2.0           readxl_1.4.0          
 [34] igraph_1.3.2           DBI_1.1.3              htmlwidgets_1.5.4     
 [37] tensorA_0.36.2         kSamples_1.2-9         stats4_4.0.5          
 [40] Rmpfr_0.8-9            paletteer_1.4.0        ellipsis_0.3.2        
 [43] crosstalk_1.2.0        survcomp_1.40.0        backports_1.4.1       
 [46] insight_0.17.1         permute_0.9-7          prismatic_1.1.0       
 [49] compositions_2.0-4     vctrs_0.4.1            Biobase_2.50.0        
 [52] remotes_2.4.2          here_1.0.1             abind_1.4-5           
 [55] cachem_1.0.6           withr_2.5.0            ggforce_0.3.3         
 [58] RVenn_1.1.0            aws.signature_0.6.0    robustbase_0.95-0     
 [61] vroom_1.5.7            emmeans_1.7.4-1        vegan_2.6-2           
 [64] prettyunits_1.1.1      svglite_2.1.0          cluster_2.1.3         
 [67] ape_5.6-2              lazyeval_0.2.2         crayon_1.5.1          
 [70] SuppDists_1.1-9.7      pkgconfig_2.0.3        labeling_0.4.2        
 [73] units_0.8-0            tweenr_1.0.2           statsExpressions_1.3.2
 [76] pkgload_1.2.4          blme_1.0-5             rlang_1.0.2           
 [79] globals_0.15.0         lifecycle_1.0.1        sandwich_3.0-2        
 [82] survivalROC_1.0.3      selbal_0.1.0           phyloseq_1.34.0       
 [85] modelr_0.1.8           cellranger_1.1.0       rprojroot_2.0.3       
 [88] polyclip_1.10-0        datawizard_0.4.1       Matrix_1.4-1          
 [91] carData_3.0-5          Rhdf5lib_1.12.1        boot_1.3-28           
 [94] zoo_1.8-10             reprex_2.0.1           base64enc_0.1-3       
 [97] whisker_0.4            processx_3.6.1         viridisLite_0.4.0     
[100] PMCMRplus_1.9.4        bootstrap_2019.6       parameters_0.18.1     
[103] getPass_0.2-2          KernSmooth_2.23-20     rhdf5filters_1.2.1    
[106] Biostrings_2.58.0      classInt_0.4-7         multcompView_0.1-8    
[109] arm_1.12-2             parallelly_1.32.0      S4Vectors_0.28.1      
[112] ggsignif_0.6.3.9000    rmeta_3.0              aws.s3_0.3.21         
[115] scales_1.2.0           memoise_2.0.1          magrittr_2.0.3        
[118] plyr_1.8.7             zlibbioc_1.36.0        compiler_4.0.5        
[121] kableExtra_1.3.4       lme4_1.1-29            cli_3.3.0             
[124] ade4_1.7-19            XVector_0.30.0         listenv_0.8.0         
[127] patchwork_1.1.1        ps_1.7.1               MASS_7.3-57           
[130] mgcv_1.8-40            tidyselect_1.1.2       stringi_1.7.6         
[133] highr_0.9              yaml_2.3.5             ANCOM_2.2             
[136] ggrepel_0.9.1          grid_4.0.5             sass_0.4.1            
[139] tools_4.0.5            future.apply_1.9.0     parallel_4.0.5        
[142] rstudioapi_0.13        foreach_1.5.2          logging_0.10-108      
[145] git2r_0.30.1           gridExtra_2.3          prodlim_2019.11.13    
[148] farver_2.1.0           digest_0.6.29          lava_1.6.10           
[151] shiny_1.7.1            BWStest_0.2.2          networkD3_0.4         
[154] Rcpp_1.0.8.3           car_3.1-0              broom_0.8.0           
[157] performance_0.9.1      later_1.3.0            httr_1.4.2            
[160] sf_1.0-7               effectsize_0.7.0       colorspace_2.0-3      
[163] rvest_1.0.2            brio_1.1.3             fs_1.5.2              
[166] IRanges_2.24.1         splines_4.0.5          ggVennDiagram_1.2.1   
[169] rematch2_2.1.2         multtest_2.46.0        systemfonts_1.0.4     
[172] plotly_4.10.0          sessioninfo_1.2.2      gmp_0.6-5             
[175] xtable_1.8-4           jsonlite_1.8.0         nloptr_2.0.3          
[178] zeallot_0.1.0          R6_2.5.1               broom.mixed_0.2.9.4   
[181] pillar_1.7.0           htmltools_0.5.2        mime_0.12             
[184] glue_1.6.2             fastmap_1.1.0          minqa_1.2.4           
[187] class_7.3-20           codetools_0.2-18       pkgbuild_1.3.1        
[190] mvtnorm_1.1-3          furrr_0.3.0            utf8_1.2.2            
[193] lattice_0.20-45        bslib_0.3.1            logger_0.2.2          
[196] curl_4.3.2             survival_3.3-1         rmarkdown_2.14        
[199] desc_1.4.1             biomformat_1.18.0      merTools_0.5.2        
[202] munsell_0.5.0          e1071_1.7-11           rhdf5_2.34.0          
[205] iterators_1.0.14       ggstatsplot_0.9.3      haven_2.5.0           
[208] reshape2_1.4.4         gtable_0.3.0           bayestestR_0.12.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
