<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />


<title>Taxonomy: Differential Abundance</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Taxonomy: Differential Abundance</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">3/23/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree2a1fe845b3637c85682f64ab5aea8d2164e1192ftargetblank2a1fe84a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/2a1fe845b3637c85682f64ab5aea8d2164e1192f" target="_blank">2a1fe84</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4tree2a1fe845b3637c85682f64ab5aea8d2164e1192ftargetblank2a1fe84a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/2a1fe845b3637c85682f64ab5aea8d2164e1192f" target="_blank">2a1fe84</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  analysis/figure/

Unstaged changes:
    Modified:   analysis/420_Taxonomy_DA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/420_Taxonomy_DA.Rmd</code>) and HTML (<code>docs/420_Taxonomy_DA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c20090b3a3c8e714d8be91e4ea67aa5d3ff7c522/docs/420_Taxonomy_DA.html" target="_blank">c20090b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-01
</td>
<td>
Compiled taxa ranking html + several bug fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5917250ea4ce78208d3a41d2c612085024734455/analysis/420_Taxonomy_DA.Rmd" target="_blank">5917250</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-31
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e253850975c479f491140a3a60d9e54a3eca44ff/docs/420_Taxonomy_DA.html" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/420_Taxonomy_DA.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/420_Taxonomy_DA.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/11c1e7d2e4c2b76ccb9e08a96305fa7f34ee48de/analysis/420_Taxonomy_DA.Rmd" target="_blank">11c1e7d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
Code reorganized towards new structure
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import-and-exploration" class="section level2">
<h2>Data import and exploration</h2>
<p>This section should explain anyithing that may have happened during the first fase of importing and formatting data, such as any modification or additional filtering step.</p>
<pre class="r"><code># mymre &lt;- here::here(&quot;data&quot;,&quot;mre.rds&quot;) %&gt;%
#   readRDS(.)

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;)
# metadata &lt;-get_meta(mymre)
metadata &lt;- here::here(&quot;Metadata&quot;, &quot;2022_04_28_clean_metadata_LIMS.csv&quot;) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

link_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here(&quot;../WMGSPipeline&quot;))</code></pre>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre class="r"><code>mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()</code></pre>
<pre class="r"><code># Any homebrew support function that may be written will be placed here</code></pre>
<p>First we’ll start with a non-compositional approach which may provide potential candidate taxa for DA to be confirmed with more astringent and specific compositional approaches.</p>
</div>
<div id="non-compositional-analysis" class="section level1">
<h1>Non-compositional analysis</h1>
<p>In this case, we’ll test for group as categorical variables. We may extend the analysis towards other cat_vars, but those were more important for basal comparisons, which already shown no great differences were found among other potential confusor variables.</p>
<div id="cross-group-differential-abundance" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.</p>
<pre class="r"><code>phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;) %&gt;% 
      phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


otu_df &lt;- 
  phyloseq::otu_table(phy) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy)[,&quot;Species&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;))


stat_list &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    
    c(0,48,96) %&gt;% 
      set_names() %&gt;% 
      map_dfr(function(tp){
        otu_df %&gt;%
        dplyr::select(-SampleID) %&gt;%
        colnames() %&gt;%
        purrr::set_names() %&gt;%
        purrr::map_dfr(function(sp) {
          df &lt;-
            otu_df %&gt;%
            dplyr::select(SampleID, !!sym(sp)) %&gt;%
            dplyr::left_join(metadata[, c(&quot;SampleID&quot;, link_var, cv, long_var), drop = F], 
                             by = &quot;SampleID&quot;) %&gt;% 
            dplyr::filter(!!sym(long_var) %in% tp) %&gt;% 
            dplyr::group_by(!!sym(long_var)) 

          
          if (length(unique(pull(.data = df, !!sym(cv)))) == 2) {
            test &lt;- df %&gt;%
              rstatix::wilcox_test(., formula(paste(sp, &quot;~&quot;, cv)), paired = F) %&gt;%
              dplyr::mutate(method = &quot;Wilcoxon&quot;)
            
          } else{
            test &lt;- df %&gt;%
              rstatix::kruskal_test(., formula(paste(sp, &quot;~&quot;, cv))) %&gt;%
              dplyr::mutate(method = &quot;Kruskal-Wallis&quot;)
          }
          
        }) %&gt;% 
      rename(Species = .y.) %&gt;% 
      group_by(!!sym(long_var)) %&gt;% 
      mutate(p.adj = p.adjust(p, &quot;BH&quot;)) %&gt;% 
      filter(p.adj &lt; .05 ) %&gt;% 
      group_by(Species) 
      }) 
}) 


DAsp &lt;- stat_list$group %&gt;% 
  pull(Species)


otu_df %&gt;% 
  select(SampleID, !!sym(DAsp)) %&gt;% 
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  group_by(group) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = !!sym(DAsp[1]),
                              pairwise.comparisons = T,
                              ylab = paste(DAsp[1], &quot;(%)&quot;), 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Only one spec</p>
<div id="comparison-of-significant-species" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<pre class="r"><code>            # dplyr::filter(!n()&lt;2) %&gt;% 
            # dplyr::arrange(!!sym(link_var), !!sym(long_var))</code></pre>
</div>
</div>
<div id="compositional-approach-ancom" class="section level1">
<h1>Compositional approach (ANCOM)</h1>
<div id="cross-group-differential-abundance-1" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.</p>
<div id="comparison-of-significant-species-1" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance-1" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5        testthat_3.1.4     RColorBrewer_1.1-3 forcats_0.5.1     
 [5] stringr_1.4.0      dplyr_1.0.8        purrr_0.3.4        readr_2.1.2       
 [9] tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             tidyselect_1.1.2       lme4_1.1-29           
  [4] htmlwidgets_1.5.4      grid_4.0.5             devtools_2.4.3        
  [7] aws.signature_0.6.0    munsell_0.5.0          effectsize_0.6.0.1    
 [10] codetools_0.2-18       future_1.25.0          withr_2.5.0           
 [13] colorspace_2.0-3       Biobase_2.50.0         phyloseq_1.34.0       
 [16] highr_0.9              logger_0.2.2           knitr_1.39            
 [19] rstudioapi_0.13        stats4_4.0.5           ggsignif_0.6.3.9000   
 [22] listenv_0.8.0          labeling_0.4.2         emmeans_1.7.3         
 [25] git2r_0.30.1           polyclip_1.10-0        datawizard_0.4.0      
 [28] bit64_4.0.5            farver_2.1.0           rhdf5_2.34.0          
 [31] rprojroot_2.0.3        TH.data_1.1-1          coda_0.19-4           
 [34] parallelly_1.31.1      vctrs_0.4.1            generics_0.1.2        
 [37] xfun_0.30              R6_2.5.1               arm_1.12-2            
 [40] BayesFactor_0.9.12-4.3 reshape_0.8.9          rhdf5filters_1.2.1    
 [43] cachem_1.0.6           assertthat_0.2.1       promises_1.2.0.1      
 [46] networkD3_0.4          scales_1.2.0           vroom_1.5.7           
 [49] multcomp_1.4-19        merTools_0.5.2         gtable_0.3.0          
 [52] globals_0.14.0         processx_3.5.3         sandwich_3.0-1        
 [55] workflowr_1.7.0        MatrixModels_0.5-0     rlang_1.0.2           
 [58] zeallot_0.1.0          splines_4.0.5          rstatix_0.7.0         
 [61] prismatic_1.1.0        broom_0.8.0            yaml_2.3.5            
 [64] reshape2_1.4.4         abind_1.4-5            modelr_0.1.8          
 [67] backports_1.4.1        httpuv_1.6.5           tools_4.0.5           
 [70] usethis_2.1.5          logging_0.10-108       ellipsis_0.3.2        
 [73] jquerylib_0.1.4        biomformat_1.18.0      selbal_0.1.0          
 [76] BiocGenerics_0.36.1    WRS2_1.1-3             sessioninfo_1.2.2     
 [79] Rcpp_1.0.8.3           plyr_1.8.7             base64enc_0.1-3       
 [82] zlibbioc_1.36.0        ps_1.7.0               prettyunits_1.1.1     
 [85] ggpubr_0.4.0           pbapply_1.5-0          correlation_0.8.0     
 [88] zoo_1.8-10             S4Vectors_0.28.1       ggrepel_0.9.1         
 [91] haven_2.5.0            cluster_2.1.3          fs_1.5.2              
 [94] here_1.0.1             furrr_0.2.3            DataExplorer_0.8.2    
 [97] magrittr_2.0.3         data.table_1.14.2      reprex_2.0.1          
[100] mvtnorm_1.1-3          whisker_0.4            pkgload_1.2.4         
[103] hms_1.1.1              patchwork_1.1.1        mime_0.12             
[106] evaluate_0.15          xtable_1.8-4           broom.mixed_0.2.9.4   
[109] readxl_1.4.0           IRanges_2.24.1         gridExtra_2.3         
[112] compiler_4.0.5         ggstatsplot_0.9.1      crayon_1.5.1          
[115] minqa_1.2.4            htmltools_0.5.2        mc2d_0.1-21           
[118] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[121] lubridate_1.8.0        aws.s3_0.3.21          DBI_1.1.2             
[124] tweenr_1.0.2           dbplyr_2.1.1           MASS_7.3-57           
[127] boot_1.3-28            Matrix_1.4-1           ade4_1.7-19           
[130] car_3.0-12             wesanderson_0.3.6      permute_0.9-7         
[133] brio_1.1.3             cli_3.3.0              parallel_4.0.5        
[136] insight_0.17.0         igraph_1.3.1           pkgconfig_2.0.3       
[139] statsExpressions_1.3.1 paletteer_1.4.0        xml2_1.3.2            
[142] foreach_1.5.2          bslib_0.3.1            blme_1.0-5            
[145] multtest_2.46.0        XVector_0.30.0         estimability_1.3      
[148] rvest_1.0.2            callr_3.7.0            digest_0.6.29         
[151] parameters_0.17.0      vegan_2.6-2            Biostrings_2.58.0     
[154] rmarkdown_2.14         cellranger_1.1.0       curl_4.3.2            
[157] gtools_3.9.2           shiny_1.7.1            nloptr_2.0.0          
[160] lifecycle_1.0.1        nlme_3.1-157           jsonlite_1.8.0        
[163] Rhdf5lib_1.12.1        carData_3.0-5          desc_1.4.1            
[166] fansi_1.0.3            pillar_1.7.0           lattice_0.20-45       
[169] fastmap_1.1.0          httr_1.4.2             pkgbuild_1.3.1        
[172] survival_3.3-1         glue_1.6.2             remotes_2.4.2         
[175] bayestestR_0.11.5.1    iterators_1.0.14       bit_4.0.4             
[178] performance_0.9.0      ggforce_0.3.3          stringi_1.7.6         
[181] sass_0.4.1             rematch2_2.1.2         memoise_2.0.1         
[184] ape_5.6-2             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
