<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Carlos Blázquez Bondia" />


<title>Taxonomy: Differential Abundance</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.5.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Taxonomy: Differential Abundance</h1>
<h4 class="author">Carlos Blázquez Bondia</h4>
<h4 class="date">3/23/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef51189b8f584a80a001462d968318e82da32fc05targetblankf51189ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/f51189b8f584a80a001462d968318e82da32fc05" target="_blank">f51189b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treef51189b8f584a80a001462d968318e82da32fc05targetblankf51189ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/f51189b8f584a80a001462d968318e82da32fc05" target="_blank">f51189b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/mre.rds

Unstaged changes:
    Modified:   analysis/420_Taxonomy_DA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/420_Taxonomy_DA.Rmd</code>) and HTML (<code>docs/420_Taxonomy_DA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/f51189b8f584a80a001462d968318e82da32fc05/analysis/420_Taxonomy_DA.Rmd" target="_blank">f51189b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
started longitudinal DA section
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/analysis/420_Taxonomy_DA.Rmd" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
Added Venn diagram and plotted stats of spps with DA in all TPs (Differential Abundance)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/420_Taxonomy_DA.html" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
<td>
Added Venn diagram and plotted stats of spps with DA in all TPs (Differential Abundance)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/74908f2328b99c6d1324c42c33659bf2b41a4abc/analysis/420_Taxonomy_DA.Rmd" target="_blank">74908f2</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a98995620eb26b2dc78546dc4003a944581d2d21/analysis/420_Taxonomy_DA.Rmd" target="_blank">a989956</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/3f2dedbb7ecf47b42abf0d5de16eb89c999523e0/analysis/420_Taxonomy_DA.Rmd" target="_blank">3f2dedb</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/0ad10f57885cf1df5eb471c896caca7e216ac3e4/analysis/420_Taxonomy_DA.Rmd" target="_blank">0ad10f5</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-04-28
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/analysis/420_Taxonomy_DA.Rmd" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Advanced on longitudinal DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/docs/420_Taxonomy_DA.html" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
<td>
Advanced on longitudinal DA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c20090b3a3c8e714d8be91e4ea67aa5d3ff7c522/docs/420_Taxonomy_DA.html" target="_blank">c20090b</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-01
</td>
<td>
Compiled taxa ranking html + several bug fixes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/5917250ea4ce78208d3a41d2c612085024734455/analysis/420_Taxonomy_DA.Rmd" target="_blank">5917250</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-03-31
</td>
<td>
Update 420_Taxonomy_DA.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/e253850975c479f491140a3a60d9e54a3eca44ff/docs/420_Taxonomy_DA.html" target="_blank">e253850</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-29
</td>
<td>
worked on NMDS
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/c51687a773a5cb2f278af76950c55a802ed234cd/analysis/420_Taxonomy_DA.Rmd" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics/advanz4/c51687a773a5cb2f278af76950c55a802ed234cd/docs/420_Taxonomy_DA.html" target="_blank">c51687a</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
fixed workflowrs and gene richness issues
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/11c1e7d2e4c2b76ccb9e08a96305fa7f34ee48de/analysis/420_Taxonomy_DA.Rmd" target="_blank">11c1e7d</a>
</td>
<td>
cblazquez
</td>
<td>
2022-03-28
</td>
<td>
Code reorganized towards new structure
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import-and-exploration" class="section level2">
<h2>Data import and exploration</h2>
<p>This section should explain anyithing that may have happened during the first fase of importing and formatting data, such as any modification or additional filtering step.</p>
<pre class="r"><code># mymre &lt;- here::here(&quot;data&quot;,&quot;mre.rds&quot;) %&gt;%
#   readRDS(.)

mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;)
# metadata &lt;-get_meta(mymre)
metadata &lt;- here::here(&quot;Metadata&quot;, &quot;2022_04_28_clean_metadata_LIMS.csv&quot;) %&gt;% 
  read.csv() %&gt;% 
  column_to_rownames(&quot;SampleID&quot;) %&gt;% 
  mutate(SampleID = rownames(.))



cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

link_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)</code></pre>
<pre class="r"><code>#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here(&quot;../WMGSPipeline&quot;))</code></pre>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<pre class="r"><code>mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()</code></pre>
<pre class="r"><code># Any homebrew support function that may be written will be placed here</code></pre>
<p>First we’ll start with a non-compositional approach which may provide potential candidate taxa for DA to be confirmed with more astringent and specific compositional approaches.</p>
</div>
<div id="non-compositional-analysis" class="section level1">
<h1>Non-compositional analysis</h1>
<p>In this case, we’ll test for group as categorical variables. We may extend the analysis towards other cat_vars, but those were more important for basal comparisons, which already shown no great differences were found among other potential confusor variables.</p>
<div id="cross-group-differential-abundance" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.</p>
<pre class="r"><code>phy &lt;- get_phyloseq(mymre, type = &quot;metaphlan&quot;) %&gt;% 
      phyloseq::filter_taxa(., function(x) sum(x &gt; 0) &gt; (0.3*length(x)), TRUE)


otu_df &lt;- 
  phyloseq::otu_table(phy) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy)[,&quot;Species&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;))


stat_list &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    
    c(0,24,48,96) %&gt;% 
      set_names() %&gt;% 
      purrr::map_dfr(function(tp){
        otu_df %&gt;%
        dplyr::select(-SampleID) %&gt;%
        colnames() %&gt;%
        purrr::set_names() %&gt;%
        purrr::map_dfr(function(sp) {
          df &lt;-
            otu_df %&gt;%
            dplyr::select(SampleID, !!sym(sp)) %&gt;%
            dplyr::left_join(metadata[, c(&quot;SampleID&quot;, link_var, cv, long_var), drop = F], 
                             by = &quot;SampleID&quot;) %&gt;% 
            dplyr::filter(!!sym(long_var) %in% tp) 

          
          if (length(unique(dplyr::pull(.data = df, !!sym(cv)))) == 2) {
            test &lt;- df %&gt;%
              rstatix::wilcox_test(., formula(paste(sp, &quot;~&quot;, cv)), paired = F) %&gt;%
              dplyr::mutate(method = &quot;Wilcoxon&quot;) 
            
          } else{
            test &lt;- df %&gt;%
              rstatix::kruskal_test(., formula(paste(sp, &quot;~&quot;, cv))) %&gt;%
              dplyr::mutate(method = &quot;Kruskal-Wallis&quot;)
          }
          
        }) %&gt;% 
      rename(species = .y.) %&gt;% 
      dplyr::mutate(long_var = tp) %&gt;% 
      group_by(long_var) %&gt;% 
      mutate(p.adj = p.adjust(p, &quot;BH&quot;)) %&gt;% 
      # filter(p.adj &lt; .05 ) %&gt;% 
      group_by(species) 
      }) 
}) 


DAsp &lt;- stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species)


## --- init MNJ
## It is interesting that Bifidobacterium species (adolescentis, longum) are consistently found as DA in all (most) timepoints
## Produce a table showing 
stat_list$group %&gt;% 
  filter(p&lt;0.05) %&gt;% 
  group_by(species) %&gt;% 
  tally() %&gt;% 
  filter(n&gt;=2) ## Species being significant (Wilcoxon unadjusted &lt;0.05)</code></pre>
<pre><code># A tibble: 15 × 2
   species                          n
   &lt;chr&gt;                        &lt;int&gt;
 1 Bifidobacterium_adolescentis     3
 2 Bifidobacterium_longum           3
 3 Clostridium_citroniae            2
 4 Coprococcus_catus                2
 5 Eubacterium_eligens              2
 6 Eubacterium_siraeum              2
 7 Eubacterium_sp_CAG_180           2
 8 Firmicutes_bacterium_CAG_110     2
 9 Flavonifractor_plautii           2
10 Lachnospira_pectinoschiza        2
11 Olsenella_scatoligenes           3
12 Oscillibacter_sp_CAG_241         2
13 Parabacteroides_distasonis       2
14 Roseburia_faecis                 2
15 Roseburia_hominis                2</code></pre>
<pre class="r"><code># otu_df %&gt;% 
#   select(SampleID,Bifidobacterium_adolescentis) %&gt;% 
#   # select(SampleID, !!sym(DAsp)) %&gt;% 
#   left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
#   group_by(group) %&gt;% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x=&quot;group&quot;,
#                               y = Bifidobacterium_adolescentis,
#                               pairwise.comparisons = T,
#                               type=&quot;np&quot;,bf.message=NULL,
#                               ylab = &quot;Bifidobacterium_adolescentis(%)&quot;, 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = &quot;time_point&quot;,plotgrid.args = NULL)
# otu_df %&gt;% 
#   select(SampleID,Bifidobacterium_longum) %&gt;% 
#   # select(SampleID, !!sym(DAsp)) %&gt;% 
#   left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
#   group_by(group) %&gt;% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x=&quot;group&quot;,
#                               y = Bifidobacterium_longum,
#                               pairwise.comparisons = F,
#                               type=&quot;np&quot;,bf.message=NULL,
#                               ylab = &quot;Bifidobacterium_longum(%)&quot;, 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = &quot;time_point&quot;,
#                               plotgrid.args = list(&quot;ncol&quot; = 4, &quot;nrow&quot; = 1))
# 

 ### There is a nice signal, consistent longitudinally, in Bifidobacterium Genus
phyloseq::otu_table(phy %&gt;% phyloseq::tax_glom(.,taxrank=&quot;Genus&quot;)) %&gt;% 
  t() %&gt;% 
  as.data.frame() %&gt;% 
  stats::setNames(phyloseq::tax_table(phy %&gt;% phyloseq::tax_glom(.,taxrank=&quot;Genus&quot;))[,&quot;Genus&quot;]) %&gt;% 
  tibble::rownames_to_column(&quot;SampleID&quot;) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\[&quot;)) %&gt;% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), &quot;\\]&quot;)) %&gt;% 
  select(SampleID,Bifidobacterium) %&gt;% 
  # select(SampleID, !!sym(DAsp)) %&gt;% 
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  group_by(group) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Bifidobacterium,
                              pairwise.comparisons = F,
                              type=&quot;np&quot;, bf.message=NULL,
                              ylab = &quot;Bifidobacterium(%)&quot;, 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL) </code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-wilcox_across_group-1">
Past versions of wilcox_across_group-1.png
</button>
</p>
<div id="fig-wilcox_across_group-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/a143db0a6058bb3d8f7be3c55092d9a7fbb54f9f/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-1.png" target="_blank">a143db0</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># %&gt;% 
  # ggsave(.,filename = &quot;~/Downloads/plot4Carlos_Bifidobacterium.pdf&quot;)

otu_df %&gt;% 
  select(SampleID,Olsenella_scatoligenes) %&gt;% 
  # select(SampleID, !!sym(DAsp)) %&gt;% 
  left_join(., metadata[,c(&quot;SampleID&quot;,&quot;group&quot;,&quot;time_point&quot;)], by=&quot;SampleID&quot;) %&gt;% 
  group_by(group) %&gt;% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x=&quot;group&quot;,
                              y = Olsenella_scatoligenes,
                              pairwise.comparisons = F,
                              type=&quot;np&quot;,bf.message=NULL,
                              ylab = &quot;Olsenella_scatoligenes(%)&quot;, 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = &quot;time_point&quot;,plotgrid.args = NULL)</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/wilcox_across_group-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-wilcox_across_group-2">
Past versions of wilcox_across_group-2.png
</button>
</p>
<div id="fig-wilcox_across_group-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/wilcox_across_group-2.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>### Show how many species are differential in at least two of the timepoints in each cat var. Risk_group should result in many species showing that kins of signal. See Venn section below.
### --- end MNJ</code></pre>
<p>Only one spec</p>
<div id="comparison-of-significant-species" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
<pre class="r"><code>maxn &lt;- DAsp %&gt;% 
  select(species,long_var) %&gt;% 
  group_by(long_var) %&gt;% 
  count() %&gt;% 
  pull(n) %&gt;% 
  max


unique(DAsp$long_var) %&gt;%
  set_names() %&gt;%
  map_dfc(function(tp) {
    vec &lt;- DAsp %&gt;%
      dplyr::select(species, long_var) %&gt;%
      dplyr::filter(long_var == tp) %&gt;%
      dplyr::pull(species)
    
    if (length(vec) &lt; maxn) {
      vec &lt;- vctrs::vec_c(vec, rep(NA, maxn - length(vec)))
      return(vec)
    } else{
      return(vec)
    }
    
  }) %&gt;%
  ggVennDiagram::ggVennDiagram(., show_intersect = T)</code></pre>
<div id="htmlwidget-15c8782a12ce3241ec2f" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-15c8782a12ce3241ec2f">{"x":{"data":[{"x":[200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,null,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,null,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,null,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940,null,60,60.6060606060606,70.7070707070707,80.8080808080808,90.9090909090909,101.010101010101,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555555,565.656565656566,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585859,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,940],"y":[103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,103.6,111.111111111111,121.212121212121,131.313131313131,141.414141414141,151.515151515152,161.616161616162,171.717171717172,181.818181818182,191.919191919192,202.020202020202,212.121212121212,222.222222222222,232.323232323232,242.424242424242,252.525252525253,262.626262626263,272.727272727273,282.828282828283,292.929292929293,303.030303030303,313.131313131313,323.232323232323,333.333333333333,343.434343434343,353.535353535354,363.636363636364,373.737373737374,383.838383838384,393.939393939394,404.040404040404,414.141414141414,424.242424242424,434.343434343434,444.444444444444,454.545454545455,464.646464646465,474.747474747475,484.848484848485,494.949494949495,505.050505050505,515.151515151515,525.252525252525,535.353535353535,545.454545454545,555.555555555556,565.656565656565,575.757575757576,585.858585858586,595.959595959596,606.060606060606,616.161616161616,626.262626262626,636.363636363636,646.464646464646,656.565656565656,666.666666666667,676.767676767677,686.868686868687,696.969696969697,707.070707070707,717.171717171717,727.272727272727,737.373737373737,747.474747474747,757.575757575758,767.676767676768,777.777777777778,787.878787878788,797.979797979798,808.080808080808,818.181818181818,828.282828282828,838.383838383838,848.484848484848,858.585858585858,868.686868686869,878.787878787879,888.888888888889,898.989898989899,909.090909090909,919.191919191919,929.292929292929,939.393939393939,948.4,null,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,null,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,400,null,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,600,null,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800,800],"type":"scatter","mode":"lines","line":{"width":null,"color":"transparent","dash":null},"hoveron":"points","hoverinfo":"none","showlegend":false,"_isGraticule":true,"xaxis":"x","yaxis":"y","frame":null},{"x":[368.395,361.896,355.51,349.244,343.105,337.099,331.234,325.516,319.951,314.547,309.311,304.248,299.366,294.673,290.173,285.875,281.785,277.909,274.256,270.83,267.64,264.691,261.991,259.547,257.365,255.451,253.814,252.459,251.393,250.624,250.157,250,257.251,264.598,272.037,279.56,287.162,294.837,302.579,310.381,318.238,326.143,334.091,342.074,350.088,358.126,366.182,374.25,382.324,390.397,398.465,406.52,414.556,422.568,430.549,438.494,446.396,454.249,462.047,469.784,477.454,485.05,492.568,500,492.821,485.819,478.997,472.357,465.905,459.643,453.574,447.703,442.033,436.566,431.307,426.26,421.427,416.812,412.418,408.25,404.31,400.602,397.13,393.896,390.905,388.16,385.665,383.422,381.435,379.709,378.245,377.049,376.123,375.47,375.095,375,368.395,368.395],"y":[395.41,399.546,403.901,408.474,413.259,418.253,423.452,428.852,434.448,440.238,446.218,452.382,458.727,465.25,471.947,478.812,485.844,493.037,500.387,507.892,515.546,523.346,531.287,539.367,547.581,555.925,564.395,572.987,581.698,590.523,599.458,608.5,612.526,616.296,619.81,623.069,626.07,628.815,631.302,633.531,635.502,637.214,638.666,639.859,640.792,641.464,641.875,642.025,641.913,641.538,640.901,640,638.836,637.407,635.714,633.756,631.533,629.044,626.288,623.266,619.976,616.419,612.594,608.5,604.167,599.609,594.832,589.845,584.652,579.262,573.68,567.914,561.97,555.854,549.574,543.136,536.546,529.812,522.941,515.938,508.81,501.565,494.208,486.747,479.188,471.538,463.804,455.992,448.109,440.162,432.157,424.101,416,407.862,399.693,391.5,395.41,395.41],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(19,43,67,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[250.081,250.323,250.723,251.281,251.993,252.858,253.875,255.041,256.354,257.813,259.416,261.16,265.066,269.516,274.494,279.987,285.977,292.451,299.393,306.788,314.62,322.875,331.537,340.591,350.022,359.814,369.953,380.424,391.21,402.297,407.948,413.67,419.458,425.313,431.231,437.211,443.251,449.35,455.504,461.713,467.974,474.286,480.646,487.053,493.505,500,506.495,512.947,519.354,525.714,532.026,538.287,544.496,550.65,556.749,562.789,568.769,574.687,580.542,586.33,592.052,597.703,608.79,619.576,630.047,640.186,649.978,659.409,668.463,677.125,685.38,693.212,700.607,707.549,714.023,720.013,725.506,730.484,734.934,738.84,740.584,742.187,743.646,744.959,746.125,747.142,748.007,748.719,749.277,749.677,749.919,750,742.749,735.402,727.963,720.44,712.838,705.163,697.421,689.619,681.762,673.857,665.909,657.926,649.912,641.874,633.818,625.75,617.676,609.603,601.535,593.48,585.444,577.432,569.451,561.506,553.604,545.751,537.953,530.216,522.546,514.95,507.432,500,492.568,485.05,477.454,469.784,462.047,454.249,446.396,438.494,430.549,422.568,414.556,406.52,398.465,390.397,382.324,374.25,366.182,358.126,350.088,342.074,334.091,326.143,318.238,310.381,302.579,294.837,287.162,279.56,272.037,264.598,257.251,250,250.081,250.081,null,507.161,514.147,520.955,527.581,534.022,540.274,546.334,552.198,557.863,563.325,568.581,573.626,578.458,583.073,587.467,591.638,595.58,599.292,602.768,606.007,609.004,611.755,614.258,616.508,618.502,620.237,621.709,622.914,623.85,624.511,624.896,625,631.605,638.104,644.49,650.756,656.895,662.901,668.766,674.484,680.049,685.453,690.689,695.752,700.634,705.327,709.827,714.125,718.215,722.091,725.744,729.17,732.36,735.309,738.009,740.453,742.635,744.549,746.186,747.541,748.607,749.376,749.843,750,757.051,764.02,770.897,777.67,784.33,790.864,797.262,803.513,809.605,815.528,821.271,826.823,832.173,837.31,842.223,846.9,850.167,853.251,856.149,858.858,861.372,863.689,865.804,867.712,869.412,870.897,872.165,873.211,874.031,874.622,874.98,875.1,875.018,874.774,874.371,873.808,873.09,872.218,871.193,870.018,868.694,867.225,865.61,863.854,861.957,859.921,857.749,855.442,850.433,844.909,838.886,832.38,825.406,817.981,810.119,801.837,793.151,784.076,774.627,764.822,754.674,744.201,733.417,722.339,716.694,710.982,705.204,699.362,693.459,687.495,681.474,675.396,669.265,663.082,656.848,650.567,644.239,637.868,631.454,625,617.032,609.067,601.11,593.164,585.233,577.319,569.428,561.562,553.726,545.923,538.156,530.43,522.747,515.112,507.529,500,507.161,507.161,null,375.095,375.47,376.123,377.049,378.245,379.709,381.435,383.422,385.665,388.16,390.905,393.896,397.13,400.602,404.31,408.25,412.418,416.812,421.427,426.26,431.307,436.566,442.033,447.703,453.574,459.643,465.905,472.357,478.997,485.819,492.821,500,507.179,514.181,521.003,527.643,534.095,540.357,546.426,552.297,557.967,563.434,568.693,573.74,578.573,583.188,587.582,591.75,595.69,599.398,602.87,606.104,609.095,611.84,614.335,616.578,618.565,620.291,621.755,622.951,623.877,624.53,624.905,625,617.854,610.579,603.182,595.674,588.061,580.354,572.56,564.689,556.749,548.748,540.696,532.601,524.471,516.316,508.143,499.962,491.782,483.611,475.457,467.329,459.236,451.187,443.191,435.255,427.388,419.6,411.899,404.294,396.792,389.404,382.137,375,375.095,375.095],"y":[614.835,621.137,627.403,633.632,639.82,645.966,652.068,658.124,664.131,670.088,675.992,681.841,693.367,704.648,715.668,726.41,736.856,746.989,756.794,766.252,775.347,784.062,792.381,800.285,807.758,814.784,821.345,827.424,833.005,838.07,840.404,842.603,844.664,846.586,848.367,850.003,851.494,852.837,854.03,855.07,855.957,856.687,857.258,857.669,857.917,858,857.917,857.669,857.258,856.687,855.957,855.07,854.03,852.837,851.494,850.003,848.367,846.586,844.664,842.603,840.404,838.07,833.005,827.424,821.345,814.784,807.758,800.285,792.381,784.062,775.347,766.252,756.794,746.989,736.856,726.41,715.668,704.648,693.367,681.841,675.992,670.088,664.131,658.124,652.068,645.966,639.82,633.632,627.403,621.137,614.835,608.5,612.526,616.296,619.81,623.069,626.07,628.815,631.302,633.531,635.502,637.214,638.666,639.859,640.792,641.464,641.875,642.025,641.913,641.538,640.901,640,638.836,637.407,635.714,633.756,631.533,629.044,626.288,623.266,619.976,616.419,612.594,608.5,612.594,616.419,619.976,623.266,626.288,629.044,631.533,633.756,635.714,637.407,638.836,640,640.901,641.538,641.913,642.025,641.875,641.464,640.792,639.859,638.666,637.214,635.502,633.531,631.302,628.815,626.07,623.069,619.81,616.296,612.526,608.5,614.835,614.835,null,179.752,184.234,188.94,193.862,198.994,204.329,209.859,215.578,221.479,227.555,233.8,240.205,246.765,253.472,260.319,267.3,274.407,281.635,288.974,296.42,303.964,311.601,319.322,327.122,334.993,342.928,350.92,358.963,367.049,375.172,383.325,391.5,395.41,399.546,403.901,408.474,413.259,418.253,423.452,428.852,434.448,440.238,446.218,452.382,458.727,465.25,471.947,478.812,485.844,493.037,500.387,507.892,515.546,523.346,531.287,539.367,547.581,555.925,564.395,572.987,581.698,590.523,599.458,608.5,604.194,599.635,594.82,589.744,584.403,578.793,572.91,566.75,560.309,553.582,546.566,539.256,531.649,523.74,515.525,507,500.488,493.855,487.107,480.252,473.293,466.238,459.093,451.862,444.554,437.173,429.725,422.217,414.655,407.044,399.39,391.7,385.255,378.851,372.489,366.171,359.898,353.674,347.5,341.378,335.31,329.298,323.344,317.449,311.616,305.847,300.144,294.508,283.447,272.681,262.224,252.093,242.303,232.87,223.81,215.138,206.869,199.021,191.608,184.646,178.151,172.139,166.624,161.623,159.321,157.152,155.12,153.226,151.473,149.862,148.394,147.073,145.9,144.877,144.006,143.289,142.728,142.325,142.082,142,142.124,142.496,143.121,144,145.137,146.535,148.197,150.125,152.323,154.793,157.539,160.562,163.868,167.457,171.333,175.5,179.752,179.752,null,399.693,407.862,416,424.101,432.157,440.162,448.109,455.992,463.804,471.538,479.188,486.747,494.208,501.565,508.81,515.938,522.941,529.812,536.546,543.136,549.574,555.854,561.97,567.914,573.68,579.262,584.652,589.845,594.832,599.609,604.167,608.5,604.167,599.609,594.832,589.845,584.652,579.262,573.68,567.914,561.97,555.854,549.574,543.136,536.546,529.812,522.941,515.938,508.81,501.565,494.208,486.747,479.188,471.538,463.804,455.992,448.109,440.162,432.157,424.101,416,407.862,399.693,391.5,387.458,383.678,380.158,376.898,373.9,371.162,368.685,366.469,364.513,362.818,361.384,360.211,359.298,358.646,358.255,358.125,358.255,358.646,359.298,360.211,361.384,362.818,364.513,366.469,368.685,371.162,373.9,376.898,380.158,383.678,387.458,391.5,399.693,399.693],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(29,63,94,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[485.853,479.045,472.419,465.978,459.726,453.666,447.802,442.137,436.675,431.419,426.374,421.542,416.927,412.533,408.363,404.42,400.708,397.232,393.993,390.996,388.245,385.742,383.492,381.498,379.763,378.291,377.086,376.15,375.489,375.104,375,382.137,389.404,396.792,404.294,411.899,419.6,427.388,435.255,443.191,451.187,459.236,467.329,475.457,483.611,491.782,499.962,508.143,516.316,524.471,532.601,540.696,548.748,556.749,564.689,572.56,580.354,588.061,595.674,603.182,610.579,617.854,625,624.896,624.511,623.85,622.914,621.709,620.237,618.502,616.508,614.258,611.755,609.004,606.007,602.768,599.292,595.58,591.638,587.467,583.073,578.458,573.626,568.581,563.325,557.863,552.198,546.334,540.274,534.022,527.581,520.955,514.147,507.161,500,492.839,485.853,485.853],"y":[184.234,188.94,193.862,198.994,204.329,209.859,215.578,221.479,227.555,233.8,240.205,246.765,253.472,260.319,267.3,274.407,281.635,288.974,296.42,303.964,311.601,319.322,327.122,334.993,342.928,350.92,358.963,367.049,375.172,383.325,391.5,387.458,383.678,380.158,376.898,373.9,371.162,368.685,366.469,364.513,362.818,361.384,360.211,359.298,358.646,358.255,358.125,358.255,358.646,359.298,360.211,361.384,362.818,364.513,366.469,368.685,371.162,373.9,376.898,380.158,383.678,387.458,391.5,383.325,375.172,367.049,358.963,350.92,342.928,334.993,327.122,319.322,311.601,303.964,296.42,288.974,281.635,274.407,267.3,260.319,253.472,246.765,240.205,233.8,227.555,221.479,215.578,209.859,204.329,198.994,193.862,188.94,184.234,179.752,175.5,179.752,184.234,184.234],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(40,84,122,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[624.905,624.53,623.877,622.951,621.755,620.291,618.565,616.578,614.335,611.84,609.095,606.104,602.87,599.398,595.69,591.75,587.582,583.188,578.573,573.74,568.693,563.434,557.967,552.297,546.426,540.357,534.095,527.643,521.003,514.181,507.179,500,507.432,514.95,522.546,530.216,537.953,545.751,553.604,561.506,569.451,577.432,585.444,593.48,601.535,609.603,617.676,625.75,633.818,641.874,649.912,657.926,665.909,673.857,681.762,689.619,697.421,705.163,712.838,720.44,727.963,735.402,742.749,750,749.843,749.376,748.607,747.541,746.186,744.549,742.635,740.453,738.009,735.309,732.36,729.17,725.744,722.091,718.215,714.125,709.827,705.327,700.634,695.752,690.689,685.453,680.049,674.484,668.766,662.901,656.895,650.756,644.49,638.104,631.605,625,624.905,624.905],"y":[399.693,407.862,416,424.101,432.157,440.162,448.109,455.992,463.804,471.538,479.188,486.747,494.208,501.565,508.81,515.938,522.941,529.812,536.546,543.136,549.574,555.854,561.97,567.914,573.68,579.262,584.652,589.845,594.832,599.609,604.167,608.5,612.594,616.419,619.976,623.266,626.288,629.044,631.533,633.756,635.714,637.407,638.836,640,640.901,641.538,641.913,642.025,641.875,641.464,640.792,639.859,638.666,637.214,635.502,633.531,631.302,628.815,626.07,623.069,619.81,616.296,612.526,608.5,599.458,590.523,581.698,572.987,564.395,555.925,547.581,539.367,531.287,523.346,515.546,507.892,500.387,493.037,485.844,478.812,471.947,465.25,458.727,452.382,446.218,440.238,434.448,428.852,423.452,418.253,413.259,408.474,403.901,399.546,395.41,391.5,399.693,399.693],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(74,153,214,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[124.9,125.02,125.378,125.969,126.789,127.835,129.103,130.588,132.287,134.196,136.311,138.628,141.142,143.851,146.749,149.833,153.1,157.777,162.69,167.827,173.177,178.729,184.472,190.395,196.488,202.738,209.136,215.67,222.33,229.103,235.98,242.949,250,250.157,250.624,251.393,252.459,253.814,255.451,257.365,259.547,261.991,264.691,267.64,270.83,274.256,277.909,281.785,285.875,290.173,294.673,299.366,304.248,309.311,314.547,319.951,325.516,331.234,337.099,343.105,349.244,355.51,361.896,368.395,375,375.104,375.489,376.15,377.086,378.291,379.763,381.498,383.492,385.742,388.245,390.996,393.993,397.232,400.708,404.42,408.363,412.533,416.927,421.542,426.374,431.419,436.675,442.137,447.802,453.666,459.726,465.978,472.419,479.045,485.853,492.839,500,492.471,484.888,477.253,469.57,461.844,454.077,446.274,438.438,430.572,422.681,414.767,406.836,398.89,390.933,382.968,375,368.546,362.132,355.761,349.433,343.152,336.918,330.735,324.604,318.526,312.505,306.541,300.638,294.796,289.018,283.306,277.661,266.583,255.799,245.326,235.178,225.373,215.924,206.849,198.162,189.881,182.019,174.594,167.62,161.114,155.091,149.567,144.558,142.251,140.079,138.043,136.146,134.39,132.775,131.306,129.982,128.807,127.782,126.91,126.192,125.629,125.226,124.982,124.9,124.9],"y":[391.7,399.39,407.044,414.655,422.217,429.725,437.173,444.554,451.862,459.093,466.238,473.293,480.252,487.107,493.855,500.488,507,515.525,523.74,531.649,539.256,546.566,553.582,560.309,566.75,572.91,578.793,584.403,589.744,594.82,599.635,604.194,608.5,599.458,590.523,581.698,572.987,564.395,555.925,547.581,539.367,531.287,523.346,515.546,507.892,500.387,493.037,485.844,478.812,471.947,465.25,458.727,452.382,446.218,440.238,434.448,428.852,423.452,418.253,413.259,408.474,403.901,399.546,395.41,391.5,383.325,375.172,367.049,358.963,350.92,342.928,334.993,327.122,319.322,311.601,303.964,296.42,288.974,281.635,274.407,267.3,260.319,253.472,246.765,240.205,233.8,227.555,221.479,215.578,209.859,204.329,198.994,193.862,188.94,184.234,179.752,175.5,171.333,167.457,163.868,160.562,157.539,154.793,152.323,150.125,148.197,146.535,145.137,144,143.121,142.496,142.124,142,142.082,142.325,142.728,143.289,144.006,144.877,145.9,147.073,148.394,149.862,151.473,153.226,155.12,157.152,159.321,161.623,166.624,172.139,178.151,184.646,191.608,199.021,206.869,215.138,223.81,232.87,242.303,252.093,262.224,272.681,283.447,294.508,300.144,305.847,311.616,317.449,323.344,329.298,335.31,341.378,347.5,353.674,359.898,366.171,372.489,378.851,385.255,391.7,391.7],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(89,89,89,1)","dash":"solid"},"fill":"toself","fillcolor":"rgba(86,177,247,1)","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[100,500,900],"y":[560,910,560],"text":["24","48","96"],"hovertext":["","",""],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[272.600353678248,500,727.399643854318,353.366830257892,500.002640315342,646.633169742108,499.998061434311],"y":[332.713058584906,726.385139817866,332.713082359295,548.571186362059,294.575969186258,548.571186362059,463.758078927749],"text":[8,3,3,2,4,7,3],"hovertext":["Prevotella_sp_AM42_24<br />Prevotella_stercorea Clostridium_bolteae<br />Roseburia_inulinivorans<br />Oscillibacter_sp_57_20<br />Erysipelatoclostridium_ramosum<br />Firmicutes_bacterium_CAG_791<br />Desulfovibrio_piger","Collinsella_stercoris<br />Bacteroides_vulgatus<br />Fusicatenibacter_saccharivorans","Collinsella_massiliensis<br />Blautia_wexlerae Ruminococcus_bromii","Parabacteroides_distasonis<br />Oscillibacter_sp_CAG_241","Eubacterium_eligens Coprococcus_catus<br />Lachnospira_pectinoschiza<br />Eubacterium_siraeum","Eubacterium_sp_CAG_180<br />Clostridium_citroniae Roseburia_faecis<br />Roseburia_hominis Flavonifractor_plautii<br />Firmicutes_bacterium_CAG_110 NA","Bifidobacterium_adolescentis<br />Bifidobacterium_longum<br />Olsenella_scatoligenes"],"textfont":{"size":14.6645669291339,"color":"rgba(0,0,0,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[200],"y":[200],"name":"99_4fd2bcaee5e464335ae6ff05378d2c88","type":"scatter","mode":"markers","opacity":0,"hoverinfo":"skip","showlegend":false,"marker":{"color":[0,1],"colorscale":[[0,"#132B43"],[0.00334448160535117,"#132B44"],[0.00668896321070234,"#132C44"],[0.0100334448160535,"#142C45"],[0.0133779264214047,"#142D45"],[0.0167224080267558,"#142D46"],[0.020066889632107,"#142D46"],[0.0234113712374582,"#142E47"],[0.0267558528428093,"#152E47"],[0.0301003344481605,"#152F48"],[0.0334448160535117,"#152F48"],[0.0367892976588629,"#152F49"],[0.040133779264214,"#153049"],[0.0434782608695652,"#16304A"],[0.0468227424749164,"#16304A"],[0.0501672240802676,"#16314B"],[0.0535117056856187,"#16314B"],[0.0568561872909699,"#16324C"],[0.060200668896321,"#17324D"],[0.0635451505016723,"#17324D"],[0.0668896321070234,"#17334E"],[0.0702341137123746,"#17334E"],[0.0735785953177258,"#17344F"],[0.0769230769230769,"#18344F"],[0.0802675585284281,"#183450"],[0.0836120401337793,"#183550"],[0.0869565217391304,"#183551"],[0.0903010033444816,"#183651"],[0.0936454849498328,"#193652"],[0.096989966555184,"#193652"],[0.100334448160535,"#193753"],[0.103678929765886,"#193754"],[0.107023411371237,"#193854"],[0.110367892976589,"#1A3855"],[0.11371237458194,"#1A3955"],[0.117056856187291,"#1A3956"],[0.120401337792642,"#1A3956"],[0.123745819397993,"#1A3A57"],[0.127090301003344,"#1B3A57"],[0.130434782608696,"#1B3B58"],[0.133779264214047,"#1B3B59"],[0.137123745819398,"#1B3B59"],[0.140468227424749,"#1C3C5A"],[0.1438127090301,"#1C3C5A"],[0.147157190635452,"#1C3D5B"],[0.150501672240803,"#1C3D5B"],[0.153846153846154,"#1C3D5C"],[0.157190635451505,"#1D3E5C"],[0.160535117056856,"#1D3E5D"],[0.163879598662207,"#1D3F5D"],[0.167224080267559,"#1D3F5E"],[0.17056856187291,"#1D3F5F"],[0.173913043478261,"#1E405F"],[0.177257525083612,"#1E4060"],[0.180602006688963,"#1E4160"],[0.183946488294314,"#1E4161"],[0.187290969899666,"#1E4261"],[0.190635451505017,"#1F4262"],[0.193979933110368,"#1F4263"],[0.197324414715719,"#1F4363"],[0.20066889632107,"#1F4364"],[0.204013377926421,"#1F4464"],[0.207357859531773,"#204465"],[0.210702341137124,"#204465"],[0.214046822742475,"#204566"],[0.217391304347826,"#204566"],[0.220735785953177,"#214667"],[0.224080267558528,"#214668"],[0.22742474916388,"#214768"],[0.230769230769231,"#214769"],[0.234113712374582,"#214769"],[0.237458193979933,"#22486A"],[0.240802675585284,"#22486A"],[0.244147157190636,"#22496B"],[0.247491638795987,"#22496C"],[0.250836120401338,"#224A6C"],[0.254180602006689,"#234A6D"],[0.25752508361204,"#234A6D"],[0.260869565217391,"#234B6E"],[0.264214046822743,"#234B6E"],[0.267558528428094,"#244C6F"],[0.270903010033445,"#244C70"],[0.274247491638796,"#244C70"],[0.277591973244147,"#244D71"],[0.280936454849498,"#244D71"],[0.28428093645485,"#254E72"],[0.287625418060201,"#254E72"],[0.290969899665552,"#254F73"],[0.294314381270903,"#254F74"],[0.297658862876254,"#254F74"],[0.301003344481605,"#265075"],[0.304347826086957,"#265075"],[0.307692307692308,"#265176"],[0.311036789297659,"#265176"],[0.31438127090301,"#275277"],[0.317725752508361,"#275278"],[0.321070234113712,"#275278"],[0.324414715719064,"#275379"],[0.327759197324415,"#275379"],[0.331103678929766,"#28547A"],[0.334448160535117,"#28547B"],[0.337792642140468,"#28557B"],[0.341137123745819,"#28557C"],[0.344481605351171,"#28567C"],[0.347826086956522,"#29567D"],[0.351170568561873,"#29567D"],[0.354515050167224,"#29577E"],[0.357859531772575,"#29577F"],[0.361204013377926,"#2A587F"],[0.364548494983278,"#2A5880"],[0.367892976588629,"#2A5980"],[0.37123745819398,"#2A5981"],[0.374581939799331,"#2A5982"],[0.377926421404682,"#2B5A82"],[0.381270903010033,"#2B5A83"],[0.384615384615385,"#2B5B83"],[0.387959866220736,"#2B5B84"],[0.391304347826087,"#2C5C85"],[0.394648829431438,"#2C5C85"],[0.397993311036789,"#2C5D86"],[0.40133779264214,"#2C5D86"],[0.404682274247492,"#2C5D87"],[0.408026755852843,"#2D5E87"],[0.411371237458194,"#2D5E88"],[0.414715719063545,"#2D5F89"],[0.418060200668896,"#2D5F89"],[0.421404682274247,"#2E608A"],[0.424749163879599,"#2E608A"],[0.42809364548495,"#2E618B"],[0.431438127090301,"#2E618C"],[0.434782608695652,"#2E618C"],[0.438127090301003,"#2F628D"],[0.441471571906354,"#2F628D"],[0.444816053511706,"#2F638E"],[0.448160535117057,"#2F638F"],[0.451505016722408,"#30648F"],[0.454849498327759,"#306490"],[0.45819397993311,"#306590"],[0.461538461538462,"#306591"],[0.464882943143813,"#306592"],[0.468227424749164,"#316692"],[0.471571906354515,"#316693"],[0.474916387959866,"#316793"],[0.478260869565217,"#316794"],[0.481605351170569,"#326895"],[0.48494983277592,"#326895"],[0.488294314381271,"#326996"],[0.491638795986622,"#326996"],[0.494983277591973,"#326997"],[0.498327759197324,"#336A98"],[0.501672240802676,"#336A98"],[0.505016722408027,"#336B99"],[0.508361204013378,"#336B99"],[0.511705685618729,"#346C9A"],[0.51505016722408,"#346C9B"],[0.518394648829431,"#346D9B"],[0.521739130434783,"#346D9C"],[0.525083612040134,"#346E9D"],[0.528428093645485,"#356E9D"],[0.531772575250836,"#356E9E"],[0.535117056856187,"#356F9E"],[0.538461538461538,"#356F9F"],[0.54180602006689,"#3670A0"],[0.545150501672241,"#3670A0"],[0.548494983277592,"#3671A1"],[0.551839464882943,"#3671A1"],[0.555183946488294,"#3772A2"],[0.558528428093645,"#3772A3"],[0.561872909698997,"#3773A3"],[0.565217391304348,"#3773A4"],[0.568561872909699,"#3773A4"],[0.57190635451505,"#3874A5"],[0.575250836120401,"#3874A6"],[0.578595317725752,"#3875A6"],[0.581939799331104,"#3875A7"],[0.585284280936455,"#3976A8"],[0.588628762541806,"#3976A8"],[0.591973244147157,"#3977A9"],[0.595317725752508,"#3977A9"],[0.598662207357859,"#3978AA"],[0.602006688963211,"#3A78AB"],[0.605351170568562,"#3A79AB"],[0.608695652173913,"#3A79AC"],[0.612040133779264,"#3A79AC"],[0.615384615384615,"#3B7AAD"],[0.618729096989967,"#3B7AAE"],[0.622073578595318,"#3B7BAE"],[0.625418060200669,"#3B7BAF"],[0.62876254180602,"#3C7CB0"],[0.632107023411371,"#3C7CB0"],[0.635451505016722,"#3C7DB1"],[0.638795986622073,"#3C7DB1"],[0.642140468227425,"#3C7EB2"],[0.645484949832776,"#3D7EB3"],[0.648829431438127,"#3D7FB3"],[0.652173913043478,"#3D7FB4"],[0.655518394648829,"#3D7FB5"],[0.658862876254181,"#3E80B5"],[0.662207357859532,"#3E80B6"],[0.665551839464883,"#3E81B6"],[0.668896321070234,"#3E81B7"],[0.672240802675585,"#3F82B8"],[0.675585284280936,"#3F82B8"],[0.678929765886288,"#3F83B9"],[0.682274247491639,"#3F83BA"],[0.68561872909699,"#4084BA"],[0.688963210702341,"#4084BB"],[0.692307692307692,"#4085BB"],[0.695652173913043,"#4085BC"],[0.698996655518395,"#4086BD"],[0.702341137123746,"#4186BD"],[0.705685618729097,"#4186BE"],[0.709030100334448,"#4187BF"],[0.712374581939799,"#4187BF"],[0.71571906354515,"#4288C0"],[0.719063545150502,"#4288C1"],[0.722408026755853,"#4289C1"],[0.725752508361204,"#4289C2"],[0.729096989966555,"#438AC2"],[0.732441471571906,"#438AC3"],[0.735785953177257,"#438BC4"],[0.739130434782609,"#438BC4"],[0.74247491638796,"#438CC5"],[0.745819397993311,"#448CC6"],[0.749163879598662,"#448DC6"],[0.752508361204013,"#448DC7"],[0.755852842809365,"#448EC8"],[0.759197324414716,"#458EC8"],[0.762541806020067,"#458FC9"],[0.765886287625418,"#458FC9"],[0.769230769230769,"#458FCA"],[0.77257525083612,"#4690CB"],[0.775919732441472,"#4690CB"],[0.779264214046823,"#4691CC"],[0.782608695652174,"#4691CD"],[0.785953177257525,"#4792CD"],[0.789297658862876,"#4792CE"],[0.792642140468227,"#4793CF"],[0.795986622073579,"#4793CF"],[0.79933110367893,"#4894D0"],[0.802675585284281,"#4894D0"],[0.806020066889632,"#4895D1"],[0.809364548494983,"#4895D2"],[0.812709030100334,"#4896D2"],[0.816053511705686,"#4996D3"],[0.819397993311037,"#4997D4"],[0.822742474916388,"#4997D4"],[0.826086956521739,"#4998D5"],[0.82943143812709,"#4A98D6"],[0.832775919732441,"#4A99D6"],[0.836120401337793,"#4A99D7"],[0.839464882943144,"#4A9AD8"],[0.842809364548495,"#4B9AD8"],[0.846153846153846,"#4B9BD9"],[0.849498327759197,"#4B9BDA"],[0.852842809364548,"#4B9BDA"],[0.8561872909699,"#4C9CDB"],[0.859531772575251,"#4C9CDB"],[0.862876254180602,"#4C9DDC"],[0.866220735785953,"#4C9DDD"],[0.869565217391304,"#4D9EDD"],[0.872909698996656,"#4D9EDE"],[0.876254180602007,"#4D9FDF"],[0.879598662207358,"#4D9FDF"],[0.882943143812709,"#4DA0E0"],[0.88628762541806,"#4EA0E1"],[0.889632107023411,"#4EA1E1"],[0.892976588628763,"#4EA1E2"],[0.896321070234114,"#4EA2E3"],[0.899665551839465,"#4FA2E3"],[0.903010033444816,"#4FA3E4"],[0.906354515050167,"#4FA3E5"],[0.909698996655518,"#4FA4E5"],[0.91304347826087,"#50A4E6"],[0.916387959866221,"#50A5E7"],[0.919732441471572,"#50A5E7"],[0.923076923076923,"#50A6E8"],[0.926421404682274,"#51A6E8"],[0.929765886287625,"#51A7E9"],[0.933110367892977,"#51A7EA"],[0.936454849498328,"#51A8EA"],[0.939799331103679,"#52A8EB"],[0.94314381270903,"#52A9EC"],[0.946488294314381,"#52A9EC"],[0.949832775919732,"#52AAED"],[0.953177257525084,"#53AAEE"],[0.956521739130435,"#53ABEE"],[0.959866220735786,"#53ABEF"],[0.963210702341137,"#53ACF0"],[0.966555183946488,"#54ACF0"],[0.96989966555184,"#54ADF1"],[0.973244147157191,"#54ADF2"],[0.976588628762542,"#54AEF2"],[0.979933110367893,"#55AEF3"],[0.983277591973244,"#55AFF4"],[0.986622073578595,"#55AFF4"],[0.989966555183947,"#55B0F5"],[0.993311036789298,"#56B0F6"],[0.996655518394649,"#56B1F6"],[1,"#56B1F7"]],"colorbar":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"thickness":23.04,"title":"count","titlefont":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"tickmode":"array","ticktext":["2","3","4","5","6","7","8"],"tickvals":[0,0.166666666666667,0.333333333333333,0.5,0.666666666666667,0.833333333333333,1],"tickfont":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"ticklen":2,"len":0.5}},"xaxis":"x","yaxis":"y","frame":null}],"layout":{"margin":{"t":16,"r":0,"b":0,"l":0},"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[60,940],"tickmode":"array","ticktext":[" 200"," 400"," 600"," 800"],"tickvals":[200,400,600,800],"categoryorder":"array","categoryarray":[" 200"," 400"," 600"," 800"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"y","scaleratio":1,"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[103.6,948.4],"tickmode":"array","ticktext":[" 200"," 400"," 600"," 800"],"tickvals":[200,400,600,800],"categoryorder":"array","categoryarray":[" 200"," 400"," 600"," 800"],"nticks":null,"ticks":"outside","tickcolor":null,"ticklen":0,"tickwidth":0,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":{"text":"","font":{"color":null,"family":null,"size":0}},"scaleanchor":"x","scaleratio":1,"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"title":{"text":"","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"93d057f1f03e2":{"fill":{},"type":"scatter"},"93d056d9e9443":{"label":{}},"93d05d54775e":{"label":{},"text":{}}},"cur_data":"93d057f1f03e2","visdat":{"93d057f1f03e2":["function (y) ","x"],"93d056d9e9443":["function (y) ","x"],"93d05d54775e":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>We can observe there are 3 species which are consistently differentially abundant between treatment groups at every point during treatment, but not at basal. It’s interesting to note that of those 3, 2 are from the bifidobacterium genera (<em>B. adolescentis</em> and <em>B.longum</em>) plus another actinobacteria: <em>Olsenella scatoligenes</em>, all of the increasing more in the DTG group although the later has little literature regarding their role on the microbiome. (hover mouse over the figure numbers to display the names of the species)</p>
<pre class="r"><code>shared_sp &lt;- 
  DAsp %&gt;% 
  select(species, long_var) %&gt;% 
  group_by(species) %&gt;% 
  count() %&gt;% 
  filter(n==3) %&gt;% 
  pull(species)


shared_sp %&gt;% 
  set_names() %&gt;% 
  purrr::map(function(sp){
    otu_df %&gt;% 
      select(SampleID, sp) %&gt;% 
      left_join(metadata[,c(&quot;SampleID&quot;, &quot;group&quot;, long_var)], by=&quot;SampleID&quot;) %&gt;% 
      group_by(!!sym(long_var)) %&gt;% 
      ggstatsplot::grouped_ggbetweenstats(., 
                                          type = &quot;np&quot;,
                                          y = !!sym(sp), 
                                          x=&quot;group&quot;, 
                                          grouping.var = !!sym(long_var))
      
  })</code></pre>
<pre><code>$Bifidobacterium_adolescentis</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-1">
Past versions of “plot_shared species-1.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-1.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Bifidobacterium_longum</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-2">
Past versions of “plot_shared species-2.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-2.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
$Olsenella_scatoligenes</code></pre>
<p><img src="figure/420_Taxonomy_DA.Rmd/plot_shared%20species-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot_shared-species-3">
Past versions of “plot_shared species-3.png”
</button>
</p>
<div id="fig-no-spaces-plot_shared-species-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics/advanz4/blob/961b1ec9f698d806778bc4bf0188fba9e357e993/docs/figure/420_Taxonomy_DA.Rmd/plot_shared species-3.png" target="_blank">961b1ec</a>
</td>
<td>
cblazquez
</td>
<td>
2022-04-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>shared_sp</code></pre>
<pre><code>[1] &quot;Bifidobacterium_adolescentis&quot; &quot;Bifidobacterium_longum&quot;      
[3] &quot;Olsenella_scatoligenes&quot;      </code></pre>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<pre class="r"><code>## This code may still be clunky and not work

endpoints &lt;- c(48,96)


stat_list_long &lt;-
  cat_vector %&gt;%
  purrr::set_names() %&gt;%
  purrr::map(function(cv) {
    
    print(cv)
    otu_df %&gt;%
      dplyr::select(-SampleID) %&gt;%
      colnames() %&gt;%
      set_names() %&gt;%
      purrr::map_dfr(function(sp) {
        
        metadata %&gt;%
          dplyr::filter(!is.na(!!sym(cv))) %&gt;%
          dplyr::pull(!!sym(cv)) %&gt;%
          unique() %&gt;%
          set_names() %&gt;%
          purrr::map_dfr(function(lv) {
            
            endpoints %&gt;%
              purrr::map_dfr(function(ep) {
                
                ids &lt;- metadata %&gt;%
                  dplyr::select(link_var, long_var, cv) %&gt;%
                  dplyr::filter(!!sym(long_var) %in% c(0, ep) &amp;
                                  !!sym(cv) == lv) %&gt;%
                  group_by(!!sym(link_var)) %&gt;%
                  dplyr::count() %&gt;%
                  filter(n &gt; 1) %&gt;%
                  pull(!!sym(link_var))
                
                
                test_df &lt;-
                  otu_df %&gt;%
                  dplyr::select(SampleID, !!sym(sp)) %&gt;%
                  dplyr::left_join(metadata[, c(&quot;SampleID&quot;,
                                                link_var,
                                                cv,
                                                long_var), drop = F],
                                   by = &quot;SampleID&quot;) %&gt;%
                  dplyr::filter(!!sym(cv) == lv) %&gt;%
                  dplyr::filter(time_point %in% c(0, ep)) %&gt;%
                  dplyr::filter(!!sym(link_var) %in% ids) 
                
                
                median_0 &lt;- median(test_df %&gt;% 
                                          dplyr::filter(!!sym(long_var) == 0) %&gt;% 
                                          dplyr::pull(!!sym(sp)))
                median_ep &lt;- median(test_df %&gt;% 
                                          dplyr::filter(!!sym(long_var) == ep) %&gt;% 
                                          dplyr::pull(!!sym(sp)))

                  test &lt;- 
                    rstatix::wilcox_test(test_df , formula(paste(sp, &quot;~&quot;, long_var)), paired = T) %&gt;%
                  mutate(level = lv,
                         median_0 = median_0,
                         median_ep = median_ep,
                         med_fc= log2(median_0/median_ep))
                  
                  return(test)
              })
            
            
            
          }) %&gt;%
          filter(p &lt; 0.05)
        ## First I&#39;ll try a selection of overall significance by Kruskal-wallis. Thos significant will be selected for a pairwise tp-to-tp comparison, to save on computational resources.
      }) %&gt;%
      rename(species = .y.)

  })</code></pre>
<pre><code>[1] &quot;group&quot;
[1] &quot;risk_group&quot;
[1] &quot;center&quot;
[1] &quot;gender&quot;
[1] &quot;ethnic_group&quot;
[1] &quot;CD4diff_48&quot;
[1] &quot;CD8diff_48&quot;
[1] &quot;CD4after_48&quot;
[1] &quot;CD8after_48&quot;
[1] &quot;CD4diff_96&quot;
[1] &quot;CD8diff_96&quot;
[1] &quot;CD4after_96&quot;
[1] &quot;CD8after_96&quot;</code></pre>
<pre class="r"><code>## Once we get the filtered species, do another loop doing pairwise wilcox for each one of the species in the &quot;reduced table&quot;

stat_list_long$group %&gt;% 
  dplyr::filter(group2 == 48) %&gt;% 
  dplyr::arrange(level) %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;, caption = &quot;DA weeks 0 v 48 by Group&quot;)</code></pre>
<table>
<caption>DA weeks 0 v 48 by Group</caption>
<thead>
<tr class="header">
<th align="left">species</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">statistic</th>
<th align="right">p</th>
<th align="left">level</th>
<th align="right">median_0</th>
<th align="right">median_ep</th>
<th align="right">med_fc</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Actinomyces_oris</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">161</td>
<td align="right">0.03830</td>
<td align="left">DRV/r</td>
<td align="right">0.000000</td>
<td align="right">0.000000</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="left">Dorea_longicatena</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">100</td>
<td align="right">0.01140</td>
<td align="left">DRV/r</td>
<td align="right">2.046715</td>
<td align="right">3.184930</td>
<td align="right">-0.6379514</td>
</tr>
<tr class="odd">
<td align="left">Desulfovibrio_piger</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">13</td>
<td align="right">0.04550</td>
<td align="left">DRV/r</td>
<td align="right">0.000000</td>
<td align="right">0.000000</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="left">Bifidobacterium_adolescentis</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">55</td>
<td align="right">0.00399</td>
<td align="left">DTG</td>
<td align="right">0.000000</td>
<td align="right">2.576615</td>
<td align="right">-Inf</td>
</tr>
<tr class="odd">
<td align="left">Olsenella_scatoligenes</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">71</td>
<td align="right">0.04310</td>
<td align="left">DTG</td>
<td align="right">0.000000</td>
<td align="right">0.004285</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Collinsella_aerofaciens</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">102</td>
<td align="right">0.01290</td>
<td align="left">DTG</td>
<td align="right">5.006980</td>
<td align="right">9.053025</td>
<td align="right">-0.8544592</td>
</tr>
<tr class="odd">
<td align="left">Collinsella_stercoris</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">112</td>
<td align="right">0.01200</td>
<td align="left">DTG</td>
<td align="right">0.030635</td>
<td align="right">0.062250</td>
<td align="right">-1.0228930</td>
</tr>
<tr class="even">
<td align="left">Oscillibacter_sp_57_20</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">111</td>
<td align="right">0.03720</td>
<td align="left">DTG</td>
<td align="right">0.091665</td>
<td align="right">0.358020</td>
<td align="right">-1.9655973</td>
</tr>
<tr class="odd">
<td align="left">Intestinibacter_bartlettii</td>
<td align="left">0</td>
<td align="left">48</td>
<td align="right">30</td>
<td align="right">30</td>
<td align="right">57</td>
<td align="right">0.04380</td>
<td align="left">DTG</td>
<td align="right">0.000000</td>
<td align="right">0.027880</td>
<td align="right">-Inf</td>
</tr>
</tbody>
</table>
<pre class="r"><code>stat_list_long$group %&gt;% 
  dplyr::filter(group2 == 96) %&gt;% 
  dplyr::arrange(level) %&gt;% 
  kableExtra::kable(format = &quot;markdown&quot;, caption = &quot;DA weeks 0 v 48 by Group&quot;)</code></pre>
<table>
<caption>DA weeks 0 v 48 by Group</caption>
<thead>
<tr class="header">
<th align="left">species</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">statistic</th>
<th align="right">p</th>
<th align="left">level</th>
<th align="right">median_0</th>
<th align="right">median_ep</th>
<th align="right">med_fc</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Actinomyces_odontolyticus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">162</td>
<td align="right">0.03490</td>
<td align="left">DRV/r</td>
<td align="right">0.00710</td>
<td align="right">0.00000</td>
<td align="right">Inf</td>
</tr>
<tr class="even">
<td align="left">Actinomyces_oris</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">125</td>
<td align="right">0.02310</td>
<td align="left">DRV/r</td>
<td align="right">0.00593</td>
<td align="right">0.00000</td>
<td align="right">Inf</td>
</tr>
<tr class="odd">
<td align="left">Odoribacter_splanchnicus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">58</td>
<td align="right">0.02730</td>
<td align="left">DRV/r</td>
<td align="right">0.00000</td>
<td align="right">0.15857</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Eubacterium_ramulus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">61</td>
<td align="right">0.03480</td>
<td align="left">DRV/r</td>
<td align="right">0.00216</td>
<td align="right">0.11372</td>
<td align="right">-5.7183109</td>
</tr>
<tr class="odd">
<td align="left">Anaerostipes_hadrus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">88</td>
<td align="right">0.02710</td>
<td align="left">DRV/r</td>
<td align="right">0.16932</td>
<td align="right">0.61036</td>
<td align="right">-1.8499080</td>
</tr>
<tr class="even">
<td align="left">Dorea_longicatena</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">97</td>
<td align="right">0.04760</td>
<td align="left">DRV/r</td>
<td align="right">1.76659</td>
<td align="right">3.00699</td>
<td align="right">-0.7673528</td>
</tr>
<tr class="odd">
<td align="left">Catenibacterium_mitsuokai</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">48</td>
<td align="right">0.03490</td>
<td align="left">DRV/r</td>
<td align="right">0.16927</td>
<td align="right">0.16266</td>
<td align="right">0.0574668</td>
</tr>
<tr class="even">
<td align="left">Actinomyces_oris</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">196</td>
<td align="right">0.00543</td>
<td align="left">DTG</td>
<td align="right">0.01315</td>
<td align="right">0.00000</td>
<td align="right">Inf</td>
</tr>
<tr class="odd">
<td align="left">Bifidobacterium_adolescentis</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">86</td>
<td align="right">0.00798</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">1.28748</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Olsenella_scatoligenes</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">68</td>
<td align="right">0.01140</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.00739</td>
<td align="right">-Inf</td>
</tr>
<tr class="odd">
<td align="left">Collinsella_massiliensis</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">97</td>
<td align="right">0.01630</td>
<td align="left">DTG</td>
<td align="right">0.00093</td>
<td align="right">0.00486</td>
<td align="right">-2.3856537</td>
</tr>
<tr class="even">
<td align="left">Bacteroides_ovatus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">64</td>
<td align="right">0.04410</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.01175</td>
<td align="right">-Inf</td>
</tr>
<tr class="odd">
<td align="left">Odoribacter_splanchnicus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">98</td>
<td align="right">0.02970</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.13995</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Prevotella_sp_CAG_279</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">84</td>
<td align="right">0.03580</td>
<td align="left">DTG</td>
<td align="right">0.00212</td>
<td align="right">0.14147</td>
<td align="right">-6.0602881</td>
</tr>
<tr class="odd">
<td align="left">Streptococcus_parasanguinis</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">339</td>
<td align="right">0.00203</td>
<td align="left">DTG</td>
<td align="right">0.06719</td>
<td align="right">0.00140</td>
<td align="right">5.5847478</td>
</tr>
<tr class="even">
<td align="left">Eubacterium_ramulus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">94</td>
<td align="right">0.03970</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.15786</td>
<td align="right">-Inf</td>
</tr>
<tr class="odd">
<td align="left">Blautia_obeum</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">134</td>
<td align="right">0.02460</td>
<td align="left">DTG</td>
<td align="right">0.23331</td>
<td align="right">0.41392</td>
<td align="right">-0.8271038</td>
</tr>
<tr class="even">
<td align="left">Blautia_wexlerae</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">391</td>
<td align="right">0.00414</td>
<td align="left">DTG</td>
<td align="right">0.56940</td>
<td align="right">0.29116</td>
<td align="right">0.9676303</td>
</tr>
<tr class="odd">
<td align="left">Ruminococcus_gnavus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">120</td>
<td align="right">0.04180</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NaN</td>
</tr>
<tr class="even">
<td align="left">Coprococcus_catus</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">112</td>
<td align="right">0.02320</td>
<td align="left">DTG</td>
<td align="right">0.08080</td>
<td align="right">0.34135</td>
<td align="right">-2.0788246</td>
</tr>
<tr class="odd">
<td align="left">Dorea_longicatena</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">110</td>
<td align="right">0.01210</td>
<td align="left">DTG</td>
<td align="right">1.04343</td>
<td align="right">2.34579</td>
<td align="right">-1.1687400</td>
</tr>
<tr class="even">
<td align="left">Clostridium_citroniae</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">90</td>
<td align="right">0.02020</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="left">Lachnospira_pectinoschiza</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">87</td>
<td align="right">0.01470</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.27482</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Erysipelatoclostridium_ramosum</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">123</td>
<td align="right">0.00483</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NaN</td>
</tr>
<tr class="odd">
<td align="left">Firmicutes_bacterium_CAG_110</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">0.01060</td>
<td align="left">DTG</td>
<td align="right">0.00000</td>
<td align="right">0.02712</td>
<td align="right">-Inf</td>
</tr>
<tr class="even">
<td align="left">Firmicutes_bacterium_CAG_83</td>
<td align="left">0</td>
<td align="left">96</td>
<td align="right">31</td>
<td align="right">31</td>
<td align="right">126</td>
<td align="right">0.01570</td>
<td align="left">DTG</td>
<td align="right">0.06155</td>
<td align="right">0.17974</td>
<td align="right">-1.5460807</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compositional-approach-ancom" class="section level1">
<h1>Compositional approach (ANCOM)</h1>
<div id="cross-group-differential-abundance-1" class="section level2">
<h2>Cross-group Differential Abundance</h2>
<p>These comparisons will be between treatment groups at the same time point: one at week 0, week 24, week 48 and 96.</p>
<div id="comparison-of-significant-species-1" class="section level3">
<h3>Comparison of significant species</h3>
<p>Following Venn’s diagram to detect group_specific DA taxa (Marc’s addition, moved here as I thought it made more sense structurally)</p>
</div>
</div>
<div id="longitudinal-within-group-differential-abundance-1" class="section level2">
<h2>Longitudinal, within-group Differential Abundance</h2>
<p>These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] metar_0.1.5        testthat_3.1.4     RColorBrewer_1.1-3 forcats_0.5.1     
 [5] stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4        readr_2.1.2       
 [9] tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] estimability_1.3       coda_0.19-4            bit64_4.0.5           
  [4] knitr_1.39             multcomp_1.4-19        wesanderson_0.3.6     
  [7] data.table_1.14.2      DataExplorer_0.8.2     generics_0.1.2        
 [10] BiocGenerics_0.36.1    callr_3.7.0            TH.data_1.1-1         
 [13] usethis_2.1.5          proxy_0.4-26           future_1.25.0         
 [16] correlation_0.8.0      bit_4.0.4              tzdb_0.3.0            
 [19] webshot_0.5.3          xml2_1.3.2             lubridate_1.8.0       
 [22] httpuv_1.6.5           assertthat_0.2.1       WRS2_1.1-3            
 [25] xfun_0.30              hms_1.1.1              jquerylib_0.1.4       
 [28] evaluate_0.15          promises_1.2.0.1       fansi_1.0.3           
 [31] dbplyr_2.1.1           readxl_1.4.0           igraph_1.3.1          
 [34] DBI_1.1.2              htmlwidgets_1.5.4      reshape_0.8.9         
 [37] stats4_4.0.5           paletteer_1.4.0        ellipsis_0.3.2        
 [40] crosstalk_1.2.0        ggpubr_0.4.0           backports_1.4.1       
 [43] insight_0.17.0         permute_0.9-7          prismatic_1.1.0       
 [46] vctrs_0.4.1            Biobase_2.50.0         remotes_2.4.2         
 [49] here_1.0.1             abind_1.4-5            cachem_1.0.6          
 [52] withr_2.5.0            ggforce_0.3.3          RVenn_1.1.0           
 [55] aws.signature_0.6.0    vroom_1.5.7            emmeans_1.7.3         
 [58] vegan_2.6-2            prettyunits_1.1.1      svglite_2.1.0         
 [61] cluster_2.1.3          lazyeval_0.2.2         ape_5.6-2             
 [64] crayon_1.5.1           pkgconfig_2.0.3        labeling_0.4.2        
 [67] units_0.8-0            tweenr_1.0.2           nlme_3.1-157          
 [70] statsExpressions_1.3.1 pkgload_1.2.4          blme_1.0-5            
 [73] devtools_2.4.3         rlang_1.0.2            globals_0.14.0        
 [76] lifecycle_1.0.1        sandwich_3.0-1         selbal_0.1.0          
 [79] phyloseq_1.34.0        modelr_0.1.8           cellranger_1.1.0      
 [82] rprojroot_2.0.3        polyclip_1.10-0        datawizard_0.4.0      
 [85] Matrix_1.4-1           mc2d_0.1-21            carData_3.0-5         
 [88] Rhdf5lib_1.12.1        boot_1.3-28            zoo_1.8-10            
 [91] reprex_2.0.1           base64enc_0.1-3        whisker_0.4           
 [94] processx_3.5.3         viridisLite_0.4.0      parameters_0.17.0     
 [97] KernSmooth_2.23-20     rhdf5filters_1.2.1     Biostrings_2.58.0     
[100] workflowr_1.7.0        classInt_0.4-3         arm_1.12-2            
[103] parallelly_1.31.1      rstatix_0.7.0          S4Vectors_0.28.1      
[106] ggsignif_0.6.3.9000    aws.s3_0.3.21          scales_1.2.0          
[109] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.7            
[112] zlibbioc_1.36.0        compiler_4.0.5         kableExtra_1.3.4      
[115] lme4_1.1-29            cli_3.3.0              ade4_1.7-19           
[118] XVector_0.30.0         listenv_0.8.0          patchwork_1.1.1       
[121] ps_1.7.0               MASS_7.3-57            mgcv_1.8-40           
[124] tidyselect_1.1.2       stringi_1.7.6          highr_0.9             
[127] yaml_2.3.5             ggrepel_0.9.1          grid_4.0.5            
[130] sass_0.4.1             tools_4.0.5            parallel_4.0.5        
[133] rstudioapi_0.13        foreach_1.5.2          logging_0.10-108      
[136] git2r_0.30.1           gridExtra_2.3          farver_2.1.0          
[139] digest_0.6.29          shiny_1.7.1            networkD3_0.4         
[142] Rcpp_1.0.8.3           car_3.0-12             broom_0.8.0           
[145] performance_0.9.0      later_1.3.0            httr_1.4.2            
[148] sf_1.0-7               effectsize_0.6.0.1     colorspace_2.0-3      
[151] rvest_1.0.2            brio_1.1.3             fs_1.5.2              
[154] IRanges_2.24.1         splines_4.0.5          ggVennDiagram_1.2.1   
[157] rematch2_2.1.2         multtest_2.46.0        systemfonts_1.0.4     
[160] plotly_4.10.0          sessioninfo_1.2.2      xtable_1.8-4          
[163] jsonlite_1.8.0         nloptr_2.0.0           zeallot_0.1.0         
[166] R6_2.5.1               broom.mixed_0.2.9.4    pillar_1.7.0          
[169] htmltools_0.5.2        mime_0.12              glue_1.6.2            
[172] fastmap_1.1.0          minqa_1.2.4            class_7.3-20          
[175] codetools_0.2-18       pkgbuild_1.3.1         mvtnorm_1.1-3         
[178] furrr_0.2.3            utf8_1.2.2             lattice_0.20-45       
[181] bslib_0.3.1            logger_0.2.2           curl_4.3.2            
[184] survival_3.3-1         rmarkdown_2.14         desc_1.4.1            
[187] biomformat_1.18.0      merTools_0.5.2         munsell_0.5.0         
[190] e1071_1.7-9            rhdf5_2.34.0           iterators_1.0.14      
[193] ggstatsplot_0.9.1      haven_2.5.0            reshape2_1.4.4        
[196] gtable_0.3.0           bayestestR_0.11.5.1   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
