<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera" />


<title>Adenovirus Analisis</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">advanz4</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="310_IGC_analysis.html">Gene richness</a>
</li>
<li>
  <a href="320_Beta_Diversity.html">Beta Diversity</a>
</li>
<li>
  <a href="335_BLMicrobiome_vs_ImmuneResponse.html">Immune and clinical BL comparisons</a>
</li>
<li>
  <a href="330_TaxaLevel_LMM.html">LMM taxa ranking</a>
</li>
<li>
  <a href="410_Taxonomy_Exploration.html">Taxonomy</a>
</li>
<li>
  <a href="420_Taxonomy_DA.html">Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Adenovirus Analisis</h1>
<h4 class="author">Marc Noguera</h4>
<h4 class="date">5/3/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>advanz4/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
read_data
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>505_Adenovirus_Analysis_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treee7644ce88f78db3d5c661c679f6f2f88f836ef01targetblanke7644cea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics/advanz4/tree/e7644ce88f78db3d5c661c679f6f2f88f836ef01" target="_blank">e7644ce</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsadvanz4treee7644ce88f78db3d5c661c679f6f2f88f836ef01targetblanke7644cea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics/advanz4/tree/e7644ce88f78db3d5c661c679f6f2f88f836ef01" target="_blank">e7644ce</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/505_Adenovirus_Analysis_cache/
    Ignored:    data/mre.rds

Untracked files:
    Untracked:  analysis/505_Adenovirus_Analysis.Rmd
    Untracked:  code/LMM.R
    Untracked:  code/update_mre.R

Unstaged changes:
    Deleted:    505_Adenovirus_Analysis.Rmd
    Modified:   analysis/310_IGC_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>mymre &lt;- mymre &lt;- aws.s3::s3readRDS(bucket = &quot;s3://mistral-wp6-advanz4&quot;, object = &quot;metagenome/WMGS/MREObject.rds&quot;)
# metadata &lt;-get_meta(mymre)


cat_df &lt;- 
  here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T)

cat_vector &lt;- cat_df %&gt;% 
  pull(CategoricalVariable)

long_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LongitudinalVariable)

link_var &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(LinkVariable)

num_var &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;%
  read.delim(., header = T)  %&gt;% 
  pull(NumericalVariable)


metadata &lt;- 
  here::here(&quot;Metadata&quot;, &quot;2022_04_28_clean_metadata_LIMS.csv&quot;) %&gt;% 
  read.csv() %&gt;% 
  mutate(ADV_p = case_when(
    is.na(ADV) ~ &quot;no&quot;, 
    !is.na(ADV) ~ &quot;yes&quot;), .after = &quot;ADV&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but it won’t be re-run if previous chunks it depends on are updated. If you need to use caching, it is highly recommended to also set <code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the file (in a chunk that is not cached). Alternatively, you can customize the option <code>dependson</code> for each individual chunk that is cached. Using either <code>autodep</code> or <code>dependson</code> will remove this warning. See the <a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>devtools::load_all(here::here(&quot;../WMGSPipeline&quot;))</code></pre>
<pre><code>ℹ Loading metar</code></pre>
<pre><code>── Attaching ──────────────────────────────────────────────── metar v-0.0.0.9 ──</code></pre>
<pre><code>suggested package selbal ✔</code></pre>
<pre><code></code></pre>
<pre><code>suggested package DataExplorer ✔</code></pre>
<pre><code></code></pre>
<pre><code>suggested package lme4 ✔</code></pre>
<pre><code></code></pre>
<pre><code>suggested package merTools ✔</code></pre>
<pre><code></code></pre>
<pre class="r"><code>mymre &lt;- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df &lt;- as.tibble(metadata) </code></pre>
<pre><code>Warning: `as.tibble()` was deprecated in tibble 2.0.0.
Please use `as_tibble()` instead.
The signature and semantics have changed, see `?as_tibble`.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<pre class="r"><code>mymre@taxa@metaphlan@phyloseq@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data &lt;- metadata %&gt;%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals &lt;- here::here(&quot;Metadata&quot;, &quot;CategoricalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@numeric_vals &lt;- here::here(&quot;Metadata&quot;, &quot;NumericalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()

mymre@metadata@longitudinal_vals &lt;- here::here(&quot;Metadata&quot;, &quot;LongitudinalVariables.txt&quot;) %&gt;% 
  read.delim(., header = T) %&gt;% 
  tibble()</code></pre>
<div id="import-data-and-variable-definition." class="section level1">
<h1>Import data and variable definition.</h1>
<p>Only a few patients had any detecatble ADV presence on their guts, which also had huge variations. Hence we expect a quantitative approach to hold little power. We’ll complement with a qualitative approach based on presence/absence of both ADV and EV.</p>
<p>First, classify each sample as ADV+/ADV-. (Do the same for Enterovirus)</p>
</div>
<div id="adenovirus-per-treatment" class="section level1">
<h1>Adenovirus per treatment</h1>
<div id="quantitative-exploration" class="section level2">
<h2>Quantitative exploration</h2>
<pre class="r"><code>create_LMM(metadata = metadata,
           num_var = &quot;ADV&quot;,
           cat_var = &quot;group&quot;,
           long_var = long_var, 
           link_var = link_var,
           breakpoints = NULL)</code></pre>
<pre><code>[[1]]
[[1]]$`DRV/r`
                 Value  Std.Error DF   t-value   p-value
(Intercept) 1499.25506 2013.21063 22 0.7447085 0.4643295
long_var      26.83348   41.94189  9 0.6397775 0.5382603

[[1]]$DTG
                 Value Std.Error DF   t-value   p-value
(Intercept) 430155.472 278786.20 27  1.542958 0.1344815
long_var     -6200.114   5336.52 13 -1.161827 0.2661931


[[2]]</code></pre>
<p><img src="figure/505_Adenovirus_Analysis.Rmd/ADV_quantitative-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]
[[3]]$`DRV/r`
                 Value  Std.Error DF   t-value   p-value
(Intercept) 1499.25506 2013.21063 22 0.7447085 0.4643295
long_var      26.83348   41.94189  9 0.6397775 0.5382603

[[3]]$DTG
                 Value Std.Error DF   t-value   p-value
(Intercept) 430155.472 278786.20 27  1.542958 0.1344815
long_var     -6200.114   5336.52 13 -1.161827 0.2661931</code></pre>
<pre class="r"><code>metadata %&gt;% 
  dplyr::select(link_var = !!sym(link_var),
                long_var = !!sym(long_var),
                cat_var = &quot;group&quot;,
                ADV) %&gt;% 
  # mutate(logADV = log10(ADV)) %&gt;% 
  ggplot2::ggplot(., aes(x=ADV, color=cat_var)) +
  # facet_wrap(~cat_var)+
  geom_density(stat=&quot;density&quot;) +
  scale_x_log10() +
  theme_minimal() +
  labs(title = &quot;Adenovirus&quot;, subtitle = &quot;Abundance distribution (log10)&quot;, x=&quot;log10(ADV)&quot;,
       color = &quot;group&quot;) +
  scale_color_brewer(palette = &quot;Set1&quot;)</code></pre>
<p><img src="figure/505_Adenovirus_Analysis.Rmd/ADV_quantitative-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>unique(metadata$group) %&gt;% 
  purrr::set_names() %&gt;% 
  purrr::map_dfr(~{
    metadata %&gt;% 
  dplyr::select(link_var = !!sym(link_var),
                long_var = !!sym(long_var),
                cat_var = &quot;group&quot;,
                ADV) %&gt;% 
      dplyr::filter(cat_var == .x) %&gt;% 
      dplyr::pull(ADV) %&gt;% 
      summary() 

}) %&gt;% 
  dplyr::mutate(lev=unique(metadata$group)) %&gt;% 
  tibble::column_to_rownames(&quot;lev&quot;) %&gt;% 
  kableExtra::kable(format=&quot;markdown&quot;)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Min.</th>
<th align="right">1st Qu.</th>
<th align="right">Median</th>
<th align="right">Mean</th>
<th align="right">3rd Qu.</th>
<th align="right">Max.</th>
<th align="right">NA’s</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">DTG</td>
<td align="right">0.0107</td>
<td align="right">0.080875</td>
<td align="right">3.0726</td>
<td align="right">185693.843</td>
<td align="right">612.0453</td>
<td align="right">7716041.0</td>
<td align="right">99</td>
</tr>
<tr class="even">
<td align="left">DRV/r</td>
<td align="right">0.0028</td>
<td align="right">0.040800</td>
<td align="right">1.6708</td>
<td align="right">2435.987</td>
<td align="right">146.4422</td>
<td align="right">37014.2</td>
<td align="right">89</td>
</tr>
</tbody>
</table>
<p>We see here that ADV follows a non-normal heavily skewed toward the left. We’ll check how do they behave considering treatment time.</p>
<pre class="r"><code>myLMMs &lt;- get_lmm_effects(data =metadata, cat_vector = c(&quot;group&quot;), num_vector = c(&quot;ADV&quot;), long_var = long_var) 

myLMMs$group$ADV$plot</code></pre>
<p><img src="figure/505_Adenovirus_Analysis.Rmd/ADV_longitudinal-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It appears clear that both treatments find a significan reduction in ADV levels, slightly harder in the DTG group than in the DRV, but it is not backed the ANOVA test. It has to be taken into account the DTG had a higher average at start, but it is probably a skew in the model caused by one outlier.</p>
</div>
</div>
<div id="adenovirus-qualitative-analysis" class="section level1">
<h1>Adenovirus qualitative analysis</h1>
<p>We will test for presence/absence and try to test for differences among groups. We’ll test for proportions of patients positive per ADV per group and time.</p>
<pre class="r"><code>cat_var = &quot;group&quot;
num_var = &quot;ADV&quot;
metadata %&gt;% 
  dplyr::select(link_var = link_var, 
                long_var = long_var, 
                cat_var = cat_var, 
                num_var = ADV_p) %&gt;% 
  dplyr::group_by(cat_var, long_var) %&gt;% 
  dplyr::count(num_var) %&gt;% 
  dplyr::mutate(long_var = as.factor(long_var)) %&gt;% 
  tidyr::pivot_wider(id_cols = c(&quot;cat_var&quot;,&quot;long_var&quot;), names_from = &quot;num_var&quot;, values_from = &quot;n&quot;) %&gt;%
  dplyr::mutate(incidence = yes*100 / (no + yes)) %&gt;% 
  ggplot(., aes(x=long_var, y = incidence, fill = num_var)) +
  facet_wrap(~cat_var) +
  geom_histogram(stat=&quot;identity&quot;, position = &quot;stack&quot;) +
  theme_minimal() +
  scale_fill_brewer(palette = &quot;Set1&quot;) +
  labs(x = &quot;week&quot;,y=&quot;incidence (%)&quot;, fill = &quot;ADV&quot;)</code></pre>
<p><img src="figure/505_Adenovirus_Analysis.Rmd/ADV_prevalence-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="adenovirus-detection-vs-microbiome-composition" class="section level3">
<h3>Adenovirus detection vs Microbiome composition</h3>
<p>Use:</p>
<ul>
<li>NMDS coordinates (NMDS1 and NMDS2) vs ADV+/ADV-</li>
<li>PERMANOVA to check significance.</li>
<li>ADV+/ADV- vs gene richness</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Fedora 34 (Workstation Edition)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libflexiblas.so.3.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] nlme_3.1-157       metar_0.1.5        testthat_3.1.4     RColorBrewer_1.1-3
 [5] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       
 [9] readr_2.1.2        tidyr_1.2.0        tibble_3.1.6       ggplot2_3.3.5     
[13] tidyverse_1.3.1   

loaded via a namespace (and not attached):
  [1] utf8_1.2.2          tidyselect_1.1.2    lme4_1.1-29        
  [4] htmlwidgets_1.5.4   grid_4.0.5          devtools_2.4.3     
  [7] munsell_0.5.0       codetools_0.2-18    future_1.25.0      
 [10] withr_2.5.0         colorspace_2.0-3    Biobase_2.50.0     
 [13] phyloseq_1.34.0     logger_0.2.2        highr_0.9          
 [16] knitr_1.39          rstudioapi_0.13     stats4_4.0.5       
 [19] ggsignif_0.6.3.9000 listenv_0.8.0       labeling_0.4.2     
 [22] git2r_0.30.1        polyclip_1.10-0     bit64_4.0.5        
 [25] farver_2.1.0        rhdf5_2.34.0        rprojroot_2.0.3    
 [28] coda_0.19-4         parallelly_1.31.1   vctrs_0.4.1        
 [31] generics_0.1.2      xfun_0.30           R6_2.5.1           
 [34] arm_1.12-2          rhdf5filters_1.2.1  cachem_1.0.6       
 [37] assertthat_0.2.1    promises_1.2.0.1    networkD3_0.4      
 [40] scales_1.2.0        vroom_1.5.7         merTools_0.5.2     
 [43] gtable_0.3.0        globals_0.14.0      processx_3.5.3     
 [46] workflowr_1.7.0     rlang_1.0.2         systemfonts_1.0.4  
 [49] splines_4.0.5       rstatix_0.7.0       broom_0.8.0        
 [52] yaml_2.3.5          reshape2_1.4.4      abind_1.4-5        
 [55] modelr_0.1.8        backports_1.4.1     httpuv_1.6.5       
 [58] tools_4.0.5         usethis_2.1.5       logging_0.10-108   
 [61] ellipsis_0.3.2      kableExtra_1.3.4    jquerylib_0.1.4    
 [64] biomformat_1.18.0   selbal_0.1.0        BiocGenerics_0.36.1
 [67] sessioninfo_1.2.2   Rcpp_1.0.8.3        plyr_1.8.7         
 [70] zlibbioc_1.36.0     ps_1.7.0            prettyunits_1.1.1  
 [73] ggpubr_0.4.0        S4Vectors_0.28.1    haven_2.5.0        
 [76] cluster_2.1.3       fs_1.5.2            here_1.0.1         
 [79] furrr_0.2.3         DataExplorer_0.8.2  magrittr_2.0.3     
 [82] data.table_1.14.2   lmerTest_3.1-3      reprex_2.0.1       
 [85] mvtnorm_1.1-3       pkgload_1.2.4       hms_1.1.1          
 [88] patchwork_1.1.1     mime_0.12           evaluate_0.15      
 [91] xtable_1.8-4        broom.mixed_0.2.9.4 readxl_1.4.0       
 [94] IRanges_2.24.1      gridExtra_2.3       compiler_4.0.5     
 [97] crayon_1.5.1        minqa_1.2.4         htmltools_0.5.2    
[100] mgcv_1.8-40         later_1.3.0         tzdb_0.3.0         
[103] lubridate_1.8.0     DBI_1.1.2           tweenr_1.0.2       
[106] dbplyr_2.1.1        MASS_7.3-57         boot_1.3-28        
[109] Matrix_1.4-1        ade4_1.7-19         car_3.0-12         
[112] wesanderson_0.3.6   permute_0.9-7       brio_1.1.3         
[115] cli_3.3.0           parallel_4.0.5      igraph_1.3.1       
[118] pkgconfig_2.0.3     numDeriv_2016.8-1.1 xml2_1.3.2         
[121] foreach_1.5.2       svglite_2.1.0       bslib_0.3.1        
[124] blme_1.0-5          multtest_2.46.0     webshot_0.5.3      
[127] XVector_0.30.0      rvest_1.0.2         callr_3.7.0        
[130] digest_0.6.29       vegan_2.6-2         Biostrings_2.58.0  
[133] rmarkdown_2.14      cellranger_1.1.0    shiny_1.7.1        
[136] nloptr_2.0.0        lifecycle_1.0.1     jsonlite_1.8.0     
[139] Rhdf5lib_1.12.1     carData_3.0-5       desc_1.4.1         
[142] viridisLite_0.4.0   fansi_1.0.3         pillar_1.7.0       
[145] lattice_0.20-45     fastmap_1.1.0       httr_1.4.2         
[148] pkgbuild_1.3.1      survival_3.3-1      glue_1.6.2         
[151] remotes_2.4.2       iterators_1.0.14    bit_4.0.4          
[154] ggforce_0.3.3       stringi_1.7.6       sass_0.4.1         
[157] memoise_2.0.1       ape_5.6-2          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
