---
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## ADVANZ4 WMGS fastq from datatransfer to project bucket
****

```{r}
if (!requireNamespace("MISTRALDM", quietly = TRUE)) {
  remotes::install_github("MicrobialGenomics/MISTRALDM@dev")
}

library(MISTRALDM)
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = here::here(),
  out.width = "100%"
)
```

### Variable definition

```{r}
# Input data path
inp_bucket <- "s3://gemirsi-datatransfer"
prefix <- "RawData" 
    
# Output data path
out_bucket <- "s3://mistral-wp6-advanz4" # This bucket already exists
out_prefix <- "metagenome/WMGS/RawData/"

# Backup data path
bk_bucket <- "s3://glacier-rawdata"

## Pattern for filtering
pattern <- "ADZ.*fastq.gz"
```

### Listing files to transfer, renaming and new path definition

```{r preprocess}
dat <- 
    aws.s3::get_bucket_df(bucket = inp_bucket, prefix = prefix, max = Inf) %>% 
    tibble::as_tibble() %>% 
    dplyr::select(Key) %>% 
    dplyr::filter(stringr::str_detect(Key, pattern = pattern)) %>% 
    dplyr::mutate(
        new_name = Key, 
        new_name = stringr::str_replace_all(new_name, "ADZ", "ADZ4"),
        new_name = stringr::str_replace_all(new_name, "_1.fastq.gz", "_R1.fastq.gz"),
        new_name = stringr::str_replace_all(new_name, "_2.fastq.gz", "_R2.fastq.gz"), 
        new_name = stringr::str_remove_all(new_name, ".*/"), 
        new_name = stringr::str_c(out_prefix, new_name)
    ) %>% 
    dplyr::filter(!duplicated(new_name))

## Head
dat 
```

```{r transfer}
## Transfer
MISTRALDM::copy_s3Objects(
  df = dat, 
  inp_bucket = inp_bucket,
  out_bucket = out_bucket, 
  rewrite = FALSE
)
```
