---
title: "Diversity_GeneRichness"
date: "`r Sys.Date()`"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(metar)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(ggpubr)
library(rstatix)
library(lme4)
library(lmerTest)
source(file = here::here("code","myFunctions.R"))

```


# Sequencing throupghput analysis

First we'll check how well the sequencing went by looking at the read count distribution among all samples and how many of them will be dropped once we rarefy the samples


```{r load_metadata_vars, eval=T, include=F}
myPath<-here::here()
setwd(myPath)
mymre<-readRDS(here::here(myPath,"data","mre.rds"))

metadata <- metar::get_metadata(mymre)@metadata_df


CategoricalVariablesDF <- get_metadata(mymre)@categorical_vals
NumericalVariablesDF <- get_metadata(mymre)@numeric_vals
LongitudinalVariablesDF <- get_metadata(mymre)@longitudinal_vals

cat_vector <- CategoricalVariablesDF$CategoricalVariable
CategoricalVariablesDF$PaletteName<-"Set1"
long_var<-LongitudinalVariablesDF$LongitudinalVariable
link_var<-LongitudinalVariablesDF$LinkVariable
t<-LongitudinalVariablesDF$Breakpoints

```



```{r get_GeneRichness_Data, echo=FALSE, message=F, warning=F}


igc<-metar::get_diversity(mymre)@igc

# We'll save this number to report the minimum read threshold.





Threshold = igc@dataTable %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>%
  as.numeric()


readCountDistribution<-igc@dataTable %>%
    select(SampleID, NumberMappedReads) %>%
    unique() %>%

    ggplot(aes(x=NumberMappedReads)) +
    geom_density(fill="grey") + 
    theme_bw() +
    geom_vline(xintercept = Threshold)



ThresholdTable <- igc@dataTable %>%
          filter(NumberMappedReads >= Threshold) %>%
          group_by(SampleID) %>%
          summarise(across(everything(), min)) %>%
          ungroup()




TableList <- cat_vector %>%
    set_names() %>%
    map( ~{

        
        ThresholdTable<-igc@dataTable %>%
          filter(NumberMappedReads >= Threshold) %>%
          group_by(SampleID) %>%
          summarise(across(everything(), min)) %>%
          ungroup() %>%
          left_join(., metadata, by="SampleID") %>%
          rename(richness = GeneNumber,
                 link_var=link_var,
                 cat_var = .x,
                 long_var = long_var)
  

        
    })

readCountDistribution

igc@dataTable %>%
          filter(!SampleID %in% ThresholdTable$SampleID) %>%
    group_by(SampleID) %>%
    summarize(maxReads = NumberMappedReads) %>%
    unique()

igc@dataTable %>%
    pull(NumberMappedReads) %>%
    summary()


```

only 6 samples have been discarded by low read count. The threshold used for filtering was the 2nd percentile, which equals 12428123 reads.



## Rarefaction and sequencing depth

We'll control first for the effect of sequencing depth over Gene richness and try to stablish a proper rarefaction threshold.

``` {r Plot_Rarefaction, echo=F, message=F, warning=F}


seq_depth_plots <- 
    cat_vector %>%
    set_names() %>%
    map(., ~ {
        
                
        myPal <- 
            CategoricalVariablesDF %>%
            filter(CategoricalVariable == .x) %>%
            pull(PaletteName)
        
        rarefaction <- 
            mymre@diversity@igc@dataTable %>%
            full_join(metadata[,c("SampleID",.x)], by="SampleID") %>%
            ggplot(., 
                   aes(x= ReadCountReal, y= GeneNumber, 
                       group = SampleID, color= !!sym(.x))) +
            geom_point() + 
            geom_line() + 
            theme_bw() + 
            geom_vline(xintercept = Threshold) +
            scale_color_brewer( palette = myPal)
        
        violin <- 
            TableList[[.x]] %>%
            ggplot(., aes(x= cat_var,y= NumberMappedReads, fill= cat_var)) +
            geom_violin(draw_quantiles = T) + 
            geom_boxplot(fill="lightgrey", width=0.2, alpha=0.4) + 
            theme_bw() + 
             ggpubr::stat_compare_means( 
                 method = "wilcox",
                 label = "p.signif", 
                 label.x.npc = 0.5) +
            scale_fill_brewer( palette = myPal) +
            geom_hline(yintercept = Threshold)

        return(list(rarefaction = rarefaction, violin = violin))
    })


ggarrange(plotlist = seq_depth_plots$group, common.legend = T)

# ggarrange(plotlist=seq_depth_plots$ARM, common.legend = T, ncol = 1, nrow = 2)


```


## Rarefaction and sequencing depth

We'll control first for the effect of sequencing depth over Gene richness and try to stablish a proper rarefaction threshold.


Data shows there is no bias created by sequencing depth between groups, and a threshold over the 2nd percentile is appropiate.


## Longitudinal evolution: LMMs

We'll start studying the longitudinal effect of treatment group over gene richness. For such purpose, we'll use Linear mixed models using patients as random effect and an interaction of treatment group and time as fixed effects with the following lme4 formula:

`richness ~ group*time_point + (1|ID)`


```{r create_LMM_Function, echo=FALSE}



create_LMM <- function(metadata, num_var, cat_var, long_var, link_var,breakpoints){
  
  myLMM <-metadata %>%
    as.data.frame() %>%
    dplyr::select(link_var = !!sym(link_var),
                  num_var = !!sym(num_var),
                  cat_var = !!sym(cat_var),
                  long_var = !!sym(long_var)) %>%
    dplyr::mutate(cat_var = as.factor(cat_var)) %>%
    myLMM_two.piece.GLMM(Datos = .,
                         Y = "num_var",
                         Time = "long_var",
                         Factor = "cat_var",
                         t = breakpoints,
                         ID = "link_var")

  
  myLMM
}


get_LMM_GraphParams <- function(Table){
  
    myList <- Table %>%
    pull(cat_var) %>%
    levels() %>%
    set_names() %>%
    map_dfr( ~ {
      
    model<-Table %>%
        dplyr::select(num_var, long_var, cat_var, link_var) %>%
        dplyr::filter(cat_var == .x) %>%
        lmerTest::lmer(formula = "num_var ~ long_var + (1|link_var)") %>%
        summary() %>%
        magrittr::extract2("coefficients") 
    
    graphparams<- Table %>%
      summarise(xmin=min(long_var),
                xmax=max(long_var),
                ymin = model[1,1] + xmin*model[2,1],
                ymax = model[1,1] + xmax*model[2,1]) %>%
      mutate(cat_var = .x) %>%
      relocate(cat_var)
  }) %>% 
      as.data.frame() 
  
    
    return(myList)
}


```




```{r plot_LMMs, message = F, warning = F, echo=F}

cat_var<-"group"
long_var <- LongitudinalVariablesDF$LongitudinalVariable[1]
link_var<-LongitudinalVariablesDF$LinkVariable[1]
num_var <- "richness"
t<-NULL
# 
# plot_list<-create_LMM(metadata = ThresholdTable,
#            num_var = "richness", 
#            cat_var = "cat_var",
#            long_var = "long_var",
#            link_var = "link_var",
#            breakpoints = NULL)

plot_list <-  c(num_var) %>%
  purrr::set_names() %>%
  purrr::map( ~ {
    
    LMM <- TableList %>%
      pluck(cat_var) %>%
      create_LMM(., 
               num_var = .x, 
               cat_var = "cat_var",
               long_var = "long_var",
               link_var = "link_var",
               breakpoints = t)
    
    pvals<-names(LMM[[1]]) %>%
      set_names() %>%
      
      map_dfr(~ {
        LMM[[1]][[.x]] %>%
          as.data.frame() %>%
          slice(2) %>%
          select("p-value", "Value") %>%
          as.numeric() %>%
          round(digits=3)
        }) %>% 
      
      t() %>%
      as.data.frame() %>%
      setNames(., nm= c("pval", "slope")) %>%
      rownames_to_column(var="cat_var")
    
    
      LMM[[2]] +
      geom_label(data=pvals, 
                 aes(alpha=NULL,label=paste("p=",pval, ", slope = ", slope, sep=""), 
                     x=-Inf, y=Inf), color="black", hjust=0.01, vjust=1 ) +
      labs(title=.x, color = cat_var)

  }
)


plot_list$richness
```

There is a pretty clear evolution in every one of the treatment groups. However the evolution rate seems pretty similar between both treatment groups. The models fail to find a great difference in gene richness evolution, althougha slight yet significant increasing trend (slope value = 0.027), can be found in the DGT group (Group 1).

However, creating a single LMM with treatment group as a random effect we can measure the actual difference in the effect of timepoint over gene richness between two groups. 


```{r compare_group_effect, warning = F, message= F, echo = F}


myUnifiedModel<- TableList %>%
  pluck(cat_var) %>%
  dplyr::rename(num_var = richness) %>%
  lmerTest::lmer(formula = "num_var ~ long_var * cat_var + (1|link_var)",data = .)

myGraphParams<- TableList %>%
  pluck(cat_var) %>%
  dplyr::rename(num_var = richness) %>%
  get_LMM_GraphParams() %>%
  mutate(across(contains("x"), ~ as.numeric(.x)))

myPval<-summary(myUnifiedModel) %>%
  magrittr::extract2("coefficients") %>%
  as.data.frame()%>%
  slice(4L) %>%
  set_names(c("Estimate", "Std.Err", "df", "t.val","pval")) %>%
  mutate(pval = round(pval,3))

TableList %>%
  pluck(cat_var) %>%
  dplyr::rename(num_var = richness) %>%
  ggplot(., aes(x =long_var, y=num_var, group=link_var, color=cat_var)) +
  geom_point(alpha=0.4) + 
  geom_line(alpha=0.4) + 
  scale_x_continuous(breaks = unique(TableList[[cat_var]]$long_var),
                     limits = c(min(TableList[[cat_var]]$long_var),
                                max(TableList[[cat_var]]$long_var))) +

  geom_segment(data=myGraphParams,aes(x=xmin,y=ymin,xend=xmax,yend=ymax, color=cat_var),inherit.aes=T,size=3) +
  scale_color_brewer(palette="Set1") +
  labs(color="Treatment", x="weeks") +
  geom_label(data=myPval, 
             aes(alpha=NULL,label=paste("ANOVA p=",pval, sep=""),
                 x=-Inf, y=Inf), color="black", hjust=0.01, vjust=1 ) +
  theme_bw() 
  
summary(myUnifiedModel) %>%
  magrittr::extract2("coefficients") %>%
  as.data.frame() %>% kableExtra::kable(., format="markdown")


```


This shows a slight effect of treatment, where the effect of time over richness in group group2 is 0.025 (St.err = 0.025) times lower than the same for group 1 (ANOVA p=0.31)


```{r boxplots, eval=F, echo=F, message=F}


### This was a first test to learn how to perform multiple comparisons in a true tidyverse fashion

plot_list <-  NumericalVariablesDF %>%
  
  dplyr::pull(NumericalVariable) %>%
  purrr::set_names() %>%
  purrr::map( ~ {
    metadata %>%
      dplyr::select(cat_var = !!sym(cat_var),
                    long_var = !!sym(long_var),
                    num_var = .x,
                    link_var = !!sym(link_var)) %>%
      ggplot(., aes(x = as.factor(long_var), y = num_var, fill = cat_var)) +
      geom_boxplot() +
      stat_compare_means(method="wilcox.test", 
                         label = "p.signif", 
                         hide.ns = T, bracket.size = .6, 
                         label.y.npc ="center") +
      labs(title = .x,
          x = "Week",
          y="counts",
          fill=cat_var) +
      theme_bw() + 
      scale_fill_brewer(palette = CategoricalVariablesDF$PaletteName[1])
  }
)

ggarrange(plotlist = plot_list, common.legend = T)



```


