---
title: "Diversity_GeneRichness"
date: "`r Sys.Date()`"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(metar)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(ggpubr)
library(rstatix)
library(lme4)
library(lmerTest)
source(file = here::here("code","myFunctions.R"))


```


# Sequencing throupghput analysis

First we'll check how well the sequencing went by looking at the read count distribution among all samples and how many of them will be dropped once we rarefy the samples


```{r load_metadata_vars, eval=T, include=F}
# mymre<-readRDS(here::here(myPath,"data","mre.rds"))




mymre <- aws.s3::s3readRDS("s3://mistral-wp6-advanz4/metagenome/WMGS/MREObject.rds") ### Check consistency between mre.rds(local and S3). You are reading s3 file, here, but in 200_ you are using local
# metadata <- metar::get_metadata(mymre)@metadata_df

metadata <- here::here("Metadata", "2022_05_30_clean_metadata_LIMS.csv") %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))

mymre@metadata@categorical_vals <-  
  here::here("Metadata","CategoricalVariables.txt") %>%
  read.table(., sep="\t") %>%
  setNames(c("CategoricalVariable","PaletteName")) %>%
  as_tibble()

mymre@metadata@metadata_df <-  tibble(metadata)


cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

link_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)


```

```{r update_mre, warning= F, message= F, echo = F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all("../WMGSPipeline/")


mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as_tibble(metadata)

mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()


```



```{r get_GeneRichness_Data, echo=FALSE, message=F, warning=F}


igc<-metar::get_diversity(mymre)@igc

# We'll save this number to report the minimum read threshold.

myigc <- igc_lmm(mymre)



Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()


readCountDistribution<-igc@dataTable %>%
    select(SampleID, NumberMappedReads) %>%
    unique() %>%

    ggplot(aes(x=NumberMappedReads)) +
    geom_density(fill="grey") + 
    theme_bw() +
    geom_vline(xintercept = Threshold)








TableList <- cat_vector %>%
    set_names() %>%
    map( ~{

        
          ThresholdTable <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          left_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=link_var,
                 cat_var = .x,
                 long_var = long_var,
                 NumberMappedReads)
  

        
    })



```




## Rarefaction and sequencing depth

We'll control first for the effect of sequencing depth over Gene richness and try to stablish a proper rarefaction threshold.

``` {r Plot_Rarefaction, echo=F, message=F, warning=F}


seq_depth_plots <- 
    cat_vector %>%
    set_names() %>%
    map(., ~ {
        
                
        myPal <- 
            cat_df %>%
            filter(CategoricalVariable == .x) %>%
            pull(PaletteName)
        
        rarefaction <- 
            mymre@diversity@igc@dataTable %>%
            full_join(metadata[,c("SampleID",.x)], by="SampleID") %>%
            ggplot(., 
                   aes(x= ReadCountReal, y= GeneNumber, 
                       group = SampleID, color= !!sym(.x))) +
            geom_point() + 
            geom_line() + 
            theme_bw() + 
            geom_vline(xintercept = Threshold) +
            scale_color_brewer( palette = myPal)
        
        violin <- 
            TableList[[.x]] %>%
            ggplot(., aes(x= cat_var,y= NumberMappedReads, fill= cat_var)) +
            geom_violin(draw_quantiles = T) + 
            geom_boxplot(fill="lightgrey", width=0.2, alpha=0.4) + 
            theme_bw() + 
             ggpubr::stat_compare_means( 
                 method = "wilcox",
                 label = "p.signif", 
                 label.x.npc = 0.5) +
            scale_fill_brewer( palette = myPal) +
            geom_hline(yintercept = Threshold)

        pltGrd <- ggarrange(rarefaction,violin, common.legend = T)
        return(pltGrd)
    })


seq_depth_plots$group


# ggarrange(plotlist=seq_depth_plots$ARM, common.legend = T, ncol = 1, nrow = 2)


```


## Rarefaction and sequencing depth

We'll control first for the effect of sequencing depth over Gene richness and try to stablish a proper rarefaction threshold.


Data shows there is no bias created by sequencing depth between groups. only 6 samples have been discarded by low read count. The threshold used for filtering was the 2nd percentile, which equals 12428123 reads.


## Longitudinal evolution: LMMs

We'll start studying the longitudinal effect of treatment group over gene richness. For such purpose, we'll use Linear mixed models using patients as random effect and an interaction of treatment group and time as fixed effects with the following lme4 formula:

`richness ~ group*time_point + (1|ID)`


```{r create_LMM_Function, echo=FALSE}



create_LMM <- function(metadata, num_var, cat_var, long_var, link_var,breakpoints){
  
  myLMM <-metadata %>%
    as.data.frame() %>%
    dplyr::select(link_var = !!sym(link_var),
                  num_var = !!sym(num_var),
                  cat_var = !!sym(cat_var),
                  long_var = !!sym(long_var)) %>%
    dplyr::mutate(cat_var = as.factor(cat_var)) %>%
    myLMM_two.piece.GLMM(Datos = .,
                         Y = "num_var",
                         Time = "long_var",
                         Factor = "cat_var",
                         t = breakpoints,
                         ID = "link_var")

  
  myLMM
}


get_LMM_GraphParams <- function(Table){
  
    myList <- Table %>%
    pull(cat_var) %>%
    unique() %>%
    set_names() %>%
    map_dfr( ~ {

    model<-Table %>%
        dplyr::select(num_var, long_var, cat_var, link_var) %>%
        dplyr::filter(cat_var == .x) %>%
        lmerTest::lmer(formula = "num_var ~ long_var + (1|link_var)") %>%
        summary() %>%
        magrittr::extract2("coefficients") 
    
    graphparams<- Table %>%
      summarise(xmin=min(long_var),
                xmax=max(long_var),
                ymin = model[1,1] + xmin*model[2,1],
                ymax = model[1,1] + xmax*model[2,1]) %>%
      mutate(cat_var = .x) %>%
      relocate(cat_var)
  }) %>% 
      as.data.frame() 
  
    
    return(myList)
}


```




We found trouble fiting the LMMs for ethnic groups due to patient A024 being the only one with the category "unknown". As the model can't estimate variance from only one observation, the code breaks.


```{r plot_LMMs, message = F, warning = F, echo=F}
num_var <- "richness"
t<-NULL
cat_var <- "group"
LMM_separate_list <-  c("richness") %>%
  purrr::set_names() %>%
  purrr::map( ~ {
    
    LMM <- TableList %>%
      pluck("group") %>%
      create_LMM(., 
               num_var = .x, 
               cat_var = "cat_var",
               long_var = "long_var",
               link_var = "link_var",
               breakpoints = t)
    
    pvals<-names(LMM[[1]]) %>%
      set_names() %>%
      
      map_dfr(~ {
        LMM[[1]][[.x]] %>%
          as.data.frame() %>%
          slice(2) %>%
          select("p-value", "Value") %>%
          as.numeric() %>%
          round(digits=3)
        }) %>% 
      
      t() %>%
      as.data.frame() %>%
      setNames(., nm= c("pval", "slope")) %>%
      rownames_to_column(var="cat_var")
    
    
      LMM[[2]] +
      geom_label(data=pvals, 
                 aes(alpha=NULL,
                     label=paste("p=",pval, ", slope = ", slope, sep=""), 
                     x=-Inf, y=Inf), color="black", hjust=0.01, vjust=1 ) +
      labs(title=.x, color = cat_var)

  }
)


LMM_separate_list$richness 

```

There is a pretty clear evolution in every one of the treatment groups. However the evolution rate seems pretty similar between both treatment groups. The models fail to find a great difference in gene richness evolution, althougha slight yet significant increasing trend (slope value = 0.027), can be found in the DGT group (Group 1).

However, creating a single LMM with treatment group as a random effect we can measure the actual difference in the effect of timepoint over gene richness between two groups. 


```{r compare_group_effect, warning = F, message= F, echo = F}

cat_vector
num_vector <- c("richness")
sam_var<-"SampleID"


myUnifiedModelList <- cat_vector %>%
  set_names() %>%
  
  map(function(cat_var) {
    print(cat_var)
    num_vector %>%
      set_names() %>%
      
      map(function(num_var) {
        dat <-
          TableList %>%
          pluck(cat_var) %>%
          select(sam_var, link_var, long_var, cat_var, num_var) %>%
          dplyr::rename(num_var = !!sym(num_var))
        
        cat_levels <- 
          dat %>% 
          dplyr::select(link_var, cat_var) %>%
          unique() %>%
          dplyr::group_by(cat_var) %>%
          dplyr::summarize(n=n()) %>%
          dplyr::filter(n>1,
                        !is.na(cat_var)) %>%
          dplyr::pull(cat_var) 
        

        dat <- 
          dat %>%
          dplyr::filter(cat_var %in% cat_levels,
                 !is.na(cat_var))
        
        
        if (length(!is.na(cat_levels)) >= 2){
        
        myUnifiedModel <- dat %>%
          lmerTest::lmer(formula = "num_var ~ long_var * cat_var + (1|link_var)", data = .)
        
        myGraphParams <- dat %>%
          get_LMM_GraphParams() %>%
          mutate(across(contains("x"), ~ as.numeric(.x)))
        
        myPval <-
          anova(myUnifiedModel) %>%

          # magrittr::extract2("coefficients") %>%
          as.data.frame() %>%
          dplyr::slice(3L) %>%
          magrittr::set_names(c("SumSq", "MeanSq", "df", "DenDF", "Fval","pval")) %>%
          dplyr::mutate(pval = round(pval, 3))
        
        myplot <- dat %>%
          ggplot(.,
                 aes(
                   x = long_var,
                   y = num_var,
                   group = link_var,
                   color = cat_var
                 )) +
          geom_point(alpha = 0.2) +
          geom_line(alpha = 0.2) +
          scale_x_continuous(breaks = unique(dat$long_var),
                             limits = c(min(dat$long_var),
                                        max(dat$long_var))) +
          
          geom_segment(
            data = myGraphParams,
            aes(
              x = xmin,
              y = ymin,
              xend = xmax,
              yend = ymax,
              color = cat_var
            ),
            inherit.aes = T,
            size = 2
          ) +
          scale_color_brewer(palette = "Set1") +
          labs(color = cat_var, 
               x = "weeks",
               y=num_var) +
          geom_label(
            data = myPval,
            aes(
              alpha = NULL,
              label = paste("ANOVA p=",
                            pval,
                            sep = ""),
              x = -Inf,
              y = Inf
            ),
            color = "black",
            hjust = 0.01,
            vjust = 1
          ) +
          theme_bw()
        
        mySummary <-
          summary(myUnifiedModel) %>%
          magrittr::extract2("coefficients") %>%
          as.data.frame() %>% 
          kableExtra::kable(., format = "markdown")
        
        return(list(plot = myplot, summary = mySummary))
        
        } else {return(NA)}
      })
    
  })

myAnovaPlotList<-names(myUnifiedModelList) %>%
  set_names() %>%
  map(~{
    myUnifiedModelList %>%
      pluck(.x) %>%
      pluck("richness") %>%
      pluck("plot")
  }) %>% 
  pluck("group")




```


This shows a slight effect of treatment, where the effect of time over richness in group group2 is 0.025 (St.err = 0.025) times lower than the same for group 1 (ANOVA p=0.31)

```{r Show_plot_list, message=F,prompt=F,echo=F, warning=F}
myAnovaPlotList
```



## Gene Richness evolution: Categorical comparisons

### Non-stratified


```{r general_boxplot, message = F, warning = F, echo = F}
dat <- TableList %>% pluck("group")
source("code/Paired_wilcox_NA_Handler.R")

long_test <- PairedWilcox_withNAs(myDF = dat,Variable = "richness",Response = "long_var",IDs = "link_var") %>% 
  rstatix::adjust_pvalue(p.col = "p") %>% 
  rstatix::add_significance(p.col = "p.adj") %>% 
  dplyr::mutate(y.position = y.position + 250000)

all_boxplot <- dat %>% 
  ggplot(., aes(x = as.factor(long_var),y=richness)) + 
  geom_boxplot(fill="lightgray") +
  theme_bw() +
  ggpubr::stat_pvalue_manual(data = long_test, label = "p.adj.signif", hide.ns = T, bracket.size = 1.1, label.size = 12) +
  labs(x = "Week") +
  theme(axis.text = element_text(size = 12),
        axis.title = element_text(size = 14))

all_boxplot
# write.csv(long_test, file = "output/general_boxplot_stats")

```

There is a general increase even without stratifying

```{r general_wilc_stats,echo = F ,message=F, warning=F}
source("code/group_comparisons.R")
mockdat <- dat %>% 
  mutate(mock = "mock") 

stats_table <- 
combn(unique(dat$long_var),2, simplify = F) %>% 
  map_dfr(function(comp){
    comp <- sort(comp)
    mockdat %>% 
    get_group_comparisons(dat = .,
                      link_var = "link_var",
                      long_var = "long_var",
                      cat_vector = "mock",
                      num_vector = "richness",
                      comps = comp,
                      type = "longitudinal"
                      ) %>% 
  pluck("mock","richness","stats") %>% 
      dplyr::mutate(group1 = comp[1], 
                    group2 = comp[2], .after=1)

  }) %>% 
      dplyr::arrange(group1,group2) %>% 
  dplyr::mutate(group1 = as.character(group1), 
                group2 = as.character(group2)) %>% 
  dplyr::full_join(long_test, by=c("group1","group2")) %>% 
  dplyr::select(cat_var, contains("group"), median1, iqr2, median2, iqr2, medianChange, iqr_change, n1, n2, statistic, p.adj) %>% 
  # kableExtra::kable(., format="markdown") %>% 
  as.data.frame()

rmarkdown::paged_table(as.data.frame(stats_table))


# write.csv(.,"output/overall_pvals.csv",quote = T,row.names = F)


```




```{r group_boxplots, echo=F, message=F}


### This was a first test to learn how to perform multiple comparisons in a true tidyverse fashion
cat_vector <- c("group")
library(rstatix)
library(ggpubr)

pal = "Accent"


dat <-
  TableList %>%
  pluck("group") %>%
  select(sam_var, link_var, long_var, cat_var, richness) 

test_cat <-
  dat %>%
  dplyr::group_by(long_var) %>%
  rstatix::wilcox_test(richness ~ cat_var, p.adjust.method = "BH") %>%
  # rstatix::adjust_pvalue(p.col = "p", method = "none") %>%
  rstatix::add_significance(p.col = "p", output.col = "p.adj.signif") %>%
  rstatix::add_xy_position("long_var") %>%
  dplyr::mutate(y.position = y.position + 250000)





test_long <-
  dat %>%
  PairedWilcox_withNAs(
    myDF = .,
    IDs = "link_var",
    Variable = "richness",
    Response = "long_var",
    Subsets = "cat_var"
  ) %>%
  group_by(cat_var) %>%
  rstatix::adjust_pvalue(p.col = "p",
                         output.col = "p.adj",
                         method = "BH") %>%
  ungroup() %>%
  # rstatix::adjust_pvalue(method = "BH") %>%
  add_significance(p.col = "p.adj") %>%
  
  # mutate(CategoricalVariable = levels(as.factor(myInnerDataFrame$CategoricalVariable))) %>%
  # add_y_position(y.trans = function(x){quantile(x, probs=c(.1))}) %>%
  mutate(
    xmin = recode(
      group1,
      `0` = 1,
      `24` = 2,
      `48` = 3
    ),
    xmax = recode(
      group2,
      `24` = 2,
      `48` = 3,
      `96` = 4
    )
  ) %>%
  mutate(xmin = case_when(cat_var == "DTG" ~ xmin + 0.2,
                          cat_var == "DRV/r" ~ xmin - 0.2),
         xmax = case_when(cat_var == "DTG" ~ xmax + 0.2,
                          cat_var == "DRV/r" ~ xmax - 0.2))


group_boxplot <-
  dat %>%
  ggplot(., aes(x = as.factor(long_var), y = richness)) +
  geom_boxplot(aes(fill = cat_var), position = "dodge") +
  ggpubr::stat_pvalue_manual(
    data = test_cat,
    label = "p.adj.signif",
    hide.ns = T,
    size = 6,
    step.increase = 0.1,
    bracket.size = 1
  ) +
  ggpubr::stat_pvalue_manual(
    data = test_long,
    label = "p.adj.signif",
    hide.ns = T,
    step.increase = 0.2,
    size = 6,
    bracket.size = 1
    
  ) +
  theme_bw() +
  scale_fill_brewer(palette = pal) +
  labs(y = "Gene richness (count)", x="weeks",fill = "group") +
  theme(
    axis.text = element_text(size = 12),
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
    legend.key.size = unit(1.3,"cm"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 12)
  )


group_boxplot

```

This is even more intereseting, as most of the significant tchange happens in the DTG group, with significant differences in all time points, and significantly higher richnes in DTG vs DRV/r is found in al timepoints except 0.

```{r group_stats,echo = F ,warning=F, message=FALSE}

group_stats <- 
combn(unique(dat$long_var),2, simplify = F) %>% 
  map_dfr(function(comp){
    comp <- sort(comp)
    mockdat %>% 
    get_group_comparisons(dat = .,
                      link_var = "link_var",
                      long_var = "long_var",
                      cat_vector = "cat_var",
                      num_vector = "richness",
                      comps = comp,
                      type = "longitudinal"
                      ) %>% 
  pluck("cat_var","richness","stats") %>% 
      dplyr::mutate(group1 = comp[1], 
                    group2 = comp[2], .after=1)

  }) %>% 
      dplyr::arrange(cat_var,group1,group2) %>% 
  dplyr::mutate(group1 = as.character(group1), 
                group2 = as.character(group2)) %>% 
  dplyr::full_join(test_long, by=c("cat_var","group1","group2")) %>% 
  dplyr::select(cat_var, contains("group"), median1, iqr2, median2, iqr2, medianChange, iqr_change, n1, n2, statistic, p.adj) %>% 
  as.data.frame()


rmarkdown::paged_table(group_stats)

```

