---
title: "Gene Richness vs CD4"
author: "Carlos Blazquez"
date: "11/11/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
library("ggpubr")
source("code/LMM.R")
source("code/myFunctions.R")


```

```{r input_data, warning=F, message=F}
metadata <- 
  here::here("Metadata") %>% 
  file.info(list.files(., full.names = T)) %>%
  rownames() %>% 
  magrittr::extract(str_detect(.,"clean_metadata")) %>% 
  sort(decreasing = T) %>% 
  magrittr::extract(1L) %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))

mymre <- aws.s3::s3readRDS(bucket = "mistral-wp6-advanz4",object = "metagenome/WMGS/MREObject.rds", region = "eu-west-1")



```

```{r update_mre, warning= F, message= F, echo = F, results=F, eval = F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here("../WMGSPipeline"))

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)




mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

cat_vector <- get_cat(mymre) %>% 
  pull(CategoricalVariable)

num_vector <- 
  get_num(mymre) %>% 
  pull(NumericalVariable)

```


## Gene Richness vs CD4

In this analysis we are interested in the relationship between CD4/CD4 recovery and gene richness. For that, several questions are of importance.

### CD4 baseline distribution

We already know that CD4 mean(sd) is similar for DTG/DRVr groups, but we have not checked the distribution. Is it unimodal or bimodal? a density plot should help.

```{r basal_CD4, warning = F, message = F}
basal_df <- 
metadata %>% 
  # dplyr::mutate(delta_cd4 = CD4[time_point == 48] - CD4[time_point == 0])
  dplyr::filter(time_point == 0) %>% 
  dplyr::select(SampleID, group, CD4)  
  
median_df <- basal_df %>% 
  dplyr::group_by(group) %>% 
  summarise(med = median(CD4))
  

peak_finder <-
  function(density, goal = "max",keep = F) {
    ### Let's define this function heere but we'll use it later
    stopifnot("goal can only be either local maxima (use: 'max') or local minima (use 'min)" = goal %in% c("min","max"))
    
    val <- case_when(goal == "max" ~ -2,
                   goal == "min" ~ 2)
    
    res <-
      density %>%
      purrr::keep(names(.) %in% c("x", "y")) %>%
      as.data.frame() %>%
      dplyr::mutate(is_peak = c(F, # 1st value can't be a local maxima, as there is no prior value to compare
                                diff(y) %>% # For each nth element of the vector, get (n+1) - n
                                  sign() %>%  # convert to -1 (n+1 is lower than n) or +1 (n+1 is higher than n)
                                  diff() == val, # if n is at a local maxima, both n+1 and n-1 are negative so [(n+1)-n] - [n - (n-1)] is -2
                                F)) # Last element can't be a local maxima either)
                    
                    if (!keep) {
                      res %>%
                        dplyr::filter(is_peak) %>%
                        dplyr::select(-is_peak)
                    } else{
                      res %>%
                        dplyr::select(-is_peak)
                    }
  }


peak_df <- unique(basal_df$group) %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(function(arm){
   basal_df %>% 
      dplyr::filter(group == arm) %>% 
      pull(CD4) %>% 
      density(n=nrow(.)) %>% 
      peak_finder(., goal = "max",keep = F) %>% 
      mutate(group = arm)

  })



  
  
  
plt <- ggplot(basal_df, aes(x=CD4, color=group)) +


    # geom_histogram(aes(fill = group), bins = 1000) +
    geom_density( alpha = .4) +
  theme_bw() +
  scale_color_brewer(palette = "Accent") +
  scale_fill_brewer(palette = "Accent") +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) +
  geom_text(data = median_df, aes(label = paste("med =",med), x = med, y = c(0.012,0.015)), color = "black") +
  # geom_vline(data = peak_df, aes(xintercept =  x),color = "black",lty = 3) +
  labs (x = "CD4 (counts)")
  # geom_errorbar(stat = "hline", xintercept = "median",
  # width=0.8,aes(xmax=..x..,xmin=..x..))

  
  
plt
peak_df

```
It appears the distribution of CD4 at BL shows two peaks in kernel density, one at 26 counts/ml and one at 229 count/ml. There is a second peak in the DRV as well, but it seems the density function smoothed it out. 


### GeneRichness at baseline distribution

We know (I think) that there is no difference in generichness between arme at baseline, but we do not know the distribution. In Guillen et al, we found a bimodal distribution in HIV patients. Is that the case of advanz4?


```{r GR_basal, warning= F, message= F}

igc<-metar::get_diversity(mymre)@igc
Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()

TableList <- cat_vector %>%
    purrr::set_names() %>%
    map( ~{

        
          ThresholdTable <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          right_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=record_id,
                 cat_var = .x,
                 long_var = time_point,
                 NumberMappedReads)
  

        
    })

basal_df <- 
  TableList %>% 
  pluck("group") %>% 
  dplyr::filter(long_var == 0)


median_df <- 
  basal_df %>% 
  dplyr::rename(group = cat_var) %>% 
  dplyr::group_by(group) %>% 
  filter(!is.na(richness)) %>% 
  summarise(med = median(richness))

  
plt <- 
basal_df %>% 
  ggplot(aes(x = richness, fill = cat_var)) +
  # geom_histogram(aes(y = ..density..),alpha = .4) +
  geom_density(alpha = .4) +
  theme_bw() +
  scale_fill_brewer(palette="Accent") +
  scale_color_brewer(palette = "Accent") +
  labs(x="gene_count", fill = "group") +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) 


peak_df <- 
basal_df %>% 
  pull(cat_var) %>% 
  unique() %>% 
  map_dfr(function(arm){
    basal_df %>% 
      dplyr::filter(cat_var == arm) %>%
      dplyr::filter(!is.na(richness)) %>% 
      dplyr::pull(richness) %>% 
      density() %>%
      peak_finder() %>% 
      dplyr::mutate(group = arm)

  })

plt
peak_df

```


It seems both groups have a similar median richness, but with somewhat different distributions, with DTG following a slightly bimodal distribution with a higher median peak than Drv/r. Although the left "step" in may look like a local maxima, it actually does not decrease on the right side.


### Gene Richness vs CD4 at baseline


Is there any association between gene richness and CD4 at baseline only, overall, and per-group(DTG and DRVr only)?
```{r GR_vs_CD4, message = F, warning=FALSE}
df <- 
basal_df %>% 
  right_join(metadata[,c("SampleID","CD4")]) %>% 
  dplyr::filter(!is.na(richness),
                !is.na(CD4))


overall <- cor.test(x= df$richness, y=df$CD4, method = "spearman") %>% 
  keep(names(.) %in% c("statistic","estimate", "p.value"))


per_group <- 
  unique(metadata$group) %>% 
  set_names() %>% 
  purrr::map(function(group){
    df_1 <- 
      df %>% 
      filter(cat_var == group) %>% 
      select(richness, CD4) 
    cor.test(x=df_1$richness, y = df_1$CD4, method = "spearman") %>% 
      keep(names(.) %in% c("statistic", "estimate", "p.value")) %>% 
    as.data.frame()
  })

overall
per_group

```

Doesn't seem to be any signifcant association between gene richness and CD4 counts at basal.

### Delta-CD4 and DeltaGR

```{r Numeric_deltas, warning=F, message=F}



deltas <- 
TableList %>% 
  pluck("group") %>% 
  dplyr::right_join(., metadata[,c("SampleID","CD4")], by="SampleID") %>% 
  dplyr::group_by(link_var) %>% 
  # dplyr::filter(long_var %in% c(0,96)) %>% 
  dplyr::filter(any(long_var == 0)) %>%
  dplyr::mutate(GR_abs = richness - richness[long_var == 0],
                GR_rel = (richness - richness[long_var == 0])/(richness + richness[long_var == 0]),
                CD4_abs = CD4- CD4[long_var == 0],
                CD4_rel = (CD4 - CD4[long_var == 0])/(CD4 + CD4[long_var == 0]))


```


We need to check if the increments in CD4 are linked to increments in GeneRichness. Let's calculate these increments in two different ways:

- Absolute: CD4(tx)-CD4(t0) or GR(tx)-GR(to)
- Relative: (CD4(tx)-CD(t0)/(CD4(tx)+CD4(t0))) or (GR(tx)-GR(t0)/(GR(tx)+GR(t0)))

Check distribution of these values and see if there is a possibility of reasonable categorization.

```{r plot_dists, warning=F, results=F, message=F, echo = F}
median_df_overall <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48, 96)) %>% 
    mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  dplyr::group_by(time_point) %>% 
  filter(!is.na(GR_rel)) %>% 
  summarise(med = median(GR_rel)) %>% 
  dplyr::mutate(med = round(med, 3))

plt_overall <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  # dplyr::filter(!is.na(GR_rel)) %>% 
  ggplot(aes(x = GR_rel)) +
  facet_wrap(~time_point) +
  geom_density() +
  theme_bw() +
  geom_vline(data = median_df_overall,aes(xintercept = med)) +
  geom_text(data = median_df_overall, aes(label = paste("med =",med), x = med, y = 1.5), color = "black") +
  labs(x="richness increase (relative)") +
  theme(strip.background = element_blank(),
        strip.text = element_text(face = "bold", size = 12))

median_df_group <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48, 96)) %>% 
  dplyr::mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  dplyr::group_by(time_point, cat_var) %>% 
  dplyr::filter(!is.na(GR_rel)) %>% 
  summarise(med = median(GR_rel)) %>% 
  dplyr::mutate(med = round(med, 3))

plt_group <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  # dplyr::filter(!is.na(GR_rel)) %>% 
  ggplot(aes(x = GR_rel, color = cat_var)) +
  facet_wrap(~time_point) +
  geom_density() +
  theme_bw() +
  geom_vline(data = median_df_group,aes(xintercept = med, color = cat_var)) +
  geom_text(data = median_df_group, aes(label = paste("med =",med), x = med, y = c(2,2.5,2.5,3))) +
  labs(x="richness increase (relative)", color = "group", y = "density") +
  theme(strip.background = element_blank(),
        strip.text = element_text(face = "bold", size = 12)) +
  scale_color_brewer(palette = "Accent")


plt_overall
plt_group

```
As we can see, the distribution of Gene Richness change vs baseline is pretty similar at both weeks 488 and 96, albeit week 96 has a slightly higher median. The differences get bigger when comparing between groups though, as DTG seems significantly bigger than DRV, especially at week 96 (DTG=0.88 vs DRV/r = 0.39 median delta. Let's test with a wilcoxon test)

```{r test_deltaGR, warning=F, message=F}
long_deltas <- 
deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  group_by(link_var) %>%
  dplyr::filter(!is.na(GR_rel)) %>% 
  dplyr::filter(n()==2) %>% 
  dplyr::group_by(cat_var, .drop= T) %>% 
  rstatix::wilcox_test(., GR_rel ~ long_var, paired = T) %>% 
  dplyr::select(-.y.) %>% 
  setNames(c("time_point", "time1", "time2", "n1","n2", "stat","p")) %>% 
  as.data.frame() %>% 
  kableExtra::kable(format = "markdown")


cat_deltas <- 
deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  group_by(long_var) %>% 
  dplyr::filter(!is.na(GR_rel)) %>% 
  rstatix::wilcox_test(., GR_rel ~ cat_var) %>% 
  dplyr::select(-.y.) %>% 
  setNames(c("time_point", "group1", "group2", "n1","n2", "stat","p")) %>% 
  as.data.frame() %>% 
  kableExtra::kable(format = "markdown")

long_deltas
cat_deltas

```

The difference in deltas between groups is significant at week 96 but not at 48. However, no significant differences were found longitudinally between groups.

# Correlations between CD4 and GR

Let's check if there is a correlation between both variables. First overall:

```{r overall_corrs, warning=F, message=F}
absolute_cors <- 
deltas %>% 
  dplyr::filter(!long_var == 0) %>% 
  correlation::cor_test(., "CD4_abs", "GR_abs", method ="spearman")

plot_overall_abs <- deltas %>% 
  ggplot(aes(x=CD4_abs, y = GR_abs)) +
  geom_point(aes(color = cat_var)) +
  geom_smooth(method = "glm") +
  theme_bw() + 
  labs(color = "group", title = "Absolute")


relative_cors <- 
deltas %>% 
  dplyr::filter(!long_var == 0) %>% 
  correlation::cor_test(., "CD4_rel", "GR_rel", method ="spearman")

plot_overall_rel <- deltas %>% 
  ggplot(aes(x=CD4_rel, y = GR_rel)) +
  geom_point(aes(color = cat_var)) +
  geom_smooth(method = "glm") +
  theme_bw() +
  labs(color = "group", title = "relative")


absolute_cors


deltas %>% 
  glm(data = ., formula = "CD4_rel ~ GR_rel")

relative_cors
ggarrange(plot_overall_abs, plot_overall_rel, common.legend = T)

```
No overall correlations could be found. Let's check split per timepoint:


```{r correlations, warning = F, message=F}
c(48,96) %>% 
  purrr::set_names() %>% 
  purrr::map(function(tp){
  deltas %>% 
  dplyr::filter(long_var == tp) %>% 
  correlation::cor_test(., "CD4_rel", "GR_rel", method ="spearman")

  })
```

No Meaningful correlations could be found neither overall nor at specific timepoints.

# Interactions between CD4 and GR: LMMs

To have a better view of wether both variables interact, we will perform LMMs. First on overall:

```{r LMMs_overall, warning=F,message=FALSE}
lmm_abs <- 
deltas %>% 
  mutate(mock = 1) %>% ## The function needs a cat_var, so we'll create a mock one
  dplyr::filter(!is.na(CD4_abs), !is.na(GR_abs), !long_var == 0) %>% 
  # dplyr::rename(record_id = link_var) %>% 
  create_LMM(., num_var = "CD4_abs", long_var = "GR_abs", link_var =  "link_var", cat_var = "mock")
stat_abs <- lmm_abs$stats
plt_abs <- lmm_abs$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = "delta richness", y = "delta CD4", title = "Absolute") +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))



lmm_rel <- 
deltas %>% 
  mutate(mock = 1) %>% ## The function needs a cat_var, so we'll create a mock one
  dplyr::filter(!is.na(CD4_rel), !is.na(GR_rel), !long_var == 0) %>% 
  # dplyr::rename(record_id = link_var) %>% 
  create_LMM(., num_var = "CD4_rel", long_var = "GR_rel", link_var =  "link_var", cat_var = "mock")
stat_rel <- lmm_rel$stats
plt_rel <- lmm_rel$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = "delta richness", y = "delta CD4", title = "Relative")+
  facet_wrap(NULL) +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))


ggpubr::ggarrange(plt_abs, plt_rel)


```

nothing could be observed overall. Let's check per group:

```{r LMMs_group, warning=F,message=FALSE}
lmm_abs <- 
deltas %>% 
  dplyr::filter(!is.na(CD4_abs), !is.na(GR_abs), !long_var == 0) %>% 
  # dplyr::rename(record_id = link_var) %>% 
  create_LMM(., num_var = "CD4_abs", long_var = "GR_abs", link_var =  "link_var", cat_var = "cat_var")
stat_abs <- lmm_abs$stats
plt_abs <- lmm_abs$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = "delta richness", y = "delta CD4", title = "Absolute") +
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))



lmm_rel <- 
deltas %>% 
  dplyr::filter(!is.na(CD4_rel), !is.na(GR_rel), !long_var == 0) %>% 
  # dplyr::rename(record_id = link_var) %>% 
  create_LMM(., num_var = "CD4_rel", long_var = "GR_rel", link_var =  "link_var", cat_var = "cat_var")
stat_rel <- lmm_rel$stats
plt_rel <- lmm_rel$plot +
  scale_x_continuous(n.breaks = 4) +
  labs(x = "delta richness", y = "delta CD4", title = "Relative")+
  theme(axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12, angle = 90))


ggpubr::ggarrange(plt_abs, plt_rel, nrow = 2, ncol = 1)


```

No quantitative difference could be found between delta GR and delta CD4. The only thing to look at now is to check wether there is a significant GR differences between those who recovered more or less than 150 CD4-

# Comparisons by CD4 recovery
We did not find many signals correlating numerical absolute and relative numbers of CD4 and Gene Richness. Now we'r going to check if there is any actual difference in Gene richness in those with CD4 recovery < 150.
The patients have been stratified in 3 groups: those who reached an increase of 150 counts/ml at week 48 (48), those who did at week 96 (96) and those who didn't surpass this threshold at all (No)

```{r richness_CD4_categorical}
df <- 
deltas %>% 
  left_join(metadata[,c("SampleID", "CD4diff_48", "CD4diff_96")], by = "SampleID") %>% 
  # mutate(CD4diff_48 = recode(CD4diff_48, "150_H" = "48_H", "150_L" = "48_L"),
  #        CD4diff_96 = recode(CD4diff_96, "150_H" = "96_H", "150_L" = "96_L")) %>% 
  mutate(recovery = case_when(
    CD4diff_48 == "150_H" ~ "48",
    CD4diff_48 == "150_L" & CD4diff_96 == "150_H" ~ "96",
    CD4diff_48 == "150_L" & CD4diff_96 == "150_L" ~ "No",
    is.na(CD4diff_48) & CD4diff_96 == "150_H" ~ "96",
    is.na(CD4diff_48) & CD4diff_96 == "150_L" ~ "No")) %>% 
  filter(!is.na(recovery), !long_var == 0)%>% 
  dplyr::mutate(rec_group = paste(recovery, cat_var, sep = "_"),
                recovery = factor(recovery, levels = c("No", "96","48")))


color_df <- 
  data.frame(group = unique(df$cat_var)) %>% 
  dplyr::mutate( color = case_when(group == "DTG" ~ viridis::magma(2)[1],
                               group == "bDRV" ~ viridis::magma(2)[2]))


test_cat <- 
df %>% 
  dplyr::mutate(rec_group = paste(recovery, cat_var, sep = "_")) %>% 
  group_by(long_var) %>% 
  rstatix::wilcox_test(richness ~ rec_group) %>% 
  rstatix::add_xy_position(x = "long_var", group = "rec_group")
plt_abs <-
  df %>% 
  dplyr::select(long_var, richness, cat_var, recovery) %>% 
  dplyr::filter(!is.na(richness)) %>% 
  group_by(recovery, cat_var, long_var) %>% 
  dplyr::summarise(med_richness = median(richness, na.rm = T),
                   rich_sd = sd(richness, na.rm = T)) %>% 
  mutate(rec_group = paste(recovery, cat_var, sep = "_")) %>% 
  ggplot(., aes(x = factor(long_var), y = med_richness)) +
  # geom_boxplot(aes(fill = cat_var)) +
  # geom_jitter(aes(colour = rec_group )) +
  geom_point(aes(group = rec_group, shape = cat_var, color = recovery, alpha = recovery),size = 3, position = position_dodge(.8)) +
  geom_line(aes(lty = cat_var, group = rec_group, color = recovery, alpha = recovery), position = position_dodge(0.8), lwd = 1) +
  # geom_line(data = df, aes(x = factor(long_var), y = richness, group = link_var, color = recovery, lty = cat_var), inherit.aes = F, alpha = .4) +
  geom_errorbar(aes(ymin = med_richness - rich_sd, ymax = med_richness + rich_sd, group = rec_group, color = recovery, alpha = recovery ), width=.2, position = position_dodge(0.8)) +
  scale_alpha_manual(values = c("No",96,48),breaks = c(.6,.8,1)) +
  labs(x = "weeks", y = "gene richness") +
  # scale_color_brewer(palette = "Dark2") +
  scale_color_viridis_d(option = "magma", begin = .1, end = .8, direction = 1) +
  theme_bw() +
  stat_pvalue_manual(test_cat,inhertit.aes = F ,hide.ns = T) +
  theme(
    axis.title = element_text(face = "bold")
  )

ggsave(filename = "richness_reconstitution.svg",path = here::here("output","figures"), plot = plt_abs,device = "svg", width =16, height = 10, units = "cm")





lmm_abs <- 
df %>% 
 get_lmm_effects(., cat_vector = c("recovery"),num_vector = "richness", link_var = "link_var", long_var = "long_var")  



plt_abs
lmm_plt_abs <- 
lmm_abs$recovery$richness$plot +
  scale_color_viridis_d(option = "magma", begin = .1, end = .8, direction = 1) 

ggsave(plot = lmm_plt_abs, device = "svg",path = here::here("output", "figures"), width = 16, height = 10, units = "cm", dpi = 300, filename = "LMMs_GR.svg")
lmm$recovery$richness$summary
```

No differences in gene richness could be found between those who recovered more than <150 CD4+/ml at week 48 and those who did at week 96. We'll test for deltaGR now:

```{r delta_GR_CD4_cat, warning = F, message=F}

df <- 
deltas %>% 
  left_join(metadata[,c("SampleID", "CD4diff_48", "CD4diff_96")], by = "SampleID") %>% 
  # mutate(CD4diff_48 = recode(CD4diff_48, "150_H" = "48_H", "150_L" = "48_L"),
  #        CD4diff_96 = recode(CD4diff_96, "150_H" = "96_H", "150_L" = "96_L")) %>% 
  mutate(recovery = case_when(
    CD4diff_48 == "150_H" ~ "48",
    CD4diff_48 == "150_L" & CD4diff_96 == "150_H" ~ "96",
    CD4diff_48 == "150_L" & CD4diff_96 == "150_L" ~ "No",
    is.na(CD4diff_48) & CD4diff_96 == "150_H" ~ "96",
    is.na(CD4diff_48) & CD4diff_96 == "150_L" ~ "No")) %>% 
  filter(!is.na(recovery), !long_var == 0)

test_cat <- 
df %>% 
  group_by(long_var) %>% 
  rstatix::wilcox_test(GR_rel ~ recovery) %>% 
  rstatix::add_xy_position(x = "long_var", group = "recovery")

plt_delta <- 
df %>% 
  dplyr::select(long_var, GR_rel, cat_var, recovery) %>% 
  dplyr::filter(!is.na(GR_rel)) %>% 
  group_by(recovery, cat_var, long_var) %>% 
  dplyr::summarise(med_richness = median(GR_rel, na.rm = T),
                   rich_sd = sd(GR_rel, na.rm = T)) %>% 
  mutate(rec_group = paste(recovery, cat_var, sep = "_")) %>% 

  ggplot(., aes(x = factor(long_var), y = med_richness)) +
  # geom_boxplot(aes(fill = cat_var)) +
  # geom_jitter(aes(colour = rec_group )) +
  geom_point(aes(group = rec_group, shape = cat_var, color = recovery, alpha = recovery),size = 3, position = position_dodge(.8)) +
  geom_line(aes(lty = cat_var, group = rec_group, color = recovery, alpha = recovery), position = position_dodge(0.8), lwd = 1) +
  # geom_line(data = df, aes(x = factor(long_var), y = richness, group = link_var, color = recovery, lty = cat_var), inherit.aes = F, alpha = .4) +
  geom_errorbar(aes(ymin = med_richness - rich_sd, ymax = med_richness + rich_sd, group = rec_group, color = recovery, alpha = recovery ), width=.2, position = position_dodge(0.8)) +
  scale_alpha_manual(values = c("No",96,48),breaks = c(.6,.8,1)) +
  labs(x = "weeks", y = "gene richness increase (relative)") +
  # scale_color_brewer(palette = "Dark2") +
  scale_color_viridis_d(option = "magma", begin = .1, end = .8, direction = 1) +
  theme_bw() +
  stat_pvalue_manual(test_cat,inhertit.aes = F ,hide.ns = T) +
  theme(
    axis.title = element_text(face = "bold")
  )


ggsave(filename = "richness_reconstitution_relative.svg",path = here::here("output","figures"), plot = plt_delta, device = "svg", width =16, height = 10, units = "cm")





lmm_delta <- 
df %>% 
 get_lmm_effects(., cat_vector = c("recovery"),num_vector = "GR_rel", link_var = "link_var", long_var = "long_var")


plt_delta
lmm_plt_delta <- 
lmm_delta$recovery$GR_rel$plot +
  labs(y = "gene richness increase (relative)") +
  scale_color_viridis_d(option = "magma", begin = .1, end = .8, direction = 1) 

lmm_plt_delta
ggsave(plot = lmm_plt_delta, device = "svg",path = here::here("output", "figures"), width = 16, height = 10, units = "cm", dpi = 300, filename = "LMMs_GRrel.svg")


lmm_delta$recovery$GR_rel$summary
```


```{r save_plots, warning = F, message = F}
plt_abs <- plt_abs +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),
        plot.title = element_text(size = 12, face = "bold")) +
  labs(title = "A")
plt_delta <- plt_delta +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),
        plot.title = element_text(size = 12, face = "bold")) +
  labs(title = "B")

linePlots <- 
ggpubr::ggarrange(plt_abs, plt_delta, common.legend = T, legend = "right")


lmm_plt_abs <- 
lmm_plt_abs +
  labs(title = "C", y = "gene richness") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),
        axis.title.y = element_text(face = "bold"),
        plot.title = element_text(size = 12, face = "bold"))

lmm_plt_delta <- 
  lmm_plt_delta +
  labs(title = "D") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10),
        axis.title.y = element_text(face = "bold"),
        plot.title = element_text(size = 12, face = "bold")) 

Lmms <- 
  ggpubr::ggarrange(lmm_plt_abs, lmm_plt_delta, common.legend = T, legend = "right")

plot_grid <- 
ggpubr::ggarrange(linePlots, Lmms, nrow = 2, ncol = 1) %>% 
  annotate_figure(bottom = text_grob(label = "Weeks", face = "bold", size = 12)) 

ggsave(filename = "recovery_grid.svg", device = "svg", 
       path = here::here("output", "figures"), 
       width = 24, height = 16, units = "cm", dpi = 300)


```


Finally, we are going to tests if overall having an increase higher than 150 counts is tied to different deltaGR.

```{r diff_CD4_test, warning = F, message = F}

dif48_abs <- 
df %>%
  ungroup() %>% 
  dplyr::filter(long_var == 48) %>% 
  rstatix::wilcox_test(GR_abs ~ CD4diff_48)


dif48_rel <- 
  df %>%
  ungroup() %>%
  dplyr::filter(long_var == 48) %>% 
  rstatix::wilcox_test(GR_rel ~ CD4diff_48)
dif_48 <- rbind(dif48_abs, dif48_rel)


plt_48_abs <- 
df %>% 
  ungroup() %>% 
  dplyr::filter(long_var %in% c(48),
                !is.na(CD4diff_48)) %>% 
  ggplot(aes(y = GR_abs, fill = CD4diff_48)) +
  geom_boxplot() +
    theme_bw() +
  labs(title = "Absolute")

plt_48_rel <- 
df %>% 
  ungroup() %>% 
  dplyr::filter(long_var %in% c(48),
                !is.na(CD4diff_48)) %>% 
  ggplot(aes(y = GR_rel, fill = CD4diff_48)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = "Relative")

dif96_abs <- 
  df %>%
  ungroup() %>%
  dplyr::filter(long_var == 96) %>% 
  rstatix::wilcox_test(GR_abs ~ CD4diff_96)


dif96_rel <- 
  df %>%
  ungroup() %>%   
  dplyr::filter(long_var == 96) %>% 
  rstatix::wilcox_test(GR_rel ~ CD4diff_96)
dif_96 <- rbind(dif96_abs, dif96_rel)


plt_96_abs <- 
df %>% 
  ungroup() %>% 
  dplyr::filter(long_var %in% c(96),
                !is.na(CD4diff_96)) %>% 
  ggplot(aes(y = GR_abs, fill = CD4diff_96)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = "Absolute")


plt_96_rel <- 
df %>% 
  ungroup() %>% 
  dplyr::filter(long_var %in% c(96),
                !is.na(CD4diff_96)) %>% 
  ggplot(aes(y = GR_rel, fill = CD4diff_96)) +
  geom_boxplot() + 
  theme_bw() +
  labs(title = "Relative")



dif_48
ggpubr::ggarrange(plt_48_abs, plt_48_rel, common.legend = T)

dif_96
ggpubr::ggarrange(plt_96_abs, plt_96_rel, common.legend = T)


```

No differences could be observed between those who had >150 CD4 count increase at either week 96 or week 48.

```{r Chisq_test, warning=F,message=F}
contingency_table <- 
df %>% 
  ungroup() %>% 
  dplyr::select(cat_var,recovery) %>% 
  dplyr::group_by(cat_var, recovery) %>% 
  tally() %>% 
  pivot_wider(values_from = n, names_from = recovery) 


contingency_table

contingency_table %>% 
  column_to_rownames("cat_var") %>% 
  chisq.test()



```

It appears, belonging to either treatment group does not affect the odds of recovering <150 CD4 counts at week 48 or 96.

And plot:

- Values vs timepoint, overall and per-group(DTG and DRVr only). Fit ANOVA/t-test vs DTG/DRVr at each timepoint to check for differential increases per ARM
- AbsCD4 vs AbsGR, overall and per-group(DTG and dRVr only). Fit LM
- RelCD4 vs RelGR, overall and per-group(DTG and DRVr only). Fit LM

### CAtegorized CD4

This has partially been done. Split patients between 150>CD4(t48)/(t96)>150, and check if DeltaGR is different for groups and fisher.test of these groups vs arm is significant.
