---
title: "Gene Richness vs CD4"
author: "Marc Noguera"
date: "11/11/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
source("code/LMM.R")
source("code/myFunctions.R")

```

```{r input_data, warning=F, message=F}
metadata <- 
  here::here("Metadata") %>% 
  file.info(list.files(., full.names = T)) %>%
  rownames() %>% 
  magrittr::extract(str_detect(.,"clean_metadata")) %>% 
  sort(decreasing = T) %>% 
  magrittr::extract(1L) %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))

mymre <- aws.s3::s3readRDS(bucket = "mistral-wp6-advanz4",object = "metagenome/WMGS/MREObject.rds", region = "eu-west-1")



```

```{r update_mre, warning= F, message= F, echo = F, results=F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here("../WMGSPipeline"))

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

cat_vector <- get_cat(mymre) %>% 
  pull(CategoricalVariable)

```


## Gene Richness vs CD4

In this analysis we are interested in the relationship between CD4/CD4 recovery and gene richness. For that, several questions are of importance.

### CD4 baseline distribution

We already know that CD4 mean(sd) is similar for DTG/DRVr groups, but we have not checked the distribution. Is it unimodal or bimodal? a density plot should help.

```{r basal_CD4}
basal_df <- 
metadata %>% 
  # dplyr::mutate(delta_cd4 = CD4[time_point == 48] - CD4[time_point == 0])
  dplyr::filter(time_point == 0) %>% 
  dplyr::select(SampleID, group, CD4)  
  
median_df <- basal_df %>% 
  dplyr::group_by(group) %>% 
  summarise(med = median(CD4))
  

peak_finder <-
  function(density, keep = F) {
    ### Let's define this function heere but we'll use it later
    res <-
      density %>%
      purrr::keep(names(.) %in% c("x", "y")) %>%
      as.data.frame() %>%
      dplyr::mutate(is_peak = c(F, # 1st value can't be a local maxima, as there is no prior value to compare
                                diff(y) %>% # For each nth element of the vector, get (n+1) - n
                                  sign() %>%  # convert to -1 (n+1 is lower than n) or +1 (n+1 is higher than n)
                                  diff() == -2, # if n is at a local maxima, both n+1 and n-1 are negative so [(n+1)-n] - [n - (n-1)] is -2
                                F)) # Last element can't be a local maxima either)
                    
                    if (!keep) {
                      res %>%
                        dplyr::filter(is_peak) %>%
                        dplyr::select(-is_peak)
                    } else{
                      res %>%
                        dplyr::select(-is_peak)
                    }
  }


peak_df <- unique(basal_df$group) %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(function(arm){
   basal_df %>% 
      dplyr::filter(group == arm) %>% 
      pull(CD4) %>% 
      density(n=nrow(.)) %>% 
      peak_finder(., keep = F) %>% 
      mutate(group = arm)

  })



  
  
  
plt <- ggplot(basal_df, aes(x=CD4, color=group)) +


    # geom_histogram(aes(fill = group), bins = 1000) +
    geom_density( alpha = .4) +
  theme_bw() +
  scale_color_brewer(palette = "Accent") +
  scale_fill_brewer(palette = "Accent") +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) +
  geom_text(data = median_df, aes(label = paste("med =",med), x = med, y = c(0.012,0.015)), color = "black")
  # geom_vline(data = peak_df, aes(xintercept =  x),color = "black",lty = 3) +)
  labs (x = "CD4 (counts)")
  # geom_errorbar(stat = "hline", xintercept = "median",
  # width=0.8,aes(xmax=..x..,xmin=..x..))

  
  
plt
peak_df

```
It appears the distribution of CD4 at BL shows two peaks in kernel density, one at 26 counts/ml and one at 229 count/ml. There is a second peak in the DRV as well, but it seems the density function smoothed it out. 


### GeneRichness at baseline distribution

We know (I think) that there is no difference in generichness between arme at baseline, but we do not know the distribution. In Guillen et al, we found a bimodal distribution in HIV patients. Is that the case of advanz4?


```{r GR_basal, warning= F, message= F}

igc<-metar::get_diversity(mymre)@igc
Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()

TableList <- "group" %>%
    set_names() %>%
    map( ~{

        
          ThresholdTable <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          right_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=record_id,
                 cat_var = .x,
                 long_var = time_point,
                 NumberMappedReads)
  

        
    }) %>% 
  purrr::pluck("group") 

basal_df <- 
  TableList %>% 
  dplyr::filter(long_var == 0)


median_df <- 
  basal_df %>% 
  dplyr::rename(group = cat_var) %>% 
  dplyr::group_by(group) %>% 
  filter(!is.na(richness)) %>% 
  summarise(med = median(richness))

  
plt <- 
basal_df %>% 
  ggplot(aes(x = richness, fill = cat_var)) +
  # geom_histogram(aes(y = ..density..),alpha = .4) +
  geom_density(alpha = .4) +
  theme_bw() +
  scale_fill_brewer(palette="Accent") +
  scale_color_brewer(palette = "Accent") +
  labs(x="gene_count", fill = "group") +
  geom_vline(data = median_df, aes(xintercept =  med, color = group), lwd = 1.5, lty = 2) 


peak_df <- 
basal_df %>% 
  pull(cat_var) %>% 
  unique() %>% 
  map_dfr(function(arm){
    basal_df %>% 
      dplyr::filter(cat_var == arm) %>%
      dplyr::filter(!is.na(richness)) %>% 
      dplyr::pull(richness) %>% 
      density() %>%
      peak_finder() %>% 
      dplyr::mutate(group = arm)

  })

plt
peak_df

```


It seems both groups have a similar median richness, but with somewhat different distributions, with DTG following a slightly bimodal distribution with a higher median peak than Drv/r. Although the left "step" in may look like a local maxima, it actually does not decrease on the right side.


### Gene Richness vs CD4 at baseline


Is there any association between gene richness and CD4 at baseline only, overall, and per-group(DTG and DRVr only)?
```{r GR_vs_CD4, message = F, warning=FALSE}
df <- 
basal_df %>% 
  right_join(metadata[,c("SampleID","CD4")]) %>% 
  dplyr::filter(!is.na(richness),
                !is.na(CD4))


overall <- cor.test(x= df$richness, y=df$CD4, method = "spearman") %>% 
  keep(names(.) %in% c("statistic","estimate", "p.value"))


per_group <- 
  unique(metadata$group) %>% 
  set_names() %>% 
  purrr::map(function(group){
    df_1 <- 
      df %>% 
      filter(cat_var == group) %>% 
      select(richness, CD4) 
    cor.test(x=df_1$richness, y = df_1$CD4, method = "spearman") %>% 
      keep(names(.) %in% c("statistic", "estimate", "p.value")) %>% 
    as.data.frame()
  })

overall
per_group

```

Doesn't seem to be any signifcant association between gene richness and CD4 counts at basal.

### Delta-CD4 and DeltaGR

```{r Numeric_deltas, warning=F, message=F}



deltas <- 
TableList %>% 
  dplyr::right_join(., metadata[,c("SampleID","CD4")], by="SampleID") %>% 
  dplyr::group_by(link_var) %>% 
  # dplyr::filter(long_var %in% c(0,96)) %>% 
  dplyr::filter(any(long_var == 0)) %>%
  dplyr::mutate(GR_abs = richness - richness[long_var == 0],
                GR_rel = (richness - richness[long_var == 0])/(richness + richness[long_var == 0]),
                CD4_abs = CD4- CD4[long_var == 0],
                CD4_rel = (CD4 - CD4[long_var == 0])/(CD4 + CD4[long_var == 0]))


```


We need to check if the increments in CD4 are linked to increments in GeneRichness. Let's calculate these increments in two different ways:

- Absolute: CD4(tx)-CD4(t0) or GR(tx)-GR(to)
- Relative: (CD4(tx)-CD(t0)/(CD4(tx)+CD4(t0))) or (GR(tx)-GR(t0)/(GR(tx)+GR(t0)))

Check distribution of these values and see if there is a possibility of reasonable categorization.

```{r plot_dists, warning=F, results=F, message=F, echo = F}
median_df_overall <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48, 96)) %>% 
    mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  dplyr::group_by(time_point) %>% 
  filter(!is.na(GR_rel)) %>% 
  summarise(med = median(GR_rel)) %>% 
  dplyr::mutate(med = round(med, 3))

plt_overall <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  # dplyr::filter(!is.na(GR_rel)) %>% 
  ggplot(aes(x = GR_rel)) +
  facet_wrap(~time_point) +
  geom_density() +
  theme_bw() +
  geom_vline(data = median_df_overall,aes(xintercept = med)) +
  geom_text(data = median_df_overall, aes(label = paste("med =",med), x = med, y = 1.5), color = "black") +
  labs(x="richness increase (relative)") +
  theme(strip.background = element_blank(),
        strip.text = element_text(face = "bold", size = 12))

median_df_group <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48, 96)) %>% 
  dplyr::mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  dplyr::group_by(time_point, cat_var) %>% 
  dplyr::filter(!is.na(GR_rel)) %>% 
  summarise(med = median(GR_rel)) %>% 
  dplyr::mutate(med = round(med, 3))

plt_group <- 
  deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  mutate(time_point = recode(long_var, `48` = "week 48", `96` = "week 96")) %>% 
  # dplyr::filter(!is.na(GR_rel)) %>% 
  ggplot(aes(x = GR_rel, color = cat_var)) +
  facet_wrap(~time_point) +
  geom_density() +
  theme_bw() +
  geom_vline(data = median_df_group,aes(xintercept = med, color = cat_var)) +
  geom_text(data = median_df_group, aes(label = paste("med =",med), x = med, y = c(2,2.5,2.5,3))) +
  labs(x="richness increase (relative)", color = "group", y = "density") +
  theme(strip.background = element_blank(),
        strip.text = element_text(face = "bold", size = 12)) +
  scale_color_brewer(palette = "Accent")


plt_overall
plt_group

```
As we can see, the distribution of Gene Richness change vs baseline is pretty similar at both weeks 488 and 96, albeit week 96 has a slightly higher median. The differences get bigger when comparing between groups though, as DTG seems significantly bigger than DRV, especially at week 96 (DTG=0.88 vs DRV/r = 0.39 median delta. Let's test with a wilcoxon test)

```{r test_deltaGR, warning=F, message=F}
long_deltas <- 
deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  group_by(link_var) %>%
  dplyr::filter(!is.na(GR_rel)) %>% 
  dplyr::filter(n()==2) %>% 
  dplyr::group_by(cat_var, .drop= T) %>% 
  rstatix::wilcox_test(., GR_rel ~ long_var, paired = T) %>% 
  dplyr::select(-.y.) %>% 
  setNames(c("time_point", "time1", "time2", "n1","n2", "stat","p")) %>% 
  as.data.frame() %>% 
  kableExtra::kable(format = "markdown")


cat_deltas <- 
deltas %>% 
  dplyr::filter(long_var %in% c(48,96)) %>% 
  group_by(long_var) %>% 
  dplyr::filter(!is.na(GR_rel)) %>% 
  rstatix::wilcox_test(., GR_rel ~ cat_var) %>% 
  dplyr::select(-.y.) %>% 
  setNames(c("time_point", "group1", "group2", "n1","n2", "stat","p")) %>% 
  as.data.frame() %>% 
  kableExtra::kable(format = "markdown")

long_deltas
cat_deltas

```

The difference in deltas between groups is significant at week 96 but not at 48. However, no significant differences were found longitudinally between groups.


```{r correlations, warning = F, message=F}

```

And plot:

- Values vs timepoint, overall and per-group(DTG and DRVr only). Fit ANOVA/t-test vs DTG/DRVr at each timepoint to check for differential increases per ARM
- AbsCD4 vs AbsGR, overall and per-group(DTG and dRVr only). Fit LM
- RelCD4 vs RelGR, overall and per-group(DTG and DRVr only). Fit LM

### CAtegorized CD4

This has partially been done. Split patients between 150>CD4(t48)/(t96)>150, and check if DeltaGR is different for groups and fisher.test of these groups vs arm is significant.
