---
title: "CMV vs Gene Richness"
author: "Carlos Bl√°zquez Bondia"
date: '2022-11-30'
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, echo=F, warning=F, message=F}

library(tidyverse)
library(ggplot2)
library(rstatix)
library(ggpubr)
library(RColorBrewer)
library(vegan)
library(gridExtra)


```



```{r load_metadata_vars, eval=T, include=F}
# mymre<-readRDS(here::here(myPath,"data","mre.rds"))


mymre <- aws.s3::s3readRDS(bucket = "mistral-wp6-advanz4",object = "metagenome/WMGS/MREObject.rds", region = "eu-west-1")
### Check consistency between mre.rds(local and S3). You are reading s3 file, here, but in 200_ you are using local
# metadata <- metar::get_metadata(mymre)@metadata_df

metadata <- here::here("Metadata", "2022_11_30_clean_metadata_LIMS.csv") %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))

mymre@metadata@categorical_vals <-  
  here::here("Metadata","CategoricalVariables.txt") %>%
  read.table(., sep="\t") %>%
  setNames(c("CategoricalVariable","PaletteName")) %>%
  as_tibble()

mymre@metadata@metadata_df <-  tibble(metadata)


cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

link_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)


```

```{r update_mre, warning= F, message= F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all("../WMGSPipeline/")


mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as_tibble(metadata)

mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()
mymre@taxa@metaphlan@phyloseq@sam_data <- phyloseq::sample_data(metadata)


```



```{r get_GeneRichness_Data, echo=FALSE, message=F, warning=F}

knitr::read_chunk(path = here::here("analysis","310_IGC_analysis.Rmd"), labels = "get_GeneRichness_Data")
igc<-metar::get_diversity(mymre)@igc

# We'll save this number to report the minimum read threshold.

myigc <- igc_lmm(mymre)



Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()


readCountDistribution<-igc@dataTable %>%
    select(SampleID, NumberMappedReads) %>% 
    unique() %>%

    ggplot(aes(x=NumberMappedReads)) +
    geom_density(fill="grey") + 
    theme_bw() +
    geom_vline(xintercept = Threshold)



TableList <- cat_vector %>%
    set_names() %>%
    map( ~{

        
          ThresholdTable <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          right_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=link_var,
                 cat_var = .x,
                 long_var = long_var,
                 NumberMappedReads)
  

        
    })



```

## presence of CMV at BL interaction with GR

First we'll compare those samples at basal in gene richness as whether or not they have CMV presence.

```{r CMV_Basal_vs_GR, message = F, warning= F}
working_df <- 
  TableList %>% 
  purrr::pluck("group") %>% 
  left_join(metadata[,c("SampleID","cmv_bl")])

bl_df <- 
working_df %>% 
  dplyr::filter(long_var == 0) %>% 
  dplyr::mutate(has_cmv = case_when(
    cmv_bl == 0 ~ "no",
    cmv_bl > 0 ~ "yes"
  ))

bl_df %>% 
  ggplot(aes(x = has_cmv, y = richness, fill = has_cmv)) +
  geom_violin(alpha = 1, draw_quantiles = c(.25,.5,.75), trim = F) +
  ggpubr::stat_compare_means(method="wilcox.test") +
  theme_bw() +
  labs(x = "CMV presence", fill = "CMV presence")


bl_df %>% 
  ggplot(aes(x=richness, y = cmv_bl,color=cat_var)) +
  geom_point() +
  geom_smooth(method = "glm")


cor.test(x=bl_df$richness, y=bl_df$cmv_bl)

```

Overall, doesn't seem like there is a difference in Gene Richness between CMV+ and CMV-, although the p.val is close to 0.05.

## CMV along the study

We'll perform a quick view of how CMV evolves along the timepoints and try to find if they correlate with Gene Richness

```{r CMV_tp, echo=FALSE, message = F, warning = F}

GR_df <- 
  metadata %>%
  dplyr::left_join(TableList$group[,c("SampleID","richness")], by = "SampleID") %>% 
  dplyr::select(time_point, richness, CMV_VL, group ) 


ggplot(GR_df, aes(x = as.factor(time_point), y = CMV_VL, color = group)) +
  geom_violin() +
  theme_bw() +
  labs(color = "group")

```


Apparently, there's barely any CMV viral load outside of baseline. It appears there is some value at week 48 but a closer look shows all entries at this timepoint have a VL of 35, which is the standard result when measurement is below LLOQ (<35). We'll stick to BL comparisons.


# Beta diversity ordination by CMV presence

```{r NMDS_CMV, message = F, warning=FALSE}
bl_df <- 
  metadata %>% 
  mutate(cmv_bl_c = case_when(cmv_bl == 0 ~ "no", cmv_bl > 0 ~ "yes")) %>% 
  dplyr::filter(time_point == 0) 

basal_mre <- 
  metar::filter_samples(mre = mymre, sample_ids = bl_df$SampleID)
basal_mre@metadata@metadata_df <- tibble(bl_df)
basal_mre@metadata@categorical_vals <- tibble(CategoricalVariable = c("cmv_bl_c", "time_point"), PaletteName = c("Accent", "Set1"))
add_num(basal_mre, c("HIV_VL","CMV_VL"), palette = c("Set1","Set1"))

phy <- get_phyloseq(basal_mre, type = "metaphlan")
phy@sam_data <- phyloseq::sample_data(bl_df)

```



```{r Get_DistMats, eval=T, include=T, message = F, warning = F, results = "hide"}

getAbundances<-function(phyloseq, level){
  
  collapsedPhyloseq<- phyloseq::tax_glom(phyloseq, level)
  
  OtuTable<- phyloseq::otu_table(collapsedPhyloseq) %>%
    magrittr::set_rownames(as.character(phyloseq::tax_table(collapsedPhyloseq)[,level])) %>%
    as.data.frame()
  return(OtuTable)
  
}



myAbundanceDF<- getAbundances(phy, "Species")

myBrayDistMat <-vegdist(wisconsin(sqrt(t(myAbundanceDF))))

myNMDS <- myAbundanceDF %>%
  t() %>% 
    as.data.frame() %>%
    # vegan::vegdist(., method = "bray") %>%
 
    metaMDS(.,trymax=800,  distance = "bray",
            k = 2) %>%
    pluck("points") %>%
    as.data.frame() 


```

```{r NMDS_Group, warning= F, message= F, results='hide', eval = T}

cat_df <- tibble(CategoricalVariable = c("cmv_bl_c","time_point"), PaletteName = c("Accent","Set1"))


cat_var <- "cmv_bl_c"
link_var <- "record_id"
id_var <- "SampleID"


myNMDS_DF <- myNMDS %>%
  rownames_to_column(var = id_var) %>%
  dplyr::full_join(., bl_df, by =id_var) %>%
  dplyr::select(SampleID = id_var,
                NMDS1 = MDS1,
                NMDS2 = MDS2,
                linkVar = !!sym(link_var),
                catVar = cmv_bl_c,
                shape = CD4diff_48)


  myXmin<-min(myNMDS$MDS1)
  # +(max(dim_NMDS$points[,1])-min(dim_NMDS$points[,1]))/20
  myYmin<-max(myNMDS$MDS2)-(max(myNMDS$MDS1)-min(myNMDS$MDS1))




myAdonis<- adonis2(myBrayDistMat~ catVar, data = myNMDS_DF) %>%
  slice(1L) %>%
  as.data.frame() %>%
  magrittr::set_rownames(cat_var)


myNMDSplot<-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point(aes(color=catVar, shape = as.factor(shape)), size = 3) +
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour="black"),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face="italic"),
          legend.title=element_text(size=11))+
        stat_ellipse(geom="polygon",alpha=0.15,color="black",aes(fill=catVar),level=0.95)+
    # ggtitle(expression(atop("Species composition",atop(italic("all samples"))))) +
    theme(plot.title=element_text(lineheight=1,face="bold",size=13))+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var) +
  scale_color_brewer(palette = "Accent") +
  scale_fill_brewer(palette = "Accent") + 
  annotation_custom(gridExtra::tableGrob(round(myAdonis,3),
                                         theme = ttheme_default(base_size = 10)),ymin = 1)




myNMDSplot

```



While it seems the samples with detectable CMV are more disprerse along the ordination, Adonis seems unable to find a significance. It may be due lo low sample count. Interestingly enough, there seems to be a correlation between CMV presence at basal and CD4 recovery. Most shockingly, presence of CMV seems to be correlated with a higher chance of recovering CD4 in more than 150 counts/ml. It could however mean that patients with CMV have a higher CD4 activation as they are trying to contain the virus.


```{r cmv_cd4, warning=F, message=FALSE}
t_df <- 
  myNMDS_DF %>% 
  dplyr::select(shape, catVar) %>% 
  filter(!is.na(shape)) %>% 
  mutate(across(everything(), as.factor)) %>% 
  group_by(shape,catVar) %>% 
  summarise(n=n()) %>% 
  spread(shape, n) %>% 
  column_to_rownames("catVar") 

test_chi <- 
  t_df %>% 
  as.matrix() %>%
  chisq_test(x = .)

```



```{r intra_group_distances, warning = F, message= F}
cmv_pos <- bl_df %>% 
  dplyr::filter(cmv_bl_c == "yes") %>% 
  dplyr::pull(SampleID)

cmv_neg <- bl_df %>% 
  dplyr::filter(cmv_bl_c == "no") %>% 
  dplyr::pull(SampleID)


upper_distmat <- as.matrix(myBrayDistMat)
upper_distmat[lower.tri(upper_distmat)] <- NA


distance_df <- 
upper_distmat %>% 
  as.matrix() %>% 
  as.data.frame() %>% 
  # dplyr::filter(!rownames(.)==colnames(.)) %>%
  rownames_to_column("Sample1") %>% 
  pivot_longer(cols = !Sample1, names_to = "Sample2", values_to = "dist") %>% 

    mutate(cmv = case_when(
      Sample1 %in% cmv_pos & Sample2 %in% cmv_pos ~ "yes",
      Sample1 %in% cmv_neg & Sample2 %in% cmv_neg ~ "no"
    )) %>% 
    dplyr::filter(Sample1 != Sample2,
                  !is.na(cmv),
                  !is.na(dist)) 


distance_df %>% 

  group_by(cmv) %>% 
  count()


distance_df %>% 
  sample_n_by(cmv, size = 406) %>%
  ggstatsplot::ggbetweenstats(.,
                              x = "cmv",
                              y = "dist",
                              type = "np") +
  labs(y = "pairwise within-group distances")

```

On the other hand though, A non-parametric mean comparison on the within-group distances shows a significant difference between both groups. Even when the CMV+ group is randomly subsampled to match the N of CMV-.
