---
title: "317_GeneRichness_Inflammation"
author: "Carlos Bl√°zquez Bondia"
date: '2022-12-15'
output:
  pdf_document: default
  html_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
source("code/LMM.R")
source("code/myFunctions.R")
source("code/group_comparisons.R")
library(rstatix)
library(ggpubr)

```


```{r utils, echo = F, warning=F, message=FALSE}
peak_finder <-
  function(density, goal = "max",keep = F) {
    ### Let's define this function heere but we'll use it later
    stopifnot("goal can only be either local maxima (use: 'max') or local minima (use 'min)" = goal %in% c("min","max"))
    
    val <- case_when(goal == "max" ~ -2,
                   goal == "min" ~ 2)
    
    res <-
      density %>%
      purrr::keep(names(.) %in% c("x", "y")) %>%
      as.data.frame() %>%
      dplyr::mutate(is_peak = c(F, # 1st value can't be a local maxima, as there is no prior value to compare
                                diff(y) %>% # For each nth element of the vector, get (n+1) - n
                                  sign() %>%  # convert to -1 (n+1 is lower than n) or +1 (n+1 is higher than n)
                                  diff() == val, # if n is at a local maxima, both n+1 and n-1 are negative so [(n+1)-n] - [n - (n-1)] is -2
                                F)) # Last element can't be a local maxima either)
                    
                    if (!keep) {
                      res %>%
                        dplyr::filter(is_peak) %>%
                        dplyr::select(-is_peak)
                    } else{
                      res %>%
                        dplyr::select(-is_peak)
                    }
  }


```

```{r load_metadata_vars, eval=T, message = F}
# mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds", region = "eu-west-1")
# metadata <-get_meta(mymre)
mymre <- here::here("data","filt_mre.rds") %>%
  readRDS(.)
metadata <- 
  here::here("Metadata") %>% 
  file.info(list.files(., full.names = T)) %>%
  rownames() %>% 
  magrittr::extract(str_detect(.,"clean_metadata")) %>% 
  sort(decreasing = T) %>% 
  magrittr::extract(1L) %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))



cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

id_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)


```


```{r update_mre, warning= F, message= F, eval=F, results='hide'}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all("../WMGSPipeline")
mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- tibble(metadata)

mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
# 
# mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
#   phyloseq::sample_data(.)


mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

phy <- get_phyloseq(mymre, type = "metaphlan")

```




# Basal exploration of inflammation markers

Perform basic distributin analysis of each marker at baseline, test for imbalances between groups which may be creating a bias.

```{r basal_inflammation, message = F, warning= F}
library(compareGroups)

metadata %>% 
  dplyr::select(record_id, time_point, group, num_var) %>% 
  dplyr::filter(time_point == 0) %>% 
  compareGroups::compareGroups(formula(paste("group", "~", paste(num_var, collapse = " + "), sep = " ")), data = .,) %>% 
  compareGroups::createTable(.,show.n = T)
      
```
No significant changes could be observed in any inflammation marker, except for CD4 counts, which is still a farily small change.

# Gene richness correlations

Now to the main point of this document. The first thing to do is to assess any correlation between initial GR and inflammation markers.

```{r read_GR, echo = F, warning=F, message=F, results='hide'}
igc<-metar::get_diversity(mymre)@igc
Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()


igc_df <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          right_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var= record_id,
                 cat_var = group,
                 long_var = time_point,
                 NumberMappedReads, 
                 num_var) %>% 
  mutate(sCD14 = as.numeric(sCD14))


```

First let's look how richness distributes across time, this will allow us to choose the best correlation and possibly stratify the population between low and high gene richness.

```{r plot_distributions, warning = F, message = F}
peaks_df <- 
c(0,24,48,96) %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(function(tp){
    igc_df %>% 
      dplyr::filter(!is.na(richness), long_var == tp) %>% 
      dplyr::pull(richness) %>% 
      density() %>% 
      peak_finder(., goal = "min") %>% 
      dplyr::mutate(long_var = tp)
  })


igc_df %>% 
  dplyr::filter(!is.na(richness)) %>% 
  ggplot(aes(x = richness, color = as.factor(long_var))) +
  geom_density(lwd = 2) +
  scale_color_brewer(palette = "YlOrRd") +
  geom_vline(data = peaks_df, aes(xintercept = x, color = as.factor(long_var)), lty = 2) +
  geom_text(data = peaks_df, aes(x= x, y = y*.9, color = as.factor(long_var), label = round(x,3))) +
  theme_bw() + 
  labs(color = "week")



```
 

It seems richness is bimodal, with both peaks becoming increasingly separate as the study goes on. There seems to be no shift on the overall distribution. Instead, patients with really low microbiome counts tend to move towards one of the the two peaks (LGC and HGC) to check wether patients are inherently prone to move towards either peak, we could check differences in BL GR between LGC and HGC.

```{r test_basal_peaks, warning=FALSE, message=FALSE}
peak <-
  igc_df %>%
  dplyr::filter(!is.na(richness), long_var == 96) %>%
  dplyr::pull(richness) %>%
  density() %>%
  peak_finder(., goal = "min") %>%
  dplyr::mutate(long_var = 96) %>%
  dplyr::filter(x == max(x)) %>% 
  dplyr::pull(x)

igc_df %>% 
  group_by(link_var) %>% 
  dplyr::filter(long_var %in% c(0,96)) %>% 
  dplyr::filter(n() == 2) %>% 
  dplyr::mutate(gr0 = richness[long_var == 0],
                gr96 = richness[long_var == 96]) %>%
  dplyr::select(SampleID,link_var, cat_var, gr0, gr96) %>% 
  unique() %>% 
  ungroup() %>% 
  dplyr::filter(!is.na(gr0), !is.na(gr96)) %>% 
  cor_test(., gr96, gr0, method = "spearman")

igc_df <- 
igc_df %>% 
  group_by(link_var) %>% 
  dplyr::filter(long_var %in% c(0,96)) %>% 
  dplyr::filter(n() == 2) %>% 
  dplyr::mutate(gr0 = richness[long_var == 0],
                gr96 = richness[long_var == 96]) %>%
  dplyr::select(link_var, gr0, gr96) %>% 
  unique() %>% 
  dplyr::filter(!is.na(gr0), !is.na(gr96)) %>% 
  dplyr::mutate( gr_group = case_when(gr96 >= peak ~ "HGC",
                                      gr96 < peak ~ "LGC")) %>%
  right_join(igc_df, by = "link_var")

igc_df %>% 
  ggplot(aes(x = cat_var, y = gr0, fill = gr_group)) +
  geom_boxplot() +
  stat_compare_means(method = "wilcox.test") +
  theme_bw() +
  labs( y = "Gene richness at BL", x = "group", fill = "GR cluster 96")



```



Create a corr table and heatmap. This will be a good exploratory analysis to see which inflammation markers change with the microbiome. This is a longitudinal study, which means time may be an important variable we need to account with the models. A good exploratory first test will be to correlate markers with gene richness at basal. Then a correlation of deltas may give information on those who grow more depending on gene richness.


```{r basal_corrs, warning= F, message= F}


num_var %>%
  purrr::set_names() %>%
  purrr::map_dfr(function(nv) {
    print(nv)
    igc_df %>%
      dplyr::filter(long_var == 0) %>%
      dplyr::select(SampleID, cat_var, nv, richness) %>% 
      dplyr::mutate(!!sym(nv) := as.numeric(!!sym(nv))) %>% 
      ungroup() %>% 
      cor_test("richness", nv,  method = "spearman")
  })

```

Apparently, no correlation could be found between GR and inflammation markers at basal. Let's look for the longitudinal approach.

```{r corrr_deltas, warning=F, message=FALSE}


num_var %>% 
  purrr::set_names() %>% 
  purrr::map(function(nv){
    print(nv)
    igc_df %>% 
      group_by(link_var) %>% 
      dplyr::filter(long_var %in% c(0,48)) %>% 
      dplyr::filter(n() == 2) %>% 
      dplyr::mutate(num_var = as.numeric(!!sym(nv))) %>% 
      dplyr::mutate(delta_nv := num_var[long_var == 48]/num_var[long_var == 0],
                    delta_gr = richness[long_var == 48]/richness[long_var == 0]) %>% 

      filter(!is.na(num_var)) %>% 
      group_by(cat_var,.drop = T) %>%
      # ungroup() %>% 
      rstatix::wilcox_test(num_var ~ long_var)
  })

res_list <- 
num_var[!num_var %in% c("CD4", "CD8","CD8_CD38_DR", "ADV")] %>% 
  purrr::set_names() %>% 
  purrr::map(function(nv){
    print(nv)
    igc_df %>% 
      dplyr::mutate(num_var := as.numeric(!!sym(nv))) %>% 
      filter(!is.na(num_var)) %>% 
    get_comp_boxplots(dat = .,cat_var = "cat_var",num_var = "num_var", long_var = "long_var",link_var = "link_var",y_trans = "sqrt", pal = "Accent") %>% 
      purrr::pluck("plot") +
      labs(y = nv, fill = "group")
    
  })

ggarrange(plotlist = res_list, nrow = 2, ncol = 2)

```

There are significant differences in both CRP and sCD14, which were lower in the DTG group. CRP became significant at week 96 only, while sCD14 became significant after 48 weeks, and kept doing so at week 96

# Inflammation markers by gene richness

## Baseline

First let's check if Gene Richness had any effect on the baseline levels of inflammation markers.

```{r GR_inflammation_BL, warning=FALSE, message=FALSE}
filt_vars <- num_var[!num_var %in% c("CD4", "CD8","CD8_CD38_DR", "ADV")]

deltas <- 
igc_df %>% 
  dplyr::select(SampleID, link_var, long_var, cat_var, richness,filt_vars) %>% 
  group_by(link_var) %>% 
  dplyr::filter(long_var %in% c(0,96)) %>% 
  dplyr::filter(n() == 2) %>% 
  dplyr::mutate(delta_richness = (richness-richness[long_var == 0]) / (richness + richness[long_var == 0]),
    across(.cols = filt_vars, 
                       .fns =  ~ .x  / .x[long_var == 0], 
                       .names = "delta_{.col}" )
                )  %>% 
  dplyr::ungroup() %>% 
  dplyr::select(SampleID, link_var, cat_var, long_var, contains("delta"))

new_numvars <- paste("delta",filt_vars,sep= "_")
new_numvars %>% 
  purrr::set_names() %>%
  purrr::map(~ {
    deltas %>% 
      ggplot(., aes(x=!!sym(.x), y = delta_richness, colour = cat_var)) +
      geom_point() +
      theme_bw()
  }) %>% 
  ggpubr::ggarrange(plotlist = ., common.legend = T)
  


lmms <- 
  deltas %>%  
  dplyr::mutate(cat_var = "mock") %>% 
get_lmm_effects(
  .,
  cat_vector = "cat_var",
  long_var = "delta_richness",
  num_vector = new_numvars,
  link_var = "link_var"
) 


new_numvars %>%
  purrr::set_names() %>%
  purrr::map(~ {
    deltas %>%
      dplyr::mutate(cat_var = "mock") %>%
      dplyr::filter(!is.na({.x}), 
                    !is.na(delta_richness),
                    long_var == 96) %>%
      create_LMM(
        .,
        num_var = .x,
        long_var = "delta_richness",
        cat_var = "cat_var",
        link_var = "link_var",
        breakpoints = NULL
      ) %>% 
      purrr::pluck("plot") +
      scale_x_continuous(breaks = c(0,2,4,6,8)) +
      # scale_y_continuous(trans = "log2") +
      theme(axis.title.x = element_text(size = 12),
            axis.title.y = element_text(size = 12, angle = 90))
  }) %>%
  ggpubr::ggarrange(plotlist = .)


deltas %>% 
  dplyr::filter(long_var == 96) %>% 
  dplyr::select(delta_richness, new_numvars) %>% 
  cor_test(use = "complete.obs") %>% 
  dplyr::filter()

```


# Differences by groups

```{r GR_group_boxplots, warning = F, message = F}
igc_df <-
  igc_df %>%
  dplyr::filter(!is.na(gr_group)) %>%
  mutate(cat_var2 = paste(gr_group, cat_var, sep = "_"))


my_list <- 
num_var %>% 
  purrr::set_names() %>% 
  purrr::map(function(nv){
    igc_df %>% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %>% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = "_")) %>% 
      # get_lmm_effects()
      get_comp_boxplots(cat_var = "cat_var2", num_var = nv, long_var = "long_var",link_var = "link_var")
  })

myPlotList <- map(my_list, pluck, "plot")
ggarrange(plotlist = myPlotList, common.legend = T)




cat <- 
num_var %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(function(nv){
    
    obj <- 
    igc_df %>% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %>% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = "_")) %>% 
      get_group_comparisons(.,link_var = "link_var", long_var = "long_var",cat_vector = "cat_var2",num_vector = nv,type = "categorical", comps = unique(igc_df$cat_var2))
    
    objtest <- obj %>% pluck("cat_var2",nv,"test")
    objstats <- obj %>% pluck("cat_var2",nv,"stats") %>% 
      mutate(stat = paste(round(median,2)," [",round(iqr,2),"]", sep = ""))
  
    objtest %>% 
      pivot_longer(cols = c("group1","group2")) %>% 
      dplyr::rename(cat_var = value) %>%
      dplyr::left_join(objstats[,c("cat_var","long_var","stat")], by = c("cat_var", "long_var")) %>% 
      pivot_wider(names_from = "name", values_from = c("cat_var", "stat")) %>% 
      dplyr::select(long_var, group1 = cat_var_group1, group2 = cat_var_group2, contains("stat"), p.adj)
    
  })

cat %>% 
  dplyr::filter(p.adj < 0.05)


long <- 
num_var %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(function(nv){
    
    obj <- 
    igc_df %>% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %>% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = "_")) %>% 
      get_group_comparisons(.,link_var = "link_var", long_var = "long_var",cat_vector = "cat_var2",num_vector = nv,type = "longitudinal", comps = c(0,48))
    
    obj %>% 
      pluck("cat_var2",nv,"test") %>% 
      rstatix::adjust_pvalue( method = "BH", p.col = "p") %>% 
      full_join(obj$cat_var2[[nv]]$stats, by = "cat_var")%>% 
      dplyr::mutate(.y. = nv) %>% 
      dplyr::select(cat_var, group1, group2,num_var = .y., contains("median"), contains("iqr"), "p.adj")
  })


long %>%
  dplyr::filter(p.adj < 0.05 )



num_var %>% 
  purrr::set_names() %>% 
  purrr::map(function(nv){
    igc_df %>% 
      dplyr::filter(!is.na(!!sym(nv)), !is.na(gr_group)) %>% 
      mutate(cat_var2 = paste(gr_group, cat_var, sep = "_")) %>% 
      ggplot(aes(x = long_var, y = !!sym(nv), color = cat_var, lty = gr_group)) +
      stat_smooth(method = "loess", alpha = .01) +
      theme_bw() +
      labs(x = "weeks") +
      scale_color_brewer(palette = "Set2") +
      scale_x_continuous(breaks = c(0,24,48,96))
      
      }) %>% 
  ggarrange(plotlist = ., common.legend = T)




```



