---
title: "Beta Diversity and NMDS"
output:
  pdf_document: default
  html_document: default
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r load_libraries,echo=FALSE, warning=F, message=FALSE}
library(metar)
library(phyloseq)
library(tidyverse)
library(vegan)
library(ggrepel)
library(cluster)
require(gridExtra)
library(ggpubr)
library(rstatix)
library(RColorBrewer)
source("code/basal_distances.R")
source("code/Paired_wilcox_NA_Handler.R")
source("code/myFunctions.R")
```

## Including Plots

You can also embed plots, for example:

```{r load_metadata_vars, eval=T, message = F}
mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds", region = "eu-west-1")
# metadata <-get_meta(mymre)
metadata <- here::here("Metadata", "2022_09_14_clean_metadata_LIMS.csv") %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))



cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

id_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)


```


```{r update_mre, warning= F, message= F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.
devtools::load_all("../WMGSPipeline")
mymre@taxa@virgo <- SpquantSet()
mymre@gene_function@humann <- metar::GeneFunSet()
mymre@taxa@metaphlan@phyloseq_sec

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- tibble(metadata)

mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
# 
# mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
#   phyloseq::sample_data(.)


mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

phy <- get_phyloseq(mymre, type = "metaphlan")

```




```{r Get_DistMats, eval=T, include=T, results = 'hide', eval = T}

getAbundances<-function(phyloseq, level){
  
  collapsedPhyloseq<- phyloseq::tax_glom(phyloseq, level)
  
  OtuTable<- phyloseq::otu_table(collapsedPhyloseq) %>%
    magrittr::set_rownames(as.character(phyloseq::tax_table(collapsedPhyloseq)[,level])) %>%
    as.data.frame()
  return(OtuTable)
  
}




myAbundanceDF<- getAbundances(phy, "Species")


myNMDS <- myAbundanceDF %>%
  t() %>% 
    as.data.frame() %>%
    # vegan::vegdist(., method = "bray") %>%
 
    metaMDS(.,trymax=800,  distance = "bray",
            k = 2) %>%
    pluck("points") %>%
    as.data.frame() 

  # NMDS = data.frame(NMDS1 = dim_NMDS$points[,1], NMDS2 = dim_NMDS$points[,2])

BiPlotCoords<- envfit(myNMDS, t(myAbundanceDF), perm=999) 

BiPlotCoordsDF <- c("MDS1","MDS2","r","pvals") %>%
  set_names() %>%
  
  map_dfr(., ~ {
    
    if(.x %in% c("MDS1","MDS2")){
      BiPlotCoords$vectors$arrows[,.x] %>%
        as.numeric()
    } else{
      BiPlotCoords$vectors %>%
        pluck(.x) %>%
        as.numeric()
    } 
      
  }) %>%
  as.data.frame() %>%
  set_names(nm=c("NMDS1","NMDS2","r","p")) %>%
  dplyr::mutate(Dim1 = NMDS1*sqrt(r),
                Dim2 = NMDS2*sqrt(r),
                species = rownames(myAbundanceDF)) 
  

myBrayDistMat <-vegdist(wisconsin(sqrt(t(myAbundanceDF))))


```

## NMDS by Treatment group

One of the variables: **CD4diff_96** has only one level. That means all patients in the study had an increase of CD4 counts higher than 50 cells/ml between weeks 96 and 0. As this variable only has one level it will cause the following code to break (as it performs conrast analysis). Hence, it will be removed from the following analysis. However, we'll look at the distribution of the CD4 increases as a distribution:

```{r CD4diff_dist, warning=F, message=F}
metadata %>% 
  dplyr::filter(time_point %in% c(0,48),
                !is.na(CD4diff_96)) %>%
  dplyr::select(record_id, time_point, CD4) %>%
  dplyr::group_by(record_id) %>%
  dplyr::summarise(dif = diff(CD4)) %>%
  ggplot(., aes(x= dif)) +
  geom_density()


```


```{r NMDS_Group, warning= F, message= F, results='hide', eval = T`}

cat_df <- cat_df %>%
  filter(!CategoricalVariable == "CD4diff_96")
cat_var <- "group"



link_var <- "record_id"
id_var <- "SampleID"

myPal<-RColorBrewer::brewer.pal(name =  cat_df$PaletteName[1],
                                n=length(unique(metadata[,cat_var])))
BiPlotCoordsDF<-BiPlotCoordsDF %>%
  mutate(Dim1s = Dim1*3,
         Dim2s = Dim2*3)


myNMDS_DF <- myNMDS %>%
  rownames_to_column(var = id_var) %>%
  dplyr::full_join(., metadata, by =id_var) %>%
  dplyr::select(SampleID = id_var,
                NMDS1 = MDS1,
                NMDS2 = MDS2,
                linkVar = !!sym(link_var),
                catVar = !!sym(cat_var),
                LongVar = long_var)


  myXmin<-min(myNMDS$MDS1)
  # +(max(dim_NMDS$points[,1])-min(dim_NMDS$points[,1]))/20
  myYmin<-max(myNMDS$MDS2)-(max(myNMDS$MDS1)-min(myNMDS$MDS1))




myAdonis<- adonis(myBrayDistMat~ catVar, data = myNMDS_DF) %>%
  pluck("aov.tab") %>%
  as.data.frame() %>%
  slice(1L) %>%
  magrittr::set_rownames(cat_var)


myNMDSplot<-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point(aes(color=catVar)) +
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour="black"),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face="italic"),
          legend.title=element_text(size=11))+
        stat_ellipse(geom="polygon",alpha=0.15,color="black",aes(fill=catVar),level=0.95)+
    # ggtitle(expression(atop("Species composition",atop(italic("all samples"))))) +
    theme(plot.title=element_text(lineheight=1,face="bold",size=13))+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var) +
  scale_color_brewer(palette = "Accent") +
  scale_fill_brewer(palette = "Accent") + annotation_custom(gridExtra::tableGrob(round(myAdonis,3),
                                         theme = ttheme_default(base_size = 10)),ymin = 1.9)





myNMDSBiplot<-ggplot(myNMDS_DF, aes(x=NMDS1, y=NMDS2)) +
    geom_point(aes(colour=catVar, shape = as.character(LongVar))) +
    scale_colour_manual(values = myPal, labels = c("DGT", "DRV/r")) +
    scale_fill_manual(values=myPal, labels = c("DGT", "DRV/r"))+
    theme_bw()+
    theme(panel.border=element_blank(),
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),

          axis.line=element_line(colour="black"),
          axis.text.x=element_text(size=11),
          axis.text.y=element_text(size=11),
          axis.title.x=element_text(size=11),
          axis.title.y=element_text(size=11),

          legend.text=element_text(size=11,face="italic"),
          legend.title=element_text(size=11))+

    ggtitle(expression(atop("Species composition",atop(italic("all samples"))))) +
    theme(plot.title=element_text(lineheight=1,face="bold",size=13))+
    stat_ellipse(geom="polygon",
                 alpha=0.15,color="black",
                 aes(fill=catVar),level=0.95)+
    # scale_size_manual(values = c(1:length(unique(myNMDS_DF[,"LongVar"])))/1.5) +
    geom_segment(data=BiPlotCoordsDF,aes(x=0,xend=Dim1s,y=0,yend=Dim2s),
                 arrow = arrow(length = unit(0.2,"cm")),colour="darkgrey")+
    geom_text_repel(data=BiPlotCoordsDF,aes(Dim1s,Dim2s,label=species),
                    size=2.5, box.padding = unit(0.25, "lines"),
                    segment.colour="black",
                    segment.alpha=0.5,
                    segment.size=0.10)+
    coord_fixed()+
    labs(fill = cat_var, color=cat_var, shape = long_var) +
  annotation_custom(gridExtra::tableGrob(round(myAdonis,3),
                                         theme = ttheme_default(base_size = 8)),ymin = 3.5, xmin=2.4)
myNMDSplot

```

```{r Print_NMDS, message=F, warning=F}
myNMDSplot
```


When we look at the most simple ordination before colouring by categorical variables, we see no clear separated clusters between samples. They all appear to be mostly concentrated in one big cluster that randomly spreads in the positive direction for both NMDS components.

```{r plot_basic_NMDS, warning = F, include = F, eval = F, message=F, warning=F}
mymre <- metar::metaphlan_nmds(mre = mymre, tax_level = "Species", save_files = F)

# mymre@taxa@metaphlan@nmds$Species$top_50$categorical %>%
  # discard(., names(.) == "record_id")

mymre@taxa@metaphlan@nmds
metar::get_diversity(mymre, type = "metaphlan", res_slot = "nmds")

```


In a more formal analysis, the silhouette coefficient was tested for different number of clusters. This analysis found the biggest coefficient was found for n=1 clustrs, with massive drops between 2-3 and 5-6.

```{r silhouette_analysis, warning=F, message=F}




myBrayDistMat <- phyloseq::otu_table(phy) %>%
  t() %>%
  sqrt() %>%
  vegan::wisconsin() %>%
  vegan::vegdist()



c(1:11) %>%
  set_names() %>%
  map_df(., ~{
    cluster::pam(myBrayDistMat, k = .x) %>%
      pluck("silinfo") %>%
      pluck("avg.width")
  }) %>%
  t() %>%
  as_data_frame() %>%
  rownames_to_column(var="clusters") %>%
  set_names(c("clusters", "coefficient")) %>%
  ggplot(., aes(x=as.numeric(clusters), y=coefficient, group=1)) +
  geom_point() + 
  geom_line() + 
  scale_x_continuous(breaks=c(1:10)) +
  theme_bw() +
  theme(panel.grid.minor = element_blank()) + 
  labs(title = "Silhouette coefficient by number of clusters", 
       x="nº Clusters", y = "Silhouette coefficient")
  

```


In the same fashion, when we start identifying samples by group and Timepoint, we see no clear separation, although samples from group 1 (+DGT) group appear to be more clustered together than group 2 (DRV+RLT), which show a more diffuse "central" cluster and are more dispersed. When looking at weight of the variables, the concentric spread of the arrows can be split in two sides, one more enriched in Prevotella and another in bacteroides, which spread mostly around the 2nd component of the NMDS.

However, the Adonis test found the variable group to have a significant effect over sample distances (p<0.001), although it only accounted for a small portion of the variance

```{r plot_biplot, eval=F,include=T, message=F, warning=F}


myNMDSBiplot
```


# Distances to basal

```{r First_differences, message = F, warning = F}

phy <- get_phyloseq(mymre, type = "metaphlan") 

otu_df <- 
  phyloseq::otu_table(phy) %>% 
  t() %>% 
  as.data.frame() %>% 
  stats::setNames(phyloseq::tax_table(phy)[,"Species"]) %>% 
  # tibble::rownames_to_column("SampleID") %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\[")) %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\]"))  



lmms <- 
otu_df %>% 
  vegan::vegdist(x = ., method = "bray") %>% 
  distance_from_basal(dist_mat = ., metadata = metadata, 
                      link_var = "record_id",
                      long_var = "time_point",
                      cat_var = "group",
                      basal = 0) 

lab <- lmms %>% 
  pluck("stat") %>% 
  names() %>% 
  set_names() %>% 
  map_dfr(~ {
    lmms %>% 
      purrr::pluck("stat",.x) %>% 
      as.data.frame() %>% 
      dplyr::select(Value, pval = `p-value`) %>% 
      dplyr::slice(2L) %>% 
        round(3) %>% 
      dplyr::mutate(group =.x)
})

plot <- 
  lmms %>% 
  pluck("plot")+
  scale_x_continuous(breaks = unique(metadata$time_point)) +
  scale_color_brewer(palette = "Accent") +
  scale_fill_brewer(palette = "Accent") 


plot
lab

```

Sample A0023 apparently jups to distance = 1 from the very start

# Pairwise distances evolution

It has been suggested the human microbiome follows the "Anna Karenina" principle which states: *"All happy families are alike; each unhappy family is unhappy in its own way."*. In microbiome it translates to healthy patients having somewhat similar microbiomes and when the immune system breaks down, the microbiome escapes a stable configuration and changes without any specific directin. Hence, in this setting, pairwise distances at basal should be higher and decrease as the immune system recomposes and the microbimes move towards a "healthy" composition which is more constant among patients.


```{r pairwise_distances_tp, warning=F, message=FALSE}

upper_distmat <- as.matrix(myBrayDistMat)
upper_distmat[lower.tri(upper_distmat)] <- NA
sample_vc <- metadata$time_point
names(sample_vc) <- metadata$SampleID

distance_df <- 
upper_distmat %>% 
  as.matrix() %>% 
  as.data.frame() %>% 
  # dplyr::filter(!rownames(.)==colnames(.)) %>%
  rownames_to_column("Sample1") %>% 
  pivot_longer(cols = !Sample1, names_to = "Sample2", values_to = "dist") %>% 

    mutate(tp1 = sample_vc[Sample1],
           tp2 = sample_vc[Sample2]) %>% 
    dplyr::filter(Sample1 != Sample2,
                  tp1 == tp2) 




distance_df %>% 
  # sample_n_by(cmv, size = 406) %>%
  ggstatsplot::ggbetweenstats(.,
                              x = "tp1",
                              y = "dist",
                              type = "np") +
  labs(y = "pairwise within-group distances",
       x = "week")


```

Looks like the hypotesis holds. Beta diversity drastically decreases from baseline to week 24 (medians 074 vs 0.66) and becomes somewhat consistent afterwards, decreasing very slowly as it only becomes relevant after 96 weeks. This suggests the microbiome makes ost of the recovery quickly after which it stabilizes, changling very slowly, if at all.

```{r first_dist_boxplots, warning= F, message=F}
first_dif <- 
  otu_df %>% 
  vegan::vegdist(x = ., method = "bray") %>% 
  distance_from_basal(dist_mat = ., metadata = metadata, 
                      link_var = "record_id",
                      long_var = "time_point",
                      cat_var = "group", basal= 0) %>% 
  pluck("data")

test_cat <-
  first_dif %>%
  dplyr::group_by(time_point) %>%
  rstatix::wilcox_test(distance ~ group, p.adjust.method = "BH") %>%
  # rstatix::adjust_pvalue(p.col = "p", method = "none") %>%
  rstatix::add_significance(p.col = "p", output.col = "p.adj.signif") %>%
  rstatix::add_xy_position("time_point") %>%
  dplyr::mutate(y.position = y.position + 0.2)





test_long <-
  first_dif %>%
  PairedWilcox_withNAs(
    myDF = .,
    IDs = "record_id",
    Variable = "distance",
    Response = "time_point",
    Subsets = "group"
  ) %>%
  group_by(group) %>%
  rstatix::adjust_pvalue(p.col = "p",
                         output.col = "p.adj",
                         method = "BH") %>%
  ungroup() %>%
  # rstatix::adjust_pvalue(method = "BH") %>%
  add_significance(p.col = "p.adj") %>%
  
  # mutate(CategoricalVariable = levels(as.factor(myInnerDataFrame$CategoricalVariable))) %>%
  # add_y_position(y.trans = function(x){quantile(x, probs=c(.1))}) %>%
  mutate(
    xmin = recode(
      group1,
      `24` = 1,
      `48` = 2
    ),
    xmax = recode(
      group2,
      `24` = 1,
      `48` = 2,
      `96` = 3
    )
  ) %>%
  mutate(xmin = case_when(group == "DTG" ~ xmin + 0.2,
                          group == "DRV/r" ~ xmin - 0.2),
         xmax = case_when(group == "DTG" ~ xmax + 0.2,
                          group == "DRV/r" ~ xmax - 0.2))



boxplots <- 
 
  ggplot(first_dif, aes(x = as.factor(time_point), y = distance)) +
  geom_boxplot(position = "dodge")+
  scale_color_brewer(palette = "Set1") +
  scale_fill_brewer(palette = "Set1") +
  stat_pvalue_manual(test_cat, label = "p.adj.signif", step.increase = .01, hide.ns = T) +
  stat_pvalue_manual(test_long, label = "p.adj.signif", step.increase = .02, hide.ns = T) +
  theme_bw() + 
  labs(x = "Weeks", y = "distance from basal") +
  scale_fill_brewer(palette = "Accent")


lmms

boxplots


```

