---
title: "Baseline microbiome vs Immune response"
author: "Marc Noguera"
date: "3/29/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r Baseline microbiome vs Immune response, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
library(patchwork)

```


```{r load_all, warning=F, message=F}

# This chunk is only for executing a load_all on the reoot directory of the metar repository.
# This is a workaround for a bug present in metar::filter_samples in which fails to find an 
# inherited method for fun_input(), a step in filtering samples in the Gene Function Slot. This 
# chunk will be removed as soon as this issue is solved.

devtools::load_all(here::here("../","WMGSPipeline"))

# Load all function arguments should be the path to where metar is installed. Change it
# accordingly.

```



```{r Setup, warning=F, message=F}

# mymre <- here::here("data","mre.rds") %>%
#   readRDS(.)

mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds")
# metadata <-get_meta(mymre)
metadata <-
  here::here("Metadata") %>% 
    list.files(pattern = "clean_metadata", full.names = T) %>%
  file.info() %>% 
  dplyr::arrange(desc(ctime)) %>% 
  rownames() %>% 
  dplyr::first() %>% 
  read.csv()




cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

id_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)



```



```{r update_mre, warning= F, message= F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()



```


```{r build_basal_mre, message = F, warning = F}
basal_meta <- metadata %>%
  filter(time_point == 0) 

basal_mre <- filter_samples(mre = mymre, sample_ids =  basal_meta$SampleID)
# basal_mre@metadata@categorical_vals <-remove_cat(basal_mre, "CD4diff_96")



# cat_vector %>% 
#   set_names() %>% 
#   map_chr(~{
#     metadata %>% 
#       filter(time_point == 0) %>% 
#       pull(!!sym(.x)) %>% 
#       unique() %>% 
#       length()
#   })

```





# Baseline differences in microbiome


## Alpha diversity

```{r baseline_richness, warning = F, message = F}

GR <- get_diversity(mymre, type = "igc", res_slot = "dataTable") 



Threshold <- GR %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% ### Again, you are using the full data table to calculate percentiles. This is wrong.
  as.numeric()


richness_df <-
  GR %>%
  filter(ReadCountReal >= Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
  group_by(SampleID) %>%
  summarise(across(everything(), min)) %>%
  ungroup() %>%
  select(SampleID, GeneNumber) %>%
  left_join(., metadata, by="SampleID") %>%
  select(id_var = !!sym(id_var),
         long_var = !!sym(long_var),
         cat_vector,
         GeneNumber) %>%
  filter(long_var == 0) 

# cat_vector %>% 
#   set_names() %>% 
#   map(function(cat_var){
#     richness_df %>%
#       dplyr::select(id_var,
#                     cat_var = !!sym(cat_var),
#                     GeneNumber) %>%
#             filter(!is.na(cat_var)) %>%
#       ggplot(., aes(x=cat_var, y=GeneNumber, fill = cat_var))+
#       geom_violin(draw_quantiles = T) +
#       geom_boxplot(fill="lightgrey", width=0.5, na.rm=T) +
#       geom_point() +
#       scale_fill_brewer(palette="Set1") +
#       ggpubr::stat_compare_means(method="wilcox",label = "p.signif", hide.ns = T) +
#       theme_bw() + 
#       labs(fill = cat_var)
#     
#   })

  
plotlist <- cat_vector %>% 
  set_names() %>% 
  map(function(cat_var){
    richness_df %>%
      dplyr::select(id_var,
                    cat_var = !!sym(cat_var),
                    GeneNumber) %>%
            filter(!is.na(cat_var)) %>%
            ggstatsplot::ggbetweenstats(
              data = .,
              x = cat_var,
              y = GeneNumber,
              bf.message = FALSE,
              type = "nonparametric",
              pairwise.comparisons = T,
              xlab = cat_var, 
              ggtheme = ggplot2::theme_bw()
              
            )
  })
igc_plotlist <- 
plotlist %>% 
  purrr::keep(names(.) %in% c("group", "CD4diff_48","CD4diff_96")) %>% 
ggpubr::ggarrange(plotlist = ., ncol = 1, nrow = 3)

```

No differences in basal Gene richness were found between patients with an increase of CD4 higher than 50 counts/ul and whose with lower increase. The same could be observed in CD8 counts.


```{r Immune_delta_Distributions, warning=F, message=F}



ids <- metadata %>%
  dplyr::group_by(record_id) %>%
  dplyr::filter(time_point %in% c(0,48)) %>%
  dplyr::filter(!is.na(CD4)) %>%
  dplyr::filter(!is.na(CD8)) %>% 
  dplyr::arrange(time_point) %>%
  dplyr::summarise(n=n()) %>%
  dplyr::filter(n==2) %>%
  dplyr::pull(record_id)


deltas <- metadata %>% 
  # dplyr::filter(record_id %in% ids) %>%
  dplyr::filter(time_point %in% c(0,48,96)) %>% 
  dplyr::select(record_id, time_point,CD4,CD8) %>%
  dplyr::mutate(ratio = CD4/CD8) %>% 
  dplyr::group_by(record_id) %>% 
  dplyr::arrange(record_id,time_point) %>%
  tidyr::pivot_wider(id_cols = record_id, names_from= time_point, values_from = c(CD4,CD8, ratio)) %>%
  dplyr::mutate(
    deltaCD4_48 = CD4_48 - CD4_0, 
    deltaCD8_48 = CD8_48 - CD8_0,
    deltaCD4_96 = CD4_96 - CD4_0, 
    deltaCD8_96 = CD8_96 - CD8_0,
    deltaratio_48 = ratio_48 - ratio_0,
    deltaratio_96 = ratio_96 - ratio_0) %>%
  dplyr::select(record_id, contains("delta"))



deltas %>%
  pivot_longer(cols=c(deltaCD4_48,deltaCD8_48,deltaCD4_96,deltaCD8_96, deltaratio_48, deltaratio_96)) %>%
  mutate(term = str_remove_all(name, "delta|_48|_96"),
         tp = str_remove_all(name, ".*(?<=_)")) %>%
  ggplot(., aes(color=tp)) +
  geom_density(aes(x=value), alpha=0.4) +
  facet_wrap(~ term, scales="free", ncol = 2) +
  # geom_density(data=pivot_longer(cols=c(deltaCD4,deltaCD8)),aes(x=value), alpha=0.4) +
  theme_bw() +
  scale_color_brewer(palette = "Set1")+

  labs(color="week")


deltas %>% 
  select(contains("delta")) %>% 
  colnames() %>% 
  set_names() %>% 
  map(~{
    summary(deltas[,.x])
  })
  


```


The last piece of output (summaries) tell there is a great ammount of missing data. This is due to a combination of many patients missing a sample and a few not having entries for analytical data:

```{r count_missing_samples, warning=F, message=F}
no0 <- metadata %>% 
  filter(!record_id %in% metadata[metadata$time_point == 0, ]$record_id) %>%
  pull(record_id)

##patients without lymphocite data but with microbiome sample
no48_CD4 <- metadata %>%
  filter(is.na(CD4) ) %>%
  filter(time_point == 48) %>% 
  pull(record_id)  

no48_CD8 <- metadata %>%
  filter(is.na(CD8) & !is.na(SampleID)) %>%
  filter(time_point == 48) %>% 
  pull(record_id)  


no96_CD4 <- metadata %>%
  filter(is.na(CD4)& !is.na(SampleID)) %>%
  filter(time_point == 96) %>% 
  pull(record_id)

no96_CD8 <- metadata %>%
  filter(is.na(CD8)& !is.na(SampleID)) %>%
  filter(time_point == 96) %>% 
  pull(record_id)


# patients without sample
noS48 <- metadata %>%
  filter(!record_id %in% metadata[metadata$time_point == 48, ]$record_id) %>% 
  pull(record_id)

noS96 <- metadata %>%
  filter(!record_id %in% metadata[metadata$time_point == 96, ]$record_id) %>% 
  pull(record_id)



totalMissing_CD4_48 <- length(unique(c(no0, no48_CD4, noS48)))
totalMissing_CD4_96 <- length(unique(c(no0,no96_CD4, noS96)))
totalMissing_CD8_48 <- length(unique(c(no0, no48_CD8, noS48)))
totalMissing_CD8_96 <- length(unique(c(no0,no96_CD8, noS96)))

totalMissing_CD4_48
totalMissing_CD4_96
totalMissing_CD8_96
totalMissing_CD8_96
```

Adding the different kind of missing data checks out with the summaries. The biggest issue came with patients not having no sample whatsoever for such timepoints. Of which: 25 didn't have a basal sample, 22 didnt have sample for week 48 and 30 for week 96.


delta CD4 threshold was too low. To keep the study clinically interpretable We'll do a cutoff as a round number around the mean. In this case <150/>150 according to the mITT design results of the parent study.

Basal gene richness vs delta CD4_48 as continuous variables

# Microbiome structure at basal

```{r get_basal_mre, warning=F, message= F}



basal_mre <- metar::metaphlan_nmds(basal_mre, tax_level="Species")



```

```{r numeric_nmds_cors, message=F, warning=F}
nmdscor <- basal_mre@taxa@metaphlan@nmds$Species$top_50$numeric 

nmdscor

```


```{r basal_NMDS,eval=t ,warning=FALSE, message=FALSE}


group <- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$group & 
    scale_color_brewer(palette="Set1") &
    scale_fill_brewer(palette = "Set1") 
  



cluster <- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$cluster


ggpubr::ggarrange(group,cluster, labels = c("group", "cluster"))

```


NMDS shows no basal significant differences per group for either com,ponent. An unsupervised clustering created 2 well differentiated groups. While ADONIS per cluster was significant, we'll look whether the numerical variables have something to do with them.

```{r adonis_table, echo=F, warning=F, message = F}
kableExtra::kable(basal_mre@taxa@metaphlan@nmds$Species$top_50$all_adonis, format = "markdown")
```



# NMDS correlation with numerical variables
```{r NMDS_numeric_correlations, warning= F, message= F}
basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_numeric %>% 
  ggpubr::ggarrange(plotlist=.)




```



## Categorical variables vs unsupervised clustering
```{r categorical_vs_cluster, message=F, warning=F, fig.show='hold', out.width="40%"}

basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical %>% 
  names() %>% 
  set_names() %>% 
  map(~{
    df <- 
    basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical %>% 
      pluck(.x) %>% 
      pluck("data") %>%
      filter(!is.na(!!sym(.x)))
    
    pal <- cat_df %>% 
      filter(CategoricalVariable == .x) %>% 
      pull(PaletteName)
    
    
    test <- df %>%
      dplyr::select(!!sym(.x), cluster) %>%
      table() %>% 
      as.matrix() %>% 
      fisher.test(df) %>% 
      pluck("p.value") %>% 
      as.numeric() %>% 
      round(.,2)
      
    plt <-   
      df %>% 
      ggplot(., aes(x = as.factor(cluster),fill=!!sym(.x)))+
      geom_bar(stat="count") +
      theme_bw() +
      labs(x="cluster",
           title=.x,
            subtitle = glue::glue("P.value = {test} (Fisher test)")) +
      scale_fill_brewer(palette = pal)
  }) 
  



```

The variabe correspoinding to the change in CD4 counts after 96 weeks **CD4diff_96** was dropped due to having only one value. This means all patients had an increase of CD4>50, which breaks any comparison we try to make. 



# DIfferential Abundance

We'll look into which genera are more correlated per group at baseline. We'll perform both classic and compositional approaches.

```{r wilcox_da, message= F, warning=F}

phy_basal <- get_phyloseq(basal_mre, type = "metaphlan") %>% 
    phyloseq::filter_taxa(., function(x) sum(x > 0) > (0.3*length(x)), TRUE)


otu_df <- 
  phyloseq::otu_table(phy_basal) %>% 
  t() %>% 
  as.data.frame() %>% 
  setNames(phyloseq::tax_table(phy_basal)[,"Species"]) %>% 
  rownames_to_column("SampleID") %>% 
  setNames(.,str_remove_all(colnames(.), "\\[")) %>% 
  setNames(.,str_remove_all(colnames(.), "\\]"))

basal_meta <- get_meta(basal_mre)

stat_list <-
  cat_vector %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    otu_df %>%
      dplyr::select(-SampleID) %>%
      colnames() %>%
      purrr::set_names() %>%
      purrr::map_dfr(function(sp) {
        df <-
          otu_df %>%
          dplyr::select(SampleID, !!sym(sp)) %>%
          dplyr::left_join(basal_meta[, c("SampleID", cv), drop = F], by = "SampleID")
        
        if (length(unique(df[, cv])) == 2) {
          test <- df %>%
            rstatix::wilcox_test(., formula(paste(sp, "~", cv)), paired = F) %>%
            dplyr::mutate(method = "Wilcoxon")
          
        } else{
          test <- df %>%
            rstatix::kruskal_test(., formula(paste(sp, "~", cv))) %>%
            dplyr::mutate(method = "Kruskal-Wallis")
        }
        
        
      }) %>%
      dplyr::mutate(p.adj = p.adjust(.$p, "BH")) %>%
      dplyr::filter(p.adj < 0.05)
  }) %>% 
  purrr::keep(., ~nrow(.) > 0)



stat_list %>% 
  map(~{
    kableExtra::kable(.x, format = "markdown") 
  })


```


The only categoricalvariables with Differential abundance at basal are Gender and Risk group. This is hardly surprising as microbiome differences between gender and sexual orientations have been widely reported. But that also means no significant signal could be detected that is significant to this study. More specifically, there is no bacterial signal that may predict increase of CD4/CD8 after 48-96 weeks of treatment.



```{r ANCOM}
phy_c<- basal_mre@taxa@metaphlan@phyloseq_sec %>% 
    phyloseq::filter_taxa(., function(x) sum(x > 0) > (0.3*length(x)), TRUE)


basal_mre@taxa@metaphlan@phyloseq_sec <- phy_c

basal_mre <- metar::metaphlan_ancom(mre = basal_mre, tax_level = "Species", save_files = F)


res_list <- basal_mre@taxa@metaphlan@ancom$Species %>% 
  map(~{
    .x %>% 
      pluck("stats") %>% 
      filter(detected_0.7 == TRUE)
  }) %>% 
  purrr::keep(., ~nrow(.) > 0)
basal_mre@taxa@metaphlan@ancom$Species[[names(res_list)]]$plot

res_list

```

ANCOM was even more astringent and ony found 2 significant species between hts and msm. 



