---
title: "Baseline microbiome vs Immune response"
author: "Marc Noguera"
date: "3/29/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r Baseline microbiome vs Immune response, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
library(patchwork)
```


```{r Setup, warning=F, message=F}

# mymre <- here::here("data","mre.rds") %>%
#   readRDS(.) 

mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds")
metadata <-get_meta(mymre)

cat_vector <- get_cat(mymre) %>% 
  pull(CategoricalVariable)

long_var <- get_lon(mymre) %>%
  pull(LongitudinalVariable)

id_var <- get_lon(mymre) %>%
  pull(LinkVariable)


```

# Baseline differences in microbiome


## Alpha diversity

```{r baseline_richness, warning = F, message = F}

GR <- get_diversity(mymre, type = "igc", res_slot = "dataTable") 



Threshold <- GR %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% ### Again, you are using the full data table to calculate percentiles. This is wrong.
  as.numeric()


richness_df <-
  GR %>%
  filter(ReadCountReal >= Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
  group_by(SampleID) %>%
  summarise(across(everything(), min)) %>%
  ungroup() %>%
  select(SampleID, GeneNumber) %>%
  left_join(., metadata, by="SampleID") %>%
  select(id_var = !!sym(id_var),
         long_var = !!sym(long_var),
         cat_vector,
         GeneNumber) %>%
  filter(long_var == 0) 


c("CD4diff_48", "CD8diff_48") %>%
  set_names() %>% 
  map(function(cat_var){
    richness_df %>%
      dplyr::select(id_var,
                    cat_var = !!sym(cat_var),
                    GeneNumber) %>%
            filter(!is.na(cat_var)) %>%
      ggplot(., aes(x=cat_var, y=GeneNumber, fill = cat_var))+
      geom_violin(draw_quantiles = T) +
      # geom_boxplot(fill="lightgrey", width=0.5, na.rm=T) +
      geom_point() +
      scale_fill_brewer(palette="Set1") +
      ggpubr::stat_compare_means(method="wilcox",label = "p.signif", hide.ns = T) +
      theme_bw() + 
      labs(fill = cat_var)
    
  })

  
  


```

No differences in basal Gene richness were found between patients with an increase of CD4 higher than 50 counts/ul and whose with lower increase. The same could be observed in CD8 counts.


```{r Immune_delta_Distributions, warning=F, message=F}



ids <- metadata %>%
  dplyr::group_by(record_id) %>%
  dplyr::filter(time_point %in% c(0,48)) %>%
  dplyr::filter(!is.na(CD4)) %>%
  dplyr::filter(!is.na(CD8)) %>% 
  dplyr::arrange(time_point) %>%
  dplyr::summarise(n=n()) %>%
  dplyr::filter(n==2) %>%
  dplyr::pull(record_id)


deltas <- metadata %>% 
  # dplyr::filter(record_id %in% ids) %>%
  dplyr::filter(time_point %in% c(0,48,96)) %>% 
  dplyr::select(record_id, time_point,CD4,CD8) %>%
  dplyr::mutate(ratio = CD4/CD8) %>% 
  dplyr::group_by(record_id) %>% 
  dplyr::arrange(record_id,time_point) %>%
  tidyr::pivot_wider(id_cols = record_id, names_from= time_point, values_from = c(CD4,CD8, ratio)) %>%
  dplyr::mutate(
    deltaCD4_48 = CD4_48 - CD4_0, 
    deltaCD8_48 = CD8_48 - CD8_0,
    deltaCD4_96 = CD4_96 - CD4_0, 
    deltaCD8_96 = CD8_96 - CD8_0,
    deltaratio_48 = ratio_48 - ratio_0,
    deltaratio_96 = ratio_96 - ratio_0) %>%
  dplyr::select(record_id, contains("delta"))



deltas %>%
  pivot_longer(cols=c(deltaCD4_48,deltaCD8_48,deltaCD4_96,deltaCD8_96, deltaratio_48, deltaratio_96)) %>%
  mutate(term = str_remove_all(name, "delta|_48|_96"),
         tp = str_remove_all(name, ".*(?<=_)")) %>%
  ggplot(., aes(color=tp)) +
  geom_density(aes(x=value), alpha=0.4) +
  facet_wrap(~ term, scales="free", ncol = 2) +
  # geom_density(data=pivot_longer(cols=c(deltaCD4,deltaCD8)),aes(x=value), alpha=0.4) +
  theme_bw() +
  scale_color_brewer(palette = "Set1")+

  labs(color="week")


deltas %>% 
  select(contains("delta")) %>% 
  colnames() %>% 
  set_names() %>% 
  map(~{
    summary(deltas[,.x])
  })
  


```


The last piece of output (summaries) tell there is a great ammount of missing data. This is due to a combination of many patients missing a sample and a few not having entries for analytical data:

```{r count_missing_samples, warning=F, message=F}
no0 <- metadata %>% 
  filter(!record_id %in% metadata[metadata$time_point == 0, ]$record_id) %>%
  pull(record_id)

##patients without lymphocite data but with microbiome sample
no48_CD4 <- metadata %>%
  filter(is.na(CD4) ) %>%
  filter(time_point == 48) %>% 
  pull(record_id)  

no48_CD8 <- metadata %>%
  filter(is.na(CD8) & !is.na(SampleID)) %>%
  filter(time_point == 48) %>% 
  pull(record_id)  


no96_CD4 <- metadata %>%
  filter(is.na(CD4)& !is.na(SampleID)) %>%
  filter(time_point == 96) %>% 
  pull(record_id)

no96_CD8 <- metadata %>%
  filter(is.na(CD8)& !is.na(SampleID)) %>%
  filter(time_point == 96) %>% 
  pull(record_id)


# patients without sample
noS48 <- metadata %>%
  filter(!record_id %in% metadata[metadata$time_point == 48, ]$record_id) %>% 
  pull(record_id)

noS96 <- metadata %>%
  filter(!record_id %in% metadata[metadata$time_point == 96, ]$record_id) %>% 
  pull(record_id)



totalMissing_CD4_48 <- length(unique(c(no0, no48_CD4, noS48)))
totalMissing_CD4_96 <- length(unique(c(no0,no96_CD4, noS96)))
totalMissing_CD8_48 <- length(unique(c(no0, no48_CD8, noS48)))
totalMissing_CD8_96 <- length(unique(c(no0,no96_CD8, noS96)))

totalMissing_CD4_48
totalMissing_CD4_96
totalMissing_CD8_96
totalMissing_CD8_96
```

Adding the different kind of missing data checks out with the summaries. The biggest issue came with patients not having no sample whatsoever for such timepoints. Of which: 25 didn't have a basal sample, 22 didnt have sample for week 48 and 30 for week 96.


delta CD4 threshold was too low. To keep the study clinically interpretable We'll do a cutoff as a round number around the mean. In this case <150/>150 according to the mITT design results of the parent study.

Basal gene richness vs delta CD4_48 as continuous variables

# Microbiome structure at basal

```{r get_basal_mre, warning=F, message= F}
basal_samples <- metadata %>%
  filter(time_point == 0) %>% 
  pull(SampleID) %>% 
  as.character()

basal_mre <- filter_samples(mymre, basal_samples)

  
  
basal_mre@metadata@categorical_vals <- get_cat(mymre) %>% 
  filter(CategoricalVariable %in% "group")


basal_mre <- metar::metaphlan_nmds(basal_mre, tax_level="Species")


```



```{r basal_NMDS,eval=t ,warning=FALSE, message=FALSE}


group <- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$group & 
    scale_color_brewer(palette="Set1") &
    scale_fill_brewer(palette = "Set1")



cluster <- basal_mre@taxa@metaphlan@nmds$Species$top_50$categorical$cluster




```


NMDS shows no basal significant differences per group for either com,ponent. An unsupervised clustering created 2 well differentiated groups. While ADONIS per cluster was significant, we'll look whether the numerical variables have something to do with them.

```{r adonis _table, echo=F, warning=F, message = F}
kableExtra::kable(basal_mre@taxa@metaphlan@nmds$Species$top_50$all_adonis, format = "markdown")
```



# NMDS correlation with numerical variables
```{r NMDS_numeric_correlations, warning= F, message= F}
basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_numeric

 

```

# Numerical variables vs unsupervised clustering


```{r numerical_vs_cluster}


basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_numeric

```


# Categorical variables vs unsupervised clustering
```{r numerical_vs_cluster}


basal_mre@taxa@metaphlan@nmds$Species$top_50$cluster_categorical

```






## Baseline microbiome vs Immune response

The goal of this analysis is to describe the patients baseline microbiome vs immune response either using:

- Increases in CD4, categorized in a useful way (Delta_week48 for instance)
- Decreased in CD8, categorized in a useful way

It may be useful to try to classify patients solely on CD4/CD8/immune markers and the contrast this classification with microbiome composition

Also it is relevant to show that the baseline microbiom was NOT different between ARMs (DTG/DRVr)


