---
title: "Taxonomy: Differential Abundance"
author: "Carlos Bl√°zquez Bondia"
date: "3/23/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)

```

## Data import and exploration

This section should explain anyithing that may have happened during the first fase of importing and formatting data, such as any modification or additional filtering step.

```{r Setup, warning=F, message=F}

# mymre <- here::here("data","mre.rds") %>%
#   readRDS(.)

mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds")
# metadata <-get_meta(mymre)
metadata <- here::here("Metadata", "2022_04_28_clean_metadata_LIMS.csv") %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.))



cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

link_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)



```



```{r update_mre, warning= F, message= F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here("../WMGSPipeline"))

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()



```


```{r declare_support_functions}
# Any homebrew support function that may be written will be placed here

```


First we'll start with a non-compositional approach which may provide potential candidate taxa for DA to be confirmed with more astringent and specific compositional approaches.


# Non-compositional analysis

In this case, we'll test for group as categorical variables. We may extend the analysis towards other cat_vars, but those were more important for basal comparisons, which already shown no great differences were found among other potential confusor variables. 

## Cross-group Differential Abundance

These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.
```{r wilcox_across_group, warning=F, message=F}

phy <- get_phyloseq(mymre, type = "metaphlan") %>% 
      phyloseq::filter_taxa(., function(x) sum(x > 0) > (0.3*length(x)), TRUE)


otu_df <- 
  phyloseq::otu_table(phy) %>% 
  t() %>% 
  as.data.frame() %>% 
  stats::setNames(phyloseq::tax_table(phy)[,"Species"]) %>% 
  tibble::rownames_to_column("SampleID") %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\[")) %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\]"))


stat_list <-
  cat_vector %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    
    c(0,24,48,96) %>% 
      set_names() %>% 
      purrr::map_dfr(function(tp){
        otu_df %>%
        dplyr::select(-SampleID) %>%
        colnames() %>%
        purrr::set_names() %>%
        purrr::map_dfr(function(sp) {
          df <-
            otu_df %>%
            dplyr::select(SampleID, !!sym(sp)) %>%
            dplyr::left_join(metadata[, c("SampleID", link_var, cv, long_var), drop = F], 
                             by = "SampleID") %>% 
            dplyr::filter(!!sym(long_var) %in% tp) 

          
          if (length(unique(dplyr::pull(.data = df, !!sym(cv)))) == 2) {
            test <- df %>%
              rstatix::wilcox_test(., formula(paste(sp, "~", cv)), paired = F) %>%
              dplyr::mutate(method = "Wilcoxon") 
            
          } else{
            test <- df %>%
              rstatix::kruskal_test(., formula(paste(sp, "~", cv))) %>%
              dplyr::mutate(method = "Kruskal-Wallis")
          }
          
        }) %>% 
      rename(species = .y.) %>% 
      dplyr::mutate(long_var = tp) %>% 
      group_by(long_var) %>% 
      mutate(p.adj = p.adjust(p, "BH")) %>% 
      # filter(p.adj < .05 ) %>% 
      group_by(species) 
      }) 
}) 


DAsp <- stat_list$group %>% 
  filter(p<0.05) %>% 
  group_by(species)


## --- init MNJ
## It is interesting that Bifidobacterium species (adolescentis, longum) are consistently found as DA in all (most) timepoints
## Produce a table showing 
stat_list$group %>% 
  filter(p<0.05) %>% 
  group_by(species) %>% 
  tally() %>% 
  filter(n>=2) ## Species being significant (Wilcoxon unadjusted <0.05)

# otu_df %>% 
#   select(SampleID,Bifidobacterium_adolescentis) %>% 
#   # select(SampleID, !!sym(DAsp)) %>% 
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
#   group_by(group) %>% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Bifidobacterium_adolescentis,
#                               pairwise.comparisons = T,
#                               type="np",bf.message=NULL,
#                               ylab = "Bifidobacterium_adolescentis(%)", 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = "time_point",plotgrid.args = NULL)
# otu_df %>% 
#   select(SampleID,Bifidobacterium_longum) %>% 
#   # select(SampleID, !!sym(DAsp)) %>% 
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
#   group_by(group) %>% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Bifidobacterium_longum,
#                               pairwise.comparisons = F,
#                               type="np",bf.message=NULL,
#                               ylab = "Bifidobacterium_longum(%)", 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = "time_point",
#                               plotgrid.args = list("ncol" = 4, "nrow" = 1))
# 

 ### There is a nice signal, consistent longitudinally, in Bifidobacterium Genus
phyloseq::otu_table(phy %>% phyloseq::tax_glom(.,taxrank="Genus")) %>% 
  t() %>% 
  as.data.frame() %>% 
  stats::setNames(phyloseq::tax_table(phy %>% phyloseq::tax_glom(.,taxrank="Genus"))[,"Genus"]) %>% 
  tibble::rownames_to_column("SampleID") %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\[")) %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\]")) %>% 
  select(SampleID,Bifidobacterium) %>% 
  # select(SampleID, !!sym(DAsp)) %>% 
  left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
  group_by(group) %>% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x="group",
                              y = Bifidobacterium,
                              pairwise.comparisons = F,
                              type="np", bf.message=NULL,
                              ylab = "Bifidobacterium(%)", 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = "time_point",plotgrid.args = NULL) 
# %>% 
  # ggsave(.,filename = "~/Downloads/plot4Carlos_Bifidobacterium.pdf")

otu_df %>% 
  select(SampleID,Olsenella_scatoligenes) %>% 
  # select(SampleID, !!sym(DAsp)) %>% 
  left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
  group_by(group) %>% 
  ggstatsplot::grouped_ggbetweenstats(.,
                              x="group",
                              y = Olsenella_scatoligenes,
                              pairwise.comparisons = F,
                              type="np",bf.message=NULL,
                              ylab = "Olsenella_scatoligenes(%)", 
                              ggtheme = ggplot2::theme_bw(),
                              grouping.var = "time_point",plotgrid.args = NULL)


  
### Show how many species are differential in at least two of the timepoints in each cat var. Risk_group should result in many species showing that kins of signal. See Venn section below.
### --- end MNJ

```

Only one spec


### Comparison of significant species

Following Venn's diagram to detect group_specific DA taxa (Marc's addition, moved here as I thought it made more sense structurally)
```{r wilcox_Venn_Group, warning=F, message=F}

### MNJ: Provide a table of species theat appear in the Venn's diagram.

maxn <- DAsp %>% 
  select(species,long_var) %>% 
  group_by(long_var) %>% 
  count() %>% 
  pull(n) %>% 
  max


unique(DAsp$long_var) %>%
  set_names() %>%
  map_dfc(function(tp) {
    vec <- DAsp %>%
      dplyr::select(species, long_var) %>%
      dplyr::filter(long_var == tp) %>%
      dplyr::pull(species)
    
    if (length(vec) < maxn) {
      vec <- vctrs::vec_c(vec, rep(NA, maxn - length(vec)))
      return(vec)
    } else{
      return(vec)
    }
    
  }) %>%
  ggVennDiagram::ggVennDiagram(., show_intersect = T)
```

We can observe there are 3 species which are consistently differentially abundant between treatment groups at every point during treatment, but not at basal. It's interesting to note that of those 3, 2 are from the bifidobacterium genera (*B. adolescentis* and *B.longum*) plus another actinobacteria: *Olsenella scatoligenes*, all of the increasing more in the DTG group although the later has little literature regarding their role on the microbiome. (hover mouse over the figure numbers to display the names of the species)


```{r plot_shared species, warning=F, message=FALSE}

shared_sp <- 
  DAsp %>% 
  select(species, long_var) %>% 
  group_by(species) %>% 
  count() %>% 
  filter(n==3) %>% 
  pull(species)


shared_sp %>% 
  set_names() %>% 
  purrr::map(function(sp){
    otu_df %>% 
      select(SampleID, sp) %>% 
      left_join(metadata[,c("SampleID", "group", long_var)], by="SampleID") %>% 
      group_by(!!sym(long_var)) %>% 
      ggstatsplot::grouped_ggbetweenstats(., 
                                          type = "np",
                                          y = !!sym(sp), 
                                          x="group", 
                                          grouping.var = !!sym(long_var))
      
  })

shared_sp

```


## Longitudinal, within-group Differential Abundance
These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.
```{r wilcox_within_group, message=F, warning=F}

## This code may still be clunky and not work

endpoints <- c(48,96)



stat_list_long <-
  cat_vector %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    print(cv)
    otu_df %>%
      dplyr::select(-SampleID) %>%
      colnames() %>%
      set_names() %>%
      purrr::map_dfr(function(sp) {
        metadata %>%
          dplyr::filter(!is.na(!!sym(cv))) %>%
          dplyr::pull(!!sym(cv)) %>%
          unique() %>%
          set_names() %>%
          purrr::map_dfr(function(lv) {
            endpoints %>%
              purrr::map_dfr(function(ep) {
                ids <- metadata %>%
                  dplyr::select(link_var, long_var, cv) %>%
                  dplyr::filter(!!sym(long_var) %in% c(0, ep) &
                                  !!sym(cv) == lv) %>%
                  group_by(!!sym(link_var)) %>%
                  dplyr::count() %>%
                  filter(n > 1) %>%
                  pull(!!sym(link_var))
                
                
                test_df <-
                  otu_df %>%
                  dplyr::select(SampleID, !!sym(sp)) %>%
                  dplyr::left_join(metadata[, c("SampleID",
                                                link_var,
                                                cv,
                                                long_var), drop = F],
                                   by = "SampleID") %>%
                  dplyr::filter(!!sym(cv) == lv) %>%
                  dplyr::filter(time_point %in% c(0, ep)) %>%
                  dplyr::filter(!!sym(link_var) %in% ids)
                
                
                mean_0 <- mean(test_df %>%
                                 dplyr::filter(!!sym(long_var) == 0) %>%
                                 dplyr::pull(!!sym(sp)))
                mean_ep <- mean(test_df %>%
                                  dplyr::filter(!!sym(long_var) == ep) %>%
                                  dplyr::pull(!!sym(sp)))
                
                test <-
                  rstatix::wilcox_test(test_df , formula(paste(sp, "~", long_var)), paired = T) %>%
                  mutate(
                    level = lv,
                    mean_0 = mean_0,
                    mean_ep = mean_ep,
                    mean_fc = mean_ep / mean_0
                  )
                
                return(test)
              })
            
            
            
          }) %>%
          filter(p < 0.05)
        ## First I'll try a selection of overall significance by Kruskal-wallis. Thos significant will be selected for a pairwise tp-to-tp comparison, to save on computational resources.
      }) %>%
      rename(species = .y.)
    
  })

## Once we get the filtered species, do another loop doing pairwise wilcox for each one of the species in the "reduced table"

stat_list_long$group %>% 
  dplyr::filter(group2 == 48) %>% 
  dplyr::arrange(level) %>% 
  kableExtra::kable(format = "markdown", caption = "DA weeks 0 v 48 by Group")

stat_list_long$group %>% 
  dplyr::filter(group2 == 96) %>% 
  dplyr::arrange(level) %>% 
  kableExtra::kable(format = "markdown", caption = "DA weeks 0 v 48 by Group")


```




```{r plot_DA_species, warning=F, message=F}

group_plots <- 
  stat_list_long$group %>% 
  pull(species) %>% 
  unique() %>% 
  set_names() %>% 
  map(function(sp){
    otu_df %>% 
      dplyr::select(SampleID,!!sym(sp)) %>% 
      dplyr::left_join(metadata[c("SampleID",link_var, long_var, "group")]) %>% 
      dplyr::filter(time_point %in% c(0,48,96)) %>% 
      ggstatsplot::grouped_ggbetweenstats(., grouping.var = "group", 
                                          x="time_point",
                                          y=!!sym(sp), 
                                          type = "np",
                                          pairwise.comparisons = T,
                                          ggtheme = ggplot2::theme_bw(),
                                          palette="Set1")
      
    
    
  })

group_plots

```



# Compositional approach (ANCOM)

## Cross-group Differential Abundance

These comparisons will be between treatment groups at the same time point: one at week 0, week 24, week 48 and 96.
```{r ANCOM_across_group, warning=F, message=F}

```

### Comparison of significant species

Following Venn's diagram to detect group_specific DA taxa (Marc's addition, moved here as I thought it made more sense structurally)
```{r ANCOM_Venn_Group, warning=F, message=F}

```

## Longitudinal, within-group Differential Abundance
These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.

```{r ANCOM_within_group, warning=F, message= F}

```

