---
title: "Taxonomy: Differential Abundance"
author: "Carlos Bl√°zquez Bondia"
date: "3/23/2022"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
source("code/LMM.R")
source("code/myFunctions.R")
source("code/group_comparisons.R")
library(rstatix)
library(ggpubr)
devtools::load_all(here::here("../WMGSPipeline"))

```

## Data import and exploration

This section should explain anyithing that may have happened during the first fase of importing and formatting data, such as any modification or additional filtering step.

```{r Setup, warning=F, message=F}

mymre <- here::here("data","filt_mre.rds") %>%
  readRDS(.)
metadata <- 
  here::here("Metadata") %>% 
  file.info(list.files(., full.names = T)) %>%
  rownames() %>% 
  magrittr::extract(str_detect(.,"clean_metadata")) %>% 
  sort(decreasing = T) %>% 
  magrittr::extract(1L) %>% 
  read.csv() %>% 
  column_to_rownames("SampleID") %>% 
  mutate(SampleID = rownames(.)) %>% 
  dplyr::mutate(group_bmi = factor(
    case_when(BMI < 18.5 ~ "underweight",
                                BMI >= 18.5 & BMI <=25 ~ "healthy_weight",
                                BMI > 25 ~ "overweight"),
    levels = c("underweight", "healthy_weight", "overweight")
    )) 




cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

link_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)



```



```{r update_mre, warning= F, message= F, eval = F}

#### This chunk is for updating the mre internally as new variables come out. Once the final mre is done this chunk will be removed.

devtools::load_all(here::here("../WMGSPipeline"))

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
# mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
#   phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@taxa@metaphlan@phyloseq@sam_data <- phyloseq::sample_data(metadata)
mymre@taxa@metaphlan@phyloseq_sec@sam_data <- phyloseq::sample_data(metadata)

```


```{r declare_support_functions}
# Any homebrew support function that may be written will be placed here

```


First we'll start with a non-compositional approach which may provide potential candidate taxa for DA to be confirmed with more astringent and specific compositional approaches.


# Non-compositional analysis

In this case, we'll test for group as categorical variables. We may extend the analysis towards other cat_vars, but those were more important for basal comparisons, which already shown no great differences were found among other potential confusor variables. 

## Cross-group Differential Abundance

These comparisons will be between treatment groups at the same time point: one at week 0, week 48 and 96.
```{r wilcox_across_group, warning=F, message=F, cache = T}

phy <- get_phyloseq(mymre, type = "metaphlan") %>% 
      phyloseq::filter_taxa(., function(x) sum(x > 0) > (0.3*length(x)), TRUE)


otu_df <- 
  phyloseq::otu_table(phy) %>% 
  t() %>% 
  as.data.frame() %>% 
  stats::setNames(phyloseq::tax_table(phy)[,"Species"]) %>% 
  tibble::rownames_to_column("SampleID") %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\[")) %>% 
  stats::setNames(.,stringr::str_remove_all(colnames(.), "\\]"))

cat_vector <- c("group", "group_bmi")

stat_list <-
  cat_vector %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    
    c(0,24,48,96) %>% 
      set_names() %>% 
      purrr::map_dfr(function(tp){
        otu_df %>%
        dplyr::select(-SampleID) %>%
        colnames() %>%
        purrr::set_names() %>%
        purrr::map_dfr(function(sp) {
          df <-
            otu_df %>%
            dplyr::select(SampleID, !!sym(sp)) %>%
            dplyr::left_join(metadata[, c("SampleID", link_var, cv, long_var), drop = F], 
                             by = "SampleID") %>% 
            dplyr::filter(!!sym(long_var) %in% tp) 

          
          if (length(unique(dplyr::pull(.data = df, !!sym(cv)))) == 2) {
            test <- df %>%
              rstatix::wilcox_test(., formula(paste(sp, "~", cv)), paired = F) %>%
              dplyr::mutate(method = "Wilcoxon") 
            
          } else{
            test <- df %>%
              rstatix::kruskal_test(., formula(paste(sp, "~", cv))) %>%
              dplyr::mutate(method = "Kruskal-Wallis")
          }
          
        }) %>% 
      rename(species = .y.) %>% 
      dplyr::mutate(long_var = tp) %>% 
      group_by(long_var) %>% 
      mutate(p.adj = p.adjust(p, "BH")) %>% 
      # filter(p.adj < .05 ) %>% 
      group_by(species) 
      }) 
}) 


DAsp <- stat_list$group %>% 
  filter(p<0.05) %>% 
  group_by(species)


stat_list$group_bmi %>% 
  dplyr::filter(p<0.05)

## --- init MNJ
## It is interesting that Bifidobacterium species (adolescentis, longum) are consistently found as DA in all (most) timepoints
## Produce a table showing 
stat_list$group %>% 
  filter(p<0.05) %>% 
  group_by(species) %>% 
  tally() %>% 
  filter(n>=2) ## Species being significant (Wilcoxon unadjusted <0.05)

# otu_df %>%
#   select(SampleID,Bifidobacterium_adolescentis) %>%
#   # select(SampleID, !!sym(DAsp)) %>%
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>%
#   group_by(group) %>%
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Bifidobacterium_adolescentis,
#                               pairwise.comparisons = T,
#                               type="np",bf.message=NULL,
#                               ylab = "Bifidobacterium_adolescentis(%)",
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = "time_point",plotgrid.args = NULL)
# otu_df %>%
#   select(SampleID,Bifidobacterium_longum) %>%
#   # select(SampleID, !!sym(DAsp)) %>%
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>%
#   # group_by(group) %>%
#   dplyr::ungroup() %>% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Bifidobacterium_longum,
#                               pairwise.comparisons = F,
#                               type="np",bf.message=NULL,
#                               ylab = "Bifidobacterium_longum(%)",
#                               ggtheme = ggplot2::theme_bw(),
#                               # grouping.var = "time_point",
#                               plotgrid.args = list("ncol" = 4, "nrow" = 1)) +
#   facet_wrap()
# 

 ### There is a nice signal, consistent longitudinally, in Bifidobacterium Genus
# phyloseq::otu_table(phy %>% phyloseq::tax_glom(.,taxrank="Genus")) %>% 
#   t() %>% 
#   as.data.frame() %>% 
#   stats::setNames(phyloseq::tax_table(phy %>% phyloseq::tax_glom(.,taxrank="Genus"))[,"Genus"]) %>% 
#   tibble::rownames_to_column("SampleID") %>% 
#   stats::setNames(.,stringr::str_remove_all(colnames(.), "\\[")) %>% 
#   stats::setNames(.,stringr::str_remove_all(colnames(.), "\\]")) %>% 
#   select(SampleID,Bifidobacterium) %>% 
#   # select(SampleID, !!sym(DAsp)) %>% 
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
#   group_by(group) %>% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Bifidobacterium,
#                               pairwise.comparisons = F,
#                               type="np", bf.message=NULL,
#                               ylab = "Bifidobacterium(%)", 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = "time_point",plotgrid.args = NULL) 
# # %>% 
#   # ggsave(.,filename = "~/Downloads/plot4Carlos_Bifidobacterium.pdf")
# 
# otu_df %>% 
#   select(SampleID,Olsenella_scatoligenes) %>% 
#   # select(SampleID, !!sym(DAsp)) %>% 
#   left_join(., metadata[,c("SampleID","group","time_point")], by="SampleID") %>% 
#   group_by(group) %>% 
#   ggstatsplot::grouped_ggbetweenstats(.,
#                               x="group",
#                               y = Olsenella_scatoligenes,
#                               pairwise.comparisons = T,
#                               type="np",bf.message=NULL,
#                               ylab = "Olsenella_scatoligenes(%)", 
#                               ggtheme = ggplot2::theme_bw(),
#                               grouping.var = "time_point",plotgrid.args = NULL)
### Show how many species are differential in at least two of the timepoints in each cat var. Risk_group should result in many species showing that kins of signal. See Venn section below.
### --- end MNJ

```



```{r plots_poster, echo = F, message=FALSE, warning=FALSE, results='hide'}
# b.adolescentis <- 

metadata <- 
  metadata %>% 
  dplyr::mutate(group = str_replace(group, "bDRV", "bDRV"))

B.a <- 
otu_df %>% 
  # select(SampleID,DAsp$species) %>% 
  # select(SampleID, !!sym(DAsp)) %>% 
  left_join(., metadata[,c("SampleID","group","time_point", "record_id")], by="SampleID") %>% 
  group_by(group) %>% 
  get_comp_boxplots(.,cat_var = "group", 
                    num_var = "Bifidobacterium_adolescentis", 
                    long_var = "time_point", 
                    link_var = "record_id", 
                    y_trans = "sqrt",
                    include.pvals = "both") %>% 
  purrr::pluck("plot") +
  theme_bw() +
  labs(y = expression(sqrt("Abundance")), title = "B.adolescentis") + 
  theme(plot.title = element_text(face = "bold.italic", size = 14),
        legend.position = "right", 
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 14, face = "bold", vjust = -1.5),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_blank(),
        text = element_text(family = "serif"))  +
  scale_fill_manual(values = c("#AAAAAA", "#000000"))


B.l <- 
otu_df %>% 
  # select(SampleID,DAsp$species) %>% 
  # select(SampleID, !!sym(DAsp)) %>% 
  left_join(., metadata[,c("SampleID","group","time_point", "record_id")], by="SampleID") %>% 
  group_by(group) %>% 
  get_comp_boxplots(.,
                    cat_var = "group",
                    num_var = "Bifidobacterium_longum", 
                    long_var = "time_point", 
                    link_var = "record_id",
                    y_trans = "sqrt",
                    include.pvals = "both") %>% 
  purrr::pluck("plot") +
  # scale_y_sqrt(breaks = seq(1,10,1)) +
  # coord_trans(y = "sqrt") +
  # scale_y_continuous() +
  theme_bw() +
  labs(y = expression(sqrt("Abundance")), title = "B.longum") + 
  theme(plot.title = element_text(face = "bold.italic", size = 14),
        legend.position = "right", 
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 14, face = "bold", vjust = -1.5),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_blank(), 
        text = element_text(family = "serif")) +
  scale_fill_manual(values = c("#AAAAAA", "#000000"))

O.s <- 
otu_df %>% 
  # select(SampleID,DAsp$species) %>% 
  # select(SampleID, !!sym(DAsp)) %>% 
  left_join(., metadata[,c("SampleID","group","time_point", "record_id")], by="SampleID") %>% 
  # group_by(group) %>% 
  get_comp_boxplots(.,
                    cat_var = "group",
                    num_var = "Olsenella_scatoligenes",
                    long_var = "time_point",
                    link_var = "record_id",
                    y_trans = "sqrt",
                    include.pvals = "both") %>% 
  pluck("plot") +
  theme_bw() +
  labs(y = expression(sqrt("Abundance")), title = "O.scatoligenes") + 
  theme(plot.title = element_text(face = "bold.italic", size = 14),
        legend.position = "right", 
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 14, face = "bold", vjust = -1.5),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 12, face = "bold"),
        text = element_text(family = "serif")) +
  scale_fill_manual(values = c("#AAAAAA", "#000000"))

plot_sps <- 
ggarrange(B.a, B.l, O.s, common.legend = T, nrow = 3, ncol = 1) %>% 
  annotate_figure(., left = text_grob(label = expression(sqrt("Abundance")),rot = 90,face = "bold", size = 14),
                  bottom = text_grob(label = "Week", face = "bold", size = 14))+
  scale_fill_manual(values = c("#AAAAAA", "#000000"))

igc<-metar::get_diversity(mymre)@igc

Threshold <-  igc@dataTable %>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads,0.02)) %>% 
  as.numeric()
igc_df <- igc@dataTable %>%
                filter(NumberMappedReads >= Threshold) %>%
                filter(ReadCountReal>=Threshold) %>% ### Get the Mapped Read count (per-sample) just above the threshold
                group_by(SampleID) %>%
                summarise(across(everything(), min)) %>%
                ungroup() %>%
          right_join(., metadata, by="SampleID") %>%
          select(SampleID,
                 richness = GeneNumber,
                 link_var=link_var,
                 cat_var = group,
                 long_var = long_var,
                 NumberMappedReads)

igc_bp <- 
  igc_df %>% 
    get_comp_boxplots(.,
                    cat_var = "cat_var",
                    num_var = "richness",
                    long_var = "long_var",
                    link_var = "link_var",
                    include.pvals = "both") %>% 
  purrr::pluck("plot") +
  labs(y = "gene count", fill = "group", title = "gene richness") +
  scale_fill_manual(values = c("#AAAAAA", "#000000")) +
  theme(plot.title = element_text(face = "bold", size = 14),
        legend.position = "right", 
        axis.title.x = element_blank(),
        axis.title.y = element_text(size = 14),
        axis.text.x = element_text(size = 12),
        axis.text.y = element_text(size = 12, face = "bold"),
        text = element_text(family = "serif"))




plt <- 
ggarrange(igc_bp,B.a, B.l, O.s, common.legend = T, ncol = 1, align = "hv", widths = c(1.2,1))%>% 
  annotate_figure(., bottom = text_grob(label = "Week", face = "bold", size = 14))



plt
ggsave(plot = plt, filename = "output/figures/poster_boxplots.svg", width = 14, height = 28, units = "cm")
```

Bifidobacterium adolescentis, Bifidobacterium longum and Olsenella scatoligenes were significant in all timepoints. Let's check the actual abundances per group and timepoint:

```{r species_group_comparisons, warning = F, message=F}

sp_vec <- c("Bifidobacterium_adolescentis","Bifidobacterium_longum","Olsenella_scatoligenes")


sp_res <- 
otu_df %>% 
  left_join(metadata, by = "SampleID") %>% 
  # select(SampleID, record_id, time_point, group, abundance = !!sym(sp)) %>% 
  get_group_comparisons(., 
                        link_var = "record_id",
                        long_var = "time_point",
                        cat_vector = "group",
                        num_vector = c("Bifidobacterium_adolescentis","Bifidobacterium_longum","Olsenella_scatoligenes"),
                        comps = c("bDRV", "DTG"),
                        type = "categorical")


sp_res$group$Bifidobacterium_adolescentis %>%  pluck("stats") %>% as.data.frame()
sp_res$group$Bifidobacterium_longum %>%  pluck("stats") %>% as.data.frame()
sp_res$group$Olsenella_scatoligenes %>%  pluck("stats") %>% as.data.frame()



```



### Comparison of significant species

Following Venn's diagram to detect group_specific DA taxa (Marc's addition, moved here as I thought it made more sense structurally)
```{r wilcox_Venn_Group, warning=F, message=F}

### MNJ: Provide a table of species theat appear in the Venn's diagram.

maxn <- DAsp %>% 
  select(species,long_var) %>% 
  group_by(long_var) %>% 
  count() %>% 
  pull(n) %>% 
  max


unique(DAsp$long_var) %>%
  set_names() %>%
  map_dfc(function(tp) {
    vec <- DAsp %>%
      dplyr::select(species, long_var) %>%
      dplyr::filter(long_var == tp) %>%
      dplyr::pull(species) 
    
    if (length(vec) < maxn) {
      vec <- vctrs::vec_c(vec, rep(NA, maxn - length(vec)))
      return(vec)
    } else{
      return(vec)
    }
    
  }) %>%
  ggVennDiagram::ggVennDiagram(., show_intersect = T)
```

We can observe there are 3 species which are consistently differentially abundant between treatment groups at every point during treatment, but not at basal. It's interesting to note that of those 3, 2 are from the bifidobacterium genera (*B. adolescentis* and *B.longum*) plus another actinobacteria: *Olsenella scatoligenes*, all of the increasing more in the DTG group although the later has little literature regarding their role on the microbiome. (hover mouse over the figure numbers to display the names of the species)


```{r plot_shared species, warning=F, message=FALSE}

shared_sp <- 
  DAsp %>% 
  select(species, long_var) %>% 
  group_by(species) %>% 
  count() %>% 
  filter(n==3) %>% 
  pull(species)


shared_sp %>% 
  set_names() %>% 
  purrr::map(function(sp){
    otu_df %>% 
      select(SampleID, sp) %>% 
      left_join(metadata[,c("SampleID", "group", long_var)], by="SampleID") %>% 
      group_by(!!sym(long_var)) %>% 
      ggstatsplot::grouped_ggbetweenstats(., 
                                          type = "np",
                                          y = !!sym(sp), 
                                          x="group", 
                                          grouping.var = !!sym(long_var))
      
  })

shared_sp

```


## Longitudinal, within-group Differential Abundance
These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.
```{r wilcox_within_group, message=F, warning=F, cache = F}

## This code may still be clunky and not work

endpoints <- c(48,96)

future::plan("multicore")

stat_list_long <-
  cat_vector %>%
  purrr::set_names() %>%
  furrr::future_map(function(cv) {

    otu_df %>%
      dplyr::select(-SampleID) %>%
      colnames() %>%
      purrr::set_names() %>%
      purrr::map_dfr(function(sp) {
        metadata %>%
          dplyr::filter(!is.na(!!sym(cv))) %>%
          dplyr::pull(!!sym(cv)) %>%
          unique() %>%
          purrr::set_names() %>%
          purrr::map_dfr(function(lv) {
            endpoints %>%
              purrr::map_dfr(function(ep) {
                ids <- metadata %>%
                  dplyr::select(link_var, long_var, cv) %>%
                  dplyr::filter(!!sym(long_var) %in% c(0, ep) &
                                  !!sym(cv) == lv) %>%
                  dplyr::group_by(!!sym(link_var)) %>%
                  dplyr::count() %>%
                  dplyr::filter(n > 1) %>%
                  dplyr::pull(!!sym(link_var))
                
# At the end it sounded better to just go and perform the pairwise comparisons.                
                test_df <-
                  otu_df %>%
                  dplyr::select(SampleID, !!sym(sp)) %>%
                  dplyr::left_join(metadata[, c("SampleID",
                                                link_var,
                                                cv,
                                                long_var), drop = F],
                                   by = "SampleID") %>%
                  dplyr::filter(!!sym(cv) == lv &
                                  time_point %in% c(0, ep) &
                                  !!sym(link_var) %in% ids) %>%
                  dplyr::arrange(!!sym(link_var), !!sym(long_var))
                
                
                mean_0 <- mean(test_df %>%
                                 dplyr::filter(!!sym(long_var) == 0) %>%
                                 dplyr::pull(!!sym(sp)))
                mean_ep <- mean(test_df %>%
                                  dplyr::filter(!!sym(long_var) == ep) %>%
                                  dplyr::pull(!!sym(sp)))
                max <- max(test_df %>%
                             dplyr::filter(!!sym(long_var) == ep) %>%
                             dplyr::pull(!!sym(sp)))
                
                test <-
                  rstatix::wilcox_test(test_df , formula(paste(sp, "~", long_var)), paired = T) %>%
                  mutate(
                    level = lv,
                    mean_0 = mean_0,
                    mean_ep = mean_ep,
                    mean_fc = mean_ep / mean_0,
                    max = max
                  )
                
                return(test)
              })
            
            
            
          }) %>%
          filter(p < 0.05)
      }) %>%
      rename(species = .y.)
    
  })



names(stat_list_long) %>% 
  set_names() %>% 
  map(~{ stat_list_long %>% 
      pluck(.x) %>% 
      dplyr::filter(group2 == 48) %>% 
  dplyr::arrange(level) %>% 
  kableExtra::kable(format = "markdown", caption = paste0("DA weeks 0 v 48 by ", .x))})



 "group" %>%
  set_names() %>%
  map( ~ {
    stat_list_long %>% 
      pluck(.x) %>% 
      dplyr::filter(group2 == 96) %>%
      dplyr::arrange(level) %>%
      kableExtra::kable(format = "markdown", caption = paste0("DA weeks 0 v 96 by delta CD4 at ", .x))
    
  })

```



# Constant DA species: Venn diagrams

As we can see, under exploratory conditions there are many DA species across all groups. Plotting them all would be quite computationally expensive and plotting so many longitudinal plots may be even confusing. We'll do additional filtering. Instead of a more astringent p-value, we'll try to select DA species more functionally. First will be Venn diagramns for DA species for each group level and time point, trying to pinpoint those who consistently changed across the study:

```{r venn_long_group, warning=F, message=F, eval=T}





### MNJ: Provide a table of species theat appear in the Venn's diagram.

comps_group <- list("DTG"=c("DTG_48","DTG_96"),
              "bDRV" = c("bDRV_48", "bDRV_96"))  # Those are the group intersections I considered most interesting

Venn_Group <- stat_list_long$group %>% # Construct list for Venn diagram
      dplyr::select(species, group2, level) %>%
      dplyr::mutate(row=row_number()) %>% 
      tidyr::pivot_wider(., values_from = "species", names_from = c("level", "group2")) %>% 
      dplyr::select(-row) %>%
      map(~{
        .x[!is.na(.x)]
      }) 

group_diff <- names(comps_group) %>% # DA species common between 48 and 96 for each group
  set_names() %>% 
  map(~ {
    Venn_Group %>% 
      ggVennDiagram::Venn(.) %>% 
      ggVennDiagram::discern_overlap(., slice = comps_group[[.x]])
  }) 



Venn_Group %>% # Plot Venn
  ggVennDiagram::ggVennDiagram(., show_intersect = T)
    

## The following is the same for CD4 deltas at 48 and 96 weeks

# comps_delta <- list("150_H"=c("150_H_48","150_H_96"),
#               "150_L" = c("150_L_48", "150_L_96"))
# 
# 
# Venn_delta48 <- stat_list_long$CD4diff_48 %>%
#       dplyr::select(species, group2, level) %>%
#       dplyr::mutate(row=row_number()) %>% 
#       tidyr::pivot_wider(., values_from = "species", names_from = c("level", "group2")) %>% 
#       dplyr::select(-row) %>%
# 
#       map(~{
#         .x[!is.na(.x)]
#         
#       }) 
# delta48_diff <- names(comps_delta) %>% 
#   set_names() %>% 
#   map(~ {
#     Venn_delta48 %>% 
#       ggVennDiagram::Venn(.) %>% 
#       ggVennDiagram::discern_overlap(., slice = comps_delta[[.x]])
#   }) 
# 
#     
# Venn_delta96 <- stat_list_long$CD4diff_96 %>%
#       dplyr::select(species, group2, level) %>%
#       dplyr::mutate(row=row_number()) %>% 
#       tidyr::pivot_wider(., values_from = "species", names_from = c("level", "group2")) %>% 
#       dplyr::select(-row) %>%
# 
#       map(~{
#         .x[!is.na(.x)]
#         
#       }) 
# delta96_diff <- names(comps_delta) %>% 
#   set_names() %>% 
#   map(~ {
#     Venn_delta96 %>% 
#       ggVennDiagram::Venn(.) %>% 
#       ggVennDiagram::discern_overlap(., slice = comps_delta[[.x]])
#   }) 

   
```


The species we found are interesting as these are the ones that may be differential between both treatment grous. We'll plot the longitudinal analysis for these selected subsets (DA at both TP 48 and 96 only for DTG, and the same for bDRV)


```{r plot_DA_species, warning=F, message=F, eval=T}


#Code is functional bu we still need to filter the input tables.



# prepare_venn_results <- function(res, cat_var, otu_df, metadata, long_var){} # Should i write it as a function?
cat_var = "group"

venn_group_res <- group_diff %>% 
  names() %>% 
  purrr::set_names() %>% 
  purrr::map(function(lv){
    group_diff[[lv]] %>% 
      purrr::set_names() %>% 
      purrr::map(function(sp){
        otu_df %>% 
          dplyr::select(SampleID,!!sym(sp)) %>% 
          dplyr::left_join(., metadata[,c("SampleID", cat_var, long_var, link_var)], by="SampleID") %>% 
          dplyr::filter(!!sym(cat_var) == lv)
      })
  })

cv <- "group"
pal <- RColorBrewer::brewer.pal(2,"Set1")[c(1,2)] %>% 
  setNames(c("bDRV","DTG"))
long_plots <-
  venn_group_res %>%
  names() %>%
  set_names() %>%
  map(function(lv)
  {
    names(venn_group_res[[lv]]) %>%
      set_names() %>%
      map(function(sp)
      {
        groupmax <-
          venn_group_res %>%
          purrr::pluck(lv, sp) %>%
          dplyr::summarise(max = max(!!sym(sp)))
        
        stat_params <-
          stat_list_long %>%
          purrr::pluck(cv) %>%
          dplyr::filter(species == sp) %>%
          dplyr::mutate(group2 = as.factor(group2)) %>%
          rstatix::add_xy_position() %>%
          dplyr::rename(cat_var = level) %>%
          dplyr::mutate(xmax = case_when(group2 == 48 ~ 2, group2 == 96 ~ 3),
                        y.position = 1.1 * max) %>% 
          dplyr::mutate(p = round(p,3))
        
        venn_group_res %>%
          purrr::pluck(lv, sp) %>%
          dplyr::arrange(!!sym(link_var)) %>%
          dplyr::rename(long_var = !!sym(long_var),
                        sp = !!sym(sp)) %>%
          dplyr::filter(long_var %in% c(0, stat_params$group2)) %>%
          dplyr::mutate(long_var = as.factor(long_var)) %>%
          ggplot(., aes(x = long_var, y = sp)) +
          geom_violin(width = 1.5,
                      trim = T,
                      fill = pal[lv]) +
          geom_boxplot(width = 0.1, fill = "white") +
          geom_line(aes(group = !!sym(link_var)), color = "grey") +
          scale_fill_brewer(palette = "Set1") +
          ggpubr::stat_pvalue_manual(
            stat_params,
            label = "p = {p}",
            xmin = "xmin",
            xmax = "xmax",
            y = "y.position"
          ) +
          theme_bw() +
          labs(x = "week", y = "RA (%)", title = sp)
        
        
      })
    
  })

```


## DA (DTG group)

```{r DTG_longPlots, echo=F, warning=FALSE, message=FALSE}


ggpubr::ggarrange(plotlist = long_plots$DTG, ncol=2)


```

## DA (DTG group)

```{r DRV_longPlots, echo=F, warning=FALSE, message=FALSE}


ggpubr::ggarrange(plotlist = long_plots$`bDRV` , ncol=2)


```


# Compositional approach (ANCOM)

## Cross-group Differential Abundance

These comparisons will be between treatment groups at the same time point: one at week 0, week 24, week 48 and 96.
```{r ANCOM_across_group, warning=F, message=F}
mymre@gene_function <- GeneFunctionSet()

mymre <- metar:::metaphlan_ancom(mymre, tax_level = "Species", save_files = F)

ancom_group <- get_taxa(mymre, type = "metaphlan",result_slot = "ancom") %>% 
  purrr::pluck("Species", "group","plot")

cat_ancom_list <- 
c(0,48,96) %>% 
  purrr::set_names() %>% 
  purrr::map(function(tp){
    keep_ids <- 
    metadata %>% 
      dplyr::filter(time_point == tp) %>% 
      dplyr::pull(SampleID)
    
    
    my_ancom <-
      metar::filter_samples(mymre, sample_ids = keep_ids) %>%
      metar::metaphlan_ancom(., tax_level = "Species", save_files = F) %>%
      metar::get_taxa(., type = "metaphlan",result_slot = "ancom") %>% 
      purrr::pluck("Species", "group","plot")

})


names(cat_ancom_list) %>% 
  purrr::set_names() %>% 
  purrr::map(function(tp){
    cat_ancom_list %>% 
      purrr::pluck(tp, "data") %>% 
      dplyr::filter(detected_0.6 == T)
  })

```


Interestingly tough, no species were found significant at any timepoint between both groups from a compositional point of view.


### Comparison of significant species

Following Venn's diagram to detect group_specific DA taxa (Marc's addition, moved here as I thought it made more sense structurally)
```{r ANCOM_Venn_Group, warning=F, message=F}

```

## Longitudinal, within-group Differential Abundance
These comparisons test differences between time points within the same group. This is a longitudinal approach so it requires a paired test, which account for dependece between data points.

```{r ANCOM_within_group, warning=F, message= F}
cat_vars <- data.frame(CategoricalVariable = "time_point", PaketteName = "Set1")
cat_vec <- c("group")
ancom_longlist <- 
cat_vec %>% 
  purrr::set_names() %>%
  purrr::map(function(cv) {
    metadata %>%
      dplyr::pull(!!sym(cv)) %>%
      unique() %>%
      purrr::set_names() %>%
      purrr::map(function(lv) {
        
        c(48, 96) %>%
          set_names() %>%
          purrr::map(function(tp) {
            
            ids <-
              metadata %>%
              dplyr::filter(!!sym(cv) == lv &
                            !!sym(long_var) %in% c(0,tp)) %>%
              dplyr::pull(SampleID)
            
            my_ancom <- 
              metar::filter_samples(mymre, sample_ids = ids) %>%
              metar::get_phyloseq(object = ., type = "metaphlan_counts") %>% 
              metar::run_ancom(phy = ., categorical_vars = cat_vars[1,], tax_level = "Species",save_files = F)
            
            plt <- 
              my_ancom$time_point$plot +
              labs(title=paste("ANCOM ", lv, " group", sep=""))
            stats <- 
              my_ancom$time_point$stats %>% 
              dplyr::filter(detected_0.7 == T) %>% 
              mutate(comp = paste(cv, "_0v",tp, sep=""))
            return(list(plot = plt, stats = stats))
          })
      })
  })

# pull significan

DTG_plotlist <- list(ancom_longlist$group$DTG$`48`$plot, 
                     ancom_longlist$group$DTG$`96`$plot)

ggpubr::ggarrange(plotlist = DTG_plotlist)

RTV_plotlist <- list(ancom_longlist$group$`bDRV`$`48`$plot,
                     ancom_longlist$group$`bDRV`$`96`$plot)

ggpubr::ggarrange(plotlist = RTV_plotlist)



statlist <- rbind(ancom_longlist$group$DTG$`48`$stats, 
                     ancom_longlist$group$DTG$`96`$stats,
                      ancom_longlist$group$`bDRV`$`48`$stats,
                     ancom_longlist$group$`bDRV`$`96`$stats)

kableExtra::kable(statlist, format = "markdown")

```

Interestingly though, only one species (Streptococcus parasanguinis) appeared as statistically significant DA, only after 96 weeks in the DTG group. This species was also identified in the non-compositional analysis, being in the intersection between DTG(0-48), DTG(0-96) and bDRV(0-96).

```{r plot_strep, message = FALSE, warning= FALSE}
phy <- metar::get_phyloseq(mymre, type = "metaphlan")



phyloseq::otu_table(phy) %>% 
  t() %>% 
  as.data.frame() %>% 
  stats::setNames(phyloseq::tax_table(phy)[,"Species"]) %>% 
  tibble::rownames_to_column("SampleID") %>% 
  dplyr::select(SampleID,SampleID,statlist$taxa_id) %>% 
  dplyr::inner_join(metadata[,c("SampleID","record_id" ,"time_point","group")], by = "SampleID") %>% 
  create_LMM(data = .,
             breakpoints = NULL, 
             cat_var = "group",
             num_var= statlist$taxa_id, 
             long_var = "time_point", 
             link_var = "record_id") 
  



```
 This is backed by LMMs which present a really small albeit significant reduction in this strain in the DTG group.

```{r oral_microbiome}
oral_db <- here::here("Metadata","HOMD_oral_microbiome.txt") %>% 
  read.delim(., header = T, skip = 1) %>% 
  dplyr::select(Genus,Species) %>% 
  mutate(name = paste(Genus, Species, sep="_")) %>% 
  dplyr::pull(name)
  

stat_list_oral <- 
names(stat_list_long) %>% 
  purrr::set_names() %>% 
  purrr::map(function(cv){
    stat_list_long %>% 
      purrr::pluck(cv) %>% 
      dplyr::filter(species %in% oral_db) %>% 
      dplyr::pull(species) %>% 
      unique()
  })


stat_list_oral$group %>% 
  purrr::set_names() %>% 
  purrr::map(function(sp){
    otu_df %>% 
      dplyr::select(SampleID, sp) %>% 
      inner_join(metadata[,c("SampleID","record_id","group","time_point")], by="SampleID") %>% 
  ggstatsplot::grouped_ggbetweenstats(., grouping.var = "group", x = "time_point", y = !!sym(sp),type = "np", pairwise.comparisons = T)

  })
  



```

