---
title: "Adenovirus Analisis"
author: "Marc Noguera, Carlos Bl√°zquez"
date: "5/3/2022"
output: html_document
editor_options:
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(RColorBrewer)
library(metar)
source(here::here("code","LMM.R"))
source(here::here("code","myFunctions.R"))
```

```{r read_data, warning=FALSE, message=FALSE, cache=F}

mymre <- mymre <- aws.s3::s3readRDS(bucket = "s3://mistral-wp6-advanz4", object = "metagenome/WMGS/MREObject.rds")
# metadata <-get_meta(mymre)


cat_df <- 
  here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T)

cat_vector <- cat_df %>% 
  pull(CategoricalVariable)

long_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LongitudinalVariable)

link_var <- here::here("Metadata", "LongitudinalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(LinkVariable)

num_var <- here::here("Metadata", "NumericalVariables.txt") %>%
  read.delim(., header = T)  %>% 
  pull(NumericalVariable)


metadata <- 
  here::here("Metadata", "2022_05_30_clean_metadata_LIMS.csv") %>% 
  read.csv() %>% 
  mutate(ADV_p = case_when(
    is.na(ADV) ~ "no", 
    !is.na(ADV) ~ "yes"), .after = "ADV") %>% 
  mutate(rownames = SampleID) %>% 
  column_to_rownames("rownames")



```


```{r mre_update,message = F, warning = F, echo= F}
devtools::load_all(here::here("../WMGSPipeline"))

mymre <- filter_samples(mre = mymre, sample_ids = metadata$SampleID)

mymre@metadata@metadata_df <- as.tibble(metadata) 
mymre@taxa@metaphlan@phyloseq@sam_data <- metadata %>%
  phyloseq::sample_data(.)
mymre@taxa@metaphlan@phyloseq_sec@sam_data <- metadata %>%
  phyloseq::sample_data(.)




mymre@metadata@categorical_vals <- here::here("Metadata", "CategoricalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@numeric_vals <- here::here("Metadata", "NumericalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()

mymre@metadata@longitudinal_vals <- here::here("Metadata", "LongitudinalVariables.txt") %>% 
  read.delim(., header = T) %>% 
  tibble()




```



# Import data and variable definition.

Only a few patients had any detecatble ADV presence on their guts, which also had huge variations. Hence we expect a quantitative approach to hold little power. We'll complement with a qualitative approach based on presence/absence of both ADV and EV.

First, classify each sample as ADV+/ADV-. (Do the same for Enterovirus).

# Adenovirus per treatment


A quick view at the value distribution should inform of normality and ossible outliers

```{r first_exploration, warning=F, message = F}
metadata %>% 
  dplyr::select(link_var = !!sym(link_var),
                long_var = !!sym(long_var),
                cat_var = "group",
                ADV) %>% 

  ggplot2::ggplot(., aes(x=ADV, color=cat_var)) +
  geom_density(stat="density") +
  theme_minimal() +
  labs(title = "Adenovirus", subtitle = "Abundance distribution (log10)", x="log10(ADV)",
       color = "group") +
  scale_color_brewer(palette = "Set1")

```



It is clear there are outliers on the data which heavily skew the DTG group. This is due to one specific sample from the DTG group with an outstanding ammount of ADV. 
```{r pick_outlier, warning = F, message = F, echo = F}
metadata %>% 
  dplyr::filter(.[,"ADV"] == max(.[,"ADV"], na.rm=T)) %>%
  select(SampleID, record_id, time_point, ADV) %>% 
  kableExtra::kable(., format = "markdown")

```

This one will be taken out from quantitative analysis from now on.


```{r filtered_distribution, message = F, warning = F}
maxADV <- metadata %>% 
  pull(ADV) %>% 
  max(., na.rm = T)

metadata_f <-
  metadata %>%
  dplyr::filter((ADV != maxADV) %>% 
  replace_na(T))




metadata_f %>% 
  dplyr::select(link_var = !!sym(link_var),
                long_var = !!sym(long_var),
                cat_var = "group",
                ADV) %>% 

  ggplot2::ggplot(., aes(x=ADV, color=cat_var)) +
  geom_density(stat="density") +
  theme_minimal() +
  labs(title = "Adenovirus", subtitle = "Abundance distribution (log10)", x="log10(ADV)",
       color = "group") +
  scale_color_brewer(palette = "Set1")

```

Now the data is still heavily skewed but we can see a more reasonable distribution, with majority of 0s but those with actual abundance have more "evenly" distributed values.

## Quantitative exploration

```{r ADV_quantitative, message= FALSE, warning= FALSE}


 



LMMlist <- metadata_f %>% 
  create_LMM(data = .,
           num_var = "ADV",
           cat_var = "group",
           long_var = long_var, 
           link_var = link_var,
           breakpoints = NULL)


LMMlist$stats





unique(metadata_f$group) %>% 
  purrr::set_names() %>% 
  purrr::map_dfr(~{
    metadata_f %>% 
  dplyr::select(link_var = !!sym(link_var),
                long_var = !!sym(long_var),
                cat_var = "group",
                ADV) %>% 
      dplyr::filter(cat_var == .x) %>% 
      dplyr::pull(ADV) %>% 
      summary() 

}) %>% 
  dplyr::mutate(lev=unique(metadata_f$group)) %>% 
  tibble::column_to_rownames("lev") %>% 
  kableExtra::kable(format="markdown")

  

```
We see here that ADV follows a non-normal heavily skewed toward the left. We'll check how do they behave considering treatment time.

```{r ADV_longitudinal, message=FALSE, warning=FALSE}


myLMMs <- get_lmm_effects(data =metadata_f, 
                          cat_vector = c("group"), 
                          num_vector = c("ADV"), 
                          long_var = long_var, 
                          link_var = "record_id") 

myLMMs$group$ADV$plot



```

It appears both groups have opposite trends, as the DRV/r group tends to increase while the DTG decreases. However, it is not only backed ANOVA test, but the increase in the DRV/r group should be taken with caution, as most patients actually decrease excep for one who has a massive increase between weeks 48 and 96. There is also little coninuity in the data, as mmost patients only present some levels of ADV at only one time point, those gaps negatively affecting the test's robustness.

# Adenovirus qualitative analysis

We will test for presence/absence and try to test for differences among groups. We'll test for proportions of patients positive per ADV per group and time. We'll first visualize the evolution of the data. Afterwards, an exact Fisher's test will be used between baseline and weeks 48/96 for each group.

```{r ADV_prevalence, message= FALSE, warning=FALSE}
cat_var = "group"
num_var = "ADV"
link_var = "record_id"

#Exploratory histogram
metadata %>% 
  dplyr::select(link_var = link_var, 
                long_var = long_var, 
                cat_var = cat_var, 
                num_var = ADV_p) %>% 
  dplyr::group_by(cat_var, long_var) %>% 
  dplyr::count(num_var) %>% 
  dplyr::mutate(long_var = as.factor(long_var)) %>% 
  tidyr::pivot_wider(id_cols = c("cat_var","long_var"), names_from = "num_var", values_from = "n") %>%
  dplyr::mutate(incidence = yes*100 / (no + yes)) %>% 
  ggplot(., aes(x=long_var, y = incidence, fill = cat_var)) +
  # facet_wrap(~cat_var) +
  geom_histogram(stat="identity", position = "dodge") +
  theme_minimal() +
  scale_fill_brewer(palette = "Set1") +
  labs(x = "week",y="ADV incidence (%)", fill = "group") 
  

## Fisher's test


unique(metadata[,cat_var]) %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    a <- 
    metadata %>%
      select(cat_var = cat_var,
             long_var = long_var,
             ADV_p) %>%
      dplyr::filter(cat_var == cv) %>%
      dplyr::group_by(long_var) %>%
      dplyr::count(ADV_p) %>%
      dplyr::ungroup() %>%
      tidyr::pivot_wider(
        id_cols = c("long_var"),
        names_from = "ADV_p",
        values_from = "n"
      ) %>%
      tibble::column_to_rownames("long_var") %>%
      as.matrix() %>%
      fisher.test()
    
  })



```


Although there is a difference in ADV incidence between baselina and week 96 for both groups, no decreasing trend over tie could be observe, as ADV seems to spike in the following weeks of starting treatment. Interestingly, this spike happens at week 24 for RTV/r and week 48 for DTG.

No differences in presence/absence ratios per time points  were found in either group, according to Fisher's test.



### Adenovirus detection vs Microbiome composition


```{r gene_richness_corr, message=FALSE, warning=FALSE    }

threshold <-   
  metar::get_diversity(mymre, type = "igc", res_slot = "metaphlan3") %>%
  purrr::pluck("dataTable")%>%
  dplyr::select(SampleID, NumberMappedReads) %>%
  unique() %>%
  summarise(Q = quantile(NumberMappedReads, 0.02)) %>%
  as.numeric()

igc <-   
  metar::get_diversity(mymre, type = "igc", res_slot = "metaphlan3") %>%
  purrr::pluck("dataTable") %>% 
  filter(NumberMappedReads >= threshold) %>%
  filter(ReadCountReal >= threshold) %>%
  group_by(SampleID) %>%
  summarise(across(everything(), min)) %>%
  ungroup() %>% 
  select(SampleID, GeneNumber)


metadata_f %>% 
  dplyr::select(SampleID, 
                link_var = "record_id", 
                # long_var = long_var,
                cat_var = cat_var,
                ADV) %>% 
  dplyr::right_join(igc, by="SampleID") %>% 
  # mutate(ADV = log10(ADV)) %>%
  get_lmm_effects(., cat_vector = cat_var, num_vector = "ADV", long_var = "GeneNumber",link_var = "link_var") %>% 
  purrr::pluck("group","ADV","plot") +
  scale_x_continuous() +
  # scale_y_log10() +
  labs(x = "gene count")



```

Nothing conclussive can be extracted by correlating gene richness to ADV presence, except that both groups evolution are close to a perfect match. This is mainly due to how much ADV levels vary beween patients and how discontinuous the data is. Most patients persent ADV in only one or two timepoints, which affects the robustness of any longitudinal approach. An qualitative approach may be more useful.

```{r ADV_presence_igc, warning = F, message = F, fig.width=8, fig.height=10}

c(0,48,96) %>% 
  purrr::set_names() %>% 
  purrr::map(function(tp){
  
    group_difs <- metadata %>% 
  dplyr::filter(time_point == tp) %>% 
  dplyr::select(SampleID, 
                link_var = "record_id", 
                # long_var = long_var,
                cat_var = cat_var,
                ADV_p) %>%
  dplyr::left_join(igc, by="SampleID") %>% 
  dplyr::filter(!is.na(ADV_p)) %>% 
  dplyr::group_by(cat_var) %>% 
  rstatix::wilcox_test(., GeneNumber ~ ADV_p) %>% 
  rstatix::add_xy_position(x = "cat_var") %>% 
  # mutate(xmin = c(0.8,1.2),
  #        xmax=c(1.8,2.2)) %>% 
  rstatix::add_significance() %>% 
  dplyr::mutate(ADV_p = "yes")  # This dows nothing, but plotting fails without it as it keeps looking for "ADV-p" because it is stated in aes()
  
  


metadata %>% 
    dplyr::filter(time_point == tp) %>% 
  dplyr::select(SampleID, 
                link_var = "record_id", 
                # long_var = long_var,
                cat_var = cat_var,
                ADV_p) %>%
  dplyr::left_join(igc, by="SampleID") %>% 
  dplyr::filter(!is.na(ADV_p)) %>% 
  ggplot(aes(x=cat_var, y = GeneNumber, fill = ADV_p)) +
  geom_boxplot(color="black", outlier.shape = NA) +
  geom_point(size = 0.7, position = position_jitterdodge(jitter.width = .2), shape = 3) +
  theme_bw() +
  labs(x = "group", y = "gene count", fill = "ADV", title = paste("ADV/group, week ",tp,sep="")) +
  scale_fill_brewer(palette = "Set1") +
  ggpubr::stat_pvalue_manual(data = group_difs,label = "p = ,{p}", hide.ns = F, size = 3)

  }) %>% 
  ggpubr::ggarrange(plotlist = ., common.legend = T)




```
Here we see a similar signal. Gene richness increased more in the DTG than in the RTV/r group, but no differences were found between those with and without presence of Adenovirus. Low amount of patients with ADV and great variation may be taking a toll on Wilcoxon tests sensitivity.

# NMDS 

Last, we will look for links between overall microbiome structure and Adenovirus incidence.

```{r construct_nmds, warning = F, message = F, cache = F}

# Add the categorical var ADV_p to the mre

mymre@metadata@categorical_vals <- rbind(mymre@metadata@categorical_vals, c("ADV_p", "Set1") )

mymre <- metaphlan_nmds(mymre, tax_level = "Species", top_n = 50, save_files = F)

ADV_nmds <- get_taxa(mymre, type="metaphlan", result_slot = "nmds") %>% 
  purrr::pluck("Species", "top_50","categorical","ADV_p")
```

No significant clstering can be observed.

```{r plot_nmds, warning=F, message= F, echo = F}
ADV_nmds

```




```{r nmds_ADV, warning=F, message = F, cache= F, fig.width= 10, fig.height= 8, eval=F}

NMDS_long_list <- c("ADV_p") %>%
  purrr::set_names() %>%
  purrr::map(function(cv) {
    c("NMDS1", "NMDS2") %>%
      purrr::set_names() %>%
      purrr::map(function(comp) {
        ADV_nmds$data %>%
          dplyr::select(SampleID, NMDS = comp) %>%
          dplyr::inner_join(metadata_f[, c("SampleID", "ADV_p","time_point" ,"record_id", cv)], by =
                              "SampleID") %>%
          get_lmm_effects(data = ., cat_vector = cv, num_vector = "NMDS", long_var = "time_point", link_var = "record_id") %>% purrr::pluck(cv,"NMDS","plot") +
          scale_x_continuous(n.breaks = 10) +
          # ggplot(., aes(x = time_point, y = NMDS, fill = !!sym(cv),color = !!sym(cv))) +
          # geom_point() +
          # geom_smooth(, method = "lm")+
        labs(x = "week", y = comp)
        
      }) %>%
      ggpubr::ggarrange(plotlist = ., common.legend = T)
    
  })


NMDS_long_list$ADV_p


```

We observe no temporal tendency towards any direction in the NMDS by ADV presence, while group do



Use:

* NMDS coordinates (NMDS1 and NMDS2) vs ADV+/ADV-
* PERMANOVA to check significance.
* ADV+/ADV- vs gene richness


